<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

  <!--radix_placeholder_meta_tags-->
  <title>: No 3. Diversity</title>

  <meta property="description" itemprop="description" content="In this section of the workflow we look at the diversity of the water samples collected from Cayo Roldan (hypoxic) and Cayo Coral (normoxic)."/>

  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>
  <link rel="icon" type="image/vnd.microsoft.icon" href="assets/favicon.ico"/>


  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content=": No 3. Diversity"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="In this section of the workflow we look at the diversity of the water samples collected from Cayo Roldan (hypoxic) and Cayo Coral (normoxic)."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content=""/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content=": No 3. Diversity"/>
  <meta property="twitter:description" content="In this section of the workflow we look at the diversity of the water samples collected from Cayo Roldan (hypoxic) and Cayo Coral (normoxic)."/>

  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Tools for microbiome analysis in r. Version 1.9.97;citation_publication_date=2017;citation_author=Leo Lahti;citation_author=Shetty Sudarshan;citation_author=Tools for microbiome analysis in r. Version 1.9.97"/>
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["No 3. Diversity"]},{"type":"character","attributes":{},"value":["In this section of the workflow we look at the diversity of the water samples collected from Cayo Roldan (hypoxic) and Cayo Coral (normoxic).\n"]},{"type":"character","attributes":{},"value":["assets/cite.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>

  <script type="application/javascript">

  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }

  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }

  function createFuseIndex() {

    // create fuse index
    var options = {
      keys: [
        { name: 'title', weight: 20 },
        { name: 'categories', weight: 15 },
        { name: 'description', weight: 10 },
        { name: 'contents', weight: 5 },
      ],
      ignoreLocation: true,
      threshold: 0
    };
    var fuse = new window.Fuse([], options);

    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });

        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }

  window.document.addEventListener("DOMContentLoaded", function (event) {

    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;

    createFuseIndex()
      .then(function(fuse) {

        // make search box visible
        searchEl.classList.remove('hidden');

        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
                ? `<img src="${offsetURL(suggestion.preview)}"</img>`
                : '';
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description || ''}
                  </div>
                  <div class="search-item-preview">
                    ${img}
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
        // remove inline display style on autocompleter (we want to
        // manage responsive display via css)
        $('.algolia-autocomplete').css("display", "");
      })
      .catch(function(error) {
        console.log(error);
      });

  });

  </script>

  <style type="text/css">

  .nav-search {
    font-size: x-small;
  }

  /* Algolioa Autocomplete */

  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #0F2E3D solid;
    width: 180px;
  }


  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      display: none;
      visibility: hidden;
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      display: inline-block;
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }

  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }

  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    font-size: 1rem;
    margin-block-start: 0;
    margin-block-end: 5px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }

  </style>


  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refHtml = $('#ref-' + $(this).attr('data-cites')).html();
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.8/header-attrs.js"></script>
  <script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
  <script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
  <link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
  <script src="site_libs/datatables-binding-0.18/datatables.js"></script>
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
  <script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
  <script src="site_libs/jszip-1.10.20/jszip.min.js"></script>
  <link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
  <script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
  <link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
  <script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Inconsolata|Source+Sans+Pro|Work+Sans:200,300,400,500|Karla:400,400i,700,700i');
  /********************************/
  /* controls general properties of site*/
  /********************************/

  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    background-color: #FEFEFE;
  }

  d-title h1{
  font-size: 2.1em;
  font-family: 'Work Sans';
  font-weight: 300;
  }

  d-article h1 {
    font-size: 1.8em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 300;
  }

  d-article h2 {
    font-size: 1.65em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 500;
  }

  d-article h3 {
    margin-top: 1em;
  }
  d-byline .authors-affiliations {
    grid-column-end: span 3;
    grid-column-gap: 8px;
  }

  img.orcid-id {
    margin: 0;
  }

  a.name:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  a.affiliation:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  blockquote p {
  	font-size: 1.1em;
  	font-family: 'Work Sans';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }

  d-article li {
    margin: 0px 0px 4px;
  }

  /********************************/
  /* controls CODE*/
  /********************************/

  /** pre.sourceCode controls code block without lang **/

  /** Controls r code  **/

  pre.sourceCode.r {
    margin-top: 0px;
     color: #333333;
     background-color: #E8E8E8;
     border-radius: 3px;
     font-size: 0.80em;
     font-family: 'Monaco', 'PT Mono', monospace;
     overflow-x: scroll !important;
     border-left: 6px solid #006ddb;
     border-top: 1px solid #B7B7B7;
     border-right: 1px solid #B7B7B7;
     border-bottom: 1px solid #B7B7B7;
     padding: 18px 0px 18px 20px;
   }

   pre.sourceCode.bash {
     margin-top: 0px;
      color: #333333;
      background-color: #E8E8E8;
      border-radius: 3px;
      font-size: 0.80em;
      font-family: 'Monaco', 'PT Mono', monospace;
      overflow-x: scroll !important;
      border-left: 6px solid #D55E00;
      border-top: 1px solid #B7B7B7;
      border-right: 1px solid #B7B7B7;
      border-bottom: 1px solid #B7B7B7;
      padding: 18px 0px 18px 20px;
    }
  /** Controls code output **/

  d-article pre {
     margin-top: 0px;
      color: #333333;
      background-color: #FAFAFA;
      font-size: 0.8em;
      border: 1px solid #ccc;
      overflow-x: scroll !important;
      padding:  10px 10px 10px 18px;
      border-radius: 5px;

    }

  /** Controls inline code  **/

  p code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  d-article aside code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-family: 'Monaco', 'PT Mono', monospace;
    color: #BF2727;
  }

  /********************************/
  /* controls NAV BAR and HEADERS*/
  /********************************/

  .nav-dropdown-content .nav-dropdown-header {
    text-transform: uppercase;
    color: #686868;
  }

  .nav-dropdown-content a {
    font-family: 'Work Sans'
  }
  .nav-dropdown .nav-dropbtn {
    color: #FEFEFE;
    font-weight: 400;
  }

  .distill-site-header .logo img {
    margin-top: 10px;
    max-height: 30px;
    margin-bottom: auto;
  }

  .distill-site-nav {
      color: #FEFEFE;
      background-color: #143D59;
      font-size: 0.95em;
      font-weight: 400;
      min-width: 165px;
      position: absolute;
      font-family: 'Karla';
      letter-spacing: .1em;

  }

  .article-banner {
    width: auto;
    height: auto;
    border: 0;
  }

  img[src="assets/banner.png"] {
    border: 0;
    padding: 0;
    width: 100%;
    margin: 100px 10px 0px 25px;
  }

  /*************************************************
  *  code for side by side krona plot
  **************************************************/

  .column {
    float: left;
    width: 48%;
    padding: 5px;
  }

  /* Clear floats after image containers */
  .row::after {
    content: "";
    clear: both;
    display: table;
  }

  div.krona {
    width: 100%;
  	height: 90%;
    background-color: white;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    margin-bottom: 25px;
  	margin-top: 5px;
  }

  div.container-krona {
    text-align: center;
    padding: 1px 10px;
  	color: #414141;
  	font-size: 1.48em;
  	line-height: 0.8em;
  	font-family: 'Inconsolata';
  }

  div.container-krona p {
    margin: 20px;
  }

  div.krona img {
    border:1px solid white;
    margin: 25px;
  }

  img {
    margin: 25px;
  }

  /*************************************************
  *  code for LIGHTBOX MARKUP
  **************************************************/

  .thumbnail {
    max-width: 100%;
    position: relative;
  }

  .lightbox {
  	/** Default lightbox to hidden */
  	display: none;

  	/** Position and style */
  	position: fixed;
  	z-index: 999;
  	width: 100%;
  	height: 100%;
  	text-align: center;
  	top: 0;
  	left: 0;
  	background: rgba(0,0,0,0.8);
  }

  .lightbox img {
  	/** Pad the lightbox image */
  	position: fixed;
  	top: 0;
  	left: 0;
  	right: 0;
  	bottom: 0;
  	max-width: 0%;
  	max-height: 0%;
  	margin: auto;
  	box-sizing: border-box;
  }

  .lightbox:target {
  	/** Remove default browser outline */
  	outline: none;

  	/** Unhide lightbox **/
  	display: block;
  }

  .lightbox:target img {
  	max-height: 100%;
  	max-width: 100%;
  }

  /*************************************************
  *  code for DETAILS fold down
  **************************************************/

  details {
    background-color: #ccd7df;
    cursor: pointer;
    color: #282828;
    margin-bottom:1vh;
    padding: 1em;
    border: none;
    text-align: left;
    outline: none;
    font-family: 'Karla'
  }

  .active, details:hover {
    transition: max-height 0.2s ease-out;
    outline: none;

    /* background-color: #acc2c2; */
  }

  details {
    display: flex;
    justify-content: space-between;
    align-content: center;
    border-radius: 4px;
    outline: none;

  }

  details.active {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    outline: none;

  }

  summary::-webkit-details-marker {
    display: none;
    outline: none;

  }

  summary:focus {
    outline: none;
  }

  summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f0fe";
  }

  details[open] summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f146";
  }

  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"No 3. Diversity","description":"In this section of the workflow we look at the diversity of the water samples collected from Cayo Roldan (hypoxic) and Cayo Coral (normoxic).","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a class="logo" href="index.html">
<img src="assets/icon.png" alt="Logo"/>
</a>
<a href="index.html" class="title"></a>
</div>
<div class="nav-right">
<a href="index.html">
<i class="fa fa-home" aria-hidden="true"></i>
</a>
<a href="intro.html">OVERVIEW</a>
<a href="field.html">FIELD</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
16S rRNA
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Processing</span>
<a href="16s-dada2.html">1. DADA2</a>
<a href="16s-data-prep.html">2. Data Prep</a>
<hr/>
<span class="nav-dropdown-header">Analyses</span>
<a href="16s-water.html">3. Diversity</a>
<a href="16s-da-asvs.html">4. Differentially Abundant ASVs</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
METAGENOMIC
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Setup</span>
<a href="mg-setup.html">1. Working Envrionment</a>
<a href="mg-databases.html">2. Annotation Databases</a>
<hr/>
<span class="nav-dropdown-header">Read Processing</span>
<a href="mg-workflow-1.html">3. Assembly &amp; Annotations</a>
<a href="mg-workflow-2.html">4. Assembley &amp; Annotation Summary</a>
<hr/>
<span class="nav-dropdown-header">MAG Analysis</span>
<a href="mg-binning.html">5. Binning</a>
<a href="mag02-phylogenomics.html">6a. MAG02 Phylogenomics</a>
<a href="mag04-phylogenomics.html">6b. MAG04 Phylogenomics</a>
<a href="mg-function.html">7. KEGG/KOFam Analysis</a>
</div>
</div>
<a href="data-availability.html">DATA</a>
<a href="som.html">SOM</a>
<a href="https://github.com/hypocolypse/web/" aria-label="Link to source">
<i class="fab fa-github" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>No 3. Diversity</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p><p>In this section of the workflow we look at the diversity of the water samples collected from Cayo Roldan (hypoxic) and Cayo Coral (normoxic).</p></p>
</div>


<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#data-availability">Data Availability</a></li>
<li><a href="#workflow-overview">Workflow Overview</a></li>
<li><a href="#summary-of-water-samples">Summary of Water Samples</a></li>
<li><a href="#diversity">Diversity</a>
<ul>
<li><a href="#taxonomic-diversity">Taxonomic Diversity</a></li>
<li><a href="#hypoxic-vs.-normoxic">Hypoxic vs.Normoxic</a></li>
<li><a href="#during-vs.-after-the-event">During vs.After the Event</a></li>
<li><a href="#source-code">Source Code</a></li>
<li><a href="#data-availability-1">Data Availability</a></li>
</ul></li>
</ul>
</nav>
</div>
<details>
<summary>
Show setup information.
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">0199</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq); <span class="fu">packageVersion</span>(<span class="st">&quot;phyloseq&quot;</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings); <span class="fu">packageVersion</span>(<span class="st">&quot;Biostrings&quot;</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(gdata)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(DT, ggplot2, dplyr, microbiome, tidyverse, </span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>               data.table, plyr, stringr, labdsv, reshape,  </span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">install =</span> <span class="cn">FALSE</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">scipen=</span><span class="dv">999</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_current<span class="sc">$</span><span class="fu">get</span>(<span class="fu">c</span>(</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache&quot;</span>,</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.path&quot;</span>,</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.rebuild&quot;</span>,</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;dependson&quot;</span>,</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;autodep&quot;</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>))</span></code></pre></div>
</div>
</details>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="data-availability">Data Availability</h1>
<blockquote>
<p>In order to run this workflow, you either need to run the previous workflows or grab the files below from figshare.</p>
</blockquote>
<p>All files needed to run this workflow can be downloaded from figshare. These are the output files from the <a href="16s-data-prep.html">Data Prep Workflow</a>.</p>
<iframe src="https://widgets.figshare.com/articles/12403925/embed?show_title=1" width="568" height="351" frameborder="0">
</iframe>
<p>File names and descriptions:</p>
<p><strong>seq_table.txt</strong>, <strong>tax_table.txt</strong>, <strong>asv.fasta</strong>: Sequence tables, taxonomy tables, and ASV fasta files from the full phyloseq object (before removing contaminants) and the trimmed dataset (after removing contaminants).</p>
<p><strong>16s-data_prep.rdata</strong>: contains all variables and phyloseq objects from the data prep pipeline. To see the Objects, in R run load(16s-data_prep.rdata, verbose=TRUE)</p>
<h1 id="workflow-overview">Workflow Overview</h1>
<p>This script will process the curated water sample data from the <a href="16s-data-prep.html">Data Prep Workflow</a>.</p>
<h1 id="summary-of-water-samples">Summary of Water Samples</h1>
<p>Now that we have a phyloseq object containing the water samples only, we can summarize the data in the water phyloseq object. Again, we use the <code>summarize_phyloseq</code> from the <a href="https://github.com/microbiome/microbiome/">microbiome R package</a> <span class="citation" data-cites="lahti2017microbiome">(<a href="#ref-lahti2017microbiome" role="doc-biblioref">Lahti, Sudarshan, and others 2017</a>)</span> as we did before.</p>
<p>First, load the R objects that contain the sample data, sequence table, and taxonomy table (saved at the end of the previous section) and merge them into a single phyloseq object. Doing it this way keeps the memory footprint low.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>sample_d <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;rdata/16s-data-prep/ps_water-sample.rds&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>seqtab <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;rdata/16s-data-prep/ps_water-seqtab.rds&quot;</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>taxtab <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;rdata/16s-data-prep/ps_water-taxtab.rds&quot;</span>)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>ps_water <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(sample_d, seqtab, taxtab)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>ps_water_o <span class="ot">&lt;-</span> ps_water</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>ps_water</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1272 taxa and 8 samples ]
sample_data() Sample Data:       [ 8 samples by 5 sample variables ]
tax_table()   Taxonomy Table:    [ 1272 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<p>Notice we made a copy of the <code>ps_water</code> phyloseq object called <code>ps_water_o</code>. This is because we are going to change the names of <code>NA</code> taxa below and we want a copy of the original object before making changes. Next, we can save copies of the sequence and taxonomy tables just in case we want a quick look at these data.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">tax_table</span>(ps_water),</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;tables/16s-water/water_tax_table.txt&quot;</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ps_water)),</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;tables/16s-water/water_seq_table.txt&quot;</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span></code></pre></div>
</div>
<p>And generate a fasta file for all ASVs in the water samples.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create fasta file from tax_table</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>table2format <span class="ot">&lt;-</span> <span class="fu">tax_table</span>(ps_water)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co">#retain only the column with the sequences</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>table2format_trim <span class="ot">&lt;-</span> table2format[, <span class="dv">7</span>]</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>table2format_trim_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">row.names</span>(table2format_trim),</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>                                   table2format_trim)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(table2format_trim_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ASV_ID&quot;</span>, <span class="st">&quot;ASV_SEQ&quot;</span>)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">#format fasta</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>table2format_trim_df<span class="sc">$</span>ASV_ID <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;ASV&quot;</span>, <span class="st">&quot;&gt;ASV&quot;</span>, table2format_trim_df<span class="sc">$</span>ASV_ID)</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(table2format_trim_df, <span class="st">&quot;tables/16s-water/water_asv.fasta&quot;</span>,</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\r</span><span class="st">&quot;</span>, <span class="at">col.names =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">fileEncoding =</span> <span class="st">&quot;UTF-8&quot;</span>)</span></code></pre></div>
</div>
<p>Lets go ahead and a sample variable called <code>oxstate</code> to the sample data frame of the water phyloseq object so we can compare the taxonomy of normoxic (Coral Caye) vs.hypoxic (Cayo Roldan) samples. Note one of the Cayo Roldan samples (WCR0) was collected after the hypoxic event when oxygen levels had returned to normal. Therefore WCR0 is considered a normoxic sample.</p>
<p>One last thing to do is to compare diversity across the two sites <em>before</em> and <em>after</em> the hypoxic event. For this we add a sample variable called <code>period</code> denoting whether the sample was collected during or after the event.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(ps_water)<span class="sc">$</span>oxstate <span class="ot">&lt;-</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">&quot;normoxic&quot;</span>, <span class="st">&quot;normoxic&quot;</span>, <span class="st">&quot;normoxic&quot;</span>, <span class="st">&quot;normoxic&quot;</span>,</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;normoxic&quot;</span>, <span class="st">&quot;hypoxic&quot;</span>, <span class="st">&quot;hypoxic&quot;</span>, <span class="st">&quot;hypoxic&quot;</span>)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(ps_water)<span class="sc">$</span>period <span class="ot">&lt;-</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">&quot;cc_after&quot;</span>, <span class="st">&quot;cc_during&quot;</span>, <span class="st">&quot;cc_during&quot;</span>, <span class="st">&quot;cc_during&quot;</span>,</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;cr_after&quot;</span>, <span class="st">&quot;cr_during&quot;</span>, <span class="st">&quot;cr_during&quot;</span>, <span class="st">&quot;cr_during&quot;</span>)</span></code></pre></div>
</div>
<blockquote>
<p>The <code>ps_water</code> dataset contains 1272 ASVs, 180557 total reads, 8 samples, and 5 sample variables.</p>
</blockquote>
<p>And here is a summary of just water data samples.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<table>
<thead>
<tr class="header">
<th>Metric</th>
<th>Results</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>12804</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>33311</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>180557</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>22570</td>
</tr>
<tr class="odd">
<td>Median number of reads</td>
<td>24120</td>
</tr>
<tr class="even">
<td>Min. number of ASVs</td>
<td>237</td>
</tr>
<tr class="odd">
<td>Max. number of ASVs</td>
<td>552</td>
</tr>
<tr class="even">
<td>Total number of ASVs</td>
<td>1272</td>
</tr>
<tr class="odd">
<td>Average number of ASVs</td>
<td>317</td>
</tr>
<tr class="even">
<td>Median number of ASVs</td>
<td>295.5</td>
</tr>
<tr class="odd">
<td>Sparsity</td>
<td>0.751</td>
</tr>
<tr class="even">
<td>Any ASVs sum to 1 or less?</td>
<td>TRUE</td>
</tr>
<tr class="odd">
<td>Number of singleton ASVs</td>
<td>68</td>
</tr>
<tr class="even">
<td>Percent of ASVs that are singletons</td>
<td>5.346</td>
</tr>
<tr class="odd">
<td>Number of sample variables are:</td>
<td>5 (SamName, TYPE, SITE, oxstate, period)</td>
</tr>
</tbody>
</table>
<p>We can also generate a summary table of total reads &amp; ASVs for each sample. You can sort the table or download a copy. Here is the code to generate the data for the table. First, we create data frames that hold total reads and ASVs for each sample. We can also do a quick calculation of alpha diversity using the Shannon and InvSimpson indices.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> <span class="fu">sample_sums</span>(ps_water)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(total_reads, <span class="at">make.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> total_reads <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;Sample_ID&quot;</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>total_asvs <span class="ot">&lt;-</span> <span class="fu">estimate_richness</span>(ps_water,</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>                                <span class="at">measures =</span> <span class="fu">c</span>(</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;Observed&quot;</span>, <span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;InvSimpson&quot;</span>))</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>total_asvs <span class="ot">&lt;-</span> total_asvs <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;Sample_ID&quot;</span>)</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>total_asvs<span class="sc">$</span>Sample_ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>, <span class="st">&#39;-&#39;</span>, total_asvs<span class="sc">$</span>Sample_ID)</span></code></pre></div>
</div>
<p>And then we merge these two data frames with the sample data frame. We will use the <code>meta</code> command from the microbiome package to convert the <code>sample_data</code> to a data frame.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>sam_details <span class="ot">&lt;-</span> <span class="fu">meta</span>(<span class="fu">sample_data</span>(ps_water))</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sam_details) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sam_details) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Sample_ID&quot;</span>, <span class="st">&quot;Type&quot;</span>, <span class="st">&quot;Site&quot;</span>, <span class="st">&quot;Oxstate&quot;</span>, <span class="st">&quot;Period&quot;</span>)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>merge_tab <span class="ot">&lt;-</span> <span class="fu">merge</span>(sam_details, total_reads, <span class="at">by =</span> <span class="st">&quot;Sample_ID&quot;</span>)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>merge_tab2 <span class="ot">&lt;-</span> <span class="fu">merge</span>(merge_tab, total_asvs, <span class="at">by =</span> <span class="st">&quot;Sample_ID&quot;</span>)</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(merge_tab2) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Sample&lt;br/&gt;ID&quot;</span>, <span class="st">&quot;Type&quot;</span>, <span class="st">&quot;Site&quot;</span>, <span class="st">&quot;Oxstate&quot;</span>, <span class="st">&quot;Period&quot;</span>,</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;total&lt;br/&gt;reads&quot;</span>, <span class="st">&quot;total&lt;br/&gt;ASVs&quot;</span>, <span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;InvSimpson&quot;</span>)</span></code></pre></div>
</div>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="mex738zeiysgiy3fysb8" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="mex738zeiysgiy3fysb8">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Total reads &amp; ASVs by sample.<\/em>\n<\/caption>","data":[["WCC0","WCC1","WCC2","WCC3","WCR0","WCR1","WCR2","WCR3"],["W","W","W","W","W","W","W","W"],["CC","CC","CC","CC","CR","CR","CR","CR"],["normoxic","normoxic","normoxic","normoxic","normoxic","hypoxic","hypoxic","hypoxic"],["cc_after","cc_during","cc_during","cc_during","cr_after","cr_during","cr_during","cr_during"],[24290,24278,23962,19817,28616,12804,13479,33311],[305,287,287,237,323,242,304,552],[3.79586821583198,3.40949079884578,3.61556802457059,3.60334996135797,3.19266288364594,3.87638453942529,4.45854339112232,4.59793953134247],[8.64499926459407,5.452464523952,6.75399022762029,6.9033065773033,4.3230831643379,15.3718361806848,37.7312581395943,35.1458238662054]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Sample<br/>ID<\/th>\n      <th>Type<\/th>\n      <th>Site<\/th>\n      <th>Oxstate<\/th>\n      <th>Period<\/th>\n      <th>total<br/>reads<\/th>\n      <th>total<br/>ASVs<\/th>\n      <th>Shannon<\/th>\n      <th>InvSimpson<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Brti","buttons":["copy","csv","excel"],"pageLength":8,"lengthMenu":[5,10],"columnDefs":[{"targets":7,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\");\n  }"},{"targets":8,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\");\n  }"},{"className":"dt-right","targets":[5,6,7,8]}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\nvar value=data[8]; $(this.api().cell(row, 8).node()).css({'font-size':'80%'});\n}"}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.rowCallback"],"jsHooks":[]}</script>
</div>
<h1 id="diversity">Diversity</h1>
<h2 id="taxonomic-diversity">Taxonomic Diversity</h2>
<p>Lets first take a look at the taxonomic diversity of the dataset. The code to generate this table is a little gross.</p>
<h3 id="rename-na-taxonomic-ranks">0. Rename <em>NA</em> taxonomic ranks</h3>
<p>Phyloseq has an odd way of dealing with taxonomic ranks that have no valuein other words, <em>NA</em> in the tax table. The first thing we are going to do before moving forward is to change all of the <em>NA</em> to have a value of the next highest classified rank. For example, <code>ASV5</code> is not classified at the Genus level but is at Family level (Saprospiraceae). So, we change the Genus name to <em>Family_Saprospiraceae</em>. The code for this is hidden here but comes from these two posts on the phyloseq GitHub, both by <a href="https://github.com/MSMortensen">MSMortensen</a>: issue <a href="https://github.com/joey711/phyloseq/issues/850#issuecomment-394771087">#850</a> and issue <a href="https://github.com/joey711/phyloseq/issues/990#issuecomment-424618425">#990</a>. If you want to access the code chunk, check out the source code linked at the bottom of this page.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;Kingdom&quot; &quot;Phylum&quot;  &quot;Class&quot;   &quot;Order&quot;   &quot;Family&quot;  &quot;Genus&quot;  
[7] &quot;ASV_SEQ&quot; &quot;ASV_ID&quot; </code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="choose-a-rank">1. Choose a rank</h3>
<p>To make it easier to change the code later on, we can assign the taxonomic rank we are interested into a variable, in the case <code>TRANK</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>TRANK <span class="ot">&lt;-</span> <span class="st">&quot;Class&quot;</span></span></code></pre></div>
</div>
<h3 id="generate-the-asv-reads-table">2. Generate the ASV &amp; reads table</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>tax_asv <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="fu">tax_table</span>(ps_water)[, TRANK],</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>                 <span class="at">exclude =</span> <span class="cn">NULL</span>, <span class="at">dnn =</span> <span class="st">&quot;Taxa&quot;</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>tax_asv <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(tax_asv, <span class="at">make.names =</span> <span class="cn">TRUE</span>, <span class="at">stringsAsFactors=</span><span class="cn">FALSE</span>)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>tax_reads <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">tax_table</span>(ps_water)[, TRANK])</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>tax_reads <span class="ot">&lt;-</span> <span class="fu">apply</span>(<span class="fu">otu_table</span>(ps_water), <span class="at">MARGIN =</span> <span class="dv">1</span>, <span class="cf">function</span>(x)</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">tapply</span>(x, <span class="at">INDEX =</span> tax_reads, <span class="at">FUN =</span> sum, <span class="at">na.rm =</span> <span class="cn">TRUE</span>, <span class="at">simplify =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>tax_reads <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(tax_reads, <span class="at">make.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>tax_reads <span class="ot">&lt;-</span> <span class="fu">cbind</span>(tax_reads, <span class="at">reads =</span> <span class="fu">rowSums</span>(tax_reads))</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>tax_reads <span class="ot">&lt;-</span> tax_reads[<span class="dv">9</span>]</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>tax_reads <span class="ot">&lt;-</span> <span class="fu">setDT</span>(tax_reads, <span class="at">keep.rownames =</span> <span class="cn">TRUE</span>)[]</span></code></pre></div>
</div>
<h3 id="merge-the-two-tables">3. Merge the two tables</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>taxa_read_asv_tab <span class="ot">&lt;-</span> <span class="fu">merge</span>(tax_reads, tax_asv, <span class="at">by.x =</span> <span class="st">&quot;rn&quot;</span>, <span class="at">by.y =</span> <span class="st">&quot;Taxa&quot;</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>top_reads <span class="ot">&lt;-</span> <span class="fu">top_n</span>(taxa_read_asv_tab, <span class="at">n =</span> <span class="dv">8</span>, <span class="at">wt =</span> reads)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>top_asvs <span class="ot">&lt;-</span> <span class="fu">top_n</span>(taxa_read_asv_tab, <span class="at">n =</span> <span class="dv">8</span>, <span class="at">wt =</span> Freq)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(taxa_read_asv_tab) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Taxa&quot;</span>, <span class="st">&quot;total reads&quot;</span>, <span class="st">&quot;total ASVs&quot;</span>)</span></code></pre></div>
</div>
<h3 id="diversity-table-by-class">4. Diversity table by Class</h3>
<div class="layout-chunk" data-layout="l-body">
<div id="9sgooxcmkzw565dtk7or" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="9sgooxcmkzw565dtk7or">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">Total reads &amp; ASVs by taxa<\/caption>","data":[["Acidimicrobiia","Actinobacteria","Alphaproteobacteria","Anaerolineae","Bacilli","Bacteroidia","Bdellovibrionia","Campylobacteria","Cloacimonadia","Clostridia","Cyanobacteriia","Dadabacteriia","Deferribacteres","Dehalococcoidia","Desulfobacteria","Desulfobulbia","Desulfovibrionia","Desulfuromonadia","Fusobacteriia","Gammaproteobacteria","Holophagae","Ignavibacteria","Kapabacteria","Kingdom_Bacteria","Kiritimatiellae","Lentisphaeria","Myxococcia","Nanoarchaeia","Negativicutes","Nitrososphaeria","OM190","Oligoflexia","Omnitrophia","Phycisphaerae","Phylum_Desulfobacterota","Phylum_Latescibacterota","Phylum_Margulisbacteria","Phylum_Marinimicrobia_(SAR406_clade)","Phylum_NB1-j","Phylum_Patescibacteria","Phylum_Proteobacteria","Phylum_SAR324_clade(Marine_group_B)","Phylum_Spirochaetota","Phylum_Verrucomicrobiota","Phylum_WPS-2","Pla3_lineage","Planctomycetes","Polyangia","Rhodothermia","Spirochaetia","Subgroup_22","Synergistia","Syntrophobacteria","Thermoanaerobaculia","Thermoplasmata","Thermotogae","Vampirivibrionia","Verrucomicrobiae","vadinHA49"],[4745,37,33227,7,422,31857,769,2423,241,302,66365,142,64,38,381,94,506,397,88,19928,8,37,40,521,376,80,211,107,3,23,859,76,4,836,2,38,14,3212,5,3,38,1360,15,2,46,51,1728,54,1671,192,2,1,2,4,5876,9,30,982,6],[13,3,182,2,10,290,46,8,6,30,71,1,5,2,9,8,20,30,7,203,2,2,4,52,7,12,4,9,1,3,16,8,1,26,1,2,4,20,1,1,7,7,1,1,3,2,30,4,6,14,1,1,1,1,34,1,2,32,2]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Taxa<\/th>\n      <th>total reads<\/th>\n      <th>total ASVs<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":0},{"className":"dt-right","targets":[1,2]}],"dom":"Blfrtip","buttons":["csv","copy"],"scrollX":true,"scrollCollapse":true,"scrollY":true,"pageLength":5,"scroller":true,"lengthMenu":[5,10,25,60],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<p>Here we can see that:</p>
<ul>
<li>Cyanobacteriia, Alphaproteobacteria, Bacteroidia, Gammaproteobacteria, Thermoplasmata, Acidimicrobiia, Phylum_Marinimicrobia_(SAR406_clade), Campylobacteria contain the <strong>most reads</strong> and</li>
<li>Bacteroidia, Gammaproteobacteria, Alphaproteobacteria, Cyanobacteriia, Kingdom_Bacteria, Bdellovibrionia, Thermoplasmata, Verrucomicrobiae contain the <strong>most ASVs</strong>.</li>
</ul>
<h2 id="hypoxic-vs.-normoxic">Hypoxic vs.Normoxic</h2>
<p>How does the taxonomic composition of normoxic compare to hypoxic samples? To look at this, we will combine samples by the <code>oxstate</code> variable and then calculate the <em>relative abundance</em> for each taxon for both oxygen states. Note one of the Cayo Roldan samples (WCR0) was collected after the hypoxic event when oxygen levels had returned to normal. Therefore, WCR0 is considered a normoxic sample.</p>
<p>There are a few steps we need to run for this analysis.</p>
<h3 id="choose-a-rank-1">1. Choose a rank</h3>
<p>Again, we can select a rank.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>TRANK <span class="ot">&lt;-</span> <span class="st">&quot;Class&quot;</span></span></code></pre></div>
</div>
<h3 id="calculate-the-averages-merge-samples-by-oxstate">2. Calculate the averages &amp; merge samples by oxstate</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>ps_water_AVG <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(ps_water, <span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x))</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>ps_water_ox <span class="ot">&lt;-</span> <span class="fu">merge_samples</span>(ps_water, <span class="st">&quot;oxstate&quot;</span>)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>SD_BAR_w <span class="ot">&lt;-</span> <span class="fu">merge_samples</span>(<span class="fu">sample_data</span>(ps_water_AVG), <span class="st">&quot;oxstate&quot;</span>)</span></code></pre></div>
</div>
<h3 id="merge-taxa-by-rank">3. Merge taxa by rank</h3>
<p>We will set a variable for the taxonomic rank of choice. If you want to choose a different rank, be sure to change the variable assignment.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>mdata_phy_w <span class="ot">&lt;-</span> <span class="fu">tax_glom</span>(ps_water_ox, <span class="at">taxrank =</span> TRANK, <span class="at">NArm =</span> <span class="cn">FALSE</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>mdata_phyrel_w <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(mdata_phy_w, <span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x))</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>meltd_w <span class="ot">&lt;-</span> <span class="fu">psmelt</span>(mdata_phyrel_w)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>meltd_w[[TRANK]] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(meltd_w[[TRANK]])</span></code></pre></div>
</div>
<h3 id="calculate-the-total-relative-abundance-for-all-taxa">4. Calculate the total relative abundance for all taxa</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>means_w <span class="ot">&lt;-</span> <span class="fu">ddply</span>(meltd_w, <span class="sc">~</span><span class="fu">get</span>(TRANK), <span class="cf">function</span>(x) <span class="fu">c</span>(<span class="at">mean =</span> <span class="fu">mean</span>(x<span class="sc">$</span>Abundance)))</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(means_w) <span class="ot">&lt;-</span> <span class="fu">c</span>(TRANK, <span class="st">&quot;mean&quot;</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>means_w<span class="sc">$</span>mean <span class="ot">&lt;-</span> <span class="fu">round</span>(means_w<span class="sc">$</span>mean, <span class="at">digits =</span> <span class="dv">5</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>taxa_means_w <span class="ot">&lt;-</span> means_w[<span class="fu">order</span>(<span class="sc">-</span>means_w<span class="sc">$</span>mean), ]  <span class="co"># this orders in decending fashion</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>taxa_means_w <span class="ot">&lt;-</span> <span class="fu">format</span>(taxa_means_w, <span class="at">scientific =</span> <span class="cn">FALSE</span>)  <span class="co"># ditch the sci notation</span></span></code></pre></div>
</div>
<h3 id="relative-abundance-by-class">5. Relative abundance by Class</h3>
<div class="layout-chunk" data-layout="l-body">
<div id="t9xg4oo3fbjqjqu1ogjm" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="t9xg4oo3fbjqjqu1ogjm">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">Relative abundance by taxonomic rank. <\/caption>","data":[["Cyanobacteriia","Bacteroidia","Alphaproteobacteria","Gammaproteobacteria","Thermoplasmata","Acidimicrobiia","Campylobacteria","Phylum_Marinimicrobia_(SAR406_clade)","Rhodothermia","Planctomycetes","Phylum_SAR324_clade(Marine_group_B)","OM190","Verrucomicrobiae","Phycisphaerae","Desulfovibrionia","Bdellovibrionia","Kingdom_Bacteria","Desulfobacteria","Desulfuromonadia","Clostridia","Bacilli","Cloacimonadia","Kiritimatiellae","Spirochaetia","Myxococcia","Desulfobulbia","Nanoarchaeia","Dadabacteriia","Fusobacteriia","Deferribacteres","Lentisphaeria","Oligoflexia","Kapabacteria","Phylum_Latescibacterota","Ignavibacteria","Pla3_lineage","Actinobacteria","Vampirivibrionia","Polyangia","Phylum_WPS-2","Phylum_Proteobacteria","Dehalococcoidia","Phylum_Spirochaetota","Nitrososphaeria","Thermotogae","Holophagae","Phylum_Margulisbacteria","Anaerolineae","Phylum_NB1-j","vadinHA49","Omnitrophia","Phylum_Desulfobacterota","Thermoanaerobaculia","Negativicutes","Phylum_Patescibacteria","Phylum_Verrucomicrobiota","Subgroup_22","Synergistia","Syntrophobacteria"],["0.30801","0.19233","0.19169","0.14199","0.02620","0.02150","0.02032","0.01398","0.01286","0.01087","0.00686","0.00605","0.00590","0.00516","0.00425","0.00414","0.00355","0.00320","0.00256","0.00253","0.00215","0.00202","0.00192","0.00161","0.00103","0.00075","0.00068","0.00065","0.00063","0.00054","0.00052","0.00036","0.00034","0.00032","0.00031","0.00031","0.00026","0.00025","0.00022","0.00020","0.00017","0.00016","0.00013","0.00010","0.00008","0.00007","0.00006","0.00005","0.00004","0.00004","0.00002","0.00002","0.00002","0.00001","0.00001","0.00001","0.00001","0.00001","0.00001"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Class<\/th>\n      <th>mean<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":"_all"}],"dom":"Blfrtip","buttons":["csv","copy"],"scrollX":true,"scrollCollapse":true,"scrollY":true,"pageLength":10,"scroller":true,"lengthMenu":[10,25,60],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<p><br/></p>
<h3 id="group-taxa-into-other">6. Group taxa into Other</h3>
<p>Time to group low abundance taxa into an <em>Other</em> category for the bar graph. Since we cannot possibly display all 59 different taxa in one figure, we need to collapse the low abundance groups.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>TAXAN <span class="ot">&lt;-</span> <span class="dv">8</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>top_perc_w <span class="ot">&lt;-</span> <span class="fu">top_n</span>(taxa_means_w, <span class="at">n =</span> TAXAN, <span class="at">wt =</span> mean)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>top_perc_w<span class="sc">$</span>mean <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">as.numeric</span>(top_perc_w<span class="sc">$</span>mean), <span class="at">digits =</span> <span class="dv">5</span>)</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>min_top_perc_w <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">as.numeric</span>(<span class="fu">min</span>(top_perc_w<span class="sc">$</span>mean)), <span class="at">digits =</span> <span class="dv">5</span>)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>top_perc_w_list <span class="ot">&lt;-</span> top_perc_w[,<span class="dv">1</span>]</span></code></pre></div>
</div>
<p>Here we decided to look at the abundance of the top 8, which sets a relative abundance cutoff at 0.01398. Anything lower than 0.01398 will be grouped into <em>Other</em>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>Other_w <span class="ot">&lt;-</span> means_w[means_w<span class="sc">$</span>mean <span class="sc">&lt;</span> min_top_perc_w, ][[TRANK]]</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>meltd_w[meltd_w[[TRANK]] <span class="sc">%in%</span> Other_w, ][[TRANK]] <span class="ot">&lt;-</span> <span class="st">&quot;Other&quot;</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>samp_names_w <span class="ot">&lt;-</span> <span class="fu">aggregate</span>(meltd_w<span class="sc">$</span>Abundance, <span class="at">by =</span> <span class="fu">list</span>(meltd_w<span class="sc">$</span>Sample), <span class="at">FUN =</span> sum)[, <span class="dv">1</span>]</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>.e_w <span class="ot">&lt;-</span> <span class="fu">environment</span>()</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>meltd_w[, TRANK] <span class="ot">&lt;-</span> <span class="fu">factor</span>(meltd_w[, TRANK], <span class="fu">sort</span>(<span class="fu">unique</span>(meltd_w[, TRANK])))</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>meltd_w1 <span class="ot">&lt;-</span> meltd_w[<span class="fu">order</span>(meltd_w[, TRANK]), ]</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>target <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Cyanobacteriia&quot;</span>, <span class="st">&quot;Alphaproteobacteria&quot;</span>,</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;Bacteroidia&quot;</span>, <span class="st">&quot;Gammaproteobacteria&quot;</span>,</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;Thermoplasmata&quot;</span>, <span class="st">&quot;Phylum_Marinimicrobia_(SAR406_clade)&quot;</span>,</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;Acidimicrobiia&quot;</span>, <span class="st">&quot;Campylobacteria&quot;</span>, <span class="st">&quot;Other&quot;</span>)</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>meltd_w1[[TRANK]] <span class="ot">&lt;-</span> <span class="fu">reorder.factor</span>(meltd_w1[[TRANK]], <span class="at">new.order =</span> target)</span></code></pre></div>
</div>
<h3 id="construct-a-bar-graph">7. Construct a bar graph</h3>
<p>Finally.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="figures/16s-water/bar-norm-v-hypox.png" width="624" /></p>
</div>
<p>We can also look at this in tabular format. To accomplish the task, we use the <code>aggregate</code> command to group abundance values by oxygen state <em>and</em> taxonomic group.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>hyp_v_norm <span class="ot">&lt;-</span> <span class="fu">aggregate</span>(meltd_w1<span class="sc">$</span>Abundance,</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>                 <span class="at">by =</span> <span class="fu">list</span>(<span class="at">Rank =</span> meltd_w1[[TRANK]],</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">Sample =</span> meltd_w1<span class="sc">$</span>Sample),</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">FUN =</span> sum)</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>hyp_v_norm<span class="sc">$</span>x <span class="ot">&lt;-</span> <span class="fu">round</span>(hyp_v_norm<span class="sc">$</span>x, <span class="at">digits =</span> <span class="dv">4</span>)</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>hyp_v_norm <span class="ot">&lt;-</span> <span class="fu">spread</span>(hyp_v_norm, Sample, x)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="wewuidn73q2ceico4vyw" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="wewuidn73q2ceico4vyw">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Relative abundance (reads) of normoxic\n                                     vs hypoxic samples by dominant taxa.<\/em>\n<\/caption>","data":[["Cyanobacteriia","Alphaproteobacteria","Bacteroidia","Gammaproteobacteria","Thermoplasmata","Phylum_Marinimicrobia_(SAR406_clade)","Acidimicrobiia","Campylobacteria","Other"],[0.1328,0.2142,0.2391,0.235,0.0075,0.0028,0.0075,0.0406,0.1205],[0.4832,0.1691,0.1456,0.049,0.0449,0.0252,0.0356,0,0.0475]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Rank<\/th>\n      <th>hypoxic<\/th>\n      <th>normoxic<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Brti","buttons":["copy","csv","excel"],"pageLength":9,"columnDefs":[{"className":"dt-right","targets":[1,2]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[9,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<p><br/></p>
<h2 id="during-vs.-after-the-event">During vs.After the Event</h2>
<p>What did taxonomic composition look like during and after the event at <strong>each site</strong>? To look at this, we will combine samples by the <code>period</code> variable and then calculate the <em>relative abundance</em> for each taxon. The code for this analysis is basically the same as the code above where we compared hypoxic to normoxic samples. Therefore, all of the code is hidden. See the link at the bottom of the page for the page source code if you are interested in the code we used here. We go through the same steps:</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<ol type="1">
<li>Choose a rank.</li>
<li>Merge samples by period.</li>
<li>Merge taxa by rank.</li>
<li>Calculate the total relative abundance for all taxa.</li>
<li>Relative abundance by Class. <em>Values are slightly different because averaging across a different grouping of samples</em>.</li>
<li>Group taxa into Other.</li>
<li>Construct a bar graph.</li>
</ol>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div id="zswbg7x7f8eqj526xuox" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="zswbg7x7f8eqj526xuox">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">Relative abundance by taxonomic rank. <\/caption>","data":[["Cyanobacteriia","Alphaproteobacteria","Bacteroidia","Gammaproteobacteria","Thermoplasmata","Acidimicrobiia","Phylum_Marinimicrobia_(SAR406_clade)","Campylobacteria","Planctomycetes","Phylum_SAR324_clade(Marine_group_B)","Rhodothermia","Verrucomicrobiae","OM190","Bdellovibrionia","Phycisphaerae","Kingdom_Bacteria","Bacilli","Desulfovibrionia","Desulfuromonadia","Kiritimatiellae","Desulfobacteria","Myxococcia","Clostridia","Cloacimonadia","Dadabacteriia","Spirochaetia","Nanoarchaeia","Desulfobulbia","Fusobacteriia","Polyangia","Oligoflexia","Lentisphaeria","Phylum_WPS-2","Pla3_lineage","Deferribacteres","Dehalococcoidia","Actinobacteria","Nitrososphaeria","Kapabacteria","Ignavibacteria","Phylum_Latescibacterota","Phylum_Proteobacteria","Vampirivibrionia","Phylum_Margulisbacteria","Phylum_Spirochaetota","Anaerolineae","Thermotogae","vadinHA49","Holophagae","Phylum_Patescibacteria","Thermoanaerobaculia","Phylum_NB1-j","Phylum_Verrucomicrobiota","Syntrophobacteria","Negativicutes","Omnitrophia","Phylum_Desulfobacterota","Subgroup_22","Synergistia"],["0.39723","0.18158","0.16630","0.09498","0.03706","0.02885","0.01791","0.01016","0.00923","0.00918","0.00721","0.00510","0.00428","0.00412","0.00401","0.00251","0.00226","0.00212","0.00200","0.00167","0.00160","0.00131","0.00127","0.00101","0.00087","0.00081","0.00069","0.00041","0.00041","0.00041","0.00039","0.00034","0.00032","0.00031","0.00027","0.00024","0.00018","0.00018","0.00017","0.00016","0.00016","0.00015","0.00013","0.00007","0.00006","0.00004","0.00004","0.00004","0.00003","0.00003","0.00003","0.00002","0.00002","0.00002","0.00001","0.00001","0.00001","0.00001","0.00000"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Class<\/th>\n      <th>mean<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":"_all"}],"dom":"Blfrtip","buttons":["csv","copy"],"scrollX":true,"scrollCollapse":true,"scrollY":true,"pageLength":9,"scroller":true,"lengthMenu":[9,25,60],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<p>Again, Time to group low abundance taxa into an <em>Other</em> category for the bar graph. Since we cannot possibly display all 59 different taxa in one figure, we need to collapse the low abundance groups.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body-outset">

</div>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="figures/16s-water/bar-before-v-after.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div id="kqtr8k9xeqyk0sb1s4h4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="kqtr8k9xeqyk0sb1s4h4">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Relative abundance (reads) of normoxic\n                                     vs hypoxic samples by dominant taxa.<\/em>\n<\/caption>","data":[["Cyanobacteriia","Alphaproteobacteria","Bacteroidia","Gammaproteobacteria","Thermoplasmata","Phylum_Marinimicrobia_(SAR406_clade)","Acidimicrobiia","Campylobacteria","Other"],[0.4056,0.2126,0.1714,0.0483,0.0704,0.0171,0.0424,0,0.0323],[0.47,0.1705,0.1561,0.0503,0.0432,0.0292,0.0354,0,0.0453],[0.5805,0.129,0.0987,0.0463,0.0272,0.0226,0.0301,0,0.0656],[0.1328,0.2142,0.2391,0.235,0.0075,0.0028,0.0075,0.0406,0.1205]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Crawl Caye (after)<\/th>\n      <th>Crawl Caye (during)<\/th>\n      <th>Cayo Roldan (after)<\/th>\n      <th>Cayo Roldan (during)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Brti","buttons":["copy","csv","excel"],"pageLength":9,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[9,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>And thats it for this part of the 16S rRNA part of the workflow. Next, we drill a bit deeper into the data set and look at differentially abundant ASVs between normoxic vs.hypoxic samples.</p>
<p><br></p>
<div class="post-nav">
<div class="post-nav-item">
<div class="meta-nav">
Previous
</div>
<p><a href="16s-data-prep.html" rel="next">N<sup><u>o</u></sup> 2. Data Preparation</a></p>
</div>
</div>
<div class="post-nav">
<div class="post-nav-item">
<div class="meta-nav">
Next
</div>
<p><a href="16s-da-asvs.html" rel="prev">N<sup><u>o</u></sup> 4. Differentially Abundant ASVs</a></p>
</div>
</div>
<h2 class="appendix" id="source-code">Source Code</h2>
<p>The source code for this page can be accessed on GitHub by <a href="https://github.com/hypocolypse/web/blob/master/16s-water.Rmd">clicking this link</a>.</p>
<h2 class="appendix" id="data-availability-1">Data Availability</h2>
<p>Output files from this workflow available on figshare at <a href="https://doi.org/10.25573/data.12794258">doi:10.25573/data.12794258</a>. The <em>16s-water.rdata</em> file contains all variables and phyloseq objects from this pipeline and is needed for the next workflow (as are the .rds files).</p>
<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-lahti2017microbiome" class="csl-entry" role="doc-biblioentry">
Lahti, Leo, Shetty Sudarshan, and others. 2017. <span>Tools for Microbiome Analysis in r. Version 1.9.97.</span> <a href="https://github.com/microbiome/microbiome">https://github.com/microbiome/microbiome</a>.
</div>
</div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/hypocolypse/web/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/hypocolypse/web/">https://github.com/hypocolypse/web/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p> Copyright 2021 metacrobe &amp; The HYPOCOLYPSE Project.</p>
<p>Site constructed using <a href="https://rstudio.github.io/distill/">Distill for R Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->


</body>

</html>
