<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

  <!--radix_placeholder_meta_tags-->
  <title>: No 3. Diversity</title>
  
  <meta property="description" itemprop="description" content="In this section of the workflow we look at the diversity of the water samples collected from Cayo Roldan (hypoxic) and Crawl Caye (normoxic)."/>
  
  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>
  <link rel="icon" type="image/vnd.microsoft.icon" href="assets/favicon.ico"/>
  
  <meta name="article:author" content="Jarrod J Scott"/>
  
  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content=": No 3. Diversity"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="In this section of the workflow we look at the diversity of the water samples collected from Cayo Roldan (hypoxic) and Crawl Caye (normoxic)."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content=""/>
  
  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content=": No 3. Diversity"/>
  <meta property="twitter:description" content="In this section of the workflow we look at the diversity of the water samples collected from Cayo Roldan (hypoxic) and Crawl Caye (normoxic)."/>
  
  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Tools for microbiome analysis in r. Version 1.9.97;citation_publication_date=2017;citation_author=Leo Lahti;citation_author=Shetty Sudarshan;citation_author=Tools for microbiome analysis in r. Version 1.9.97"/>
  <!--radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["No 3. Diversity"]},{"type":"character","attributes":{},"value":["In this section of the workflow we look at the diversity of the water samples collected from Cayo Roldan (hypoxic) and Crawl Caye (normoxic).\n"]},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name"]}},"value":[{"type":"character","attributes":{},"value":["Jarrod J Scott"]}]}]},{"type":"character","attributes":{},"value":["assets/cite.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  
  <script type="application/javascript">
  
    window.headroom_prevent_pin = false;
  
    window.document.addEventListener("DOMContentLoaded", function (event) {
  
      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');
  
      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });
  
      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });
  
      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>
  
  <style type="text/css">
  
  /* Theme (user-documented overrideables for nav appearance) */
  
  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #455a64;
    font-size: 15px;
    font-weight: 300;
  }
  
  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }
  
  .distill-site-nav a:hover {
    color: white;
  }
  
  @media print {
    .distill-site-nav {
      display: none;
    }
  }
  
  .distill-site-header {
  
  }
  
  .distill-site-footer {
  
  }
  
  
  /* Site Header */
  
  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }
  
  
  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }
  
  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }
  
  .distill-site-header .title {
    font-size: 18px;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }
  
  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }
  
  
  
  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }
  
  
  .distill-site-header .nav-toggle {
    display: none;
  }
  
  .nav-dropdown {
    display: inline-block;
    position: relative;
  }
  
  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }
  
  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }
  
  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .nav-dropdown-active {
    display: block;
  }
  
  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }
  
  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }
  
  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }
  
  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative;}
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }
  
  /* Site Footer */
  
  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }
  
  /* Headroom */
  
  d-title {
    padding-top: 6rem;
  }
  
  @media print {
    d-title {
      padding-top: 4rem;
    }
  }
  
  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }
  
  .headroom--transition {
    transition: all .4s ease-in-out;
  }
  
  .headroom--unpinned {
    top: -100px;
  }
  
  .headroom--pinned {
    top: 0;
  }
  
  </style>
  
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->
  
  <style type="text/css">
  
  body {
    background-color: white;
  }
  
  .pandoc-table {
    width: 100%;
  }
  
  .pandoc-table>caption {
    margin-bottom: 10px;
  }
  
  .pandoc-table th:not([align]) {
    text-align: left;
  }
  
  .pagedtable-footer {
    font-size: 15px;
  }
  
  .html-widget {
    margin-bottom: 2.0em;
  }
  
  .l-screen-inset {
    padding-right: 16px;
  }
  
  .l-screen .caption {
    margin-left: 10px;
  }
  
  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .shaded-content {
    background: white;
  }
  
  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }
  
  .hidden {
    display: none !important;
  }
  
  d-article {
    padding-bottom: 30px;
  }
  
  d-appendix {
    padding-top: 30px;
  }
  
  d-article>p>img {
    width: 100%;
  }
  
  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }
  
  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  /* CSS for table of contents */
  
  .d-toc {
    color: rgba(0,0,0,0.8);
    font-size: 0.8em;
    line-height: 1em;
  }
  
  .d-toc-header {
    font-size: 0.6rem;
    font-weight: 400;
    color: rgba(0, 0, 0, 0.5);
    text-transform: uppercase;
    margin-top: 0;
    margin-bottom: 1.3em;
  }
  
  .d-toc a {
    border-bottom: none;
  }
  
  .d-toc ul {
    padding-left: 0;
  }
  
  .d-toc li>ul {
    padding-top: 0.8em;
    padding-left: 16px;
    margin-bottom: 0.6em;
  }
  
  .d-toc ul,
  .d-toc li {
    list-style-type: none;
  }
  
  .d-toc li {
    margin-bottom: 0.9em;
  }
  
  .d-toc-separator {
    margin-top: 20px;
    margin-bottom: 2em;
  }
  
  .d-article-with-toc {
    border-top: none;
    padding-top: 0;
  }
  
  
  
  /* Tweak code blocks (note that this CSS is repeated above in an injection
     into the d-code shadow dom) */
  
  d-code {
    overflow-x: auto !important;
  }
  
  pre.d-code code.d-code {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }
  
  pre.text-output {
  
    font-size: 12px;
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  @media(min-width: 768px) {
  
  d-code {
    overflow-x: visible !important;
  }
  
  pre.d-code code.d-code  {
      padding-left: 18px;
      font-size: 14px;
  }
  pre.text-output {
    font-size: 14px;
  }
  }
  
  /* Figure */
  
  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }
  
  .figure img {
    width: 100%;
  }
  
  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }
  
  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }
  
  
  
  /* Tweak 1000px media break to show more text */
  
  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }
  
    .grid {
      grid-column-gap: 16px;
    }
  
    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }
  
  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }
  
    .grid {
      grid-column-gap: 32px;
    }
  }
  
  
  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */
  
  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }
  
  
  /* Social footer */
  
  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }
  
  .disqus-comments {
    margin-right: 30px;
  }
  
  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
  
  #disqus_thread {
    margin-top: 30px;
  }
  
  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }
  
  .article-sharing a:hover {
    border-bottom: none;
  }
  
  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .subscribe p {
    margin-bottom: 0.5em;
  }
  
  
  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }
  
  
  /* Improve display for browsers without grid (IE/Edge <= 15) */
  
  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }
  
  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }
  
  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }
  
  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }
  
  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }
  
  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
  
  
  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }
  
  .downlevel .footnotes ol {
    padding-left: 13px;
  }
  
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }
  
  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }
  
  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }
  
  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  </style>
  
  <script type="application/javascript">
  
  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }
  
  // show body when load is complete
  function on_load_complete() {
  
    // set body to visible
    document.body.style.visibility = 'visible';
  
    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }
  
    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }
  
  function init_distill() {
  
    init_common();
  
    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);
  
    // create d-title
    $('.d-title').changeElementType('d-title');
  
    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);
  
    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();
  
    // move posts container into article
    $('.posts-container').appendTo($('d-article'));
  
    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');
  
    // create d-bibliography
    var bibliography = $('<d-bibliography></d-bibliography>');
    $('#distill-bibliography').wrap(bibliography);
  
    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;
  
    // replace citations with <d-cite>
    $('.citation').each(function(i, val) {
      appendix = true;
      var cites = $(this).attr('data-cites').split(" ");
      var dt_cite = $('<d-cite></d-cite>');
      dt_cite.attr('key', cites.join());
      $(this).replaceWith(dt_cite);
    });
    // remove refs
    $('#refs').remove();
  
    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();
  
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-toc a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });
  
    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');
  
    // replace code blocks with d-code
    $('pre>code').each(function(i, val) {
      var code = $(this);
      var pre = code.parent();
      var clz = "";
      var language = pre.attr('class');
      if (language) {
        // map unknown languages to "clike" (without this they just dissapear)
        if ($.inArray(language, ["bash", "clike", "css", "go", "html",
                                 "javascript", "js", "julia", "lua", "markdown",
                                 "markup", "mathml", "python", "svg", "xml"]) == -1)
          language = "clike";
        language = ' language="' + language + '"';
        var dt_code = $('<d-code block' + language + clz + '></d-code>');
        dt_code.text(code.text());
        pre.replaceWith(dt_code);
      } else {
        code.addClass('text-output').unwrap().changeElementType('pre');
      }
    });
  
    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {
  
      // capture layout
      var layout = $(this).attr('data-layout');
  
      // apply layout to markdown level block elements
      var elements = $(this).children().not('d-code, pre.text-output, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });
  
  
      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });
  
    // load distill framework
    load_distill_framework();
  
    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {
  
      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;
  
      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');
  
      // table of contents
      if (have_authors) // adjust border if we are in authors
        $('.d-toc').parent().addClass('d-article-with-toc');
  
      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');
  
      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }
  
      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');
  
      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");
  
      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }
  
       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }
  
      // inject pre code styles (can't do this with a global stylesheet b/c a shadow root is used)
      $('d-code').each(function(i, val) {
        var style = document.createElement('style');
        style.innerHTML = 'pre code { padding-left: 10px; font-size: 12px; border-left: 2px solid rgba(0,0,0,0.1); } ' +
                          '@media(min-width: 768px) { pre code { padding-left: 18px; font-size: 14px; } }';
        if (this.shadowRoot)
          this.shadowRoot.appendChild(style);
      });
  
      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();
  
      // clear polling timer
      clearInterval(tid);
  
      // show body now that everything is ready
      on_load_complete();
    }
  
    var tid = setInterval(distill_post_process, 50);
    distill_post_process();
  
  }
  
  function init_downlevel() {
  
    init_common();
  
     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));
  
    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
  
    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();
  
    // remove toc
    $('.d-toc-header').remove();
    $('.d-toc').remove();
    $('.d-toc-separator').remove();
  
    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });
  
  
    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);
  
    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);
  
    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();
  
    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();
  
    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));
  
    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });
  
    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));
  
    $('body').addClass('downlevel');
  
    on_load_complete();
  }
  
  
  function init_common() {
  
    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};
  
        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });
  
        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);
  
    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});
  
    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });
  
      }
    });
  
    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });
  
    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }
  
    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');
  
    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");
  
    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();
  
    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }
  
  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });
  
  </script>
  
  <!--/radix_placeholder_distill-->
  <script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
  <script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
  <link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
  <script src="site_libs/datatables-binding-0.13/datatables.js"></script>
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
  <script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
  <script src="site_libs/jszip-1.10.20/jszip.min.js"></script>
  <link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
  <script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
  <link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
  <script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Roboto:300|Nunito+Sans:300|Inconsolata|Source+Sans+Pro');
  /* controls general properties of site*/
  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    background-color: #FAFAFA;
  }
  
  d-title h1{
  font-size: 2.5em;
  font-family: Roboto;
  font-weight: 500;
  }
  
  d-article h1, h2 {
    font-family: Roboto;
  }
  
  d-article h3 {
    margin-top: 1em;
  }
  
  blockquote p {
  	font-size: 1.3em;
  	font-family: 'Lato';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }
  
  d-code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    border-left: 5px solid #006ddb;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
  }
  
  pre.text-output {
    margin-top: 0px;
     color: #333333;
     background-color: #FAFAFA;
     font-size: 0.8em
   }
  
  pre {
    border: 1px solid #ccc;
    padding: 0px 10px 10px 18px;
    margin: 1em 0px;
    border-radius: 5px;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    text-transform: none;
  }
  
  d-article li {
    margin: 0px 0px 4px;
  }
  
  /*************************************************
  *  code for side by side krona plot
  **************************************************/
  
  .column {
    float: left;
    width: 48%;
    padding: 5px;
  }
  
  /* Clear floats after image containers */
  .row::after {
    content: "";
    clear: both;
    display: table;
  }
  
  div.krona {
    width: 100%;
  	height: 90%;
    background-color: white;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    margin-bottom: 25px;
  	margin-top: 5px;
  }
  
  div.container-krona {
    text-align: center;
    padding: 1px 10px;
  	color: #414141;
  	font-size: 1.48em;
  	line-height: 0.8em;
  	font-family: 'Inconsolata';
  }
  
  div.container-krona p {
    margin: 20px;
  }
  
  div.krona img {
    border:1px solid white;
    margin: 25px;
  }
  
  img {
    margin: 25px;
  }
  
  /** LIGHTBOX MARKUP **/
  .thumbnail {
    max-width: 100%;
    position: relative;
  }
  
  .lightbox {
  	/** Default lightbox to hidden */
  	display: none;
  
  	/** Position and style */
  	position: fixed;
  	z-index: 999;
  	width: 100%;
  	height: 100%;
  	text-align: center;
  	top: 0;
  	left: 0;
  	background: rgba(0,0,0,0.8);
  }
  
  .lightbox img {
  	/** Pad the lightbox image */
  	position: fixed;
  	top: 0;
  	left: 0;
  	right: 0;
  	bottom: 0;
  	max-width: 0%;
  	max-height: 0%;
  	margin: auto;
  	box-sizing: border-box;
  }
  
  .lightbox:target {
  	/** Remove default browser outline */
  	outline: none;
  
  	/** Unhide lightbox **/
  	display: block;
  }
  
  .lightbox:target img {
  	max-height: 100%;
  	max-width: 100%;
  }
  
  .distill-site-header .logo img {
    margin-top: 10px;
    max-height: 30px;
    margin-bottom: 10px;
  }
  
  .distill-site-nav {
      color: rgba(255, 255, 255, 0.8);
      background-color: #143D59;
      font-size: 16px;
      font-weight: 300;
    }
  }
  
  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"No 3. Diversity","description":"In this section of the workflow we look at the diversity of the water samples collected from Cayo Roldan (hypoxic) and Crawl Caye (normoxic).","authors":[{"author":"Jarrod J Scott","authorURL":"#","affiliation":"&nbsp;","affiliationURL":"#"}]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a class="logo" href="index.html">
<img src="assets/icon.png"/>
</a>
<a href="index.html" class="title"></a>
</div>
<div class="nav-right">
<a href="index.html">
<i class="fa fa-home"></i>
</a>
<a href="intro.html">Introduction</a>
<a href="field.html">A. Field Analyses</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
B. 16S rRNA
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">16S rRNA Processing</span>
<a href="16s-dada2.html">1. DADA2</a>
<a href="16s-data-prep.html">2. Data Prep</a>
<hr/>
<span class="nav-dropdown-header">16S rRNA Analyses</span>
<a href="16s-water.html">3. Diversity</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
C. Metagenomic
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Setup</span>
<a href="mg-setup.html">1. Working Envrionment</a>
<a href="mg-databases.html">2. Annotation Databases</a>
<hr/>
<span class="nav-dropdown-header">Read Processing</span>
<a href="mg-workflow-1.html">3. Assembly &amp; Annotations</a>
<a href="mg-workflow-2.html">4. Assembley &amp; Annotation Summary</a>
<hr/>
<span class="nav-dropdown-header">MAG Analysis</span>
<a href="mg-binning.html">5. Binning</a>
<a href="mag02-phylogenomics.html">6. MAG02 Phylogenomics</a>
</div>
</div>
<a href="synthesis.html">D. Synthesis</a>
<a href="data-availability.html">Data</a>
<a href="https://github.com/hypocolypse/web/">
<i class="fa fa-github fa-lg"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>No 3. Diversity</h1>
<p>In this section of the workflow we look at the diversity of the water samples collected from Cayo Roldan (hypoxic) and Crawl Caye (normoxic).</p>
</div>

<div class="d-byline">
  Jarrod J Scott  
  

</div>

<div class="d-article">
<h3 class="d-toc-header">Table of Contents</h3>
<nav class="d-toc" id="TOC">
<ul>
<li><a href="#summary-of-water-samples">Summary of Water Samples</a></li>
<li><a href="#diversity">Diversity</a><ul>
<li><a href="#taxonomic-diversity">Taxonomic Diversity</a></li>
<li><a href="#hypoxic-vs.normoxic">Hypoxic vs. Normoxic</a></li>
<li><a href="#during-vs.after-the-event">During vs. After the event</a></li>
<li><a href="#source-code">Source Code</a></li>
</ul></li>
</ul>
</nav>
<hr class="d-toc-separator"/>
<h1 id="summary-of-water-samples">Summary of Water Samples</h1>
<p>Now that we have a phyloseq object containing the water samples only, we can summarize the data in the water phyloseq object. Again, we use the <code>summarize_phyloseq</code> from the <a href="https://github.com/microbiome/microbiome/">microbiome R package</a><span class="citation" data-cites="lahti2017microbiome">(Lahti, Sudarshan, and others <a href="#ref-lahti2017microbiome">2017</a>)</span> as we did before.</p>
<p>First, load the R objects that contain the sample data, sequence table, and taxonomy table (saved at the end of the previous section) and merge them into a single phyloseq object. Doing it this way keeps the memory footprint low.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
remove(list = ls())
sample_d &lt;- readRDS(&quot;rdata/16s-water/ps_water-sample.rds&quot;)
seqtab &lt;- readRDS(&quot;rdata/16s-water/ps_water-seqtab.rds&quot;)
taxtab &lt;- readRDS(&quot;rdata/16s-water/ps_water-taxtab.rds&quot;)
ps_water &lt;- merge_phyloseq(sample_d, seqtab, taxtab)
ps_water</code></pre>
<pre><code>
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1272 taxa and 8 samples ]
sample_data() Sample Data:       [ 8 samples by 3 sample variables ]
tax_table()   Taxonomy Table:    [ 1272 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<p>Next, we can save copies of the sequence and taxonomy tables just in case we want a quick look at these data.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
write.table(tax_table(ps_water),
            &quot;tables/16s-water/water_tax_table.txt&quot;, sep=&quot;\t&quot;,
            quote = FALSE, col.names=NA)
write.table(t(otu_table(ps_water)),
            &quot;tables/16s-water/water_seq_table.txt&quot;, sep=&quot;\t&quot;,
            quote = FALSE, col.names=NA)</code></pre>
</div>
<p>And generate a fasta file for all ASVs in the water samples.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
# Create fasta file from tax_table
table2format &lt;- tax_table(ps_water)
#retain only the column with the sequences
table2format_trim &lt;- table2format[, 7]
table2format_trim_df &lt;- data.frame(row.names(table2format_trim),
                                   table2format_trim)
colnames(table2format_trim_df) &lt;- c(&quot;ASV_ID&quot;, &quot;ASV_SEQ&quot;)
#format fasta
table2format_trim_df$ASV_ID &lt;- sub(&quot;ASV&quot;, &quot;&gt;ASV&quot;, table2format_trim_df$ASV_ID)

write.table(table2format_trim_df, &quot;tables/16s-water/water_asv.fasta&quot;,
            sep = &quot;\r&quot;, col.names = FALSE, row.names = FALSE,
            quote = FALSE, fileEncoding = &quot;UTF-8&quot;)</code></pre>
</div>
<p>Let’s go ahead and a sample variable called <code>oxstate</code> to the sample data frame of the water phyloseq object so we can compare the taxonomy of normoxic (Coral Caye) vs. hypoxic (Cayo Roldan) samples. Note one of the Cayo Roldan samples (WCR0) was collected after the hypoxic event when oxygen levels had returned to normal. Therefore WCR0 is considered a normoxic sample.</p>
<p>One last thing to do is to compare diversity across the two sites <em>before</em> and <em>after</em> the hypoxic event. For this we add a sample variable called <code>period</code> denoting whether the sample was collected during or after the event.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
sample_data(ps_water)$oxstate &lt;-
  c(&quot;normoxic&quot;, &quot;normoxic&quot;, &quot;normoxic&quot;, &quot;normoxic&quot;,
    &quot;normoxic&quot;, &quot;hypoxic&quot;, &quot;hypoxic&quot;, &quot;hypoxic&quot;)
sample_data(ps_water)$period &lt;-
  c(&quot;cc_after&quot;, &quot;cc_during&quot;, &quot;cc_during&quot;, &quot;cc_during&quot;,
    &quot;cr_after&quot;, &quot;cr_during&quot;, &quot;cr_during&quot;, &quot;cr_during&quot;)</code></pre>
</div>
<blockquote>
<p>The <code>ps_water</code> dataset contains 1272 ASVs, 180557 total reads, 8 samples, and 5 sample variables.</p>
</blockquote>
<p>And here is a summary of just water data samples.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<table>
<thead>
<tr class="header">
<th>Metric</th>
<th>Results</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>12804</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>33311</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>180557</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>22570</td>
</tr>
<tr class="odd">
<td>Median number of reads</td>
<td>24120</td>
</tr>
<tr class="even">
<td>Min. number of ASVs</td>
<td>237</td>
</tr>
<tr class="odd">
<td>Max. number of ASVs</td>
<td>552</td>
</tr>
<tr class="even">
<td>Total number of ASVs</td>
<td>1272</td>
</tr>
<tr class="odd">
<td>Average number of ASVs</td>
<td>317</td>
</tr>
<tr class="even">
<td>Median number of ASVs</td>
<td>295.5</td>
</tr>
<tr class="odd">
<td>Sparsity</td>
<td>0.751</td>
</tr>
<tr class="even">
<td>Any ASVs sum to 1 or less?</td>
<td>TRUE</td>
</tr>
<tr class="odd">
<td>Number of singleton ASVs</td>
<td>68</td>
</tr>
<tr class="even">
<td>Percent of ASVs that are singletons</td>
<td>5.346</td>
</tr>
<tr class="odd">
<td>Number of sample variables are:</td>
<td>5 (SamName, TYPE, SITE, oxstate, period)</td>
</tr>
</tbody>
</table>
<p>We can also generate a summary table of total reads &amp; ASVs for each sample. You can sort the table or download a copy. Here is the code to generate the data for the table. First, we create data frames that hold total reads and ASVs for each sample. We can also do a quick calculation of alpha diversity using the Shannon and InvSimpson indices.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
total_reads &lt;- sample_sums(ps_water)
total_reads &lt;- as.data.frame(total_reads, make.names = TRUE)
total_reads &lt;- total_reads %&gt;% rownames_to_column(&quot;Sample_ID&quot;)

total_asvs &lt;- estimate_richness(ps_water,
                                measures = c(
                                  &quot;Observed&quot;, &quot;Shannon&quot;, &quot;InvSimpson&quot;))
total_asvs &lt;- total_asvs %&gt;% rownames_to_column(&quot;Sample_ID&quot;)
total_asvs$Sample_ID &lt;- gsub(&#39;\\.&#39;, &#39;-&#39;, total_asvs$Sample_ID)</code></pre>
</div>
<p>And then we merge these two data frames with the sample data frame. We will use the <code>meta</code> command from the microbiome package to convert the <code>sample_data</code> to a data frame.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
sam_details &lt;- meta(sample_data(ps_water))
rownames(sam_details) &lt;- NULL

colnames(sam_details) &lt;- c(&quot;Sample_ID&quot;, &quot;Type&quot;, &quot;Site&quot;, &quot;Oxstate&quot;, &quot;Period&quot;)

merge_tab &lt;- merge(sam_details, total_reads, by = &quot;Sample_ID&quot;)
merge_tab2 &lt;- merge(merge_tab, total_asvs, by = &quot;Sample_ID&quot;)
colnames(merge_tab2) &lt;- c(&quot;Sample&lt;br/&gt;ID&quot;, &quot;Type&quot;, &quot;Site&quot;, &quot;Oxstate&quot;, &quot;Period&quot;,
    &quot;total&lt;br/&gt;reads&quot;, &quot;total&lt;br/&gt;ASVs&quot;, &quot;Shannon&quot;, &quot;InvSimpson&quot;)</code></pre>
</div>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="mex738zeiysgiy3fysb8" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="mex738zeiysgiy3fysb8">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Total reads &amp; ASVs by sample.<\/em>\n<\/caption>","data":[["WCC0","WCC1","WCC2","WCC3","WCR0","WCR1","WCR2","WCR3"],["W","W","W","W","W","W","W","W"],["CC","CC","CC","CC","CR","CR","CR","CR"],["normoxic","normoxic","normoxic","normoxic","normoxic","hypoxic","hypoxic","hypoxic"],["cc_after","cc_during","cc_during","cc_during","cr_after","cr_during","cr_during","cr_during"],[24290,24278,23962,19817,28616,12804,13479,33311],[305,287,287,237,323,242,304,552],[3.79586821583198,3.40949079884578,3.61556802457059,3.60334996135797,3.19266288364594,3.87638453942529,4.45854339112232,4.59793953134247],[8.64499926459407,5.452464523952,6.75399022762029,6.9033065773033,4.3230831643379,15.3718361806848,37.7312581395943,35.1458238662054]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Sample<br/>ID<\/th>\n      <th>Type<\/th>\n      <th>Site<\/th>\n      <th>Oxstate<\/th>\n      <th>Period<\/th>\n      <th>total<br/>reads<\/th>\n      <th>total<br/>ASVs<\/th>\n      <th>Shannon<\/th>\n      <th>InvSimpson<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Brti","buttons":["copy","csv","excel"],"pageLength":8,"lengthMenu":[5,10],"columnDefs":[{"targets":[7,8],"render":"function(data, type, row, meta) { return DTWidget.formatRound(data, 2, 3, \",\", \".\"); }"},{"className":"dt-right","targets":[5,6,7,8]}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\nvar value=data[8]; $(this.api().cell(row, 8).node()).css({'font-size':'80%'});\n}"}},"evals":["options.columnDefs.0.render","options.rowCallback"],"jsHooks":[]}</script>
</div>
<h1 id="diversity">Diversity</h1>
<h2 id="taxonomic-diversity">Taxonomic Diversity</h2>
<p>Let’s first take a look at the taxonomic diversity of the dataset. The code to generate this table is a little gross.</p>
<h4 id="rename-na-taxonomic-ranks">0. Rename <em>NA</em> taxonomic ranks</h4>
<p>Phyloseq has an odd way of dealing with taxonomic ranks that have no value—in other words, <em>NA</em> in the tax table. The first thing we are going to do before moving forward is to change all of the <em>NA</em> to have a value of the next highest classified rank. For example, <code>ASV5</code> is not classified at the Genus level but is at Family level (Saprospiraceae). So we change the Genus name to <em>Family_Saprospiraceae</em>. The code for this is hidden here but comes from these two posts on the phyloseq GitHub, both by <a href="https://github.com/MSMortensen">MSMortensen</a>: issue <a href="https://github.com/joey711/phyloseq/issues/850#issuecomment-394771087">#850</a> and issue <a href="https://github.com/joey711/phyloseq/issues/990#issuecomment-424618425">#990</a>. If you want to access the code chunk, check out the source code linked at the bottom of this page.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
[1] &quot;Kingdom&quot; &quot;Phylum&quot;  &quot;Class&quot;   &quot;Order&quot;   &quot;Family&quot;  &quot;Genus&quot;  
[7] &quot;ASV_SEQ&quot; &quot;ASV_ID&quot; </code></pre>
</div>
<h4 id="choose-a-rank">1. Choose a rank</h4>
<p>To make it easier to change the code later on, we can assign the taxonomic rank we are interested in to a variable, in the case <code>TRANK</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
TRANK &lt;- &quot;Class&quot;</code></pre>
</div>
<h4 id="generate-the-asv-reads-table">2. Generate the ASV &amp; reads table</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
tax_asv &lt;- table(tax_table(ps_water)[, TRANK], 
                 exclude = NULL, dnn = &quot;Taxa&quot;)
tax_asv &lt;- as.data.frame(tax_asv, make.names = TRUE, stringsAsFactors=FALSE)

tax_reads &lt;- factor(tax_table(ps_water)[, TRANK])
tax_reads &lt;- apply(otu_table(ps_water), MARGIN = 1, function(x)
{
    tapply(x, INDEX = tax_reads, FUN = sum, na.rm = TRUE, simplify = TRUE)
})
tax_reads &lt;- as.data.frame(tax_reads, make.names = TRUE)
tax_reads &lt;- cbind(tax_reads, reads = rowSums(tax_reads))
tax_reads &lt;- tax_reads[9]
tax_reads &lt;- setDT(tax_reads, keep.rownames = TRUE)[]</code></pre>
</div>
<h4 id="merge-the-two-tables">3. Merge the two tables</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
taxa_read_asv_tab &lt;- merge(tax_reads, tax_asv, by.x = &quot;rn&quot;, by.y = &quot;Taxa&quot;)
top_reads &lt;- top_n(taxa_read_asv_tab, n = 8, wt = reads)
top_asvs &lt;- top_n(taxa_read_asv_tab, n = 8, wt = Freq)

names(taxa_read_asv_tab) &lt;- c(&quot;Taxa&quot;, &quot;total reads&quot;, &quot;total ASVs&quot;)</code></pre>
</div>
<h4 id="diversity-table-by-class">4. Diversity table by Class</h4>
<div class="layout-chunk" data-layout="l-body">
<div id="9sgooxcmkzw565dtk7or" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="9sgooxcmkzw565dtk7or">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">Total reads &amp; ASVs by taxa<\/caption>","data":[["Acidimicrobiia","Actinobacteria","Alphaproteobacteria","Anaerolineae","Bacilli","Bacteroidia","Bdellovibrionia","Campylobacteria","Cloacimonadia","Clostridia","Cyanobacteriia","Dadabacteriia","Deferribacteres","Dehalococcoidia","Desulfobacteria","Desulfobulbia","Desulfovibrionia","Desulfuromonadia","Fusobacteriia","Gammaproteobacteria","Holophagae","Ignavibacteria","Kapabacteria","Kingdom_Bacteria","Kiritimatiellae","Lentisphaeria","Myxococcia","Nanoarchaeia","Negativicutes","Nitrososphaeria","OM190","Oligoflexia","Omnitrophia","Phycisphaerae","Phylum_Desulfobacterota","Phylum_Latescibacterota","Phylum_Margulisbacteria","Phylum_Marinimicrobia_(SAR406_clade)","Phylum_NB1-j","Phylum_Patescibacteria","Phylum_Proteobacteria","Phylum_SAR324_clade(Marine_group_B)","Phylum_Spirochaetota","Phylum_Verrucomicrobiota","Phylum_WPS-2","Pla3_lineage","Planctomycetes","Polyangia","Rhodothermia","Spirochaetia","Subgroup_22","Synergistia","Syntrophobacteria","Thermoanaerobaculia","Thermoplasmata","Thermotogae","Vampirivibrionia","Verrucomicrobiae","vadinHA49"],[4745,37,33227,7,422,31857,769,2423,241,302,66365,142,64,38,381,94,506,397,88,19928,8,37,40,521,376,80,211,107,3,23,859,76,4,836,2,38,14,3212,5,3,38,1360,15,2,46,51,1728,54,1671,192,2,1,2,4,5876,9,30,982,6],[13,3,182,2,10,290,46,8,6,30,71,1,5,2,9,8,20,30,7,203,2,2,4,52,7,12,4,9,1,3,16,8,1,26,1,2,4,20,1,1,7,7,1,1,3,2,30,4,6,14,1,1,1,1,34,1,2,32,2]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Taxa<\/th>\n      <th>total reads<\/th>\n      <th>total ASVs<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":0},{"className":"dt-right","targets":[1,2]}],"dom":"Blfrtip","buttons":["csv","copy"],"scrollX":true,"scrollCollapse":true,"scrollY":true,"pageLength":5,"scroller":true,"lengthMenu":[5,10,25,60],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<p>Here we can see that:</p>
<ul>
<li>Cyanobacteriia, Alphaproteobacteria, Bacteroidia, Gammaproteobacteria, Thermoplasmata, Acidimicrobiia, Phylum_Marinimicrobia_(SAR406_clade), Campylobacteria contain the <strong>most reads</strong> and</li>
<li>Bacteroidia, Gammaproteobacteria, Alphaproteobacteria, Cyanobacteriia, Kingdom_Bacteria, Bdellovibrionia, Thermoplasmata, Verrucomicrobiae contain the <strong>most ASVs</strong>.</li>
</ul>
<h2 id="hypoxic-vs.normoxic">Hypoxic vs. Normoxic</h2>
<p>How does the taxonomic composition of normoxic compare to hypoxic samples? To look at this, we will combine samples by the <code>oxstate</code> variable and then calculate the <em>relative abundance</em> for each taxa for both oxygen states. Note one of the Cayo Roldan samples (WCR0) was collected after the hypoxic event when oxygen levels had returned to normal. Therefore WCR0 is considered a normoxic sample.</p>
<p>There are a few steps we need to run for this analysis.</p>
<h4 id="choose-a-rank-1">1. Choose a rank</h4>
<p>Again, we can select a rank.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
TRANK &lt;- &quot;Class&quot;</code></pre>
</div>
<h4 id="calculate-the-averages-merge-samples-by-oxstate">2. Calculate the averages &amp; merge samples by oxstate</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
ps_water_AVG &lt;- transform_sample_counts(ps_water, function(x) x/sum(x))
ps_water_ox &lt;- merge_samples(ps_water, &quot;oxstate&quot;)
SD_BAR_w &lt;- merge_samples(sample_data(ps_water_AVG), &quot;oxstate&quot;)</code></pre>
</div>
<h4 id="merge-taxa-by-rank">3. Merge taxa by rank</h4>
<p>We will set a variable for the taxonomic rank of choice. If you want to choose a different rank, be sure to change the variable assignment.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
mdata_phy_w &lt;- tax_glom(ps_water_ox, taxrank = TRANK, NArm = FALSE)
mdata_phyrel_w &lt;- transform_sample_counts(mdata_phy_w, function(x) x/sum(x))
meltd_w &lt;- psmelt(mdata_phyrel_w)
meltd_w[[TRANK]] &lt;- as.character(meltd_w[[TRANK]])</code></pre>
</div>
<h4 id="calculate-the-total-relative-abundance-for-all-taxa">4. Calculate the total relative abundance for all taxa</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
means_w &lt;- ddply(meltd_w, ~get(TRANK), function(x) c(mean = mean(x$Abundance)))
colnames(means_w) &lt;- c(TRANK, &quot;mean&quot;)
means_w$mean &lt;- round(means_w$mean, digits = 5)
taxa_means_w &lt;- means_w[order(-means_w$mean), ]  # this orders in decending fashion
taxa_means_w &lt;- format(taxa_means_w, scientific = FALSE)  # ditch the sci notation</code></pre>
</div>
<h4 id="relative-abundance-by-class">5. Relative abundance by Class</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
datatable(
  taxa_means_w, rownames = FALSE, width = &quot;100%&quot;,
  colnames = c(TRANK, &quot;mean&quot;),
  caption = htmltools::tags$caption(
    style = &quot;caption-side: bottom; text-align: left;&quot;,
    &quot;Relative abundance by taxonomic rank. &quot;),
  elementId = &quot;t9xg4oo3fbjqjqu1ogjm&quot;,
  extensions = &quot;Buttons&quot;,
  options = list(columnDefs = list(list(
    className = &quot;dt-left&quot;, targets = &quot;_all&quot;)),
    dom = &quot;Blfrtip&quot;, buttons = c(&quot;csv&quot;, &quot;copy&quot;),
    scrollX = TRUE, scrollCollapse = TRUE, scrollY=TRUE,
    pageLength = 8,
    scroller=TRUE, lengthMenu = c(10, 25, 60)))</code></pre>
<div id="t9xg4oo3fbjqjqu1ogjm" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="t9xg4oo3fbjqjqu1ogjm">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">Relative abundance by taxonomic rank. <\/caption>","data":[["Cyanobacteriia","Bacteroidia","Alphaproteobacteria","Gammaproteobacteria","Thermoplasmata","Acidimicrobiia","Campylobacteria","Phylum_Marinimicrobia_(SAR406_clade)","Rhodothermia","Planctomycetes","Phylum_SAR324_clade(Marine_group_B)","OM190","Verrucomicrobiae","Phycisphaerae","Desulfovibrionia","Bdellovibrionia","Kingdom_Bacteria","Desulfobacteria","Desulfuromonadia","Clostridia","Bacilli","Cloacimonadia","Kiritimatiellae","Spirochaetia","Myxococcia","Desulfobulbia","Nanoarchaeia","Dadabacteriia","Fusobacteriia","Deferribacteres","Lentisphaeria","Oligoflexia","Kapabacteria","Phylum_Latescibacterota","Ignavibacteria","Pla3_lineage","Actinobacteria","Vampirivibrionia","Polyangia","Phylum_WPS-2","Phylum_Proteobacteria","Dehalococcoidia","Phylum_Spirochaetota","Nitrososphaeria","Thermotogae","Holophagae","Phylum_Margulisbacteria","Anaerolineae","Phylum_NB1-j","vadinHA49","Omnitrophia","Phylum_Desulfobacterota","Thermoanaerobaculia","Negativicutes","Phylum_Patescibacteria","Phylum_Verrucomicrobiota","Subgroup_22","Synergistia","Syntrophobacteria"],["0.30801","0.19233","0.19169","0.14199","0.02620","0.02150","0.02032","0.01398","0.01286","0.01087","0.00686","0.00605","0.00590","0.00516","0.00425","0.00414","0.00355","0.00320","0.00256","0.00253","0.00215","0.00202","0.00192","0.00161","0.00103","0.00075","0.00068","0.00065","0.00063","0.00054","0.00052","0.00036","0.00034","0.00032","0.00031","0.00031","0.00026","0.00025","0.00022","0.00020","0.00017","0.00016","0.00013","0.00010","0.00008","0.00007","0.00006","0.00005","0.00004","0.00004","0.00002","0.00002","0.00002","0.00001","0.00001","0.00001","0.00001","0.00001","0.00001"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Class<\/th>\n      <th>mean<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":"_all"}],"dom":"Blfrtip","buttons":["csv","copy"],"scrollX":true,"scrollCollapse":true,"scrollY":true,"pageLength":8,"scroller":true,"lengthMenu":[10,25,60],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<h4 id="group-taxa-into-other">6. Group taxa into ‘Other’</h4>
<p>Time to group low abundance taxa into an <em>Other</em> catagory for the bar graph. Since we cannot possibly display all 59 differnt taxa in one figure, we need to collapse the low abundance groups.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
TAXAN &lt;- 8
top_perc_w &lt;- top_n(taxa_means_w, n = TAXAN, wt = mean)
top_perc_w$mean &lt;- round(as.numeric(top_perc_w$mean), digits = 5)
min_top_perc_w &lt;- round(as.numeric(min(top_perc_w$mean)), digits = 5)
top_perc_w_list &lt;- top_perc_w[,1]</code></pre>
</div>
<p>Here we decided to look at the abundance of the top 8, which sets a relative abundance cutoff at 0.01398. Anything lower than 0.01398 will be grouped into <em>Other</em>.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
Other_w &lt;- means_w[means_w$mean &lt; min_top_perc_w, ][[TRANK]]
meltd_w[meltd_w[[TRANK]] %in% Other_w, ][[TRANK]] &lt;- &quot;Other&quot;
samp_names_w &lt;- aggregate(meltd_w$Abundance, by = list(meltd_w$Sample), FUN = sum)[, 1]
.e_w &lt;- environment()
meltd_w[, TRANK] &lt;- factor(meltd_w[, TRANK], sort(unique(meltd_w[, TRANK])))
meltd_w1 &lt;- meltd_w[order(meltd_w[, TRANK]), ]

target &lt;- c(&quot;Cyanobacteriia&quot;, &quot;Alphaproteobacteria&quot;, 
            &quot;Bacteroidia&quot;, &quot;Gammaproteobacteria&quot;, 
            &quot;Thermoplasmata&quot;, &quot;Phylum_Marinimicrobia_(SAR406_clade)&quot;, 
            &quot;Acidimicrobiia&quot;, &quot;Campylobacteria&quot;, &quot;Other&quot;)
meltd_w1[[TRANK]] &lt;- reorder.factor(meltd_w1[[TRANK]], new.order=target)</code></pre>
</div>
<h4 id="construct-a-bar-graph">7. Construct a bar graph</h4>
<p>Finally.</p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="16s-water_files/figure-html5/make_fig_n_v_h-1.png" width="624" /></p>
</div>
<p>We can also look at this in tabular format. To accomplish the task, we use the <code>aggregate</code> command to group abundance values by oxygen state <em>and</em> taxonomic group.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
hyp_v_norm &lt;- aggregate(meltd_w1$Abundance, 
                 by = list(Rank = meltd_w1[[TRANK]], 
                           Sample = meltd_w1$Sample), 
                 FUN = sum)
hyp_v_norm$x &lt;- round(hyp_v_norm$x, digits = 4)
hyp_v_norm &lt;- spread(hyp_v_norm, Sample, x)</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="wewuidn73q2ceico4vyw" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="wewuidn73q2ceico4vyw">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Relative abundance (reads) of normoxic \n                                     vs hypoxic samples by dominant taxa.<\/em>\n<\/caption>","data":[["Cyanobacteriia","Alphaproteobacteria","Bacteroidia","Gammaproteobacteria","Thermoplasmata","Phylum_Marinimicrobia_(SAR406_clade)","Acidimicrobiia","Campylobacteria","Other"],[0.1328,0.2142,0.2391,0.235,0.0075,0.0028,0.0075,0.0406,0.1205],[0.4832,0.1691,0.1456,0.049,0.0449,0.0252,0.0356,0,0.0475]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Rank<\/th>\n      <th>hypoxic<\/th>\n      <th>normoxic<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Brti","buttons":["copy","csv","excel"],"pageLength":9,"columnDefs":[{"className":"dt-right","targets":[1,2]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[9,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<p></br></p>
<h2 id="during-vs.after-the-event">During vs. After the event</h2>
<p>What did taxonomic composition look like during and after the event at <strong>each site</strong>? To look at this, we will combine samples by the <code>period</code> variable and then calculate the <em>relative abundance</em> for each taxa. The code for this analysis is basically the same as the code above where we compared hypoxic to normoxic samples. Therefore, all of the code is hidden. See the link at the bottom of the page for the page source code if you are interested in the code we used here. We go through the same steps:</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<ol type="1">
<li>Choose a rank.</li>
<li>Merge samples by period.</li>
<li>Merge taxa by rank.</li>
<li>Calculate the total relative abundance for all taxa.</li>
<li>Relative abundance by Class. <em>Values are slightly different because averaging across a different grouping</em>.</li>
<li>Group taxa into ‘Other’.</li>
<li>Construct a bar graph.</li>
</ol>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div id="zswbg7x7f8eqj526xuox" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="zswbg7x7f8eqj526xuox">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">Relative abundance by taxonomic rank. <\/caption>","data":[["Cyanobacteriia","Alphaproteobacteria","Bacteroidia","Gammaproteobacteria","Thermoplasmata","Acidimicrobiia","Phylum_Marinimicrobia_(SAR406_clade)","Campylobacteria","Planctomycetes","Phylum_SAR324_clade(Marine_group_B)","Rhodothermia","Verrucomicrobiae","OM190","Bdellovibrionia","Phycisphaerae","Kingdom_Bacteria","Bacilli","Desulfovibrionia","Desulfuromonadia","Kiritimatiellae","Desulfobacteria","Myxococcia","Clostridia","Cloacimonadia","Dadabacteriia","Spirochaetia","Nanoarchaeia","Desulfobulbia","Fusobacteriia","Polyangia","Oligoflexia","Lentisphaeria","Phylum_WPS-2","Pla3_lineage","Deferribacteres","Dehalococcoidia","Actinobacteria","Nitrososphaeria","Kapabacteria","Ignavibacteria","Phylum_Latescibacterota","Phylum_Proteobacteria","Vampirivibrionia","Phylum_Margulisbacteria","Phylum_Spirochaetota","Anaerolineae","Thermotogae","vadinHA49","Holophagae","Phylum_Patescibacteria","Thermoanaerobaculia","Phylum_NB1-j","Phylum_Verrucomicrobiota","Syntrophobacteria","Negativicutes","Omnitrophia","Phylum_Desulfobacterota","Subgroup_22","Synergistia"],["0.39723","0.18158","0.16630","0.09498","0.03706","0.02885","0.01791","0.01016","0.00923","0.00918","0.00721","0.00510","0.00428","0.00412","0.00401","0.00251","0.00226","0.00212","0.00200","0.00167","0.00160","0.00131","0.00127","0.00101","0.00087","0.00081","0.00069","0.00041","0.00041","0.00041","0.00039","0.00034","0.00032","0.00031","0.00027","0.00024","0.00018","0.00018","0.00017","0.00016","0.00016","0.00015","0.00013","0.00007","0.00006","0.00004","0.00004","0.00004","0.00003","0.00003","0.00003","0.00002","0.00002","0.00002","0.00001","0.00001","0.00001","0.00001","0.00000"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Class<\/th>\n      <th>mean<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":"_all"}],"dom":"Blfrtip","buttons":["csv","copy"],"scrollX":true,"scrollCollapse":true,"scrollY":true,"pageLength":9,"scroller":true,"lengthMenu":[9,25,60],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<p>Again, Time to group low abundance taxa into an <em>Other</em> catagory for the bar graph. Since we cannot possibly display all 59 differnt taxa in one figure, we need to collapse the low abundance groups.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="16s-water_files/figure-html5/make_fig_d_v_a-1.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div id="kqtr8k9xeqyk0sb1s4h4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="kqtr8k9xeqyk0sb1s4h4">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Relative abundance (reads) of normoxic \n                                     vs hypoxic samples by dominant taxa.<\/em>\n<\/caption>","data":[["Cyanobacteriia","Alphaproteobacteria","Bacteroidia","Gammaproteobacteria","Thermoplasmata","Phylum_Marinimicrobia_(SAR406_clade)","Acidimicrobiia","Campylobacteria","Other"],[0.4056,0.2126,0.1714,0.0483,0.0704,0.0171,0.0424,0,0.0323],[0.47,0.1705,0.1561,0.0503,0.0432,0.0292,0.0354,0,0.0453],[0.5805,0.129,0.0987,0.0463,0.0272,0.0226,0.0301,0,0.0656],[0.1328,0.2142,0.2391,0.235,0.0075,0.0028,0.0075,0.0406,0.1205]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Crawl Caye (after)<\/th>\n      <th>Crawl Caye (during)<\/th>\n      <th>Cayo Roldan (after)<\/th>\n      <th>Cayo Roldan (during)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Brti","buttons":["copy","csv","excel"],"pageLength":9,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[9,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<p></br></p>
<div class="post-nav">
<div class="post-nav-item">
<div class="meta-nav">
Previous
</div>
<p><a href="16s-diversity.html" rel="next">N<sup><u>o</u></sup> 2. Data Preparation</a></p>
</div>
</div>
<div class="post-nav">
<div class="post-nav-item">
<div class="meta-nav">
Next
</div>
<p>PROCEED to METAGENOMIC WORKFLOW<br/> <a href="mg-setup.html" rel="prev">N<sup><u>o</u></sup> 1. Setup Working Environment</a></p>
</div>
</div>
<h2 id="source-code" class="appendix">Source Code</h2>
<p>The source code for this page can be accessed on GitHub by <a href="https://github.com/hypocolypse/web/blob/master/16s-water.Rmd">clicking this link</a>.</p>
<div id="refs" class="references">
<div id="ref-lahti2017microbiome">
<p>Lahti, Leo, Shetty Sudarshan, and others. 2017. “Tools for Microbiome Analysis in R. Version 1.9.97.” <a href="https://github.com/microbiome/microbiome">https://github.com/microbiome/microbiome</a>.</p>
</div>
</div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/hypocolypse/web/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/hypocolypse/web/">https://github.com/hypocolypse/web/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<script id="distill-bibliography" type="text/bibtex">
@article{callahan2016dada2,
  title={DADA2: high-resolution sample inference from Illumina amplicon data},
  author={Callahan, Benjamin J and McMurdie, Paul J and Rosen, Michael J and Han, Andrew W and Johnson, Amy Jo A and Holmes, Susan P},
  journal={Nature Methods},
  volume={13},
  number={7},
  pages={581},
  year={2016},
  url = {https://doi.org/10.1038/nmeth.3869},
}

@article{martin2011cutadapt,
  title={Cutadapt removes adapter sequences from high-throughput sequencing reads},
  author={Martin, Marcel},
  journal={EMBnet. journal},
  volume={17},
  number={1},
  pages={10--12},
  year={2011},
  url = {https://doi.org/10.14806/ej.17.1.200}
}

@article{quast2012silva,
  title={The SILVA ribosomal RNA gene database project: improved data processing and web-based tools},
  author={Quast, Christian and Pruesse, Elmar and Yilmaz, Pelin and Gerken, Jan and Schweer, Timmy and Yarza, Pablo and Peplies, J{\"o}rg and Gl{\"o}ckner, Frank Oliver},
  journal={Nucleic Acids Research},
  volume={41},
  number={D1},
  pages={D590--D596},
  year={2012},
  url = {https://doi.org/10.1093/nar/gks1219}
}

@article{mcmurdie2013phyloseq,
  title={phyloseq: an R package for reproducible interactive analysis and graphics of microbiome census data},
  author={McMurdie, Paul J and Holmes, Susan},
  journal={PLoS One},
  volume={8},
  number={4},
  pages={e61217},
  year={2013},
  url = {https://doi.org/10.1371/journal.pone.0061217}
}

@misc{wong2011points,
  title={Points of view: Color blindness},
  author={Wong, Bang},
  journal={Nature Methods},
  volume={8},
  number={6},
  pages={441},
  year={2011},
  url = {https://doi.org/10.1038/nmeth.1618}
}

@misc{lahti2017microbiome,
  author = {Lahti, Leo and Shetty Sudarshan and others},
  title = {Tools for microbiome analysis in R. Version 1.9.97},
  year = {2017},
  url = {https://github.com/microbiome/microbiome},
}

@article{eren2015anvi,
  title={Anvi’o: an advanced analysis and visualization platform for ‘omics data},
  author={Eren, A Murat and Esen, {\"O}zcan C and Quince, Christopher and Vineis, Joseph H and Morrison, Hilary G and Sogin, Mitchell L and Delmont, Tom O},
  journal={PeerJ},
  volume={3},
  pages={e1319},
  year={2015},
  url = {https://doi.org/10.7717/peerj.1319}
}

@article{bolger2014trimmomatic,
  title={Trimmomatic: a flexible trimmer for Illumina sequence data},
  author={Bolger, Anthony M and Lohse, Marc and Usadel, Bjoern},
  journal={Bioinformatics},
  volume={30},
  number={15},
  pages={2114--2120},
  year={2014},
  url = {https://doi.org/10.1093/bioinformatics/btu170}
}

@article{koster2012snakemake,
  title={Snakemake—a scalable bioinformatics workflow engine},
  author={K{\"o}ster, Johannes and Rahmann, Sven},
  journal={Bioinformatics},
  volume={28},
  number={19},
  pages={2520--2522},
  year={2012},
  url = {https://doi.org/10.1093/bioinformatics/bts480}
}

@article{menzel2016fast,
  title={Fast and sensitive taxonomic classification for metagenomics with Kaiju},
  author={Menzel, Peter and Ng, Kim Lee and Krogh, Anders},
  journal={Nature Communications},
  volume={7},
  pages={11257},
  year={2016},
  url = {https://doi.org/10.1038/ncomms11257}
}

@article{roux2015virsorter,
  title={VirSorter: mining viral signal from microbial genomic data},
  author={Roux, Simon and Enault, Francois and Hurwitz, Bonnie L and Sullivan, Matthew B},
  journal={PeerJ},
  volume={3},
  pages={e985},
  year={2015},
  url = {https://doi.org/10.1093/bioinformatics/btu170}
}

@article{kim2016centrifuge,
  title={Centrifuge: rapid and sensitive classification of metagenomic sequences},
  author={Kim, Daehwan and Song, Li and Breitwieser, Florian P and Salzberg, Steven L},
  journal={Genome Research},
  volume={26},
  number={12},
  pages={1721--1729},
  year={2016},
  url = {https://doi.org/10.1101/gr.210641.116.}
}

@article{hyatt2010prodigal,
  title={Prodigal: prokaryotic gene recognition and translation initiation site identification},
  author={Hyatt, Doug and Chen, Gwo-Liang and LoCascio, Philip F and Land, Miriam L and Larimer, Frank W and Hauser, Loren J},
  journal={BMC Bioinformatics},
  volume={11},
  number={1},
  pages={119},
  year={2010},
  url = {https://doi.org/10.1186/1471-2105-11-119}
}

@article{edgar2004muscle,
  title={MUSCLE: multiple sequence alignment with high accuracy and high throughput},
  author={Edgar, Robert C},
  journal={Nucleic Acids Research},
  volume={32},
  number={5},
  pages={1792--1797},
  year={2004},
  url = {https://doi.org/10.1093/nar/gkh340}
}

@article{li2015megahit,
  title={MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph},
  author={Li, Dinghua and Liu, Chi-Man and Luo, Ruibang and Sadakane, Kunihiko and Lam, Tak-Wah},
  journal={Bioinformatics},
  volume={31},
  number={10},
  pages={1674--1676},
  year={2015},
  url = {https://doi.org/10.1093/bioinformatics/btv033}
}

@article{langmead2012fast,
  title={Fast gapped-read alignment with Bowtie 2},
  author={Langmead, Ben and Salzberg, Steven L},
  journal={Nature Methods},
  volume={9},
  number={4},
  pages={357},
  year={2012},
  url = {https://doi.org/10.1038/nmeth.1923}
}

@article{li2009fast,
  title={Fast and accurate short read alignment with Burrows--Wheeler transform},
  author={Li, Heng and Durbin, Richard},
  journal={Bioinformatics},
  volume={25},
  number={14},
  pages={1754--1760},
  year={2009},
  url = {https://doi.org/10.1093/bioinformatics/btp324}
}

@article{li2009sequence,
  title={The sequence alignment/map format and SAMtools},
  author={Li, Heng and Handsaker, Bob and Wysoker, Alec and Fennell, Tim and Ruan, Jue and Homer, Nils and Marth, Gabor and Abecasis, Goncalo and Durbin, Richard},
  journal={Bioinformatics},
  volume={25},
  number={16},
  pages={2078--2079},
  year={2009},
  url = {https://doi.org/10.1093/bioinformatics/btp352}
}

@article{breitwieser2018krakenuniq,
  title={KrakenUniq: confident and fast metagenomics classification using unique k-mer counts},
  author={Breitwieser, FP and Baker, DN and Salzberg, Steven L},
  journal={Genome Biology},
  volume={19},
  number={1},
  pages={1--10},
  year={2018},
  url = {https://doi.org/10.5281/zenodo.1412647}
}

@article{eren2013filtering,
  title={A filtering method to generate high quality short reads using Illumina paired-end technology},
  author={Eren, A Murat and Vineis, Joseph H and Morrison, Hilary G and Sogin, Mitchell L},
  journal={PLoS One},
  volume={8},
  number={6},
  year={2013},
  url = {https://doi.org/10.1371/journal.pone.0066643}
}

@article{alneberg2014binning,
  title={Binning metagenomic contigs by coverage and composition},
  author={Alneberg, Johannes and Bjarnason, Brynjar Sm{\'a}ri and De Bruijn, Ino and Schirmer, Melanie and Quick, Joshua and Ijaz, Umer Z and Lahti, Leo and Loman, Nicholas J and Andersson, Anders F and Quince, Christopher},
  journal={Nature Methods},
  volume={11},
  number={11},
  pages={1144--1146},
  year={2014},
  url = {https://doi.org/10.1038/nmeth.3103}
}

@article{sieber2018recovery,
  title={Recovery of genomes from metagenomes via a dereplication, aggregation and scoring strategy},
  author={Sieber, Christian MK and Probst, Alexander J and Sharrar, Allison and Thomas, Brian C and Hess, Matthias and Tringe, Susannah G and Banfield, Jillian F},
  journal={Nature Microbiology},
  volume={3},
  number={7},
  pages={836--843},
  year={2018},
}

@article{kang2019metabat,
  title={MetaBAT 2: an adaptive binning algorithm for robust and efficient genome reconstruction from metagenome assemblies},
  author={Kang, Dongwan D and Li, Feng and Kirton, Edward and Thomas, Ashleigh and Egan, Rob and An, Hong and Wang, Zhong},
  journal={PeerJ},
  volume={7},
  pages={e7359},
  year={2019},
}
@article{wu2016maxbin,
  title={MaxBin 2.0: an automated binning algorithm to recover genomes from multiple metagenomic datasets},
  author={Wu, Yu-Wei and Simmons, Blake A and Singer, Steven W},
  journal={Bioinformatics},
  volume={32},
  number={4},
  pages={605--607},
  year={2016},
}

@article{ondov2011krona,
  title={Interactive metagenomic visualization in a Web browser},
  author={Ondov, Brian D and Bergman, Nicholas H and Phillippy, Adam M},
  journal={BMC Bioinformatics},
  volume={12},
  number={1},
  pages={385},
  year={2011},
  url = {https://doi.org/10.1186/1471-2105-12-385}
}

@article{buchfink2015fast,
  title={Fast and sensitive protein alignment using DIAMOND},
  author={Buchfink, Benjamin and Xie, Chao and Huson, Daniel H},
  journal={Nature Methods},
  volume={12},
  number={1},
  pages={59--60},
  year={2015},
  url = {https://doi.org/10.1038/nmeth.3176}
}

@article{delmont2018nitrogen,
  title={Nitrogen-fixing populations of Planctomycetes and Proteobacteria are abundant in surface ocean metagenomes},
  author={Delmont, Tom O and Quince, Christopher and Shaiber, Alon and Esen, {\"O}zcan C and Lee, Sonny TM and Rapp{\'e}, Michael S and McLellan, Sandra L and L{\"u}cker, Sebastian and Eren, A Murat},
  journal={Nature Microbiology},
  volume={3},
  number={7},
  pages={804--813},
  year={2018},
  url = {https://doi.org/10.1038/s41564-018-0176-9}
}

@article{dongen2000cluster,
  title={A cluster algorithm for graphs},
  author={Dongen, Stijn},
  year={2000},
  publisher={CWI (Centre for Mathematics and Computer Science)},
  url = {https://www.micans.org/mcl/index.html?sec_software}
}

@article{jain2018high,
  title={High throughput ANI analysis of 90K prokaryotic genomes reveals clear species boundaries},
  author={Jain, Chirag and Rodriguez-R, Luis M and Phillippy, Adam M and Konstantinidis, Konstantinos T and Aluru, Srinivas},
  journal={Nature Communications},
  volume={9},
  number={1},
  pages={1--8},
  year={2018},
  url={https://doi.org/10.1038/s41467-018-07641-9}
}

@article{stamatakis2014raxml,
  title={RAxML version 8: a tool for phylogenetic analysis and post-analysis of large phylogenies},
  author={Stamatakis, Alexandros},
  journal={Bioinformatics},
  volume={30},
  number={9},
  pages={1312--1313},
  year={2014},
  url={https://doi.org/10.1093/bioinformatics/btu033}
}

@article{minh2020iq,
  title={IQ-TREE 2: New models and efficient methods for phylogenetic inference in the genomic era},
  author={Minh, Bui Quang and Schmidt, Heiko A and Chernomor, Olga and Schrempf, Dominik and Woodhams, Michael D and Von Haeseler, Arndt and Lanfear, Robert},
  journal={Molecular Biology and Evolution},
  volume={37},
  number={5},
  pages={1530--1534},
  year={2020},
  url={https://doi.org/10.1093/molbev/msaa015}
}

@article{kalyaanamoorthy2017modelfinder,
  title={ModelFinder: fast model selection for accurate phylogenetic estimates},
  author={Kalyaanamoorthy, Subha and Minh, Bui Quang and Wong, Thomas KF and von Haeseler, Arndt and Jermiin, Lars S},
  journal={Nature Methods},
  volume={14},
  number={6},
  pages={587},
  year={2017},
  url={https://doi.org/10.1038/nmeth.4285}
}

@article{quince2011removing,
  title={Removing noise from pyrosequenced amplicons},
  author={Quince, Christopher and Lanzen, Anders and Davenport, Russell J and Turnbaugh, Peter J},
  journal={BMC bioinformatics},
  volume={12},
  number={1},
  pages={38},
  year={2011},
  url={https://doi.org/10.1186/1471-2105-12-38}
}
@article{caporaso2011global,
  title={Global patterns of 16S rRNA diversity at a depth of millions of sequences per sample},
  author={Caporaso, J Gregory and Lauber, Christian L and Walters, William A and Berg-Lyons, Donna and Lozupone, Catherine A and Turnbaugh, Peter J and Fierer, Noah and Knight, Rob},
  journal={Proceedings of the National Academy of Sciences},
  volume={108},
  number={Supplement 1},
  pages={4516--4522},
  year={2011},
  url={https://doi.org/10.1073/pnas.1000080107}
}
@article{kanehisa2000kegg,
  title={KEGG: kyoto encyclopedia of genes and genomes},
  author={Kanehisa, Minoru and Goto, Susumu},
  journal={Nucleic Acids Research},
  volume={28},
  number={1},
  pages={27--30},
  year={2000},
  url={https://doi.org/10.1093/nar/28.1.27}
}

@article{kanehisa2016blastkoala,
  title={BlastKOALA and GhostKOALA: KEGG tools for functional characterization of genome and metagenome sequences},
  author={Kanehisa, Minoru and Sato, Yoko and Morishima, Kanae},
  journal={Journal of Molecular Biology},
  volume={428},
  number={4},
  pages={726--731},
  year={2016},
  url={https://doi.org/10.1016/j.jmb.2015.11.006}
}

@article{minoche2011evaluation,
  title={Evaluation of genomic high-throughput sequencing data generated on Illumina HiSeq and genome analyzer systems},
  author={Minoche, Andr{\'e} E and Dohm, Juliane C and Himmelbauer, Heinz},
  journal={Genome Biology},
  volume={12},
  number={11},
  pages={R112},
  year={2011},
  url={https://doi.org/10.1186/gb-2011-12-11-r112}
}

@article{lee2019gtotree,
  title={GToTree: a user-friendly workflow for phylogenomics},
  author={Lee, Michael D},
  journal={Bioinformatics},
  volume={35},
  number={20},
  pages={4162--4164},
  year={2019},
  url={https://doi.org/10.1093/bioinformatics/btz188}
}

@article{wang2007naive,
  title={Naive Bayesian classifier for rapid assignment of rRNA sequences into the new bacterial taxonomy},
  author={Wang, Qiong and Garrity, George M and Tiedje, James M and Cole, James R},
  journal={Applied & Environmental Microbiology},
  volume={73},
  number={16},
  pages={5261--5267},
  year={2007},
  url={https://doi.org/10.1128/AEM.00062-07}
}

@article{hoang2017ultrafast,
  author = {Hoang, Diep Thi and Chernomor, Olga and von Haeseler, Arndt and Minh, Bui Quang and Vinh, Le Sy},
  title = {UFBoot2: Improving the Ultrafast Bootstrap Approximation},
  journal = {Molecular Biology and Evolution},
  volume = {35},
  number = {2},
  pages = {518-522},
  year = {2017},
  url = {https://doi.org/10.1093/molbev/msx281}
}

@article{parada2016every,
  title={Every base matters: assessing small subunit rRNA primers for marine microbiomes with mock communities, time series and global field samples},
  author={Parada, Alma E and Needham, David M and Fuhrman, Jed A},
  journal={Environmental microbiology},
  volume={18},
  number={5},
  pages={1403--1414},
  year={2016},
  url = {https://doi.org/10.1111/1462-2920.13023}
}

@article{parks2018gtgb,
  title={A standardized bacterial taxonomy based on genome phylogeny substantially revises the tree of life},
  author={Parks, Donovan H and Chuvochina, Maria and Waite, David W and Rinke, Christian and Skarshewski, Adam and Chaumeil, Pierre-Alain and Hugenholtz, Philip},
  journal={Nature Biotechnology},
  volume={36},
  number={10},
  pages={1546-1696},
  year={2018},
  url = {https://doi.org/10.1038/nbt.4229}
}
</script>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p>© Copyright 2020 The HYPOCOLYPSE Project.</p>
<p>Site constructed using <a href="https://rstudio.github.io/distill/">Distill for R Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
