<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

  <!--radix_placeholder_meta_tags-->
  <title>: Field Analyses</title>
  
  <meta property="description" itemprop="description" content="In this section we discuss the various field analyses conducted in this study."/>
  
  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>
  <link rel="icon" type="image/vnd.microsoft.icon" href="assets/favicon.ico"/>
  
  
  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content=": Field Analyses"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="In this section we discuss the various field analyses conducted in this study."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content=""/>
  
  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content=": Field Analyses"/>
  <meta property="twitter:description" content="In this section we discuss the various field analyses conducted in this study."/>
  
  <!--/radix_placeholder_meta_tags-->
  
  <!--radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","bibliography","output"]}},"value":[{"type":"character","attributes":{},"value":["Field Analyses"]},{"type":"character","attributes":{},"value":["In this section we discuss the various field analyses conducted in this study.\n"]},{"type":"character","attributes":{},"value":["assets/cite.bib"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["distill::distill_article"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["css","toc","toc_depth"]}},"value":[{"type":"character","attributes":{},"value":["assets/styles.css"]},{"type":"logical","attributes":{},"value":[true]},{"type":"integer","attributes":{},"value":[3]}]}]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  
  <script type="application/javascript">
  
    window.headroom_prevent_pin = false;
  
    window.document.addEventListener("DOMContentLoaded", function (event) {
  
      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');
  
      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });
  
      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });
  
      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>
  
  <style type="text/css">
  
  /* Theme (user-documented overrideables for nav appearance) */
  
  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #455a64;
    font-size: 15px;
    font-weight: 300;
  }
  
  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }
  
  .distill-site-nav a:hover {
    color: white;
  }
  
  @media print {
    .distill-site-nav {
      display: none;
    }
  }
  
  .distill-site-header {
  
  }
  
  .distill-site-footer {
  
  }
  
  
  /* Site Header */
  
  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }
  
  
  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }
  
  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }
  
  .distill-site-header .title {
    font-size: 18px;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }
  
  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }
  
  
  
  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }
  
  
  .distill-site-header .nav-toggle {
    display: none;
  }
  
  .nav-dropdown {
    display: inline-block;
    position: relative;
  }
  
  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }
  
  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }
  
  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .nav-dropdown-active {
    display: block;
  }
  
  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }
  
  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }
  
  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }
  
  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative;}
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }
  
  /* Site Footer */
  
  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }
  
  /* Headroom */
  
  d-title {
    padding-top: 6rem;
  }
  
  @media print {
    d-title {
      padding-top: 4rem;
    }
  }
  
  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }
  
  .headroom--transition {
    transition: all .4s ease-in-out;
  }
  
  .headroom--unpinned {
    top: -100px;
  }
  
  .headroom--pinned {
    top: 0;
  }
  
  </style>
  
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->
  
  <style type="text/css">
  
  body {
    background-color: white;
  }
  
  .pandoc-table {
    width: 100%;
  }
  
  .pandoc-table>caption {
    margin-bottom: 10px;
  }
  
  .pandoc-table th:not([align]) {
    text-align: left;
  }
  
  .pagedtable-footer {
    font-size: 15px;
  }
  
  .html-widget {
    margin-bottom: 2.0em;
  }
  
  .l-screen-inset {
    padding-right: 16px;
  }
  
  .l-screen .caption {
    margin-left: 10px;
  }
  
  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .shaded-content {
    background: white;
  }
  
  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }
  
  .hidden {
    display: none !important;
  }
  
  d-article {
    padding-bottom: 30px;
  }
  
  d-appendix {
    padding-top: 30px;
  }
  
  d-article>p>img {
    width: 100%;
  }
  
  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }
  
  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  /* CSS for table of contents */
  
  .d-toc {
    color: rgba(0,0,0,0.8);
    font-size: 0.8em;
    line-height: 1em;
  }
  
  .d-toc-header {
    font-size: 0.6rem;
    font-weight: 400;
    color: rgba(0, 0, 0, 0.5);
    text-transform: uppercase;
    margin-top: 0;
    margin-bottom: 1.3em;
  }
  
  .d-toc a {
    border-bottom: none;
  }
  
  .d-toc ul {
    padding-left: 0;
  }
  
  .d-toc li>ul {
    padding-top: 0.8em;
    padding-left: 16px;
    margin-bottom: 0.6em;
  }
  
  .d-toc ul,
  .d-toc li {
    list-style-type: none;
  }
  
  .d-toc li {
    margin-bottom: 0.9em;
  }
  
  .d-toc-separator {
    margin-top: 20px;
    margin-bottom: 2em;
  }
  
  .d-article-with-toc {
    border-top: none;
    padding-top: 0;
  }
  
  
  
  /* Tweak code blocks (note that this CSS is repeated above in an injection
     into the d-code shadow dom) */
  
  d-code {
    overflow-x: auto !important;
  }
  
  pre.d-code code.d-code {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }
  
  pre.text-output {
  
    font-size: 12px;
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  @media(min-width: 768px) {
  
  d-code {
    overflow-x: visible !important;
  }
  
  pre.d-code code.d-code  {
      padding-left: 18px;
      font-size: 14px;
  }
  pre.text-output {
    font-size: 14px;
  }
  }
  
  /* Figure */
  
  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }
  
  .figure img {
    width: 100%;
  }
  
  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }
  
  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }
  
  
  
  /* Tweak 1000px media break to show more text */
  
  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }
  
    .grid {
      grid-column-gap: 16px;
    }
  
    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }
  
  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }
  
    .grid {
      grid-column-gap: 32px;
    }
  }
  
  
  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */
  
  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }
  
  
  /* Social footer */
  
  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }
  
  .disqus-comments {
    margin-right: 30px;
  }
  
  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
  
  #disqus_thread {
    margin-top: 30px;
  }
  
  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }
  
  .article-sharing a:hover {
    border-bottom: none;
  }
  
  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .subscribe p {
    margin-bottom: 0.5em;
  }
  
  
  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }
  
  
  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .custom p {
    margin-bottom: 0.5em;
  }
  
  
  /* Improve display for browsers without grid (IE/Edge <= 15) */
  
  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }
  
  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }
  
  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }
  
  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }
  
  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }
  
  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
  
  
  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }
  
  .downlevel .footnotes ol {
    padding-left: 13px;
  }
  
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }
  
  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }
  
  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }
  
  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  </style>
  
  <script type="application/javascript">
  
  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }
  
  // show body when load is complete
  function on_load_complete() {
  
    // set body to visible
    document.body.style.visibility = 'visible';
  
    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }
  
    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }
  
  function init_distill() {
  
    init_common();
  
    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);
  
    // create d-title
    $('.d-title').changeElementType('d-title');
  
    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);
  
    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();
  
    // move posts container into article
    $('.posts-container').appendTo($('d-article'));
  
    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');
  
    // create d-bibliography
    var bibliography = $('<d-bibliography></d-bibliography>');
    $('#distill-bibliography').wrap(bibliography);
  
    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;
  
    // replace citations with <d-cite>
    $('.citation').each(function(i, val) {
      appendix = true;
      var cites = $(this).attr('data-cites').split(" ");
      var dt_cite = $('<d-cite></d-cite>');
      dt_cite.attr('key', cites.join());
      $(this).replaceWith(dt_cite);
    });
    // remove refs
    $('#refs').remove();
  
    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();
  
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-toc a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });
  
    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');
  
    // replace code blocks with d-code
    $('pre>code').each(function(i, val) {
      var code = $(this);
      var pre = code.parent();
      var clz = "";
      var language = pre.attr('class');
      if (language) {
        // map unknown languages to "clike" (without this they just dissapear)
        if ($.inArray(language, ["bash", "clike", "css", "go", "html",
                                 "javascript", "js", "julia", "lua", "markdown",
                                 "markup", "mathml", "python", "svg", "xml"]) == -1)
          language = "clike";
        language = ' language="' + language + '"';
        var dt_code = $('<d-code block' + language + clz + '></d-code>');
        dt_code.text(code.text());
        pre.replaceWith(dt_code);
      } else {
        code.addClass('text-output').unwrap().changeElementType('pre');
      }
    });
  
    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {
  
      // capture layout
      var layout = $(this).attr('data-layout');
  
      // apply layout to markdown level block elements
      var elements = $(this).children().not('d-code, pre.text-output, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });
  
  
      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });
  
    // load distill framework
    load_distill_framework();
  
    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {
  
      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;
  
      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');
  
      // table of contents
      if (have_authors) // adjust border if we are in authors
        $('.d-toc').parent().addClass('d-article-with-toc');
  
      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');
  
      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }
  
      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');
  
      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");
  
      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }
  
       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }
  
      // inject pre code styles (can't do this with a global stylesheet b/c a shadow root is used)
      $('d-code').each(function(i, val) {
        var style = document.createElement('style');
        style.innerHTML = 'pre code { padding-left: 10px; font-size: 12px; border-left: 2px solid rgba(0,0,0,0.1); } ' +
                          '@media(min-width: 768px) { pre code { padding-left: 18px; font-size: 14px; } }';
        if (this.shadowRoot)
          this.shadowRoot.appendChild(style);
      });
  
      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();
  
      // clear polling timer
      clearInterval(tid);
  
      // show body now that everything is ready
      on_load_complete();
    }
  
    var tid = setInterval(distill_post_process, 50);
    distill_post_process();
  
  }
  
  function init_downlevel() {
  
    init_common();
  
     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));
  
    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
  
    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();
  
    // remove toc
    $('.d-toc-header').remove();
    $('.d-toc').remove();
    $('.d-toc-separator').remove();
  
    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });
  
  
    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);
  
    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);
  
    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();
  
    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();
  
    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));
  
    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });
  
    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));
  
    $('body').addClass('downlevel');
  
    on_load_complete();
  }
  
  
  function init_common() {
  
    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};
  
        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });
  
        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);
  
    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});
  
    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });
  
      }
    });
  
    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });
  
    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }
  
    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');
  
    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");
  
    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();
  
    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }
  
  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });
  
  </script>
  
  <!--/radix_placeholder_distill-->
  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Roboto:300|Nunito+Sans:300|Inconsolata|Source+Sans+Pro|Work+Sans:200,300,400,500');
  /* controls general properties of site*/
  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    background-color: #FEFEFE;
  }
  
  d-title h1{
  font-size: 2.1em;
  font-family: 'Work Sans';
  font-weight: 300;
  }
  
  d-article h1 {
    font-size: 2.2em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 300;
  }
  
  d-article h2 {
    font-size: 1.65em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 500;
  }
  
  d-article h3 {
    margin-top: 1em;
  }
  d-byline .authors-affiliations {
    grid-column-end: span 3;
    grid-column-gap: 8px;
  }
  
  a.name:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }
  
  a.affiliation:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }
  
  blockquote p {
  	font-size: 1.3em;
  	font-family: 'Lato';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }
  
  d-code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    border-left: 5px solid #006ddb;
    border-top: 1px solid #B7B7B7;
    border-right: 1px solid #B7B7B7;
    border-bottom: 1px solid #B7B7B7;
  }
  
  code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.9em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }
  
  pre.text-output {
    margin-top: 0px;
     color: #333333;
     background-color: #FAFAFA;
     font-size: 0.8em
   }
  
  pre {
    border: 1px solid #ccc;
    padding: 0px 10px 10px 18px;
    margin: 1em 0px;
    border-radius: 5px;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    text-transform: none;
  }
  
  d-article li {
    margin: 0px 0px 4px;
  }
  
  /*************************************************
  *  code for side by side krona plot
  **************************************************/
  
  .column {
    float: left;
    width: 48%;
    padding: 5px;
  }
  
  /* Clear floats after image containers */
  .row::after {
    content: "";
    clear: both;
    display: table;
  }
  
  div.krona {
    width: 100%;
  	height: 90%;
    background-color: white;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    margin-bottom: 25px;
  	margin-top: 5px;
  }
  
  div.container-krona {
    text-align: center;
    padding: 1px 10px;
  	color: #414141;
  	font-size: 1.48em;
  	line-height: 0.8em;
  	font-family: 'Inconsolata';
  }
  
  div.container-krona p {
    margin: 20px;
  }
  
  div.krona img {
    border:1px solid white;
    margin: 25px;
  }
  
  img {
    margin: 25px;
  }
  
  /** LIGHTBOX MARKUP **/
  .thumbnail {
    max-width: 100%;
    position: relative;
  }
  
  .lightbox {
  	/** Default lightbox to hidden */
  	display: none;
  
  	/** Position and style */
  	position: fixed;
  	z-index: 999;
  	width: 100%;
  	height: 100%;
  	text-align: center;
  	top: 0;
  	left: 0;
  	background: rgba(0,0,0,0.8);
  }
  
  .lightbox img {
  	/** Pad the lightbox image */
  	position: fixed;
  	top: 0;
  	left: 0;
  	right: 0;
  	bottom: 0;
  	max-width: 0%;
  	max-height: 0%;
  	margin: auto;
  	box-sizing: border-box;
  }
  
  .lightbox:target {
  	/** Remove default browser outline */
  	outline: none;
  
  	/** Unhide lightbox **/
  	display: block;
  }
  
  .lightbox:target img {
  	max-height: 100%;
  	max-width: 100%;
  }
  
  .distill-site-header .logo img {
    margin-top: 10px;
    max-height: 30px;
    margin-bottom: 10px;
  }
  
  .distill-site-nav {
      color: rgba(255, 255, 255, 0.8);
      background-color: #143D59;
      font-size: 16px;
      font-weight: 300;
    }
  }
  
  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Field Analyses","description":"In this section we discuss the various field analyses conducted in this study.","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a class="logo" href="index.html">
<img src="assets/icon.png" alt="Logo"/>
</a>
<a href="index.html" class="title"></a>
</div>
<div class="nav-right">
<a href="index.html">
<i class="fa fa-home" aria-hidden="true"></i>
</a>
<a href="intro.html">Site Overview</a>
<a href="field.html">A. Field Analyses</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
B. 16S rRNA
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">16S rRNA Processing</span>
<a href="16s-dada2.html">1. DADA2</a>
<a href="16s-data-prep.html">2. Data Prep</a>
<hr/>
<span class="nav-dropdown-header">16S rRNA Analyses</span>
<a href="16s-water.html">3. Diversity</a>
<a href="16s-da-asvs.html">4. Differentially Abundant ASVs</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
C. Metagenomic
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Setup</span>
<a href="mg-setup.html">1. Working Envrionment</a>
<a href="mg-databases.html">2. Annotation Databases</a>
<hr/>
<span class="nav-dropdown-header">Read Processing</span>
<a href="mg-workflow-1.html">3. Assembly &amp; Annotations</a>
<a href="mg-workflow-2.html">4. Assembley &amp; Annotation Summary</a>
<hr/>
<span class="nav-dropdown-header">MAG Analysis</span>
<a href="mg-binning.html">5. Binning</a>
<a href="mag02-phylogenomics.html">6. MAG02 Phylogenomics</a>
<a href="mg-function.html">7. KEGG/KOFam Analysis</a>
</div>
</div>
<a href="data-availability.html">Data</a>
<a href="https://github.com/hypocolypse/web/">
<i class="fa fa-github fa-lg" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Field Analyses</h1>
<p><p>In this section we discuss the various field analyses conducted in this study.</p></p>
</div>


<div class="d-article">
<h3 class="d-toc-header">Table of Contents</h3>
<nav class="d-toc" id="TOC">
<ul>
<li><a href="#hypoxic-reef-conditions">Hypoxic Reef Conditions</a></li>
<li><a href="#coral-bleaching">Coral Bleaching</a></li>
<li><a href="#coral-community-structure">Coral Community Structure</a></li>
<li><a href="#analysis-workflow">Analysis Workflow</a><ul>
<li><a href="#pulse-amplitude-modulated-pam-fluorometry">Pulse Amplitude Modulated (PAM) Fluorometry</a><ul>
<li><a href="#non-transformed-data">Non-Transformed Data</a></li>
</ul></li>
<li><a href="#zooxanthellae">Zooxanthellae</a><ul>
<li><a href="#non-transformed-data-1">Non-Transformed Data</a></li>
<li><a href="#transformed-data">Transformed Data</a></li>
</ul></li>
<li><a href="#chlorophyll-a-chla">Chlorophyll a (Chla)</a><ul>
<li><a href="#non-transformed-data-2">Non-Transformed Data</a></li>
<li><a href="#transformed-data-1">Transformed Data</a></li>
</ul></li>
<li><a href="#chlorophyll-c-chlc">Chlorophyll c (Chlc)</a><ul>
<li><a href="#non-transformed-data-3">Non-Transformed Data</a></li>
<li><a href="#transformed-data-2">Transformed Data</a></li>
</ul></li>
</ul></li>
<li><a href="#figures">Figures</a><ul>
<li><a href="#source-code">Source Code</a></li>
</ul></li>
</ul>
</nav>
<hr class="d-toc-separator"/>
<h2 id="hypoxic-reef-conditions">Hypoxic Reef Conditions</h2>
<p>Conditions characteristic of a hypoxic event were first detected on Sept. 20, 2017 in Bahia Almirante on the Caribbean coast of Panama. To evaluate the environmental conditions associated with the event we conducted a series of depth profiles with a YSI multiparameter sonde, measuring dissolved oxygen (DO), temperature, chlorophyll, salinity, and pH. Measurements were taken at 83 sites across Bahía Almirante on Sept. 25, 2017. We selected two focal sites across a gradient of DO concentrations, from extremely hypoxic to normoxic (ambient DO concentrations). Cayo Roldan (hypoxic site) lies at the inner reaches of Bahia Almirante where water exchange with the open ocean is restricted, experience extreme hypoxia on the benthos during the event. Cayo Coral (normoxic site) is well-flushed with open ocean water and was used as a control site because oxygenated conditions persisted during the event.</p>
<h2 id="coral-bleaching">Coral Bleaching</h2>
<p>The common lettuce coral, <em>Agaricia tenuifolia</em>, was collected from each site during the hypoxic event and evaluated for bleaching through measurements of symbiont densities, pigment content, and pulse amplitude modulated (PAM) fluorometry. Three fragments from each of six colonies were collected from each site and returned to the laboratory facilities at the Smithsonian Tropical Research Institute’s Bocas del Toro Research Station. Differences in each response variable were evaluated with a linear mixed effects model, with colony as a random effect and size as a fixed effect.</p>
<h2 id="coral-community-structure">Coral Community Structure</h2>
<p>The hypoxic and normoxic sites are the location of permanent benthic monitoring. We evaluated changes in benthic community structure from four months before, during, and one year after the hypoxic event using permanent photoquadrats. At each site, permanent quadrats were photographed every 5m along one 50m permanent transect, and community structure was determined from 50 randomly overlaid points using the online platform <a href="https://coralnet.ucsd.edu/">CoralNet</a>. Taxa were identified to the finest resolution possible, and then grouped into different functional groups for subsequent analyses.</p>
<p>We used nonmetric multidimensional scaling (nMDS) to visually represent benthic community structure at each time point, and evaluated changes over time with a permutational analysis of variance (PERMANOVA), with pairwise contrasts. The benthic community did not change significantly over time at the normoxic site, so changes in the coral community over time were evaluated only at the hypoxic site.</p>
<h1 id="analysis-workflow">Analysis Workflow</h1>
<p>Next, we present the workflows we used to analyze coral response variables specifically the Pulse Amplitude Modulated (PAM) Fluorometry, Zooxanthellae, Chlorophyll a (Chla), and Chlorophyll C (Chlc) data. For each data set, we visually assess residuals, test for normality (Shapiro-Wilk Test), and test for equality of variances (Levene’s Test). In some cases, data are subsequently transformed.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
data &lt;- read.table(&quot;tables/field/coralhypoxia.csv&quot;, 
                   header = T, sep = &quot;,&quot;)
attach(data)
class(site)</code></pre>
<pre><code>
[1] &quot;character&quot;</code></pre>
<pre class="r"><code>
levels(site)</code></pre>
<pre><code>
NULL</code></pre>
<pre class="r"><code>
class(colony)</code></pre>
<pre><code>
[1] &quot;character&quot;</code></pre>
</div>
<h2 id="pulse-amplitude-modulated-pam-fluorometry">Pulse Amplitude Modulated (PAM) Fluorometry</h2>
<p>First, we test all data for assumptions of normality and homogeneity of variances based on raw data (i.e., not transformed) and residuals. Data are normal, variances hetergeneous (transform doesn’t improve). Look at distribution of data with histogram.</p>
<h3 id="non-transformed-data">Non-Transformed Data</h3>
<h4 id="visual-assessment">Visual Assessment</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
plotNormalHistogram(yield)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-2-1.png" width="624" /></p>
</div>
<p>First run the model to generate the residuals, then look at residuals to see if they are normally distributed; mixed model with fragment nested in colony, site as fixed effect.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
lmepam &lt;- lmer(yield ~ site + (1|colony), 
               REML = FALSE, na.action = na.exclude)
res1 &lt;- resid(lmepam)</code></pre>
</div>
<p>Produce a histogram of the residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
hist(res1, main = &quot;Yield&quot;, xlab = &quot;Residuals&quot;)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-4-1.png" width="624" /></p>
<pre class="r"><code>
plotNormalHistogram(res1)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-4-2.png" width="624" /></p>
</div>
<h4 id="shapiro-wilk-normality-test">Shapiro-Wilk Normality Test</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
pam.shapiro &lt;- shapiro.test(res1) #runs a normality test on residuals
print(pam.shapiro) # null = normally distrubuted P&lt;0.05 = non-normal</code></pre>
<pre><code>
    Shapiro-Wilk normality test

data:  res1
W = 0.97021, p-value = 0.4312</code></pre>
</div>
<p>QQ plots of residuals from lme model</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
qqnorm(res1, ylab = &quot;Sample Quantiles for PAM&quot;)
qqline(res1, col = &quot;red&quot;)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-6-1.png" width="624" /></p>
</div>
<h4 id="levenes-test">Levene’s Test</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
leveneTest(yield ~ as.factor(site), data = data)</code></pre>
<pre><code>
Levene&#39;s Test for Homogeneity of Variance (center = median)
      Df F value  Pr(&gt;F)   
group  1  7.9824 0.00785 **
      34                   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>
bartlett.test(x = yield, g = site)</code></pre>
<pre><code>
    Bartlett test of homogeneity of variances

data:  yield and site
Bartlett&#39;s K-squared = 6.3431, df = 1, p-value = 0.01178</code></pre>
</div>
<p>Data is normal but not homogenous—log transform does not improve.</p>
<h4 id="mixed-effects-models">Mixed Effects Models</h4>
<p>Site is the fixed effect, colony is random, fragment is random nested in colony. Compared full model with fragment dropped. Best fit model (with lowest AIC) was the model with colony dropped. But using model that includes fragment because it’s necessary random effect.</p>
<p>When comparing models with different random effects like here—use <code>REML = FALSE</code>. But when not comparing use <code>REML = TRUE</code> (default). The full model includes site, colony, and fragment nested in colony. Colony/frag includes a term for colony alone and nests fragment in colony.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
pam1 &lt;- lmer(yield ~ site + (1|colony/frag), data = data, 
             REML = FALSE, na.action = na.exclude)
ranova(pam1)</code></pre>
<pre><code>
ANOVA-like table for random-effects: Single term deletions

Model:
yield ~ site + (1 | frag:colony) + (1 | colony)
                  npar logLik     AIC LRT Df Pr(&gt;Chisq)
&lt;none&gt;               5 65.561 -121.12                  
(1 | frag:colony)    4 65.561 -123.12   0  1          1
(1 | colony)         4 65.561 -123.12   0  1          1</code></pre>
</div>
<p>Frag nested in colony dropped—I think colony accounts for fragment nested inherently?</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
pam2 &lt;- lmer(yield ~ site + (1|colony), data = data, 
             REML = FALSE, na.action = na.exclude)
ranova(pam2)</code></pre>
<pre><code>
ANOVA-like table for random-effects: Single term deletions

Model:
yield ~ site + (1 | colony)
             npar logLik     AIC                   LRT Df Pr(&gt;Chisq)
&lt;none&gt;          4 65.561 -123.12                                    
(1 | colony)    3 65.561 -125.12 -0.000000000000028422  1          1</code></pre>
</div>
<p>Use the full model to account for nesting fragments</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
mod.AIC &lt;- AIC(pam1, pam2) 
mod.AIC </code></pre>
<pre><code>
     df       AIC
pam1  5 -121.1218
pam2  4 -123.1218</code></pre>
</div>
<p>Final models uses REML—not comparing models with different random effects.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
finalpam &lt;- lmer(yield ~ site + (1|colony), 
                 data = data, 
                 REML = TRUE, na.action = na.exclude)</code></pre>
</div>
<p>p values for random effects.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
finalrandpam &lt;- ranova(finalpam)
pam.table &lt;- anova(finalpam, type = 2)
pam.table</code></pre>
<pre><code>
Type II Analysis of Variance Table with Satterthwaite&#39;s method
       Sum Sq  Mean Sq NumDF DenDF F value      Pr(&gt;F)    
site 0.056551 0.056551     1    34  34.827 0.000001156 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>
finalrandpam</code></pre>
<pre><code>
ANOVA-like table for random-effects: Single term deletions

Model:
yield ~ site + (1 | colony)
             npar logLik     AIC                   LRT Df Pr(&gt;Chisq)
&lt;none&gt;          4 58.057 -108.11                                    
(1 | colony)    3 58.057 -110.11 -0.000000000000014211  1          1</code></pre>
</div>
<h2 id="zooxanthellae">Zooxanthellae</h2>
<p>First testing all data for assumptions of normality and homogeneity of variances. Based on raw data and residuals, CR6B is an outlier and removed—log-transform data makes data normal and homogenous.</p>
<h3 id="non-transformed-data-1">Non-Transformed Data</h3>
<h4 id="visual-assessment-1">Visual Assessment</h4>
<p>Look at distribution of the raw data with histogram.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
attach(data)
plotNormalHistogram(zoox)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-13-1.png" width="624" /></p>
</div>
<p>First run the model to generate the residuals, then look at residuals to see if normally distributed. Mixed model with fragment nested in colony, site as fixed effect. Yield is normally distributed, but variances not homogenous.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
lmezoox &lt;- lmer(zoox ~ site + (1|colony), REML = FALSE, na.action = na.exclude)
res2 &lt;- resid(lmezoox)</code></pre>
</div>
<p>Produce a histogram of the residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
hist(res2, main = &quot;Zoox&quot;, xlab = &quot;Residuals&quot;)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-15-1.png" width="624" /></p>
<pre class="r"><code>
plotNormalHistogram(res2)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-15-2.png" width="624" /></p>
</div>
<h4 id="shapiro-wilk-normality-test-1">Shapiro-Wilk Normality Test</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
zoox.shapiro &lt;- shapiro.test(res2) #runs a normality test on residuals
print(zoox.shapiro) # null = normally distrubuted (P&lt;0.05 = non-normal</code></pre>
<pre><code>
    Shapiro-Wilk normality test

data:  res2
W = 0.9031, p-value = 0.004795</code></pre>
</div>
<p>QQ plots of residuals from lme model.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
qqnorm(res2, ylab = &quot;Sample Quantiles for Zoox&quot;)
qqline(res2, col = &quot;red&quot;)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-17-1.png" width="624" /></p>
</div>
<h4 id="levenes-test-1">Levene’s Test</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
leveneTest(zoox~ as.factor(site),data = data) # p = 0.0007486</code></pre>
<pre><code>
Levene&#39;s Test for Homogeneity of Variance (center = median)
      Df F value    Pr(&gt;F)    
group  1  13.805 0.0007486 ***
      33                      
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
<h3 id="transformed-data">Transformed Data</h3>
<p>Log transform improves normality—data are homogenous. Add a new column to dataframe for log transformation. Check that log transformed data is added as a new column.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
data$zoox_log = log(zoox)
head(data)</code></pre>
<pre><code>
   site colony frag      zoox     chla      chlc     yield zoox_log
1 Coral    One    A  411086.2 2.597060 0.5928972 0.6503333 12.92656
2 Coral    One    B  199325.3 1.235098 0.4116325 0.6453333 12.20269
3 Coral    One    C  411812.2 2.620320 0.6300991 0.6513333 12.92832
4 Coral    Two    A 1089420.7 4.557184 1.1952204 0.6393333 13.90116
5 Coral    Two    B  353670.7 1.852745 0.6181671 0.6213333 12.77612
6 Coral    Two    C  891722.1 2.815479 0.7507814 0.6383333 13.70091</code></pre>
<pre class="r"><code>
attach(data)</code></pre>
</div>
<h4 id="visual-assessment-2">Visual Assessment</h4>
<p>Check normality of log-transformed data</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
plotNormalHistogram(zoox_log)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-20-1.png" width="624" /></p>
</div>
<p>Check normality of log-transformed residuals</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
lmezoox_log &lt;- lmer(zoox_log ~ site + (1|colony), 
                    REML = FALSE, 
                    na.action = na.exclude)
res3 &lt;- resid(lmezoox_log)</code></pre>
</div>
<p>Produce a histogram of the residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
hist(res3, main = &quot;zoox log&quot;, xlab = &quot;Residuals&quot;)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-22-1.png" width="624" /></p>
<pre class="r"><code>
plotNormalHistogram(res3)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-22-2.png" width="624" /></p>
</div>
<h4 id="shapiro-wilk-normality-test-2">Shapiro-Wilk Normality Test</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
zooxlog.shapiro &lt;- shapiro.test(res3) #runs a normality test on residuals
print(zooxlog.shapiro) # null = normally distrubuted (P&lt;0.05 = non-normal</code></pre>
<pre><code>
    Shapiro-Wilk normality test

data:  res3
W = 0.96613, p-value = 0.3461</code></pre>
</div>
<p>QQ plots of residuals from lme model</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
qqnorm(res3, ylab = &quot;Sample Quantiles for Zooxlog&quot;)
qqline(res3, col = &quot;red&quot;)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-24-1.png" width="624" /></p>
</div>
<h4 id="levenes-test-2">Levene’s Test</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
leveneTest(zoox_log ~ as.factor(site), data = data) # p = 0.7968</code></pre>
<pre><code>
Levene&#39;s Test for Homogeneity of Variance (center = median)
      Df F value Pr(&gt;F)
group  1  0.0674 0.7968
      33               </code></pre>
</div>
<h4 id="mixed-effects-models-1">Mixed Effects Models</h4>
<p>Site is fixed effect, colony is random, fragment is random nested in colony. Use log-transformed data. Ccolony/frag includes a term for colony alone and nests fragment in colony.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
zoox1 &lt;- lmer(zoox_log ~ site + (1|colony/frag), 
              data = data, 
              REML = FALSE, 
              na.action = na.exclude) #AIC 72.8756
ranova(zoox1)</code></pre>
<pre><code>
ANOVA-like table for random-effects: Single term deletions

Model:
zoox_log ~ site + (1 | frag:colony) + (1 | colony)
                  npar  logLik    AIC     LRT Df Pr(&gt;Chisq)
&lt;none&gt;               5 -31.438 72.876                      
(1 | frag:colony)    4 -31.438 70.876 0.00000  1     1.0000
(1 | colony)         4 -31.584 71.167 0.29147  1     0.5893</code></pre>
</div>
<p>frag nested in colony dropped—I think colony accounts for fragment nested inherently?</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
zoox2 &lt;- lmer(zoox_log ~ site + (1|colony), 
              data = data, 
              REML = FALSE, 
              na.action = na.exclude) #AIC 70.8756
ranova(zoox2)</code></pre>
<pre><code>
ANOVA-like table for random-effects: Single term deletions

Model:
zoox_log ~ site + (1 | colony)
             npar  logLik    AIC     LRT Df Pr(&gt;Chisq)
&lt;none&gt;          4 -31.438 70.876                      
(1 | colony)    3 -31.584 69.167 0.29147  1     0.5893</code></pre>
<pre class="r"><code>
mod.AIC &lt;- AIC(zoox1, zoox2) 
mod.AIC ### use the full model to account for nesting fragments</code></pre>
<pre><code>
      df     AIC
zoox1  5 72.8756
zoox2  4 70.8756</code></pre>
</div>
<p>Final models uses REML—not comparing models with different random effects.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
finalzoox &lt;- lmer(zoox_log ~ site + (1|colony), 
                  data = data, 
                  REML = TRUE, 
                  na.action = na.exclude)</code></pre>
</div>
<p>P values for random effects</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
finalrandzoox &lt;- ranova(finalzoox)
zoox.table &lt;- anova(finalzoox, type = 2) #ANOVA table
zoox.table</code></pre>
<pre><code>
Type II Analysis of Variance Table with Satterthwaite&#39;s method
     Sum Sq Mean Sq NumDF  DenDF F value        Pr(&gt;F)    
site 20.468  20.468     1 27.954  59.465 0.00000002143 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>
finalrandzoox</code></pre>
<pre><code>
ANOVA-like table for random-effects: Single term deletions

Model:
zoox_log ~ site + (1 | colony)
             npar  logLik    AIC     LRT Df Pr(&gt;Chisq)
&lt;none&gt;          4 -33.335 74.670                      
(1 | colony)    3 -33.611 73.223 0.55316  1      0.457</code></pre>
</div>
<h2 id="chlorophyll-a-chla">Chlorophyll a (Chla)</h2>
<p>First testing all data for assumptions of normality and homogeneity of variances. Based on raw data and residuals</p>
<h3 id="non-transformed-data-2">Non-Transformed Data</h3>
<h4 id="visual-assessment-3">Visual Assessment</h4>
<p>Look at distribution of the raw data with histogram.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
plotNormalHistogram(chla)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-30-1.png" width="624" /></p>
</div>
<p>First run the model to generate the residuals, then look at residuals to see if normally distributed. Mixed model with fragment nested in colony, site as fixed effect. Raw data are normal but variances not homogenous—log-transformed improves homogeneity.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
lmechla &lt;- lmer(chla ~ site + (1|colony), 
                REML = FALSE, 
                na.action = na.exclude)
res4 &lt;- resid(lmechla)</code></pre>
</div>
<p>Produce a histogram of the residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
hist(res4, main = &quot;Chla&quot;, xlab = &quot;Residuals&quot;)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-32-1.png" width="624" /></p>
<pre class="r"><code>
plotNormalHistogram(res4)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-32-2.png" width="624" /></p>
</div>
<h4 id="shapiro-wilk-normality-test-3">Shapiro-Wilk Normality Test</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
chla.shapiro &lt;- shapiro.test(res4)
print(chla.shapiro) </code></pre>
<pre><code>
    Shapiro-Wilk normality test

data:  res4
W = 0.96483, p-value = 0.3015</code></pre>
</div>
<p>QQ plots of residuals from lme model</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
qqnorm(res4, ylab = &quot;Sample Quantiles for Zoox&quot;)
qqline(res4, col = &quot;red&quot;)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-34-1.png" width="624" /></p>
</div>
<h4 id="levenes-test-3">Levene’s Test</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
leveneTest(chla ~ as.factor(site), data = data) # p = 0.01606</code></pre>
<pre><code>
Levene&#39;s Test for Homogeneity of Variance (center = median)
      Df F value  Pr(&gt;F)  
group  1  6.4203 0.01606 *
      34                  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
<p>Data are normal but variances are not homogenous.</p>
<h3 id="transformed-data-1">Transformed Data</h3>
<p>Add a new column to dataframe for log transformation.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
data$chla_log = log(chla)</code></pre>
</div>
<p>Check that log transformed data is added as a new column</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
head(data)</code></pre>
<pre><code>
   site colony frag      zoox     chla      chlc     yield zoox_log
1 Coral    One    A  411086.2 2.597060 0.5928972 0.6503333 12.92656
2 Coral    One    B  199325.3 1.235098 0.4116325 0.6453333 12.20269
3 Coral    One    C  411812.2 2.620320 0.6300991 0.6513333 12.92832
4 Coral    Two    A 1089420.7 4.557184 1.1952204 0.6393333 13.90116
5 Coral    Two    B  353670.7 1.852745 0.6181671 0.6213333 12.77612
6 Coral    Two    C  891722.1 2.815479 0.7507814 0.6383333 13.70091
   chla_log
1 0.9543802
2 0.2111501
3 0.9632964
4 1.5167048
5 0.6166685
6 1.0351322</code></pre>
<pre class="r"><code>
attach(data)</code></pre>
</div>
<h4 id="visual-assessment-4">Visual Assessment</h4>
<p>Check normality of log-transformed data.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
plotNormalHistogram(chla_log)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-38-1.png" width="624" /></p>
</div>
<p>Check normality of log-transformed residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
lmechla_log &lt;- lmer(chla_log ~ site + (1|colony), 
                    REML = FALSE, 
                    na.action = na.exclude)
res7 &lt;- resid(lmechla_log)</code></pre>
</div>
<p>Produce a histogram of the residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
hist(res7, main = &quot;chla log&quot;, xlab = &quot;Residuals&quot;)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-40-1.png" width="624" /></p>
<pre class="r"><code>
plotNormalHistogram(res7)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-40-2.png" width="624" /></p>
</div>
<h4 id="shapiro-wilk-normality-test-4">Shapiro-Wilk Normality Test</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
chlalog.shapiro &lt;- shapiro.test(res7) #runs a normality test on residuals
print(chlalog.shapiro) # null = normally distrubuted (P&lt;0.05 = non-normal</code></pre>
<pre><code>
    Shapiro-Wilk normality test

data:  res7
W = 0.95611, p-value = 0.1627</code></pre>
</div>
<p>QQ plots of residuals from lme model</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
qqnorm(res7, ylab = &quot;Sample Quantiles for chlalog&quot;)
qqline(res7, col = &quot;red&quot;)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-42-1.png" width="624" /></p>
</div>
<h4 id="levenes-test-4">Levene’s Test</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
leveneTest(chla_log ~ as.factor(site), data = data) # p = 0.7487</code></pre>
<pre><code>
Levene&#39;s Test for Homogeneity of Variance (center = median)
      Df F value Pr(&gt;F)
group  1  0.1043 0.7487
      34               </code></pre>
</div>
<h4 id="mixed-model-effects">Mixed Model Effects</h4>
<p>Site is fixed effect, colony is random, fragment is random nested in colony on log transformed data. Colony/frag includes a term for colony alone and nests fragment in colony.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
chla_log1 &lt;- lmer(chla_log ~ site+(1|colony/frag), 
                  data = data,
                  REML = FALSE, 
                  na.action = na.exclude) #AIC 55.11567
ranova(chla_log1)</code></pre>
<pre><code>
ANOVA-like table for random-effects: Single term deletions

Model:
chla_log ~ site + (1 | frag:colony) + (1 | colony)
                  npar  logLik    AIC    LRT Df Pr(&gt;Chisq)
&lt;none&gt;               5 -22.558 55.116                     
(1 | frag:colony)    4 -22.558 53.116 0.0000  1     1.0000
(1 | colony)         4 -23.274 54.549 1.4333  1     0.2312</code></pre>
</div>
<p>Frag nested in colony dropped—I think colony accounts for fragment nested inherently?</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
chla_log2 &lt;- lmer(chla_log ~ site + (1|colony), 
                  data = data, 
                  REML = FALSE, 
                  na.action = na.exclude) #AIC 53.11567
ranova(chla_log2)</code></pre>
<pre><code>
ANOVA-like table for random-effects: Single term deletions

Model:
chla_log ~ site + (1 | colony)
             npar  logLik    AIC    LRT Df Pr(&gt;Chisq)
&lt;none&gt;          4 -22.558 53.116                     
(1 | colony)    3 -23.274 52.549 1.4333  1     0.2312</code></pre>
<pre class="r"><code>
mod.AIC &lt;- AIC(chla_log1, chla_log2) 
mod.AIC ### use the full model to account for nesting fragments</code></pre>
<pre><code>
          df      AIC
chla_log1  5 55.11567
chla_log2  4 53.11567</code></pre>
</div>
<p>Final models uses REML—not comparing models with different random effects.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
finalchla_log &lt;- lmer(chla_log ~ site + (1|colony), 
                      data = data, 
                      REML = TRUE, 
                      na.action = na.exclude)</code></pre>
</div>
<p>p values for random effects.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
finalchla_logrand &lt;- ranova(finalchla_log)
chla_log.table &lt;- anova(finalchla_log, type = 2) #ANOVA table
chla_log.table</code></pre>
<pre><code>
Type II Analysis of Variance Table with Satterthwaite&#39;s method
     Sum Sq Mean Sq NumDF DenDF F value       Pr(&gt;F)    
site  7.508   7.508     1    29  39.802 0.0000006851 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>
finalchla_logrand</code></pre>
<pre><code>
ANOVA-like table for random-effects: Single term deletions

Model:
chla_log ~ site + (1 | colony)
             npar  logLik    AIC   LRT Df Pr(&gt;Chisq)
&lt;none&gt;          4 -24.908 57.816                    
(1 | colony)    3 -25.843 57.687 1.871  1     0.1714</code></pre>
</div>
<h2 id="chlorophyll-c-chlc">Chlorophyll c (Chlc)</h2>
<p>First testing all data for assumptions of normality and homogeneity of variances. Based on raw data and residuals. Data need to be log transformed for normality and variances</p>
<h3 id="non-transformed-data-3">Non-Transformed Data</h3>
<h4 id="visual-assessment-5">Visual Assessment</h4>
<p>Look at distribution of the raw data with histogram.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
plotNormalHistogram(chlc)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-48-1.png" width="624" /></p>
</div>
<p>First run the model to generate the residuals then look at residuals to see if normally distributed mixed model with fragment nested in colony, site as fixed effect yield is normally distributed, but variances not homogenous</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
lmechlc &lt;- lmer(chlc ~ site + (1|colony), 
                REML = FALSE, 
                na.action = na.exclude)
res5 &lt;- resid(lmechlc)</code></pre>
</div>
<p>Produce a histogram of the residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
hist(res5,main = &quot;chlc&quot;, xlab = &quot;Residuals&quot;)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-50-1.png" width="624" /></p>
<pre class="r"><code>
plotNormalHistogram(res5)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-50-2.png" width="624" /></p>
</div>
<h4 id="shapiro-wilk-normality-test-5">Shapiro-Wilk Normality Test</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
chlc.shapiro &lt;- shapiro.test(res5)
print(chlc.shapiro)</code></pre>
<pre><code>
    Shapiro-Wilk normality test

data:  res5
W = 0.90427, p-value = 0.004475</code></pre>
</div>
<p>QQ plots of residuals from lme model</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
qqnorm(res5, ylab = &quot;Sample Quantiles for Zoox&quot;)
qqline(res5, col = &quot;red&quot;)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-52-1.png" width="624" /></p>
</div>
<h4 id="levenes-test-5">Levene’s Test</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
leveneTest(chlc ~ as.factor(site), data = data) # p = 0.01316</code></pre>
<pre><code>
Levene&#39;s Test for Homogeneity of Variance (center = median)
      Df F value  Pr(&gt;F)  
group  1  6.8454 0.01316 *
      34                  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
<p>Data not normally distributed or homogenous</p>
<h3 id="transformed-data-2">Transformed Data</h3>
<p>Log transform improves normality and homogeneity. Add a new column to dataframe for log transformation. Check that log transformed data is added as a new column</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
data$chlc_log = log(chlc)
head(data)</code></pre>
<pre><code>
   site colony frag      zoox     chla      chlc     yield zoox_log
1 Coral    One    A  411086.2 2.597060 0.5928972 0.6503333 12.92656
2 Coral    One    B  199325.3 1.235098 0.4116325 0.6453333 12.20269
3 Coral    One    C  411812.2 2.620320 0.6300991 0.6513333 12.92832
4 Coral    Two    A 1089420.7 4.557184 1.1952204 0.6393333 13.90116
5 Coral    Two    B  353670.7 1.852745 0.6181671 0.6213333 12.77612
6 Coral    Two    C  891722.1 2.815479 0.7507814 0.6383333 13.70091
   chla_log   chlc_log
1 0.9543802 -0.5227343
2 0.2111501 -0.8876244
3 0.9632964 -0.4618782
4 1.5167048  0.1783306
5 0.6166685 -0.4809965
6 1.0351322 -0.2866407</code></pre>
<pre class="r"><code>
attach(data)</code></pre>
</div>
<h4 id="visual-assessment-6">Visual Assessment</h4>
<p>Check normality of log-transformed data</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
plotNormalHistogram(chlc_log)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-55-1.png" width="624" /></p>
</div>
<p>Check normality of log-transformed residuals</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
lmechlc_log &lt;- lmer(chlc_log ~ site + (1|colony), 
                    REML = FALSE, 
                    na.action = na.exclude)
res6 &lt;- resid(lmechlc_log)</code></pre>
</div>
<p>Produce a histogram of the residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
hist(res6, main = &quot;chlc log&quot;,xlab = &quot;Residuals&quot;)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-57-1.png" width="624" /></p>
<pre class="r"><code>
plotNormalHistogram(res6)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-57-2.png" width="624" /></p>
</div>
<h4 id="shapiro-wilk-normality-test-6">Shapiro-Wilk Normality Test</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
chlclog.shapiro &lt;- shapiro.test(res6)
print(chlclog.shapiro)</code></pre>
<pre><code>
    Shapiro-Wilk normality test

data:  res6
W = 0.98465, p-value = 0.8874</code></pre>
</div>
<p>QQ plots of residuals from lme model</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
qqnorm(res6, ylab = &quot;Sample Quantiles for chlclog&quot;)
qqline(res6, col = &quot;red&quot;)</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-59-1.png" width="624" /></p>
</div>
<h4 id="levenes-test-6">Levene’s Test</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
leveneTest(chlc_log ~ as.factor(site), data = data) # p = 0.884</code></pre>
<pre><code>
Levene&#39;s Test for Homogeneity of Variance (center = median)
      Df F value Pr(&gt;F)
group  1  0.0216  0.884
      34               </code></pre>
</div>
<h4 id="mixed-model-effects-1">Mixed Model Effects</h4>
<p>Site is fixed effect, colony is random, fragment is random nested in colony. On log-transformed data. Colony/frag includes a term for colony alone and nests fragment in colony.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
chlc_log1 &lt;- lmer(chlc_log ~ site + (1|colony/frag), 
                  data = data, 
                  REML = FALSE, 
                  na.action = na.exclude) #AIC 64.52399
ranova(chlc_log1)</code></pre>
<pre><code>
ANOVA-like table for random-effects: Single term deletions

Model:
chlc_log ~ site + (1 | frag:colony) + (1 | colony)
                  npar  logLik    AIC    LRT Df Pr(&gt;Chisq)
&lt;none&gt;               5 -27.262 64.524                     
(1 | frag:colony)    4 -27.262 62.524 0.0000  1     1.0000
(1 | colony)         4 -27.833 63.665 1.1412  1     0.2854</code></pre>
</div>
<p>frag nested in colony dropped- I think colony accounts for fragment nested inherently?</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
chlc_log2 &lt;- lmer(chlc_log ~ site+(1|colony), 
                  data = data, 
                  REML = FALSE, 
                  na.action = na.exclude) #AIC 62.52399
ranova(chlc_log2)</code></pre>
<pre><code>
ANOVA-like table for random-effects: Single term deletions

Model:
chlc_log ~ site + (1 | colony)
             npar  logLik    AIC    LRT Df Pr(&gt;Chisq)
&lt;none&gt;          4 -27.262 62.524                     
(1 | colony)    3 -27.833 61.665 1.1412  1     0.2854</code></pre>
<pre class="r"><code>
mod.AIC &lt;- AIC(chlc_log1, chlc_log2) 
mod.AIC ### use the full model to account for nesting fragments</code></pre>
<pre><code>
          df      AIC
chlc_log1  5 64.52399
chlc_log2  4 62.52399</code></pre>
</div>
<p>Final models uses REML- not comparing models with different random effects</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
finalchlc_log &lt;- lmer(chlc_log ~ site + (1|colony), 
                      data = data, 
                      REML = TRUE, 
                      na.action = na.exclude)</code></pre>
</div>
<p>p values for random effects</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
finalchlc_logrand &lt;- ranova(finalchlc_log)
chlc_log.table &lt;- anova(finalchlc_log, type = 2) #ANOVA table
chlc_log.table</code></pre>
<pre><code>
Type II Analysis of Variance Table with Satterthwaite&#39;s method
     Sum Sq Mean Sq NumDF DenDF F value     Pr(&gt;F)    
site 6.1918  6.1918     1    29  24.977 0.00002553 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>
finalchlc_logrand</code></pre>
<pre><code>
ANOVA-like table for random-effects: Single term deletions

Model:
chlc_log ~ site + (1 | colony)
             npar  logLik    AIC    LRT Df Pr(&gt;Chisq)
&lt;none&gt;          4 -29.375 66.749                     
(1 | colony)    3 -30.148 66.297 1.5476  1     0.2135</code></pre>
</div>
<h1 id="figures">Figures</h1>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
all &lt;- read.table(&quot;tables/field/coralhypoxia.csv&quot;, header = T, sep = &quot;,&quot;)
attach(all)</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
ggplot(all,aes(x=site, y=yield))+ #group tells line what to connect
  geom_boxplot()+ #make box
  labs(x=&quot;Site&quot;,y=&#39;Maximum Quantum Yield&#39;)+ 
  theme_classic()+ #makes look normal
  #scale_fill_manual(values = c(&quot;steelblue&quot;,&quot;darkred&quot;))+
  theme(text = element_text(size=14))+ #change text size
  theme(axis.text.x = element_text(colour=&quot;black&quot;, angle=0, hjust=1), #make all text black ## to put x axis label at angle: angle=45, hjust=1), 
        axis.text.y = element_text(colour=&quot;black&quot;))+
  scale_y_continuous(expand = c(0,0),
                     limits = c(0.4,0.75)) #set y axis</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-66-1.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
ggplot(all,aes(x=site, y=yield, color=colony))+ #colors the dots by colony
  geom_point(size=3)+ 
  labs(x=&quot;Site&quot;,y=&#39;Maximum Quantum Yield&#39;)+ 
  theme_classic()+ #makes look normal
  #scale_fill_manual(values = c(&quot;steelblue&quot;,&quot;darkred&quot;))+
  theme(text = element_text(size=14))+ #change text size
  theme(axis.text.x = element_text(colour=&quot;black&quot;, angle=0, hjust=1), #make all text black ## to put x axis label at angle: angle=45, hjust=1), 
        axis.text.y = element_text(colour=&quot;black&quot;))+
  scale_y_continuous(expand = c(0,0),
                     limits = c(0.4,0.75)) #set y axis</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-67-1.png" width="624" /></p>
</div>
<p>Calculate Means for Figures</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
means.all&lt;-ddply(all,c(&quot;site&quot;,&quot;colony&quot;),summarise,
                 N    = sum(!is.na(yield)),
                 yield_mean = mean(yield, na.rm=TRUE),
                 zoox_mean = mean(zoox, na.rm=TRUE),
                 chla_mean = mean(chla, na.rm=TRUE),
                 chlc_mean = mean(chlc, na.rm=TRUE),
                 sd_yield = sd(yield),
                 sd_zoox = sd(zoox),
                 sd_chla = sd(chla),
                 sd_chlc = sd(chlc),
                 yield_se = (sd(yield)/sqrt(N)),
                 zoox_se = (sd(zoox)/sqrt(N)),
                 chla_se = (sd(chla)/sqrt(N)),
                 chlc_se = (sd(chlc)/sqrt(N)))
                
head(means.all)</code></pre>
<pre><code>
   site colony N yield_mean zoox_mean chla_mean chlc_mean   sd_yield
1 Coral   Five 3  0.6531111  713619.3  3.495386 1.1783720 0.01266813
2 Coral   Four 3  0.6695556  741780.0  5.199911 1.9368219 0.01281203
3 Coral    One 3  0.6490000  340741.2  2.150826 0.5448762 0.00321455
4 Coral    Six 3  0.6153333 1360692.9  6.017438 1.9504259 0.02088327
5 Coral  Three 3  0.6686944 1295987.7  3.830870 1.1553710 0.04261458
6 Coral    Two 3  0.6330000  778271.2  3.075136 0.8547230 0.01011599
   sd_zoox   sd_chla   sd_chlc    yield_se   zoox_se   chla_se
1 192640.8 0.8182351 0.3588934 0.007313947 111221.23 0.4724083
2 221822.6 1.3905156 0.6624581 0.007397030 128069.35 0.8028146
3 122470.3 0.7931292 0.1168821 0.001855921  70708.27 0.4579133
4 418268.7 1.9781461 0.8402899 0.012056964 241487.56 1.1420832
5 734473.8 0.4410107 0.4475100 0.024603538 424048.67 0.2546176
6 380769.4 1.3707892 0.3022425 0.005840472 219837.32 0.7914255
    chlc_se
1 0.2072072
2 0.3824703
3 0.0674819
4 0.4851416
5 0.2583700
6 0.1744998</code></pre>
<pre class="r"><code>
write.csv(means.all, file = &quot;tables/field/allcoralmeans.csv&quot;, row.names=FALSE)</code></pre>
</div>
<h4 id="colony-means">Colony Means</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
ggplot(means.all,aes(x=site, y=yield_mean,fill=site)) +
  geom_boxplot() +
  labs(x = &quot;Site&quot;, y = &#39;Maximum Quantum Yield&#39;) + 
  theme_classic()+ #makes look normal
  scale_fill_manual(values = c(&quot;steelblue&quot;,&quot;darkred&quot;))+
  theme(text = element_text(size=14))+ #change text size
  theme(axis.text.x = element_text(colour=&quot;black&quot;, angle=0, hjust=1),
        axis.text.y = element_text(colour=&quot;black&quot;))+
  scale_y_continuous(expand = c(0,0),
                     limits = c(0.4,0.75)) #set y axis</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-69-1.png" width="624" /></p>
</div>
<p>set a jitter object to offset points and assign data to ggplot</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
jitter &lt;- position_jitter(width = 0.1)  
yield_plot &lt;- ggplot(means.all, aes(x = site, y = yield_mean))</code></pre>
</div>
<p>make it the kind of plot you want- this makes a box plot colored by site, then overlays a scatterplot of all the data (in this case the colony means) Close to publication quality :)</p>
<h4 id="maximum-quantum-yield-by-site-with-colony-means">Maximum Quantum Yield by site with colony means</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
yp &lt;- yield_plot+
  geom_boxplot(color=&quot;black&quot;,
               aes(fill=site),
               width=0.4)+
  scale_fill_manual(values=c(&quot;skyblue3&quot;,&quot;coral3&quot;), name= NULL, labels=c(&quot;Normoxic&quot;,&quot;Hypoxic&quot;))+
  geom_point(size=2, position=jitter)+ #aes(color=colony))+
  labs(x=&quot;Site&quot;,y=&quot;Maximum Quantum Yield&quot;)+
  theme_classic()+
  theme(legend.position = c(0.85,0.85),
        legend.title= element_text(color=&quot;transparent&quot;),
        axis.text.x = element_text(colour=&quot;black&quot;), #make all text black ## to put x axis label at angle: angle=45, hjust=1), 
        axis.text.y = element_text(colour=&quot;black&quot;),
        axis.title.y = element_text(margin = margin(t=0, r=10,b=0,l=0)),
        panel.border = element_rect(colour = &quot;black&quot;, fill=NA, size=.5))+
   # modifies the legend titletheme(text = element_text(size=14),     panel.background = element_rect(fill = &quot;transparent&quot;))+ # bg of the panel
  theme(text = element_text(size=12))+
  scale_x_discrete(labels=c(&quot;Coral&quot; = &quot;Coral Cay&quot;,&quot;Roldan&quot; = &quot;Cayo Roldan&quot;))+
  scale_y_continuous(expand = c(0,0), limits = c(0.5,0.7))</code></pre>
</div>
<h4 id="zooxanthellae-counts-by-site-with-colony-means">Zooxanthellae Counts by site with colony means</h4>
<p>assign data to ggplot</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
yield_zoox &lt;- ggplot(means.all, aes(x=site, y=zoox_mean))

zp &lt;- yield_zoox+
  geom_boxplot(color=&quot;black&quot;,
               aes(fill=site),
               width=0.4)+
  scale_fill_manual(values=c(&quot;skyblue3&quot;,&quot;coral3&quot;), name= NULL, labels=c(&quot;Normoxic&quot;,&quot;Hypoxic&quot;))+
  geom_point(size=2, position=jitter)+ #aes(color=colony))+
  labs(x=&quot;Site&quot;,
       y=(&quot;Endosymbiont Density (cells cm^-2)&quot;))+
  theme_classic()+
  theme(legend.position = &quot;none&quot;, #
        #c(0.8,0.875),
        #legend.title= element_text(color=&quot;transparent&quot;),
        axis.text.x = element_text(colour=&quot;black&quot;), #make all text black ## to put x axis label at angle: angle=45, hjust=1), 
        axis.text.y = element_text(colour=&quot;black&quot;),
        axis.title.y = element_text(margin = margin(t=0, r=10,b=0,l=0)),
        panel.border = element_rect(colour = &quot;black&quot;, fill=NA, size=.5))+
  # modifies the legend titletheme(text = element_text(size=14),     panel.background = element_rect(fill = &quot;transparent&quot;))+ # bg of the panel
  theme(text = element_text(size=12))+
  scale_x_discrete(labels=c(&quot;Coral&quot; = &quot;Coral Cay&quot;,&quot;Roldan&quot; = &quot;Cayo Roldan&quot;))+
  scale_y_continuous(labels = scientific,
                     breaks=seq(0,1.5E6, 4.5E5),
                      expand = c(0,0), 
                     limits = c(0,1.5E6))
zp  </code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-72-1.png" width="624" /></p>
</div>
<h4 id="pigment-data">Pigment Data</h4>
<p>Convert Pigment data to long format with Chla and Chlc Stacked, so can put in same figures</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
pigs &lt;- tidyr::gather(means.all, pigment, concentration,chla_mean, chlc_mean)
head(pigs)</code></pre>
<pre><code>
   site colony N yield_mean zoox_mean   sd_yield  sd_zoox   sd_chla
1 Coral   Five 3  0.6531111  713619.3 0.01266813 192640.8 0.8182351
2 Coral   Four 3  0.6695556  741780.0 0.01281203 221822.6 1.3905156
3 Coral    One 3  0.6490000  340741.2 0.00321455 122470.3 0.7931292
4 Coral    Six 3  0.6153333 1360692.9 0.02088327 418268.7 1.9781461
5 Coral  Three 3  0.6686944 1295987.7 0.04261458 734473.8 0.4410107
6 Coral    Two 3  0.6330000  778271.2 0.01011599 380769.4 1.3707892
    sd_chlc    yield_se   zoox_se   chla_se   chlc_se   pigment
1 0.3588934 0.007313947 111221.23 0.4724083 0.2072072 chla_mean
2 0.6624581 0.007397030 128069.35 0.8028146 0.3824703 chla_mean
3 0.1168821 0.001855921  70708.27 0.4579133 0.0674819 chla_mean
4 0.8402899 0.012056964 241487.56 1.1420832 0.4851416 chla_mean
5 0.4475100 0.024603538 424048.67 0.2546176 0.2583700 chla_mean
6 0.3022425 0.005840472 219837.32 0.7914255 0.1744998 chla_mean
  concentration
1      3.495386
2      5.199911
3      2.150826
4      6.017438
5      3.830870
6      3.075136</code></pre>
<pre class="r"><code>
pigments &lt;- dplyr::select(pigs,site,colony,pigment,concentration)
head(pigments)</code></pre>
<pre><code>
   site colony   pigment concentration
1 Coral   Five chla_mean      3.495386
2 Coral   Four chla_mean      5.199911
3 Coral    One chla_mean      2.150826
4 Coral    Six chla_mean      6.017438
5 Coral  Three chla_mean      3.830870
6 Coral    Two chla_mean      3.075136</code></pre>
</div>
<p>assign data to ggplot data frame “pigments”</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
pig_plot &lt;- ggplot(pigments,
                     aes(x=pigment, y=concentration, fill=site))</code></pre>
</div>
<p>Shows chl a and chl c on the x axis with site color coded</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
pig_plot+
  geom_boxplot(color=&quot;black&quot;)+
  scale_fill_manual(values=c(&quot;skyblue3&quot;,&quot;coral3&quot;), name= NULL, labels=c(&quot;Normoxic&quot;,&quot;Hypoxic&quot;))</code></pre>
<p><img src="field_files/figure-html5/unnamed-chunk-75-1.png" width="624" /></p>
</div>
<h4 id="chlorophyll-a-chla-plots-only">Chlorophyll a (Chla) Plots only</h4>
<p>assign data to ggplot set a jitter object to offset points</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
jitter2 &lt;- position_jitter(width=0.05)  

chla_plot &lt;- ggplot(means.all,
                     aes(x=site, y=chla_mean))</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
cp &lt;- chla_plot+
  geom_boxplot(color=&quot;black&quot;,
               aes(fill=site),
               width=0.4)+
  scale_fill_manual(values=c(&quot;skyblue3&quot;,&quot;coral3&quot;), name= NULL, labels=c(&quot;Normoxic&quot;,&quot;Hypoxic&quot;))+
  geom_point(size=2, position=jitter)+ #aes(color=colony))+
  labs(x=&quot;Site&quot;,
       y=(&quot;Chl a Concentration (ug cm^-2)&quot;))+
  theme_classic()+
  theme(legend.position = &quot;none&quot;,
        #c(0.8,0.875),
        #legend.title= element_text(color=&quot;transparent&quot;),
        axis.text.x = element_text(colour=&quot;black&quot;), #make all text black ## to put x axis label at angle: angle=45, hjust=1), 
        axis.text.y = element_text(colour=&quot;black&quot;),
        axis.title.y = element_text(margin = margin(t=0, r=10,b=0,l=0)),
        panel.border = element_rect(colour = &quot;black&quot;, fill=NA, size=.5))+
  # modifies the legend titletheme(text = element_text(size=14),     panel.background = element_rect(fill = &quot;transparent&quot;))+ # bg of the panel
  theme(text = element_text(size=12))+
  scale_x_discrete(labels=c(&quot;Coral&quot; = &quot;Coral Cay&quot;,&quot;Roldan&quot; = &quot;Cayo Roldan&quot;))+
  scale_y_continuous(expand = c(0,0), 
                     limits = c(0,6))</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body-outset">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Fine tuning figures for the panel plot</p>
<h4 id="maximum-quantum-yield-by-site-with-colony-means-1">Maximum Quantum Yield by site with colony means</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
ypp &lt;- yield_plot +
  geom_boxplot(color = &quot;black&quot;, aes(fill = site), width = 0.4) +
  scale_fill_manual(values = c(&quot;skyblue3&quot;, &quot;coral3&quot;), name = NULL, 
                    labels = c(&quot;Normoxic&quot;, &quot;Hypoxic&quot;)) +
  geom_point(size = 1, position = jitter) + #aes(color=colony))+
  labs(x = &quot;Site&quot;, y = (&quot;Maximum Quantum Yield&quot;)) +
  theme_classic() +
  theme(legend.position = &quot;top&quot;, legend.direction = &quot;vertical&quot;, #
        legend.title= element_text(color = &quot;transparent&quot;),
        #axis.text.x = element_blank(),
        axis.text.x = element_text(size = 7, colour=&quot;black&quot;), 
        axis.text.y = element_text(size = 8, colour = &quot;black&quot;),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size = 9, margin = margin(t = 0, r = 10, b = 0, l = 0)),
        panel.border = element_rect(colour = &quot;black&quot;, fill = NA, size = 0.5)) +
  theme(text = element_text(size = 12)) +
  scale_x_discrete(labels=c(&quot;Coral&quot; = &quot;Coral Cay&quot;, &quot;Roldan&quot; = &quot;Cayo Roldan&quot;)) +
  scale_y_continuous(expand = c(0,0), limits = c(0.5,0.7))</code></pre>
</div>
<h4 id="zooxanthellae-counts-by-site-with-colony-means-1">Zooxanthellae Counts by site with colony means</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
yzield_zoox &lt;- ggplot(means.all, aes(x = site, y = zoox_mean))

zpp &lt;- yield_zoox+
  geom_boxplot(color=&quot;black&quot;,
               aes(fill=site),
               width=0.4)+
  scale_fill_manual(values=c(&quot;skyblue3&quot;,&quot;coral3&quot;), name= NULL, labels=c(&quot;Normoxic&quot;,&quot;Hypoxic&quot;))+
  geom_point(size=1, position=jitter)+ #aes(color=colony))+
  labs(x = &quot;Site&quot;, y=(&quot;Endosymbiont Density (cells cm^-2)&quot;))+
  theme_classic() +
  theme(legend.position = &quot;none&quot;, #
        #c(0.8,0.875),
        #legend.title= element_text(color=&quot;transparent&quot;),
        axis.text.x = element_text(size = 7, colour=&quot;black&quot;), 
        axis.title.x = element_text(size = 12),
        #axis.title.x = element_blank(),
        #axis.text.x = element_blank(), 
        axis.text.y = element_text(size = 8, colour=&quot;black&quot;),
        axis.title.y = element_text(size = 9, margin = margin(t=0, r=10,b=0,l=0)),
        panel.border = element_rect(colour = &quot;black&quot;, fill=NA, size=.5)) +
  theme(text = element_text(size=12))+
  scale_x_discrete(labels=c(&quot;Coral&quot; = &quot;Coral Cay&quot;,&quot;Roldan&quot; = &quot;Cayo Roldan&quot;))+
  scale_y_continuous(labels = scientific,
                     breaks=seq(0,1.5E6, 4.5E5),
                     expand = c(0,0), 
                     limits = c(0,1.5E6))</code></pre>
</div>
<h4 id="chlorophyll-a-chla-1">Chlorophyll a (Chla)</h4>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
chla_plot &lt;- ggplot(means.all, aes(x = site, y = chla_mean))

cpp &lt;- chla_plot +
  geom_boxplot(color=&quot;black&quot;, aes(fill=site), width=0.4)+
  scale_fill_manual(values = c(&quot;skyblue3&quot;,&quot;coral3&quot;), 
                    name = NULL, labels = c(&quot;Normoxic&quot;, &quot;Hypoxic&quot;)) +
  geom_point(size=1, position=jitter) + 
  labs(x = &quot;Site&quot;, y = (&quot;Chl a Concentration (ug cm^-2)&quot;)) +
  theme_classic() +
  theme(legend.position = &quot;none&quot;,
        #legend.title= element_text(color=&quot;transparent&quot;),
        axis.text.x = element_text(size = 7, colour=&quot;black&quot;), 
        axis.text.y = element_text(size = 8, colour=&quot;black&quot;),
        axis.title.y = element_text(size = 9, margin = margin(t = 0, r = 10, b = 0, l = 0)),
        axis.title.x = element_blank(),
        #axis.title.x = element_text(size = 12),
        panel.border = element_rect(colour = &quot;black&quot;, fill=NA, size=.5)) +
  theme(text = element_text(size=12))+
  scale_x_discrete(labels=c(&quot;Coral&quot; = &quot;Coral Cay&quot;,&quot;Roldan&quot; = &quot;Cayo Roldan&quot;))+
  scale_y_continuous(expand = c(0,0), 
                     limits = c(0,6))</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="field_files/figure-html5/unnamed-chunk-83-1.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="source-code" class="appendix">Source Code</h2>
<p>The source code for this page can be accessed on GitHub by <a href="https://github.com/hypocolypse/web/blob/master/field.Rmd">clicking this link</a>.</p>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/hypocolypse/web/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/hypocolypse/web/">https://github.com/hypocolypse/web/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<script id="distill-bibliography" type="text/bibtex">
@article{callahan2016dada2,
  title={DADA2: high-resolution sample inference from Illumina amplicon data},
  author={Callahan, Benjamin J and McMurdie, Paul J and Rosen, Michael J and Han, Andrew W and Johnson, Amy Jo A and Holmes, Susan P},
  journal={Nature Methods},
  volume={13},
  number={7},
  pages={581},
  year={2016},
  url = {https://doi.org/10.1038/nmeth.3869},
}

@article{martin2011cutadapt,
  title={Cutadapt removes adapter sequences from high-throughput sequencing reads},
  author={Martin, Marcel},
  journal={EMBnet. journal},
  volume={17},
  number={1},
  pages={10--12},
  year={2011},
  url = {https://doi.org/10.14806/ej.17.1.200}
}

@article{quast2012silva,
  title={The SILVA ribosomal RNA gene database project: improved data processing and web-based tools},
  author={Quast, Christian and Pruesse, Elmar and Yilmaz, Pelin and Gerken, Jan and Schweer, Timmy and Yarza, Pablo and Peplies, J{\"o}rg and Gl{\"o}ckner, Frank Oliver},
  journal={Nucleic Acids Research},
  volume={41},
  number={D1},
  pages={D590--D596},
  year={2012},
  url = {https://doi.org/10.1093/nar/gks1219}
}

@article{mcmurdie2013phyloseq,
  title={phyloseq: an R package for reproducible interactive analysis and graphics of microbiome census data},
  author={McMurdie, Paul J and Holmes, Susan},
  journal={PLoS One},
  volume={8},
  number={4},
  pages={e61217},
  year={2013},
  url = {https://doi.org/10.1371/journal.pone.0061217}
}

@misc{wong2011points,
  title={Points of view: Color blindness},
  author={Wong, Bang},
  journal={Nature Methods},
  volume={8},
  number={6},
  pages={441},
  year={2011},
  url = {https://doi.org/10.1038/nmeth.1618}
}

@misc{lahti2017microbiome,
  author = {Lahti, Leo and Shetty Sudarshan and others},
  title = {Tools for microbiome analysis in R. Version 1.9.97},
  year = {2017},
  url = {https://github.com/microbiome/microbiome},
}

@article{eren2015anvi,
  title={Anvi’o: an advanced analysis and visualization platform for ‘omics data},
  author={Eren, A Murat and Esen, {\"O}zcan C and Quince, Christopher and Vineis, Joseph H and Morrison, Hilary G and Sogin, Mitchell L and Delmont, Tom O},
  journal={PeerJ},
  volume={3},
  pages={e1319},
  year={2015},
  url = {https://doi.org/10.7717/peerj.1319}
}

@article{bolger2014trimmomatic,
  title={Trimmomatic: a flexible trimmer for Illumina sequence data},
  author={Bolger, Anthony M and Lohse, Marc and Usadel, Bjoern},
  journal={Bioinformatics},
  volume={30},
  number={15},
  pages={2114--2120},
  year={2014},
  url = {https://doi.org/10.1093/bioinformatics/btu170}
}

@article{koster2012snakemake,
  title={Snakemake—a scalable bioinformatics workflow engine},
  author={K{\"o}ster, Johannes and Rahmann, Sven},
  journal={Bioinformatics},
  volume={28},
  number={19},
  pages={2520--2522},
  year={2012},
  url = {https://doi.org/10.1093/bioinformatics/bts480}
}

@article{menzel2016fast,
  title={Fast and sensitive taxonomic classification for metagenomics with Kaiju},
  author={Menzel, Peter and Ng, Kim Lee and Krogh, Anders},
  journal={Nature Communications},
  volume={7},
  pages={11257},
  year={2016},
  url = {https://doi.org/10.1038/ncomms11257}
}

@article{roux2015virsorter,
  title={VirSorter: mining viral signal from microbial genomic data},
  author={Roux, Simon and Enault, Francois and Hurwitz, Bonnie L and Sullivan, Matthew B},
  journal={PeerJ},
  volume={3},
  pages={e985},
  year={2015},
  url = {https://doi.org/10.1093/bioinformatics/btu170}
}

@article{kim2016centrifuge,
  title={Centrifuge: rapid and sensitive classification of metagenomic sequences},
  author={Kim, Daehwan and Song, Li and Breitwieser, Florian P and Salzberg, Steven L},
  journal={Genome Research},
  volume={26},
  number={12},
  pages={1721--1729},
  year={2016},
  url = {https://doi.org/10.1101/gr.210641.116.}
}

@article{hyatt2010prodigal,
  title={Prodigal: prokaryotic gene recognition and translation initiation site identification},
  author={Hyatt, Doug and Chen, Gwo-Liang and LoCascio, Philip F and Land, Miriam L and Larimer, Frank W and Hauser, Loren J},
  journal={BMC Bioinformatics},
  volume={11},
  number={1},
  pages={119},
  year={2010},
  url = {https://doi.org/10.1186/1471-2105-11-119}
}

@article{edgar2004muscle,
  title={MUSCLE: multiple sequence alignment with high accuracy and high throughput},
  author={Edgar, Robert C},
  journal={Nucleic Acids Research},
  volume={32},
  number={5},
  pages={1792--1797},
  year={2004},
  url = {https://doi.org/10.1093/nar/gkh340}
}

@article{li2015megahit,
  title={MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph},
  author={Li, Dinghua and Liu, Chi-Man and Luo, Ruibang and Sadakane, Kunihiko and Lam, Tak-Wah},
  journal={Bioinformatics},
  volume={31},
  number={10},
  pages={1674--1676},
  year={2015},
  url = {https://doi.org/10.1093/bioinformatics/btv033}
}

@article{langmead2012fast,
  title={Fast gapped-read alignment with Bowtie 2},
  author={Langmead, Ben and Salzberg, Steven L},
  journal={Nature Methods},
  volume={9},
  number={4},
  pages={357},
  year={2012},
  url = {https://doi.org/10.1038/nmeth.1923}
}

@article{li2009fast,
  title={Fast and accurate short read alignment with Burrows--Wheeler transform},
  author={Li, Heng and Durbin, Richard},
  journal={Bioinformatics},
  volume={25},
  number={14},
  pages={1754--1760},
  year={2009},
  url = {https://doi.org/10.1093/bioinformatics/btp324}
}

@article{li2009sequence,
  title={The sequence alignment/map format and SAMtools},
  author={Li, Heng and Handsaker, Bob and Wysoker, Alec and Fennell, Tim and Ruan, Jue and Homer, Nils and Marth, Gabor and Abecasis, Goncalo and Durbin, Richard},
  journal={Bioinformatics},
  volume={25},
  number={16},
  pages={2078--2079},
  year={2009},
  url = {https://doi.org/10.1093/bioinformatics/btp352}
}

@article{breitwieser2018krakenuniq,
  title={KrakenUniq: confident and fast metagenomics classification using unique k-mer counts},
  author={Breitwieser, FP and Baker, DN and Salzberg, Steven L},
  journal={Genome Biology},
  volume={19},
  number={1},
  pages={1--10},
  year={2018},
  url = {https://doi.org/10.5281/zenodo.1412647}
}

@article{eren2013filtering,
  title={A filtering method to generate high quality short reads using Illumina paired-end technology},
  author={Eren, A Murat and Vineis, Joseph H and Morrison, Hilary G and Sogin, Mitchell L},
  journal={PLoS One},
  volume={8},
  number={6},
  year={2013},
  url = {https://doi.org/10.1371/journal.pone.0066643}
}

@article{alneberg2014binning,
  title={Binning metagenomic contigs by coverage and composition},
  author={Alneberg, Johannes and Bjarnason, Brynjar Sm{\'a}ri and De Bruijn, Ino and Schirmer, Melanie and Quick, Joshua and Ijaz, Umer Z and Lahti, Leo and Loman, Nicholas J and Andersson, Anders F and Quince, Christopher},
  journal={Nature Methods},
  volume={11},
  number={11},
  pages={1144--1146},
  year={2014},
  url = {https://doi.org/10.1038/nmeth.3103}
}

@article{sieber2018recovery,
  title={Recovery of genomes from metagenomes via a dereplication, aggregation and scoring strategy},
  author={Sieber, Christian MK and Probst, Alexander J and Sharrar, Allison and Thomas, Brian C and Hess, Matthias and Tringe, Susannah G and Banfield, Jillian F},
  journal={Nature Microbiology},
  volume={3},
  number={7},
  pages={836--843},
  year={2018},
}

@article{kang2019metabat,
  title={MetaBAT 2: an adaptive binning algorithm for robust and efficient genome reconstruction from metagenome assemblies},
  author={Kang, Dongwan D and Li, Feng and Kirton, Edward and Thomas, Ashleigh and Egan, Rob and An, Hong and Wang, Zhong},
  journal={PeerJ},
  volume={7},
  pages={e7359},
  year={2019},
}
@article{wu2016maxbin,
  title={MaxBin 2.0: an automated binning algorithm to recover genomes from multiple metagenomic datasets},
  author={Wu, Yu-Wei and Simmons, Blake A and Singer, Steven W},
  journal={Bioinformatics},
  volume={32},
  number={4},
  pages={605--607},
  year={2016},
}

@article{ondov2011krona,
  title={Interactive metagenomic visualization in a Web browser},
  author={Ondov, Brian D and Bergman, Nicholas H and Phillippy, Adam M},
  journal={BMC Bioinformatics},
  volume={12},
  number={1},
  pages={385},
  year={2011},
  url = {https://doi.org/10.1186/1471-2105-12-385}
}

@article{buchfink2015fast,
  title={Fast and sensitive protein alignment using DIAMOND},
  author={Buchfink, Benjamin and Xie, Chao and Huson, Daniel H},
  journal={Nature Methods},
  volume={12},
  number={1},
  pages={59--60},
  year={2015},
  url = {https://doi.org/10.1038/nmeth.3176}
}

@article{delmont2018nitrogen,
  title={Nitrogen-fixing populations of Planctomycetes and Proteobacteria are abundant in surface ocean metagenomes},
  author={Delmont, Tom O and Quince, Christopher and Shaiber, Alon and Esen, {\"O}zcan C and Lee, Sonny TM and Rapp{\'e}, Michael S and McLellan, Sandra L and L{\"u}cker, Sebastian and Eren, A Murat},
  journal={Nature Microbiology},
  volume={3},
  number={7},
  pages={804--813},
  year={2018},
  url = {https://doi.org/10.1038/s41564-018-0176-9}
}

@article{dongen2000cluster,
  title={A cluster algorithm for graphs},
  author={Dongen, Stijn},
  year={2000},
  publisher={CWI (Centre for Mathematics and Computer Science)},
  url = {https://www.micans.org/mcl/index.html?sec_software}
}

@article{jain2018high,
  title={High throughput ANI analysis of 90K prokaryotic genomes reveals clear species boundaries},
  author={Jain, Chirag and Rodriguez-R, Luis M and Phillippy, Adam M and Konstantinidis, Konstantinos T and Aluru, Srinivas},
  journal={Nature Communications},
  volume={9},
  number={1},
  pages={1--8},
  year={2018},
  url={https://doi.org/10.1038/s41467-018-07641-9}
}

@article{stamatakis2014raxml,
  title={RAxML version 8: a tool for phylogenetic analysis and post-analysis of large phylogenies},
  author={Stamatakis, Alexandros},
  journal={Bioinformatics},
  volume={30},
  number={9},
  pages={1312--1313},
  year={2014},
  url={https://doi.org/10.1093/bioinformatics/btu033}
}

@article{minh2020iq,
  title={IQ-TREE 2: New models and efficient methods for phylogenetic inference in the genomic era},
  author={Minh, Bui Quang and Schmidt, Heiko A and Chernomor, Olga and Schrempf, Dominik and Woodhams, Michael D and Von Haeseler, Arndt and Lanfear, Robert},
  journal={Molecular Biology and Evolution},
  volume={37},
  number={5},
  pages={1530--1534},
  year={2020},
  url={https://doi.org/10.1093/molbev/msaa015}
}

@article{kalyaanamoorthy2017modelfinder,
  title={ModelFinder: fast model selection for accurate phylogenetic estimates},
  author={Kalyaanamoorthy, Subha and Minh, Bui Quang and Wong, Thomas KF and von Haeseler, Arndt and Jermiin, Lars S},
  journal={Nature Methods},
  volume={14},
  number={6},
  pages={587},
  year={2017},
  url={https://doi.org/10.1038/nmeth.4285}
}

@article{quince2011removing,
  title={Removing noise from pyrosequenced amplicons},
  author={Quince, Christopher and Lanzen, Anders and Davenport, Russell J and Turnbaugh, Peter J},
  journal={BMC bioinformatics},
  volume={12},
  number={1},
  pages={38},
  year={2011},
  url={https://doi.org/10.1186/1471-2105-12-38}
}
@article{caporaso2011global,
  title={Global patterns of 16S rRNA diversity at a depth of millions of sequences per sample},
  author={Caporaso, J Gregory and Lauber, Christian L and Walters, William A and Berg-Lyons, Donna and Lozupone, Catherine A and Turnbaugh, Peter J and Fierer, Noah and Knight, Rob},
  journal={Proceedings of the National Academy of Sciences},
  volume={108},
  number={Supplement 1},
  pages={4516--4522},
  year={2011},
  url={https://doi.org/10.1073/pnas.1000080107}
}
@article{kanehisa2000kegg,
  title={KEGG: kyoto encyclopedia of genes and genomes},
  author={Kanehisa, Minoru and Goto, Susumu},
  journal={Nucleic Acids Research},
  volume={28},
  number={1},
  pages={27--30},
  year={2000},
  url={https://doi.org/10.1093/nar/28.1.27}
}

@article{kanehisa2016blastkoala,
  title={BlastKOALA and GhostKOALA: KEGG tools for functional characterization of genome and metagenome sequences},
  author={Kanehisa, Minoru and Sato, Yoko and Morishima, Kanae},
  journal={Journal of Molecular Biology},
  volume={428},
  number={4},
  pages={726--731},
  year={2016},
  url={https://doi.org/10.1016/j.jmb.2015.11.006}
}

@article{minoche2011evaluation,
  title={Evaluation of genomic high-throughput sequencing data generated on Illumina HiSeq and genome analyzer systems},
  author={Minoche, Andr{\'e} E and Dohm, Juliane C and Himmelbauer, Heinz},
  journal={Genome Biology},
  volume={12},
  number={11},
  pages={R112},
  year={2011},
  url={https://doi.org/10.1186/gb-2011-12-11-r112}
}

@article{lee2019gtotree,
  title={GToTree: a user-friendly workflow for phylogenomics},
  author={Lee, Michael D},
  journal={Bioinformatics},
  volume={35},
  number={20},
  pages={4162--4164},
  year={2019},
  url={https://doi.org/10.1093/bioinformatics/btz188}
}

@article{wang2007naive,
  title={Naive Bayesian classifier for rapid assignment of rRNA sequences into the new bacterial taxonomy},
  author={Wang, Qiong and Garrity, George M and Tiedje, James M and Cole, James R},
  journal={Applied & Environmental Microbiology},
  volume={73},
  number={16},
  pages={5261--5267},
  year={2007},
  url={https://doi.org/10.1128/AEM.00062-07}
}

@article{hoang2017ultrafast,
  author = {Hoang, Diep Thi and Chernomor, Olga and von Haeseler, Arndt and Minh, Bui Quang and Vinh, Le Sy},
  title = {UFBoot2: Improving the Ultrafast Bootstrap Approximation},
  journal = {Molecular Biology and Evolution},
  volume = {35},
  number = {2},
  pages = {518-522},
  year = {2017},
  url = {https://doi.org/10.1093/molbev/msx281}
}

@article{parada2016every,
  title={Every base matters: assessing small subunit rRNA primers for marine microbiomes with mock communities, time series and global field samples},
  author={Parada, Alma E and Needham, David M and Fuhrman, Jed A},
  journal={Environmental microbiology},
  volume={18},
  number={5},
  pages={1403--1414},
  year={2016},
  url = {https://doi.org/10.1111/1462-2920.13023}
}

@article{parks2018gtgb,
  title={A standardized bacterial taxonomy based on genome phylogeny substantially revises the tree of life},
  author={Parks, Donovan H and Chuvochina, Maria and Waite, David W and Rinke, Christian and Skarshewski, Adam and Chaumeil, Pierre-Alain and Hugenholtz, Philip},
  journal={Nature Biotechnology},
  volume={36},
  number={10},
  pages={1546-1696},
  year={2018},
  url = {https://doi.org/10.1038/nbt.4229}
}

@article{foster2017metacoder,
  title={Metacoder: an R package for visualization and manipulation of community taxonomic diversity data},
  author={Foster, Zachary SL and Sharpton, Thomas J and Gr{\"u}nwald, Niklaus J},
  journal={PLoS Computational Biology},
  volume={13},
  number={2},
  pages={e1005404},
  year={2017},
  url={https://doi.org/10.1371/journal.pcbi.1005404}
}

@article{dufrene1997species,
  title={Species assemblages and indicator species: the need for a flexible asymmetrical approach},
  author={Dufr{\^e}ne, Marc and Legendre, Pierre},
  journal={Ecological Monographs},
  volume={67},
  number={3},
  pages={345--366},
  year={1997},
  url={https://doi.org/10.1890/0012-9615(1997)067[0345:SAAIST]2.0.CO;2}
}

@article{roberts2007labdsv,
  title={labdsv: Ordination and multivariate analysis for ecology},
  author={Roberts, David W and others},
  journal={R package version},
  volume={1},
  number={1},
  year={2007}
}
</script>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p>© Copyright 2020 The HYPOCOLYPSE Project.</p>
<p>Site constructed using <a href="https://rstudio.github.io/distill/">Distill for R Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
