<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

  <!--radix_placeholder_meta_tags-->
  <title>: Field Analyses</title>
  
  <meta property="description" itemprop="description" content="In this section we discuss the various field analyses conducted in this study."/>
  
  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>
  <link rel="icon" type="image/vnd.microsoft.icon" href="assets/favicon.ico"/>
  
  
  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content=": Field Analyses"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="In this section we discuss the various field analyses conducted in this study."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content=""/>
  
  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content=": Field Analyses"/>
  <meta property="twitter:description" content="In this section we discuss the various field analyses conducted in this study."/>
  
  <!--/radix_placeholder_meta_tags-->
  
  <!--radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","output","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["Field Analyses"]},{"type":"character","attributes":{},"value":["In this section we discuss the various field analyses conducted in this study.\n"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["distill::distill_article"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["css","toc","toc_depth"]}},"value":[{"type":"character","attributes":{},"value":["assets/styles.css"]},{"type":"logical","attributes":{},"value":[true]},{"type":"integer","attributes":{},"value":[3]}]}]},{"type":"character","attributes":{},"value":["assets/cite.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>
  
  <script type="application/javascript">
  
    window.headroom_prevent_pin = false;
  
    window.document.addEventListener("DOMContentLoaded", function (event) {
  
      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');
  
      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });
  
      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });
  
      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>
  
  <style type="text/css">
  
  /* Theme (user-documented overrideables for nav appearance) */
  
  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }
  
  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }
  
  .distill-site-nav a:hover {
    color: white;
  }
  
  @media print {
    .distill-site-nav {
      display: none;
    }
  }
  
  .distill-site-header {
  
  }
  
  .distill-site-footer {
  
  }
  
  
  /* Site Header */
  
  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }
  
  
  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }
  
  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }
  
  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }
  
  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }
  
  
  
  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }
  
  
  .distill-site-header .nav-toggle {
    display: none;
  }
  
  .nav-dropdown {
    display: inline-block;
    position: relative;
  }
  
  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }
  
  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }
  
  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .nav-dropdown-active {
    display: block;
  }
  
  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }
  
  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }
  
  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }
  
  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }
  
  /* Site Footer */
  
  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }
  
  /* Headroom */
  
  d-title {
    padding-top: 6rem;
  }
  
  @media print {
    d-title {
      padding-top: 4rem;
    }
  }
  
  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }
  
  .headroom--transition {
    transition: all .4s ease-in-out;
  }
  
  .headroom--unpinned {
    top: -100px;
  }
  
  .headroom--pinned {
    top: 0;
  }
  
  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }
  
  </style>
  
  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>
  
  <script type="application/javascript">
  
  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }
  
  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }
  
  function createFuseIndex() {
  
    // create fuse index
    var options = {
      keys: [
        { name: 'title', weight: 20 },
        { name: 'categories', weight: 15 },
        { name: 'description', weight: 10 },
        { name: 'contents', weight: 5 },
      ],
      ignoreLocation: true,
      threshold: 0
    };
    var fuse = new window.Fuse([], options);
  
    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });
  
        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }
  
  window.document.addEventListener("DOMContentLoaded", function (event) {
  
    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;
  
    createFuseIndex()
      .then(function(fuse) {
  
        // make search box visible
        searchEl.classList.remove('hidden');
  
        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
                ? `<img src="${offsetURL(suggestion.preview)}"</img>`
                : '';
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description || ''}
                  </div>
                  <div class="search-item-preview">
                    ${img}
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
        // remove inline display style on autocompleter (we want to
        // manage responsive display via css)
        $('.algolia-autocomplete').css("display", "");
      })
      .catch(function(error) {
        console.log(error);
      });
  
  });
  
  </script>
  
  <style type="text/css">
  
  .nav-search {
    font-size: x-small;
  }
  
  /* Algolioa Autocomplete */
  
  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #0F2E3D solid;
    width: 180px;
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      display: none;
      visibility: hidden;
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      display: inline-block;
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }
  
  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }
  
  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    font-size: 1rem;
    margin-block-start: 0;
    margin-block-end: 5px;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }
  
  </style>
  
  
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->
  
  <style type="text/css">
  
  body {
    background-color: white;
  }
  
  .pandoc-table {
    width: 100%;
  }
  
  .pandoc-table>caption {
    margin-bottom: 10px;
  }
  
  .pandoc-table th:not([align]) {
    text-align: left;
  }
  
  .pagedtable-footer {
    font-size: 15px;
  }
  
  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }
  
  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }
  
  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }
  
  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }
  
  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }
  
  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }
  
  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }
  
  .html-widget {
    margin-bottom: 2.0em;
  }
  
  .l-screen-inset {
    padding-right: 16px;
  }
  
  .l-screen .caption {
    margin-left: 10px;
  }
  
  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .shaded-content {
    background: white;
  }
  
  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }
  
  .hidden {
    display: none !important;
  }
  
  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }
  
  d-appendix {
    padding-top: 30px;
  }
  
  d-article>p>img {
    width: 100%;
  }
  
  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }
  
  d-article h3 {
    margin-top: 1.5rem;
  }
  
  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }
  
  /* Tweak code blocks */
  
  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }
  
  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }
  
  d-article div.sourceCode {
    background-color: white;
  }
  
  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }
  
  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  d-article pre a {
    border-bottom: none;
  }
  
  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }
  
  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }
  
  @media(min-width: 768px) {
  
  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }
  
  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }
  
  d-article pre {
    font-size: 14px;
  }
  
  }
  
  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  /* CSS for d-contents */
  
  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }
  
  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }
  
  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }
  
  .d-contents li {
    list-style-type: none
  }
  
  .d-contents nav > ul {
    padding-left: 0;
  }
  
  .d-contents ul {
    padding-left: 1em
  }
  
  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }
  
  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }
  
  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }
  
  .d-contents nav > ul > li > a {
    font-weight: 600;
  }
  
  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }
  
  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }
  
  
  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }
  
  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }
  
  
  /* Figure */
  
  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }
  
  .figure img {
    width: 100%;
  }
  
  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }
  
  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }
  
  /* Citations */
  
  d-article .citation {
    color: inherit;
    cursor: inherit;
  }
  
  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }
  
  /* Citation hover box */
  
  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }
  
  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }
  
  
  /* Tweak 1000px media break to show more text */
  
  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }
  
    .grid {
      grid-column-gap: 16px;
    }
  
    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }
  
  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }
  
    .grid {
      grid-column-gap: 32px;
    }
  }
  
  
  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */
  
  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }
  
  /* Include appendix styles here so they can be overridden */
  
  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }
  
  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }
  
  d-appendix h3 + * {
    margin-top: 1em;
  }
  
  d-appendix ol {
    padding: 0 0 0 15px;
  }
  
  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }
  
  d-appendix li {
    margin-bottom: 1em;
  }
  
  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  d-appendix > * {
    grid-column: text;
  }
  
  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }
  
  /* Include footnote styles here so they can be overridden */
  
  d-footnote-list {
    contain: layout style;
  }
  
  d-footnote-list > * {
    grid-column: text;
  }
  
  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }
  
  
  
  /* Anchor.js */
  
  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }
  
  /* Social footer */
  
  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }
  
  .disqus-comments {
    margin-right: 30px;
  }
  
  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
  
  #disqus_thread {
    margin-top: 30px;
  }
  
  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }
  
  .article-sharing a:hover {
    border-bottom: none;
  }
  
  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .subscribe p {
    margin-bottom: 0.5em;
  }
  
  
  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }
  
  
  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .custom p {
    margin-bottom: 0.5em;
  }
  
  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }
  
  /* Styles for posts lists (not auto-injected) */
  
  
  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }
  
  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }
  
  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }
  
  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }
  
  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }
  
  .post-preview-last {
    border-bottom: none !important;
  }
  
  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }
  
  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }
  
  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }
  
  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }
  
  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }
  
  .posts-list .metadata > * {
    display: inline-block;
  }
  
  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }
  
  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }
  
  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }
  
  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }
  
  .posts-list img {
    opacity: 1;
  }
  
  .posts-list img[data-src] {
    opacity: 0;
  }
  
  .posts-more {
    clear: both;
  }
  
  
  .posts-sidebar {
    font-size: 16px;
  }
  
  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }
  
  .sidebar-section {
    margin-bottom: 30px;
  }
  
  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }
  
  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }
  
  .categories li>a {
    border-bottom: none;
  }
  
  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }
  
  .categories .active {
    font-weight: 600;
  }
  
  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }
  
  
  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }
  
  
  /* Improve display for browsers without grid (IE/Edge <= 15) */
  
  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }
  
  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }
  
  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }
  
  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }
  
  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }
  
  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
  
  
  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }
  
  .downlevel .footnotes ol {
    padding-left: 13px;
  }
  
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }
  
  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }
  
  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }
  
  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  .downlevel .posts-list .post-preview {
    color: inherit;
  }
  
  
  
  </style>
  
  <script type="application/javascript">
  
  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }
  
  // show body when load is complete
  function on_load_complete() {
  
    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }
  
  
    // set body to visible
    document.body.style.visibility = 'visible';
  
    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }
  
    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }
  
  function init_distill() {
  
    init_common();
  
    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);
  
    // create d-title
    $('.d-title').changeElementType('d-title');
  
    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);
  
    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();
  
    // move posts container into article
    $('.posts-container').appendTo($('d-article'));
  
    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');
  
    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;
  
    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();
  
    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));
  
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });
  
    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');
  
    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {
  
      // capture layout
      var layout = $(this).attr('data-layout');
  
      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });
  
  
      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });
  
    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();
  
    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();
  
    // load distill framework
    load_distill_framework();
  
    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {
  
      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;
  
      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');
  
      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');
  
      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');
  
      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });
  
      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }
  
      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');
  
      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");
  
      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }
  
       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }
  
      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();
  
      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();
  
      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refHtml = $('#ref-' + $(this).attr('data-cites')).html();
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });
  
      // clear polling timer
      clearInterval(tid);
  
      // show body now that everything is ready
      on_load_complete();
    }
  
    var tid = setInterval(distill_post_process, 50);
    distill_post_process();
  
  }
  
  function init_downlevel() {
  
    init_common();
  
     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));
  
    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
  
    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();
  
    // remove toc
    $('.d-contents').remove();
  
    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });
  
  
    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);
  
    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);
  
    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();
  
    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();
  
    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));
  
    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });
  
    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));
  
    $('body').addClass('downlevel');
  
    on_load_complete();
  }
  
  
  function init_common() {
  
    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};
  
        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });
  
        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);
  
    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});
  
    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });
  
      }
    });
  
    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });
  
    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }
  
    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');
  
    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");
  
    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();
  
    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }
  
  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });
  
  </script>
  
  <!--/radix_placeholder_distill-->
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Inconsolata|Source+Sans+Pro|Work+Sans:200,300,400,500|Karla:400,400i,700,700i');
  /********************************/
  /* controls general properties of site*/
  /********************************/
  
  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    background-color: #FEFEFE;
  }
  
  d-title h1{
  font-size: 2.1em;
  font-family: 'Work Sans';
  font-weight: 300;
  }
  
  d-article h1 {
    font-size: 1.8em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 300;
  }
  
  d-article h2 {
    font-size: 1.65em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 500;
  }
  
  d-article h3 {
    margin-top: 1em;
  }
  d-byline .authors-affiliations {
    grid-column-end: span 3;
    grid-column-gap: 8px;
  }
  
  img.orcid-id {
    margin: 0;
  }
  
  a.name:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }
  
  a.affiliation:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }
  
  blockquote p {
  	font-size: 1.3em;
  	font-family: 'Lato';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }
  
  d-article li {
    margin: 0px 0px 4px;
  }
  
  /********************************/
  /* controls CODE*/
  /********************************/
  
  /** pre.sourceCode controls code block without lang **/
  
  /** Controls r code  **/
  
  pre.sourceCode.r {
    margin-top: 0px;
     color: #333333;
     background-color: #E8E8E8;
     border-radius: 3px;
     font-size: 0.80em;
     font-family: 'Monaco', 'PT Mono', monospace;
     overflow-x: scroll !important;
     border-left: 6px solid #006ddb;
     border-top: 1px solid #B7B7B7;
     border-right: 1px solid #B7B7B7;
     border-bottom: 1px solid #B7B7B7;
     padding: 18px 0px 18px 20px;
   }
  
   pre.sourceCode.bash {
     margin-top: 0px;
      color: #333333;
      background-color: #E8E8E8;
      border-radius: 3px;
      font-size: 0.80em;
      font-family: 'Monaco', 'PT Mono', monospace;
      overflow-x: scroll !important;
      border-left: 6px solid #D55E00;
      border-top: 1px solid #B7B7B7;
      border-right: 1px solid #B7B7B7;
      border-bottom: 1px solid #B7B7B7;
      padding: 18px 0px 18px 20px;
    }
  /** Controls code output **/
  
  d-article pre {
     margin-top: 0px;
      color: #333333;
      background-color: #FAFAFA;
      font-size: 0.8em;
      border: 1px solid #ccc;
      overflow-x: scroll !important;
      padding:  10px 10px 10px 18px;
      border-radius: 5px;
  
    }
  
  /** Controls inline code  **/
  
  p code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }
  
  d-article aside code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-family: 'Monaco', 'PT Mono', monospace;
    color: #BF2727;
  }
  
  /********************************/
  /* controls NAV BAR and HEADERS*/
  /********************************/
  
  .nav-dropdown-content .nav-dropdown-header {
    text-transform: uppercase;
    color: #686868;
  }
  
  .nav-dropdown-content a {
    font-family: 'Work Sans'
  }
  .nav-dropdown .nav-dropbtn {
    color: #FEFEFE;
    font-weight: 400;
  }
  
  .distill-site-header .logo img {
    margin-top: 10px;
    max-height: 30px;
    margin-bottom: auto;
  }
  
  .distill-site-nav {
      color: #FEFEFE;
      background-color: #143D59;
      font-size: 0.95em;
      font-weight: 400;
      min-width: 165px;
      position: absolute;
      font-family: 'Karla';
      letter-spacing: .1em;
  
  }
  
  .article-banner {
    width: auto;
    height: auto;
    border: 0;
  }
  
  img[src="assets/banner.png"] {
    border: 0;
    padding: 0;
    width: 100%;
    margin: 100px 10px 0px 25px;
  }
  
  /*************************************************
  *  code for side by side krona plot
  **************************************************/
  
  .column {
    float: left;
    width: 48%;
    padding: 5px;
  }
  
  /* Clear floats after image containers */
  .row::after {
    content: "";
    clear: both;
    display: table;
  }
  
  div.krona {
    width: 100%;
  	height: 90%;
    background-color: white;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    margin-bottom: 25px;
  	margin-top: 5px;
  }
  
  div.container-krona {
    text-align: center;
    padding: 1px 10px;
  	color: #414141;
  	font-size: 1.48em;
  	line-height: 0.8em;
  	font-family: 'Inconsolata';
  }
  
  div.container-krona p {
    margin: 20px;
  }
  
  div.krona img {
    border:1px solid white;
    margin: 25px;
  }
  
  img {
    margin: 25px;
  }
  
  /*************************************************
  *  code for LIGHTBOX MARKUP
  **************************************************/
  
  .thumbnail {
    max-width: 100%;
    position: relative;
  }
  
  .lightbox {
  	/** Default lightbox to hidden */
  	display: none;
  
  	/** Position and style */
  	position: fixed;
  	z-index: 999;
  	width: 100%;
  	height: 100%;
  	text-align: center;
  	top: 0;
  	left: 0;
  	background: rgba(0,0,0,0.8);
  }
  
  .lightbox img {
  	/** Pad the lightbox image */
  	position: fixed;
  	top: 0;
  	left: 0;
  	right: 0;
  	bottom: 0;
  	max-width: 0%;
  	max-height: 0%;
  	margin: auto;
  	box-sizing: border-box;
  }
  
  .lightbox:target {
  	/** Remove default browser outline */
  	outline: none;
  
  	/** Unhide lightbox **/
  	display: block;
  }
  
  .lightbox:target img {
  	max-height: 100%;
  	max-width: 100%;
  }
  
  /*************************************************
  *  code for DETAILS fold down
  **************************************************/
  
  details {
    background-color: #ccd7df;
    cursor: pointer;
    color: #282828;
    margin-bottom:1vh;
    padding: 1em;
    border: none;
    text-align: left;
    outline: none;
    font-family: 'Karla'
  }
  
  .active, details:hover {
    transition: max-height 0.2s ease-out;
    outline: none;
  
    /* background-color: #acc2c2; */
  }
  
  details {
    display: flex;
    justify-content: space-between;
    align-content: center;
    border-radius: 4px;
    outline: none;
  
  }
  
  details.active {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    outline: none;
  
  }
  
  summary::-webkit-details-marker {
    display: none;
    outline: none;
  
  }
  
  summary:focus {
    outline: none;
  }
  
  summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f0fe";
  }
  
  details[open] summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f146";
  }
  
  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Field Analyses","description":"In this section we discuss the various field analyses conducted in this study.","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a class="logo" href="index.html">
<img src="assets/icon.png" alt="Logo"/>
</a>
<a href="index.html" class="title"></a>
<input id="distill-search" class="nav-search hidden" type="text" placeholder="Search..."/>
</div>
<div class="nav-right">
<a href="index.html">
<i class="fa fa-home" aria-hidden="true"></i>
</a>
<a href="intro.html">OVERVIEW</a>
<a href="field.html">FIELD</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
16S rRNA
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Processing</span>
<a href="16s-dada2.html">1. DADA2</a>
<a href="16s-data-prep.html">2. Data Prep</a>
<hr/>
<span class="nav-dropdown-header">Analyses</span>
<a href="16s-water.html">3. Diversity</a>
<a href="16s-da-asvs.html">4. Differentially Abundant ASVs</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
METAGENOMIC
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Setup</span>
<a href="mg-setup.html">1. Working Envrionment</a>
<a href="mg-databases.html">2. Annotation Databases</a>
<hr/>
<span class="nav-dropdown-header">Read Processing</span>
<a href="mg-workflow-1.html">3. Assembly &amp; Annotations</a>
<a href="mg-workflow-2.html">4. Assembley &amp; Annotation Summary</a>
<hr/>
<span class="nav-dropdown-header">MAG Analysis</span>
<a href="mg-binning.html">5. Binning</a>
<a href="mag02-phylogenomics.html">6. MAG02 Phylogenomics</a>
<a href="mg-function.html">7. KEGG/KOFam Analysis</a>
</div>
</div>
<a href="data-availability.html">DATA</a>
<a href="https://github.com/hypocolypse/web/" aria-label="Link to source">
<i class="fab fa-github" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Field Analyses</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p><p>In this section we discuss the various field analyses conducted in this study.</p></p>
</div>


<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#hypoxic-reef-conditions">Hypoxic Reef Conditions</a></li>
<li><a href="#coral-bleaching">Coral Bleaching</a></li>
<li><a href="#coral-community-structure">Coral Community Structure</a></li>
<li><a href="#analysis-workflow">Analysis Workflow</a></li>
<li><a href="#pulse-amplitude-modulated-pam-fluorometry">Pulse Amplitude Modulated (PAM) Fluorometry</a><ul>
<li><a href="#non-transformed-data">Non-Transformed Data</a></li>
</ul></li>
<li><a href="#zooxanthellae">Zooxanthellae</a><ul>
<li><a href="#non-transformed-data-1">Non-Transformed Data</a></li>
<li><a href="#transformed-data">Transformed Data</a></li>
</ul></li>
<li><a href="#chlorophyll-a-chla">Chlorophyll a (Chla)</a><ul>
<li><a href="#non-transformed-data-2">Non-Transformed Data</a></li>
<li><a href="#transformed-data-1">Transformed Data</a></li>
</ul></li>
<li><a href="#chlorophyll-c-chlc">Chlorophyll c (Chlc)</a><ul>
<li><a href="#non-transformed-data-3">Non-Transformed Data</a></li>
<li><a href="#transformed-data-2">Transformed Data</a></li>
</ul></li>
<li><a href="#figures">Figures</a><ul>
<li><a href="#calculate-means-for-figures">Calculate Means for Figures</a></li>
<li><a href="#pam">PAM</a></li>
<li><a href="#zooxanthellae-counts">Zooxanthellae Counts</a></li>
<li><a href="#pigment-data">Pigment Data</a></li>
<li><a href="#fine-tuning">Fine Tuning</a></li>
</ul></li>
<li><a href="#panel-figure">Panel Figure</a></li>
<li><a href="#source-code">Source Code</a></li>
</ul>
</nav>
</div>
<h2 id="hypoxic-reef-conditions">Hypoxic Reef Conditions</h2>
<p>Conditions characteristic of a hypoxic event were first detected on Sept.20, 2017 in Bahia Almirante on the Caribbean coast of Panama. To evaluate the environmental conditions associated with the event we conducted a series of depth profiles with a YSI multiparameter sonde, measuring dissolved oxygen (DO), temperature, chlorophyll, salinity, and pH. Measurements were taken at 83 sites across Baha Almirante on Sept.25, 2017. We selected two focal sites across a gradient of DO concentrations, from extremely hypoxic to normoxic (ambient DO concentrations). Cayo Roldan (hypoxic site) lies at the inner reaches of Bahia Almirante where water exchange with the open ocean is restricted, experience extreme hypoxia on the benthos during the event. Cayo Coral (normoxic site) is well-flushed with open ocean water and was used as a control site because oxygenated conditions persisted during the event.</p>
<h2 id="coral-bleaching">Coral Bleaching</h2>
<p>The common lettuce coral, <em>Agaricia tenuifolia</em>, was collected from each site during the hypoxic event and evaluated for bleaching through measurements of symbiont densities, pigment content, and pulse amplitude modulated (PAM) fluorometry. Three fragments from each of six colonies were collected from each site and returned to the laboratory facilities at the Smithsonian Tropical Research Institutes Bocas del Toro Research Station. Differences in each response variable were evaluated with a linear mixed effects model, with colony as a random effect and size as a fixed effect.</p>
<h2 id="coral-community-structure">Coral Community Structure</h2>
<p>The hypoxic and normoxic sites are the location of permanent benthic monitoring. We evaluated changes in benthic community structure from four months before, during, and one year after the hypoxic event using permanent photoquadrats. At each site, permanent quadrats were photographed every 5m along one 50m permanent transect, and community structure was determined from 50 randomly overlaid points using the online platform <a href="https://coralnet.ucsd.edu/">CoralNet</a>. Taxa were identified to the finest resolution possible, and then grouped into different functional groups for subsequent analyses.</p>
<p>We used nonmetric multidimensional scaling (nMDS) to visually represent benthic community structure at each time point, and evaluated changes over time with a permutational analysis of variance (PERMANOVA), with pairwise contrasts. The benthic community did not change significantly over time at the normoxic site, so changes in the coral community over time were evaluated only at the hypoxic site.</p>
<h2 id="analysis-workflow">Analysis Workflow</h2>
<p>Next, we present the workflows we used to analyze coral response variables, specifically the Pulse Amplitude Modulated (PAM) Fluorometry, Zooxanthellae, Chlorophyll a (Chla), and Chlorophyll C (Chlc) data. For each data set, we visually assess residuals, test for normality (Shapiro-Wilk Test), and test for equality of variances (Levenes Test). In some cases, data are subsequently transformed.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a>data &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;tables/field/coralhypoxia.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw">attach</span>(data)</span></code></pre></div>
</div>
<h2 id="pulse-amplitude-modulated-pam-fluorometry">Pulse Amplitude Modulated (PAM) Fluorometry</h2>
<p>First, we test all data for assumptions of normality and homogeneity of variances based on raw data (i.e., not transformed) and residuals. Data are normal, variances hetergeneous (transform doesnt improve). Look at distribution of data with histogram.</p>
<h3 id="non-transformed-data">Non-Transformed Data</h3>
<h4 id="visual-assessment">Visual Assessment</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw">plotNormalHistogram</span>(yield, <span class="dt">main =</span> <span class="st">&quot;PAM&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;Yield&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-2-1.png" width="624" /></p>
</div>
<p>First, we run the model to generate the residuals, then look at residuals to see if they are normally distributed. This is a mixed model with fragment nested in colony and site as the fixed effect.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>lmepam &lt;-<span class="st"> </span><span class="kw">lmer</span>(yield <span class="op">~</span><span class="st"> </span>site <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>colony), <span class="dt">REML =</span> <span class="ot">FALSE</span>, <span class="dt">na.action =</span> na.exclude)</span>
<span id="cb3-2"><a href="#cb3-2"></a>res1 &lt;-<span class="st"> </span><span class="kw">resid</span>(lmepam)</span></code></pre></div>
</div>
<p>And then produce a histogram of the residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="kw">plotNormalHistogram</span>(res1, <span class="dt">main =</span> <span class="st">&quot;PAM&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;Residuals&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-5-1.png" width="624" /></p>
</div>
<h4 id="shapiro-wilk-normality-test">Shapiro-Wilk Normality Test</h4>
<p>Next, we run a normality test on residuals. The null hypothesis is that the data is normally distributed, while a p-value &lt; 0.05 means the data is non-normally distributed.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>pam.shapiro &lt;-<span class="st"> </span><span class="kw">shapiro.test</span>(res1)</span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="kw">print</span>(pam.shapiro)  </span></code></pre></div>
<pre><code>
    Shapiro-Wilk normality test

data:  res1
W = 0.97021, p-value = 0.4312</code></pre>
</div>
<p>Ok, the p-value is 0.4311851, meaning the result is not significant and the data are normally distributed.</p>
<p>Next, we generate QQ plots of residuals from the lme model.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="kw">qqnorm</span>(res1, <span class="dt">ylab =</span> <span class="st">&quot;Sample Quantiles for PAM&quot;</span>)</span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="kw">qqline</span>(res1, <span class="dt">col =</span> <span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-7-1.png" width="624" /></p>
</div>
<h4 id="levenes-test">Levenes Test</h4>
<p>And the Levenes Test, which tests for homogeneity of variance across groups.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="kw">leveneTest</span>(yield <span class="op">~</span><span class="st"> </span><span class="kw">as.factor</span>(site), <span class="dt">data =</span> data)</span></code></pre></div>
<pre><code>Levene&#39;s Test for Homogeneity of Variance (center = median)
      Df F value  Pr(&gt;F)   
group  1  7.9824 0.00785 **
      34                   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="kw">bartlett.test</span>(<span class="dt">x =</span> yield, <span class="dt">g =</span> site)</span></code></pre></div>
<pre><code>
    Bartlett test of homogeneity of variances

data:  yield and site
Bartlett&#39;s K-squared = 6.3431, df = 1, p-value = 0.01178</code></pre>
</div>
<p>So the data is <strong><em>normal but not homogenous</em></strong>.</p>
<h4 id="mixed-effects-models">Mixed Effects Models</h4>
<p>In our mixed model approach, <strong>site</strong> is the <em>fixed</em> effect, <strong>colony</strong> the <em>random</em> effect, and <strong>fragment</strong> is random nested in <strong>colony</strong>. We compared the full model with fragment dropped. The best fit model (with lowest AIC) was the model with colony dropped. But we use a model that includes fragment because its a necessary random effect.</p>
<p>When comparing models with different random effects like hereuse <code>REML = FALSE</code>. But when not comparing use <code>REML = TRUE</code> (default). The full model includes site, colony, and fragment nested in colony. Colony/frag includes a term for colony alone and nests fragment in colony.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>pam1 &lt;-<span class="st"> </span><span class="kw">lmer</span>(yield <span class="op">~</span><span class="st"> </span>site <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>colony<span class="op">/</span>frag), <span class="dt">data =</span> data, </span>
<span id="cb12-2"><a href="#cb12-2"></a>             <span class="dt">REML =</span> <span class="ot">FALSE</span>, <span class="dt">na.action =</span> na.exclude)</span>
<span id="cb12-3"><a href="#cb12-3"></a><span class="kw">ranova</span>(pam1)</span></code></pre></div>
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
yield ~ site + (1 | frag:colony) + (1 | colony)
                  npar logLik     AIC LRT Df Pr(&gt;Chisq)
&lt;none&gt;               5 65.561 -121.12                  
(1 | frag:colony)    4 65.561 -123.12   0  1          1
(1 | colony)         4 65.561 -123.12   0  1          1</code></pre>
</div>
<p>So we drop fragment nested in colony.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>pam2 &lt;-<span class="st"> </span><span class="kw">lmer</span>(yield <span class="op">~</span><span class="st"> </span>site <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>colony), <span class="dt">data =</span> data, </span>
<span id="cb14-2"><a href="#cb14-2"></a>             <span class="dt">REML =</span> <span class="ot">FALSE</span>, <span class="dt">na.action =</span> na.exclude)</span>
<span id="cb14-3"><a href="#cb14-3"></a><span class="kw">ranova</span>(pam2)</span></code></pre></div>
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
yield ~ site + (1 | colony)
             npar logLik     AIC                   LRT Df Pr(&gt;Chisq)
&lt;none&gt;          4 65.561 -123.12                                    
(1 | colony)    3 65.561 -125.12 -0.000000000000028422  1          1</code></pre>
</div>
<p>And then use the full model to account for nesting fragments.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a>mod.AIC &lt;-<span class="st"> </span><span class="kw">AIC</span>(pam1, pam2) </span>
<span id="cb16-2"><a href="#cb16-2"></a>mod.AIC </span></code></pre></div>
<pre><code>     df       AIC
pam1  5 -121.1218
pam2  4 -123.1218</code></pre>
</div>
<p>And the final models use REML but without comparing models with different random effects.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>finalpam &lt;-<span class="st"> </span><span class="kw">lmer</span>(yield <span class="op">~</span><span class="st"> </span>site <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>colony), </span>
<span id="cb18-2"><a href="#cb18-2"></a>                 <span class="dt">data =</span> data, </span>
<span id="cb18-3"><a href="#cb18-3"></a>                 <span class="dt">REML =</span> <span class="ot">TRUE</span>, <span class="dt">na.action =</span> na.exclude)</span></code></pre></div>
</div>
<p>p values for random effects.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>finalrandpam &lt;-<span class="st"> </span><span class="kw">ranova</span>(finalpam)</span>
<span id="cb19-2"><a href="#cb19-2"></a>pam.table &lt;-<span class="st"> </span><span class="kw">anova</span>(finalpam, <span class="dt">type =</span> <span class="dv">2</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a>pam.table</span></code></pre></div>
<pre><code>Type II Analysis of Variance Table with Satterthwaite&#39;s method
       Sum Sq  Mean Sq NumDF DenDF F value      Pr(&gt;F)    
site 0.056551 0.056551     1    34  34.827 0.000001156 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a>finalrandpam</span></code></pre></div>
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
yield ~ site + (1 | colony)
             npar logLik     AIC                   LRT Df Pr(&gt;Chisq)
&lt;none&gt;          4 58.057 -108.11                                    
(1 | colony)    3 58.057 -110.11 -0.000000000000014211  1          1</code></pre>
</div>
<h2 id="zooxanthellae">Zooxanthellae</h2>
<p>First, we test all data for assumptions of normality and homogeneity of variances. Based on the raw data and residuals, CR6B is an outlier and was removed. Log-transforming the data makes the data normal and homogeneous. We will look at both non-transformed transformed data.</p>
<h3 id="non-transformed-data-1">Non-Transformed Data</h3>
<p>Look at distribution of the raw data with histogram.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a><span class="kw">attach</span>(data)</span>
<span id="cb24-2"><a href="#cb24-2"></a><span class="kw">plotNormalHistogram</span>(zoox, <span class="dt">main =</span> <span class="st">&quot;Zooxanthellae&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;Counts&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-16-1.png" width="624" /></p>
</div>
<p>First, run the model to generate the residuals, then look at residuals to see if thay are normally distributed. Mixed model with fragment nested in colony, site as fixed effect. Yield is normally distributed, but variances not homogeneous.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a>lmezoox &lt;-<span class="st"> </span><span class="kw">lmer</span>(zoox <span class="op">~</span><span class="st"> </span>site <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>colony), <span class="dt">REML =</span> <span class="ot">FALSE</span>, <span class="dt">na.action =</span> na.exclude)</span>
<span id="cb25-2"><a href="#cb25-2"></a>res2 &lt;-<span class="st"> </span><span class="kw">resid</span>(lmezoox)</span></code></pre></div>
</div>
<p>Produce a histogram of the residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a><span class="kw">plotNormalHistogram</span>(res2, <span class="dt">main =</span> <span class="st">&quot;Zooxanthellae&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;Residuals&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-19-1.png" width="624" /></p>
</div>
<h4 id="shapiro-wilk-normality-test-1">Shapiro-Wilk Normality Test</h4>
<p>Next, we again run a normality test on residuals. And again, the null hypothesis is that the data is normally distributed, while a p-value &lt; 0.05 means the data is non-normally distributed.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a>zoox.shapiro &lt;-<span class="st"> </span><span class="kw">shapiro.test</span>(res2) <span class="co">#runs a normality test on residuals</span></span>
<span id="cb27-2"><a href="#cb27-2"></a><span class="kw">print</span>(zoox.shapiro) <span class="co"># null = normally distrubuted (P&lt;0.05 = non-normal</span></span></code></pre></div>
<pre><code>
    Shapiro-Wilk normality test

data:  res2
W = 0.9031, p-value = 0.004795</code></pre>
</div>
<p>We see the p-value (0.004795) is less than 0.05, and thus the data is non-normally distributed.</p>
<p>And a quick look at the QQ plot of residuals from lme model.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a><span class="kw">qqnorm</span>(res2, <span class="dt">ylab =</span> <span class="st">&quot;Sample Quantiles for Zoox&quot;</span>)</span>
<span id="cb29-2"><a href="#cb29-2"></a><span class="kw">qqline</span>(res2, <span class="dt">col =</span> <span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-21-1.png" width="624" /></p>
</div>
<h4 id="levenes-test-1">Levenes Test</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a><span class="kw">leveneTest</span>(zoox<span class="op">~</span><span class="st"> </span><span class="kw">as.factor</span>(site),<span class="dt">data =</span> data) <span class="co"># p = 0.0007486</span></span></code></pre></div>
<pre><code>Levene&#39;s Test for Homogeneity of Variance (center = median)
      Df F value    Pr(&gt;F)    
group  1  13.805 0.0007486 ***
      33                      
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
<h3 id="transformed-data">Transformed Data</h3>
<p>Log transformation improves normality. First, we add a new column to data frame for the log transformation and check that log transformed data is added as a new column.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a>data<span class="op">$</span>zoox_log =<span class="st"> </span><span class="kw">log</span>(zoox)</span>
<span id="cb32-2"><a href="#cb32-2"></a><span class="kw">attach</span>(data)</span>
<span id="cb32-3"><a href="#cb32-3"></a><span class="kw">colnames</span>(data)</span></code></pre></div>
<pre><code>[1] &quot;site&quot;     &quot;colony&quot;   &quot;frag&quot;     &quot;zoox&quot;     &quot;chla&quot;     &quot;chlc&quot;    
[7] &quot;yield&quot;    &quot;zoox_log&quot;</code></pre>
</div>
<h4 id="visual-assessment-1">Visual Assessment</h4>
<p>Check normality of log-transformed data.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a><span class="kw">plotNormalHistogram</span>(zoox_log, <span class="dt">main =</span> <span class="st">&quot;Zooxanthellae&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;log (counts)&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-24-1.png" width="624" /></p>
</div>
<p>Check normality of log-transformed residuals</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a>lmezoox_log &lt;-<span class="st"> </span><span class="kw">lmer</span>(zoox_log <span class="op">~</span><span class="st"> </span>site <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>colony), </span>
<span id="cb35-2"><a href="#cb35-2"></a>                    <span class="dt">REML =</span> <span class="ot">FALSE</span>, </span>
<span id="cb35-3"><a href="#cb35-3"></a>                    <span class="dt">na.action =</span> na.exclude)</span>
<span id="cb35-4"><a href="#cb35-4"></a>res3 &lt;-<span class="st"> </span><span class="kw">resid</span>(lmezoox_log)</span></code></pre></div>
</div>
<p>Produce a histogram of the residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a><span class="kw">plotNormalHistogram</span>(res3, <span class="dt">main =</span> <span class="st">&quot;Zooxanthellae log&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;Residuals&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-27-1.png" width="624" /></p>
</div>
<h4 id="shapiro-wilk-normality-test-2">Shapiro-Wilk Normality Test</h4>
<p>We then run a normality test on residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a>zooxlog.shapiro &lt;-<span class="st"> </span><span class="kw">shapiro.test</span>(res3)</span>
<span id="cb37-2"><a href="#cb37-2"></a><span class="kw">print</span>(zooxlog.shapiro)</span></code></pre></div>
<pre><code>
    Shapiro-Wilk normality test

data:  res3
W = 0.96613, p-value = 0.3461</code></pre>
</div>
<p>Now the p-value (0.3460954) is greater than 0.05, indicating the log-transformed data is normally distributed.</p>
<p>QQ plots of residuals from lme model.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1"></a><span class="kw">qqnorm</span>(res3, <span class="dt">ylab =</span> <span class="st">&quot;Sample Quantiles for Zooxlog&quot;</span>)</span>
<span id="cb39-2"><a href="#cb39-2"></a><span class="kw">qqline</span>(res3, <span class="dt">col =</span> <span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-29-1.png" width="624" /></p>
</div>
<h4 id="levenes-test-2">Levenes Test</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1"></a><span class="kw">leveneTest</span>(zoox_log <span class="op">~</span><span class="st"> </span><span class="kw">as.factor</span>(site), <span class="dt">data =</span> data)</span></code></pre></div>
<pre><code>Levene&#39;s Test for Homogeneity of Variance (center = median)
      Df F value Pr(&gt;F)
group  1  0.0674 0.7968
      33               </code></pre>
</div>
<h4 id="mixed-effects-models-1">Mixed Effects Models</h4>
<p>For the Mixed Effects Models, site is a fixed effect, colony is random, and fragment is random nested in colony. Colony/fragment includes a term for colony alone and nests fragment in colony.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1"></a>zoox1 &lt;-<span class="st"> </span><span class="kw">lmer</span>(zoox_log <span class="op">~</span><span class="st"> </span>site <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>colony<span class="op">/</span>frag), </span>
<span id="cb42-2"><a href="#cb42-2"></a>              <span class="dt">data =</span> data, </span>
<span id="cb42-3"><a href="#cb42-3"></a>              <span class="dt">REML =</span> <span class="ot">FALSE</span>, </span>
<span id="cb42-4"><a href="#cb42-4"></a>              <span class="dt">na.action =</span> na.exclude) <span class="co">#AIC 72.8756</span></span>
<span id="cb42-5"><a href="#cb42-5"></a><span class="kw">ranova</span>(zoox1)</span></code></pre></div>
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
zoox_log ~ site + (1 | frag:colony) + (1 | colony)
                  npar  logLik    AIC     LRT Df Pr(&gt;Chisq)
&lt;none&gt;               5 -31.438 72.876                      
(1 | frag:colony)    4 -31.438 70.876 0.00000  1     1.0000
(1 | colony)         4 -31.584 71.167 0.29147  1     0.5893</code></pre>
</div>
<p>Fragment nested in colony is then dropped.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1"></a>zoox2 &lt;-<span class="st"> </span><span class="kw">lmer</span>(zoox_log <span class="op">~</span><span class="st"> </span>site <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>colony), </span>
<span id="cb44-2"><a href="#cb44-2"></a>              <span class="dt">data =</span> data, </span>
<span id="cb44-3"><a href="#cb44-3"></a>              <span class="dt">REML =</span> <span class="ot">FALSE</span>, </span>
<span id="cb44-4"><a href="#cb44-4"></a>              <span class="dt">na.action =</span> na.exclude)</span>
<span id="cb44-5"><a href="#cb44-5"></a><span class="kw">ranova</span>(zoox2)</span></code></pre></div>
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
zoox_log ~ site + (1 | colony)
             npar  logLik    AIC     LRT Df Pr(&gt;Chisq)
&lt;none&gt;          4 -31.438 70.876                      
(1 | colony)    3 -31.584 69.167 0.29147  1     0.5893</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1"></a>mod.AIC &lt;-<span class="st"> </span><span class="kw">AIC</span>(zoox1, zoox2) </span>
<span id="cb46-2"><a href="#cb46-2"></a>mod.AIC</span></code></pre></div>
<pre><code>      df     AIC
zoox1  5 72.8756
zoox2  4 70.8756</code></pre>
</div>
<p>The Ffinal models uses REML but not comparing models with different random effects.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1"></a>finalzoox &lt;-<span class="st"> </span><span class="kw">lmer</span>(zoox_log <span class="op">~</span><span class="st"> </span>site <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>colony), </span>
<span id="cb48-2"><a href="#cb48-2"></a>                  <span class="dt">data =</span> data, </span>
<span id="cb48-3"><a href="#cb48-3"></a>                  <span class="dt">REML =</span> <span class="ot">TRUE</span>, </span>
<span id="cb48-4"><a href="#cb48-4"></a>                  <span class="dt">na.action =</span> na.exclude)</span></code></pre></div>
</div>
<p>P-values for random effects</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1"></a>finalrandzoox &lt;-<span class="st"> </span><span class="kw">ranova</span>(finalzoox)</span>
<span id="cb49-2"><a href="#cb49-2"></a>zoox.table &lt;-<span class="st"> </span><span class="kw">anova</span>(finalzoox, <span class="dt">type =</span> <span class="dv">2</span>) <span class="co">#ANOVA table</span></span>
<span id="cb49-3"><a href="#cb49-3"></a>zoox.table</span></code></pre></div>
<pre><code>Type II Analysis of Variance Table with Satterthwaite&#39;s method
     Sum Sq Mean Sq NumDF  DenDF F value        Pr(&gt;F)    
site 20.468  20.468     1 27.954  59.465 0.00000002143 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1"></a>finalrandzoox</span></code></pre></div>
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
zoox_log ~ site + (1 | colony)
             npar  logLik    AIC     LRT Df Pr(&gt;Chisq)
&lt;none&gt;          4 -33.335 74.670                      
(1 | colony)    3 -33.611 73.223 0.55316  1      0.457</code></pre>
</div>
<h2 id="chlorophyll-a-chla">Chlorophyll a (Chla)</h2>
<p>As before, we first test all data for assumptions of normality and homogeneity of variances.</p>
<h3 id="non-transformed-data-2">Non-Transformed Data</h3>
<h4 id="visual-assessment-2">Visual Assessment</h4>
<p>Look at distribution of the raw data with histogram.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1"></a><span class="kw">plotNormalHistogram</span>(chla, <span class="dt">main =</span> <span class="st">&quot;Chlorophyll a&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;Chlorophyll&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-35-1.png" width="624" /></p>
</div>
<p>First run the model to generate the residuals, then look at residuals to see if normally distributed. Mixed model with fragment nested in colony, site as fixed effect. Raw data are normal but variances not homogenouslog-transformed improves homogeneity.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1"></a>lmechla &lt;-<span class="st"> </span><span class="kw">lmer</span>(chla <span class="op">~</span><span class="st"> </span>site <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>colony), </span>
<span id="cb54-2"><a href="#cb54-2"></a>                <span class="dt">REML =</span> <span class="ot">FALSE</span>, </span>
<span id="cb54-3"><a href="#cb54-3"></a>                <span class="dt">na.action =</span> na.exclude)</span>
<span id="cb54-4"><a href="#cb54-4"></a>res4 &lt;-<span class="st"> </span><span class="kw">resid</span>(lmechla)</span></code></pre></div>
</div>
<p>Produce a histogram of the residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1"></a><span class="kw">plotNormalHistogram</span>(res4, <span class="dt">main =</span> <span class="st">&quot;Chlorophyll a&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;Residuals&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-38-1.png" width="624" /></p>
</div>
<h4 id="shapiro-wilk-normality-test-3">Shapiro-Wilk Normality Test</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1"></a>chla.shapiro &lt;-<span class="st"> </span><span class="kw">shapiro.test</span>(res4)</span>
<span id="cb56-2"><a href="#cb56-2"></a><span class="kw">print</span>(chla.shapiro) </span></code></pre></div>
<pre><code>
    Shapiro-Wilk normality test

data:  res4
W = 0.96483, p-value = 0.3015</code></pre>
</div>
<p>QQ plots of residuals from lme model</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1"></a><span class="kw">qqnorm</span>(res4, <span class="dt">ylab =</span> <span class="st">&quot;Sample Quantiles for Zoox&quot;</span>)</span>
<span id="cb58-2"><a href="#cb58-2"></a><span class="kw">qqline</span>(res4, <span class="dt">col =</span> <span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-40-1.png" width="624" /></p>
</div>
<h4 id="levenes-test-3">Levenes Test</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1"></a><span class="kw">leveneTest</span>(chla <span class="op">~</span><span class="st"> </span><span class="kw">as.factor</span>(site), <span class="dt">data =</span> data) <span class="co"># p = 0.01606</span></span></code></pre></div>
<pre><code>Levene&#39;s Test for Homogeneity of Variance (center = median)
      Df F value  Pr(&gt;F)  
group  1  6.4203 0.01606 *
      34                  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
<p>Data are normal but variances are not homogenous.</p>
<h3 id="transformed-data-1">Transformed Data</h3>
<p>Add a new column to dataframe for log transformation.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1"></a>data<span class="op">$</span>chla_log =<span class="st"> </span><span class="kw">log</span>(chla)</span></code></pre></div>
</div>
<p>Check that log transformed data is added as a new column</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1"></a><span class="kw">head</span>(data)</span></code></pre></div>
<pre><code>   site colony frag      zoox     chla      chlc     yield zoox_log
1 Coral    One    A  411086.2 2.597060 0.5928972 0.6503333 12.92656
2 Coral    One    B  199325.3 1.235098 0.4116325 0.6453333 12.20269
3 Coral    One    C  411812.2 2.620320 0.6300991 0.6513333 12.92832
4 Coral    Two    A 1089420.7 4.557184 1.1952204 0.6393333 13.90116
5 Coral    Two    B  353670.7 1.852745 0.6181671 0.6213333 12.77612
6 Coral    Two    C  891722.1 2.815479 0.7507814 0.6383333 13.70091
   chla_log
1 0.9543802
2 0.2111501
3 0.9632964
4 1.5167048
5 0.6166685
6 1.0351322</code></pre>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1"></a><span class="kw">attach</span>(data)</span>
<span id="cb64-2"><a href="#cb64-2"></a><span class="kw">colnames</span>(data)</span></code></pre></div>
<pre><code>[1] &quot;site&quot;     &quot;colony&quot;   &quot;frag&quot;     &quot;zoox&quot;     &quot;chla&quot;     &quot;chlc&quot;    
[7] &quot;yield&quot;    &quot;zoox_log&quot; &quot;chla_log&quot;</code></pre>
</div>
<h4 id="visual-assessment-3">Visual Assessment</h4>
<p>Check normality of log-transformed data.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1"></a><span class="kw">plotNormalHistogram</span>(chla_log, <span class="dt">main =</span> <span class="st">&quot;Chlorophyll a&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;log (Chlorophyll)&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-44-1.png" width="624" /></p>
</div>
<p>Check normality of log-transformed residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1"></a>lmechla_log &lt;-<span class="st"> </span><span class="kw">lmer</span>(chla_log <span class="op">~</span><span class="st"> </span>site <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>colony), </span>
<span id="cb67-2"><a href="#cb67-2"></a>                    <span class="dt">REML =</span> <span class="ot">FALSE</span>, </span>
<span id="cb67-3"><a href="#cb67-3"></a>                    <span class="dt">na.action =</span> na.exclude)</span>
<span id="cb67-4"><a href="#cb67-4"></a>res7 &lt;-<span class="st"> </span><span class="kw">resid</span>(lmechla_log)</span></code></pre></div>
</div>
<p>Produce a histogram of the residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1"></a><span class="kw">plotNormalHistogram</span>(res7, <span class="dt">main =</span> <span class="st">&quot;Chlorophyll a&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;Residuals&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-47-1.png" width="624" /></p>
</div>
<h4 id="shapiro-wilk-normality-test-4">Shapiro-Wilk Normality Test</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1"></a>chlalog.shapiro &lt;-<span class="st"> </span><span class="kw">shapiro.test</span>(res7) <span class="co">#runs a normality test on residuals</span></span>
<span id="cb69-2"><a href="#cb69-2"></a><span class="kw">print</span>(chlalog.shapiro) <span class="co"># null = normally distrubuted (P&lt;0.05 = non-normal</span></span></code></pre></div>
<pre><code>
    Shapiro-Wilk normality test

data:  res7
W = 0.95611, p-value = 0.1627</code></pre>
</div>
<p>QQ plots of residuals from lme model.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1"></a><span class="kw">qqnorm</span>(res7, <span class="dt">ylab =</span> <span class="st">&quot;Sample Quantiles for chlalog&quot;</span>)</span>
<span id="cb71-2"><a href="#cb71-2"></a><span class="kw">qqline</span>(res7, <span class="dt">col =</span> <span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-49-1.png" width="624" /></p>
</div>
<h4 id="levenes-test-4">Levenes Test</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1"></a><span class="kw">leveneTest</span>(chla_log <span class="op">~</span><span class="st"> </span><span class="kw">as.factor</span>(site), <span class="dt">data =</span> data) <span class="co"># p = 0.7487</span></span></code></pre></div>
<pre><code>Levene&#39;s Test for Homogeneity of Variance (center = median)
      Df F value Pr(&gt;F)
group  1  0.1043 0.7487
      34               </code></pre>
</div>
<h4 id="mixed-model-effects">Mixed Model Effects</h4>
<p>Site is fixed effect, colony is random, fragment is random nested in colony on log transformed data. Colony/frag includes a term for colony alone and nests fragment in colony.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1"></a>chla_log1 &lt;-<span class="st"> </span><span class="kw">lmer</span>(chla_log <span class="op">~</span><span class="st"> </span>site<span class="op">+</span>(<span class="dv">1</span><span class="op">|</span>colony<span class="op">/</span>frag), </span>
<span id="cb74-2"><a href="#cb74-2"></a>                  <span class="dt">data =</span> data,</span>
<span id="cb74-3"><a href="#cb74-3"></a>                  <span class="dt">REML =</span> <span class="ot">FALSE</span>, </span>
<span id="cb74-4"><a href="#cb74-4"></a>                  <span class="dt">na.action =</span> na.exclude) <span class="co">#AIC 55.11567</span></span>
<span id="cb74-5"><a href="#cb74-5"></a><span class="kw">ranova</span>(chla_log1)</span></code></pre></div>
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
chla_log ~ site + (1 | frag:colony) + (1 | colony)
                  npar  logLik    AIC    LRT Df Pr(&gt;Chisq)
&lt;none&gt;               5 -22.558 55.116                     
(1 | frag:colony)    4 -22.558 53.116 0.0000  1     1.0000
(1 | colony)         4 -23.274 54.549 1.4333  1     0.2312</code></pre>
</div>
<p>Fragment nested in colony dropped.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1"></a>chla_log2 &lt;-<span class="st"> </span><span class="kw">lmer</span>(chla_log <span class="op">~</span><span class="st"> </span>site <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>colony), </span>
<span id="cb76-2"><a href="#cb76-2"></a>                  <span class="dt">data =</span> data, </span>
<span id="cb76-3"><a href="#cb76-3"></a>                  <span class="dt">REML =</span> <span class="ot">FALSE</span>, </span>
<span id="cb76-4"><a href="#cb76-4"></a>                  <span class="dt">na.action =</span> na.exclude)</span>
<span id="cb76-5"><a href="#cb76-5"></a><span class="kw">ranova</span>(chla_log2)</span></code></pre></div>
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
chla_log ~ site + (1 | colony)
             npar  logLik    AIC    LRT Df Pr(&gt;Chisq)
&lt;none&gt;          4 -22.558 53.116                     
(1 | colony)    3 -23.274 52.549 1.4333  1     0.2312</code></pre>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1"></a>mod.AIC &lt;-<span class="st"> </span><span class="kw">AIC</span>(chla_log1, chla_log2) </span>
<span id="cb78-2"><a href="#cb78-2"></a>mod.AIC</span></code></pre></div>
<pre><code>          df      AIC
chla_log1  5 55.11567
chla_log2  4 53.11567</code></pre>
</div>
<p>Final models uses REMLnot comparing models with different random effects.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1"></a>finalchla_log &lt;-<span class="st"> </span><span class="kw">lmer</span>(chla_log <span class="op">~</span><span class="st"> </span>site <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>colony), </span>
<span id="cb80-2"><a href="#cb80-2"></a>                      <span class="dt">data =</span> data, </span>
<span id="cb80-3"><a href="#cb80-3"></a>                      <span class="dt">REML =</span> <span class="ot">TRUE</span>, </span>
<span id="cb80-4"><a href="#cb80-4"></a>                      <span class="dt">na.action =</span> na.exclude)</span></code></pre></div>
</div>
<p>p values for random effects.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1"></a>finalchla_logrand &lt;-<span class="st"> </span><span class="kw">ranova</span>(finalchla_log)</span>
<span id="cb81-2"><a href="#cb81-2"></a>chla_log.table &lt;-<span class="st"> </span><span class="kw">anova</span>(finalchla_log, <span class="dt">type =</span> <span class="dv">2</span>) <span class="co">#ANOVA table</span></span>
<span id="cb81-3"><a href="#cb81-3"></a>chla_log.table</span></code></pre></div>
<pre><code>Type II Analysis of Variance Table with Satterthwaite&#39;s method
     Sum Sq Mean Sq NumDF DenDF F value       Pr(&gt;F)    
site  7.508   7.508     1    29  39.802 0.0000006851 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1"></a>finalchla_logrand</span></code></pre></div>
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
chla_log ~ site + (1 | colony)
             npar  logLik    AIC   LRT Df Pr(&gt;Chisq)
&lt;none&gt;          4 -24.908 57.816                    
(1 | colony)    3 -25.843 57.687 1.871  1     0.1714</code></pre>
</div>
<h2 id="chlorophyll-c-chlc">Chlorophyll c (Chlc)</h2>
<p>First testing all data for assumptions of normality and homogeneity of variances. Based on raw data and residuals. Data need to be log transformed for normality and variances.</p>
<h3 id="non-transformed-data-3">Non-Transformed Data</h3>
<h4 id="visual-assessment-4">Visual Assessment</h4>
<p>Look at distribution of the raw data with histogram.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1"></a><span class="kw">plotNormalHistogram</span>(chlc, <span class="dt">main =</span> <span class="st">&quot;Chlorophyll c&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;Chlorophyll&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-55-1.png" width="624" /></p>
</div>
<p>Run the model to generate the residuals then look at residuals to see if normally distributed mixed model with fragment nested in colony, site as fixed effect yield is normally distributed, but variances not homogeneous.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1"></a>lmechlc &lt;-<span class="st"> </span><span class="kw">lmer</span>(chlc <span class="op">~</span><span class="st"> </span>site <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>colony), </span>
<span id="cb86-2"><a href="#cb86-2"></a>                <span class="dt">REML =</span> <span class="ot">FALSE</span>, </span>
<span id="cb86-3"><a href="#cb86-3"></a>                <span class="dt">na.action =</span> na.exclude)</span>
<span id="cb86-4"><a href="#cb86-4"></a>res5 &lt;-<span class="st"> </span><span class="kw">resid</span>(lmechlc)</span></code></pre></div>
</div>
<p>Produce a histogram of the residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1"></a><span class="kw">plotNormalHistogram</span>(res5, <span class="dt">main =</span> <span class="st">&quot;Chlorophyll c&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;Residuals&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-58-1.png" width="624" /></p>
</div>
<h4 id="shapiro-wilk-normality-test-5">Shapiro-Wilk Normality Test</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1"></a>chlc.shapiro &lt;-<span class="st"> </span><span class="kw">shapiro.test</span>(res5)</span>
<span id="cb88-2"><a href="#cb88-2"></a><span class="kw">print</span>(chlc.shapiro)</span></code></pre></div>
<pre><code>
    Shapiro-Wilk normality test

data:  res5
W = 0.90427, p-value = 0.004475</code></pre>
</div>
<p>QQ plots of residuals from lme model.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1"></a><span class="kw">qqnorm</span>(res5, <span class="dt">ylab =</span> <span class="st">&quot;Sample Quantiles for Zoox&quot;</span>)</span>
<span id="cb90-2"><a href="#cb90-2"></a><span class="kw">qqline</span>(res5, <span class="dt">col =</span> <span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-60-1.png" width="624" /></p>
</div>
<h4 id="levenes-test-5">Levenes Test</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1"></a><span class="kw">leveneTest</span>(chlc <span class="op">~</span><span class="st"> </span><span class="kw">as.factor</span>(site), <span class="dt">data =</span> data) <span class="co"># p = 0.01316</span></span></code></pre></div>
<pre><code>Levene&#39;s Test for Homogeneity of Variance (center = median)
      Df F value  Pr(&gt;F)  
group  1  6.8454 0.01316 *
      34                  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
<p>Data not normally distributed or homogenous</p>
<h3 id="transformed-data-2">Transformed Data</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1"></a>data<span class="op">$</span>chlc_log &lt;-<span class="st"> </span><span class="kw">log</span>(chlc)</span>
<span id="cb93-2"><a href="#cb93-2"></a><span class="kw">attach</span>(data)</span>
<span id="cb93-3"><a href="#cb93-3"></a><span class="kw">colnames</span>(data)</span></code></pre></div>
<pre><code> [1] &quot;site&quot;     &quot;colony&quot;   &quot;frag&quot;     &quot;zoox&quot;     &quot;chla&quot;     &quot;chlc&quot;    
 [7] &quot;yield&quot;    &quot;zoox_log&quot; &quot;chla_log&quot; &quot;chlc_log&quot;</code></pre>
</div>
<h4 id="visual-assessment-5">Visual Assessment</h4>
<p>Check normality of log-transformed data</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1"></a><span class="kw">plotNormalHistogram</span>(data<span class="op">$</span>chlc_log, <span class="dt">main =</span> <span class="st">&quot;Chlorophyll c (log)&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;log (Chlorophyll)&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-63-1.png" width="624" /></p>
</div>
<p>Check normality of log-transformed residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1"></a>lmechlc_log &lt;-<span class="st"> </span><span class="kw">lmer</span>(chlc_log <span class="op">~</span><span class="st"> </span>site <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>colony), </span>
<span id="cb96-2"><a href="#cb96-2"></a>                    <span class="dt">REML =</span> <span class="ot">FALSE</span>, </span>
<span id="cb96-3"><a href="#cb96-3"></a>                    <span class="dt">na.action =</span> na.exclude)</span>
<span id="cb96-4"><a href="#cb96-4"></a>res6 &lt;-<span class="st"> </span><span class="kw">resid</span>(lmechlc_log)</span></code></pre></div>
</div>
<p>Produce a histogram of the residuals.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1"></a><span class="kw">plotNormalHistogram</span>(res6, <span class="dt">main =</span> <span class="st">&quot;Chlorophyll c (log)&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;Residuals&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-66-1.png" width="624" /></p>
</div>
<h4 id="shapiro-wilk-normality-test-6">Shapiro-Wilk Normality Test</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1"></a>chlclog.shapiro &lt;-<span class="st"> </span><span class="kw">shapiro.test</span>(res6)</span>
<span id="cb98-2"><a href="#cb98-2"></a><span class="kw">print</span>(chlclog.shapiro)</span></code></pre></div>
<pre><code>
    Shapiro-Wilk normality test

data:  res6
W = 0.98465, p-value = 0.8874</code></pre>
</div>
<p>QQ plots of residuals from lme model.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1"></a><span class="kw">qqnorm</span>(res6, <span class="dt">ylab =</span> <span class="st">&quot;Sample Quantiles for chlclog&quot;</span>)</span>
<span id="cb100-2"><a href="#cb100-2"></a><span class="kw">qqline</span>(res6, <span class="dt">col =</span> <span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-68-1.png" width="624" /></p>
</div>
<h4 id="levenes-test-6">Levenes Test</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1"></a><span class="kw">leveneTest</span>(chlc_log <span class="op">~</span><span class="st"> </span><span class="kw">as.factor</span>(site), <span class="dt">data =</span> data) <span class="co"># p = 0.884</span></span></code></pre></div>
<pre><code>Levene&#39;s Test for Homogeneity of Variance (center = median)
      Df F value Pr(&gt;F)
group  1  0.0216  0.884
      34               </code></pre>
</div>
<h4 id="mixed-model-effects-1">Mixed Model Effects</h4>
<p>Site is the fixed effect, colony is random, fragment is random nested in colony. On log-transformed data. Colony/fragment includes a term for colony alone and nests fragment in colony.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1"></a>chlc_log1 &lt;-<span class="st"> </span><span class="kw">lmer</span>(chlc_log <span class="op">~</span><span class="st"> </span>site <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>colony<span class="op">/</span>frag), </span>
<span id="cb103-2"><a href="#cb103-2"></a>                  <span class="dt">data =</span> data, </span>
<span id="cb103-3"><a href="#cb103-3"></a>                  <span class="dt">REML =</span> <span class="ot">FALSE</span>, </span>
<span id="cb103-4"><a href="#cb103-4"></a>                  <span class="dt">na.action =</span> na.exclude) <span class="co">#AIC 64.52399</span></span>
<span id="cb103-5"><a href="#cb103-5"></a><span class="kw">ranova</span>(chlc_log1)</span></code></pre></div>
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
chlc_log ~ site + (1 | frag:colony) + (1 | colony)
                  npar  logLik    AIC    LRT Df Pr(&gt;Chisq)
&lt;none&gt;               5 -27.262 64.524                     
(1 | frag:colony)    4 -27.262 62.524 0.0000  1     1.0000
(1 | colony)         4 -27.833 63.665 1.1412  1     0.2854</code></pre>
</div>
<p>Fragment nested in colony dropped.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1"></a>chlc_log2 &lt;-<span class="st"> </span><span class="kw">lmer</span>(chlc_log <span class="op">~</span><span class="st"> </span>site<span class="op">+</span>(<span class="dv">1</span><span class="op">|</span>colony), </span>
<span id="cb105-2"><a href="#cb105-2"></a>                  <span class="dt">data =</span> data, </span>
<span id="cb105-3"><a href="#cb105-3"></a>                  <span class="dt">REML =</span> <span class="ot">FALSE</span>, </span>
<span id="cb105-4"><a href="#cb105-4"></a>                  <span class="dt">na.action =</span> na.exclude)</span>
<span id="cb105-5"><a href="#cb105-5"></a><span class="kw">ranova</span>(chlc_log2)</span></code></pre></div>
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
chlc_log ~ site + (1 | colony)
             npar  logLik    AIC    LRT Df Pr(&gt;Chisq)
&lt;none&gt;          4 -27.262 62.524                     
(1 | colony)    3 -27.833 61.665 1.1412  1     0.2854</code></pre>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1"></a>mod.AIC &lt;-<span class="st"> </span><span class="kw">AIC</span>(chlc_log1, chlc_log2) </span>
<span id="cb107-2"><a href="#cb107-2"></a>mod.AIC</span></code></pre></div>
<pre><code>          df      AIC
chlc_log1  5 64.52399
chlc_log2  4 62.52399</code></pre>
</div>
<p>Final models use REML.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1"></a>finalchlc_log &lt;-<span class="st"> </span><span class="kw">lmer</span>(chlc_log <span class="op">~</span><span class="st"> </span>site <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>colony), </span>
<span id="cb109-2"><a href="#cb109-2"></a>                      <span class="dt">data =</span> data, </span>
<span id="cb109-3"><a href="#cb109-3"></a>                      <span class="dt">REML =</span> <span class="ot">TRUE</span>, </span>
<span id="cb109-4"><a href="#cb109-4"></a>                      <span class="dt">na.action =</span> na.exclude)</span></code></pre></div>
</div>
<p>P-values for random effects.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1"></a>finalchlc_logrand &lt;-<span class="st"> </span><span class="kw">ranova</span>(finalchlc_log)</span>
<span id="cb110-2"><a href="#cb110-2"></a>chlc_log.table &lt;-<span class="st"> </span><span class="kw">anova</span>(finalchlc_log, <span class="dt">type =</span> <span class="dv">2</span>) <span class="co">#ANOVA table</span></span>
<span id="cb110-3"><a href="#cb110-3"></a>chlc_log.table</span></code></pre></div>
<pre><code>Type II Analysis of Variance Table with Satterthwaite&#39;s method
     Sum Sq Mean Sq NumDF DenDF F value     Pr(&gt;F)    
site 6.1918  6.1918     1    29  24.977 0.00002553 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1"></a>finalchlc_logrand</span></code></pre></div>
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
chlc_log ~ site + (1 | colony)
             npar  logLik    AIC    LRT Df Pr(&gt;Chisq)
&lt;none&gt;          4 -29.375 66.749                     
(1 | colony)    3 -30.148 66.297 1.5476  1     0.2135</code></pre>
</div>
<h2 id="figures">Figures</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1"></a>all &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;tables/field/coralhypoxia.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb114-2"><a href="#cb114-2"></a><span class="kw">attach</span>(all)</span></code></pre></div>
</div>
<h3 id="calculate-means-for-figures">Calculate Means for Figures</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1"></a>means.all &lt;-<span class="st"> </span><span class="kw">ddply</span>(all, <span class="kw">c</span>(<span class="st">&quot;site&quot;</span>, <span class="st">&quot;colony&quot;</span>), summarise,</span>
<span id="cb115-2"><a href="#cb115-2"></a>                 <span class="dt">N    =</span> <span class="kw">sum</span>(<span class="op">!</span><span class="kw">is.na</span>(yield)),</span>
<span id="cb115-3"><a href="#cb115-3"></a>                 <span class="dt">yield_mean =</span> <span class="kw">mean</span>(yield, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>),</span>
<span id="cb115-4"><a href="#cb115-4"></a>                 <span class="dt">zoox_mean =</span> <span class="kw">mean</span>(zoox, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>),</span>
<span id="cb115-5"><a href="#cb115-5"></a>                 <span class="dt">chla_mean =</span> <span class="kw">mean</span>(chla, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>),</span>
<span id="cb115-6"><a href="#cb115-6"></a>                 <span class="dt">chlc_mean =</span> <span class="kw">mean</span>(chlc, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>),</span>
<span id="cb115-7"><a href="#cb115-7"></a>                 <span class="dt">sd_yield =</span> <span class="kw">sd</span>(yield),</span>
<span id="cb115-8"><a href="#cb115-8"></a>                 <span class="dt">sd_zoox =</span> <span class="kw">sd</span>(zoox),</span>
<span id="cb115-9"><a href="#cb115-9"></a>                 <span class="dt">sd_chla =</span> <span class="kw">sd</span>(chla),</span>
<span id="cb115-10"><a href="#cb115-10"></a>                 <span class="dt">sd_chlc =</span> <span class="kw">sd</span>(chlc),</span>
<span id="cb115-11"><a href="#cb115-11"></a>                 <span class="dt">yield_se =</span> (<span class="kw">sd</span>(yield)<span class="op">/</span><span class="kw">sqrt</span>(N)),</span>
<span id="cb115-12"><a href="#cb115-12"></a>                 <span class="dt">zoox_se =</span> (<span class="kw">sd</span>(zoox)<span class="op">/</span><span class="kw">sqrt</span>(N)),</span>
<span id="cb115-13"><a href="#cb115-13"></a>                 <span class="dt">chla_se =</span> (<span class="kw">sd</span>(chla)<span class="op">/</span><span class="kw">sqrt</span>(N)),</span>
<span id="cb115-14"><a href="#cb115-14"></a>                 <span class="dt">chlc_se =</span> (<span class="kw">sd</span>(chlc)<span class="op">/</span><span class="kw">sqrt</span>(N)))</span>
<span id="cb115-15"><a href="#cb115-15"></a><span class="kw">write.csv</span>(means.all, <span class="dt">file =</span> <span class="st">&quot;tables/field/allcoralmeans.csv&quot;</span>, </span>
<span id="cb115-16"><a href="#cb115-16"></a>          <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
</div>
<h3 id="pam">PAM</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1"></a><span class="kw">ggplot</span>(all,<span class="kw">aes</span>(<span class="dt">x =</span> site, <span class="dt">y =</span> yield)) <span class="op">+</span></span>
<span id="cb116-2"><a href="#cb116-2"></a><span class="st">  </span><span class="kw">geom_boxplot</span>() <span class="op">+</span></span>
<span id="cb116-3"><a href="#cb116-3"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;Site&quot;</span>, <span class="dt">y=</span><span class="st">&#39;Maximum Quantum Yield&#39;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb116-4"><a href="#cb116-4"></a><span class="st">  </span><span class="kw">theme_classic</span>() <span class="op">+</span></span>
<span id="cb116-5"><a href="#cb116-5"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">text =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">14</span>)) <span class="op">+</span></span>
<span id="cb116-6"><a href="#cb116-6"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">angle =</span> <span class="dv">0</span>, <span class="dt">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb116-7"><a href="#cb116-7"></a>        <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>)) <span class="op">+</span></span>
<span id="cb116-8"><a href="#cb116-8"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>),</span>
<span id="cb116-9"><a href="#cb116-9"></a>                     <span class="dt">limits =</span> <span class="kw">c</span>(<span class="fl">0.4</span>,<span class="fl">0.75</span>))</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-76-1.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1"></a><span class="kw">ggplot</span>(all,<span class="kw">aes</span>(<span class="dt">x=</span>site, <span class="dt">y=</span>yield, <span class="dt">color=</span>colony)) <span class="op">+</span></span>
<span id="cb117-2"><a href="#cb117-2"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="dv">3</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb117-3"><a href="#cb117-3"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;Site&quot;</span>, <span class="dt">y =</span> <span class="st">&#39;Maximum Quantum Yield&#39;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb117-4"><a href="#cb117-4"></a><span class="st">  </span><span class="kw">theme_classic</span>() <span class="op">+</span></span>
<span id="cb117-5"><a href="#cb117-5"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">text =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">14</span>)) <span class="op">+</span></span>
<span id="cb117-6"><a href="#cb117-6"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>, <span class="dt">angle=</span><span class="dv">0</span>, <span class="dt">hjust=</span><span class="dv">1</span>),</span>
<span id="cb117-7"><a href="#cb117-7"></a>        <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>)) <span class="op">+</span></span>
<span id="cb117-8"><a href="#cb117-8"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>),</span>
<span id="cb117-9"><a href="#cb117-9"></a>                     <span class="dt">limits =</span> <span class="kw">c</span>(<span class="fl">0.4</span>,<span class="fl">0.75</span>))</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-77-1.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1"></a><span class="kw">ggplot</span>(means.all,<span class="kw">aes</span>(<span class="dt">x =</span> site, <span class="dt">y =</span> yield_mean,<span class="dt">fill =</span> site)) <span class="op">+</span></span>
<span id="cb118-2"><a href="#cb118-2"></a><span class="st">  </span><span class="kw">geom_boxplot</span>() <span class="op">+</span></span>
<span id="cb118-3"><a href="#cb118-3"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;Site&quot;</span>, <span class="dt">y =</span> <span class="st">&#39;Maximum Quantum Yield&#39;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb118-4"><a href="#cb118-4"></a><span class="st">  </span><span class="kw">theme_classic</span>() <span class="op">+</span></span>
<span id="cb118-5"><a href="#cb118-5"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;steelblue&quot;</span>,<span class="st">&quot;darkred&quot;</span>)) <span class="op">+</span></span>
<span id="cb118-6"><a href="#cb118-6"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">14</span>)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb118-7"><a href="#cb118-7"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">angle =</span> <span class="dv">0</span>, <span class="dt">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb118-8"><a href="#cb118-8"></a>        <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>)) <span class="op">+</span></span>
<span id="cb118-9"><a href="#cb118-9"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>),</span>
<span id="cb118-10"><a href="#cb118-10"></a>                     <span class="dt">limits =</span> <span class="kw">c</span>(<span class="fl">0.4</span>,<span class="fl">0.75</span>)) </span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-78-1.png" width="624" /></p>
</div>
<p>Set a jitter object to offset points and assign data to ggplot.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1"></a>jitter &lt;-<span class="st"> </span><span class="kw">position_jitter</span>(<span class="dt">width =</span> <span class="fl">0.1</span>)  </span>
<span id="cb119-2"><a href="#cb119-2"></a>yield_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(means.all, <span class="kw">aes</span>(<span class="dt">x =</span> site, <span class="dt">y =</span> yield_mean))</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1"></a>yp &lt;-<span class="st"> </span>yield_plot<span class="op">+</span></span>
<span id="cb120-2"><a href="#cb120-2"></a><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">color=</span><span class="st">&quot;black&quot;</span>,</span>
<span id="cb120-3"><a href="#cb120-3"></a>               <span class="kw">aes</span>(<span class="dt">fill=</span>site),</span>
<span id="cb120-4"><a href="#cb120-4"></a>               <span class="dt">width=</span><span class="fl">0.4</span>) <span class="op">+</span></span>
<span id="cb120-5"><a href="#cb120-5"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;skyblue3&quot;</span>,<span class="st">&quot;coral3&quot;</span>), </span>
<span id="cb120-6"><a href="#cb120-6"></a>                    <span class="dt">name=</span> <span class="ot">NULL</span>, <span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;Normoxic&quot;</span>,<span class="st">&quot;Hypoxic&quot;</span>)) <span class="op">+</span></span>
<span id="cb120-7"><a href="#cb120-7"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="dv">2</span>, <span class="dt">position=</span>jitter) <span class="op">+</span></span>
<span id="cb120-8"><a href="#cb120-8"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x=</span><span class="st">&quot;Site&quot;</span>,<span class="dt">y=</span><span class="st">&quot;Maximum Quantum Yield&quot;</span>) <span class="op">+</span></span>
<span id="cb120-9"><a href="#cb120-9"></a><span class="st">  </span><span class="kw">theme_classic</span>() <span class="op">+</span></span>
<span id="cb120-10"><a href="#cb120-10"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.85</span>,<span class="fl">0.85</span>),</span>
<span id="cb120-11"><a href="#cb120-11"></a>        <span class="dt">legend.title=</span> <span class="kw">element_text</span>(<span class="dt">color=</span><span class="st">&quot;transparent&quot;</span>),</span>
<span id="cb120-12"><a href="#cb120-12"></a>        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>), </span>
<span id="cb120-13"><a href="#cb120-13"></a>        <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>),</span>
<span id="cb120-14"><a href="#cb120-14"></a>        <span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">margin =</span> <span class="kw">margin</span>(<span class="dt">t=</span><span class="dv">0</span>, <span class="dt">r=</span><span class="dv">10</span>,<span class="dt">b=</span><span class="dv">0</span>,<span class="dt">l=</span><span class="dv">0</span>)),</span>
<span id="cb120-15"><a href="#cb120-15"></a>        <span class="dt">panel.border =</span> <span class="kw">element_rect</span>(<span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">fill=</span><span class="ot">NA</span>, <span class="dt">size=</span>.<span class="dv">5</span>)) <span class="op">+</span></span>
<span id="cb120-16"><a href="#cb120-16"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">text =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">12</span>)) <span class="op">+</span></span>
<span id="cb120-17"><a href="#cb120-17"></a><span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;Coral&quot;</span> =<span class="st"> &quot;Coral Cay&quot;</span>,<span class="st">&quot;Roldan&quot;</span> =<span class="st"> &quot;Cayo Roldan&quot;</span>)) <span class="op">+</span></span>
<span id="cb120-18"><a href="#cb120-18"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="dt">limits =</span> <span class="kw">c</span>(<span class="fl">0.5</span>,<span class="fl">0.7</span>))</span>
<span id="cb120-19"><a href="#cb120-19"></a>yp</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-80-1.png" width="624" /></p>
</div>
<h3 id="zooxanthellae-counts">Zooxanthellae Counts</h3>
<p>Assign data to ggplot.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1"></a>yield_zoox &lt;-<span class="st"> </span><span class="kw">ggplot</span>(means.all, <span class="kw">aes</span>(<span class="dt">x=</span>site, <span class="dt">y=</span>zoox_mean))</span>
<span id="cb121-2"><a href="#cb121-2"></a>zp &lt;-<span class="st"> </span>yield_zoox<span class="op">+</span></span>
<span id="cb121-3"><a href="#cb121-3"></a><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">color=</span><span class="st">&quot;black&quot;</span>,</span>
<span id="cb121-4"><a href="#cb121-4"></a>               <span class="kw">aes</span>(<span class="dt">fill=</span>site),</span>
<span id="cb121-5"><a href="#cb121-5"></a>               <span class="dt">width=</span><span class="fl">0.4</span>) <span class="op">+</span></span>
<span id="cb121-6"><a href="#cb121-6"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;skyblue3&quot;</span>,<span class="st">&quot;coral3&quot;</span>), </span>
<span id="cb121-7"><a href="#cb121-7"></a>                    <span class="dt">name=</span> <span class="ot">NULL</span>, <span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;Normoxic&quot;</span>,<span class="st">&quot;Hypoxic&quot;</span>)) <span class="op">+</span></span>
<span id="cb121-8"><a href="#cb121-8"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="dv">2</span>, <span class="dt">position=</span>jitter) <span class="op">+</span></span>
<span id="cb121-9"><a href="#cb121-9"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x=</span><span class="st">&quot;Site&quot;</span>,</span>
<span id="cb121-10"><a href="#cb121-10"></a>       <span class="dt">y=</span>(<span class="st">&quot;Endosymbiont Density (cells cm^-2)&quot;</span>)) <span class="op">+</span></span>
<span id="cb121-11"><a href="#cb121-11"></a><span class="st">  </span><span class="kw">theme_classic</span>() <span class="op">+</span></span>
<span id="cb121-12"><a href="#cb121-12"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>, <span class="co">#</span></span>
<span id="cb121-13"><a href="#cb121-13"></a>        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>),</span>
<span id="cb121-14"><a href="#cb121-14"></a>        <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>),</span>
<span id="cb121-15"><a href="#cb121-15"></a>        <span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">margin =</span> <span class="kw">margin</span>(<span class="dt">t=</span><span class="dv">0</span>, <span class="dt">r=</span><span class="dv">10</span>,<span class="dt">b=</span><span class="dv">0</span>,<span class="dt">l=</span><span class="dv">0</span>)),</span>
<span id="cb121-16"><a href="#cb121-16"></a>        <span class="dt">panel.border =</span> <span class="kw">element_rect</span>(<span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">fill=</span><span class="ot">NA</span>, <span class="dt">size=</span>.<span class="dv">5</span>)) <span class="op">+</span></span>
<span id="cb121-17"><a href="#cb121-17"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">text =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">12</span>)) <span class="op">+</span></span>
<span id="cb121-18"><a href="#cb121-18"></a><span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;Coral&quot;</span> =<span class="st"> &quot;Coral Cay&quot;</span>,<span class="st">&quot;Roldan&quot;</span> =<span class="st"> &quot;Cayo Roldan&quot;</span>)) <span class="op">+</span></span>
<span id="cb121-19"><a href="#cb121-19"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">labels =</span> scientific,</span>
<span id="cb121-20"><a href="#cb121-20"></a>                     <span class="dt">breaks=</span><span class="kw">seq</span>(<span class="dv">0</span>,<span class="fl">1.5E6</span>, <span class="fl">4.5E5</span>),</span>
<span id="cb121-21"><a href="#cb121-21"></a>                      <span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb121-22"><a href="#cb121-22"></a>                     <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="fl">1.5E6</span>))</span>
<span id="cb121-23"><a href="#cb121-23"></a>zp  </span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-81-1.png" width="624" /></p>
</div>
<h3 id="pigment-data">Pigment Data</h3>
<p>Convert Pigment data to long format with Chlorophyll a and Chlorophyll c stacked.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1"></a>pigs &lt;-<span class="st"> </span>tidyr<span class="op">::</span><span class="kw">gather</span>(means.all, pigment, concentration,chla_mean, chlc_mean)</span>
<span id="cb122-2"><a href="#cb122-2"></a>pigments &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(pigs,site,colony,pigment,concentration)</span></code></pre></div>
</div>
<p>Assign data to ggplot data frame pigments.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1"></a>pig_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(pigments,</span>
<span id="cb123-2"><a href="#cb123-2"></a>                     <span class="kw">aes</span>(<span class="dt">x=</span>pigment, <span class="dt">y=</span>concentration, <span class="dt">fill=</span>site))</span></code></pre></div>
</div>
<p>Shows Chlorophyll a and Chlorophyll c on the x axis with site color coded.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1"></a>pig_plot<span class="op">+</span></span>
<span id="cb124-2"><a href="#cb124-2"></a><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">color=</span><span class="st">&quot;black&quot;</span>) <span class="op">+</span></span>
<span id="cb124-3"><a href="#cb124-3"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;skyblue3&quot;</span>,<span class="st">&quot;coral3&quot;</span>), </span>
<span id="cb124-4"><a href="#cb124-4"></a>                    <span class="dt">name=</span> <span class="ot">NULL</span>, <span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;Normoxic&quot;</span>,<span class="st">&quot;Hypoxic&quot;</span>))</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-84-1.png" width="624" /></p>
</div>
<h4 id="chlorophyll-a-chla-plots-only">Chlorophyll a (Chla) Plots only</h4>
<p>Assign data to ggplot set a jitter object to offset points.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1"></a>jitter2 &lt;-<span class="st"> </span><span class="kw">position_jitter</span>(<span class="dt">width=</span><span class="fl">0.05</span>)  </span>
<span id="cb125-2"><a href="#cb125-2"></a>chla_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(means.all,</span>
<span id="cb125-3"><a href="#cb125-3"></a>                     <span class="kw">aes</span>(<span class="dt">x=</span>site, <span class="dt">y=</span>chla_mean))</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1"></a>cp &lt;-<span class="st"> </span>chla_plot<span class="op">+</span></span>
<span id="cb126-2"><a href="#cb126-2"></a><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">color=</span><span class="st">&quot;black&quot;</span>,</span>
<span id="cb126-3"><a href="#cb126-3"></a>               <span class="kw">aes</span>(<span class="dt">fill=</span>site),</span>
<span id="cb126-4"><a href="#cb126-4"></a>               <span class="dt">width=</span><span class="fl">0.4</span>) <span class="op">+</span></span>
<span id="cb126-5"><a href="#cb126-5"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;skyblue3&quot;</span>,<span class="st">&quot;coral3&quot;</span>), <span class="dt">name=</span> <span class="ot">NULL</span>, <span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;Normoxic&quot;</span>,<span class="st">&quot;Hypoxic&quot;</span>)) <span class="op">+</span></span>
<span id="cb126-6"><a href="#cb126-6"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="dv">2</span>, <span class="dt">position=</span>jitter) <span class="op">+</span></span>
<span id="cb126-7"><a href="#cb126-7"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x=</span><span class="st">&quot;Site&quot;</span>,</span>
<span id="cb126-8"><a href="#cb126-8"></a>       <span class="dt">y=</span>(<span class="st">&quot;Chl a Concentration (ug cm^-2)&quot;</span>)) <span class="op">+</span></span>
<span id="cb126-9"><a href="#cb126-9"></a><span class="st">  </span><span class="kw">theme_classic</span>() <span class="op">+</span></span>
<span id="cb126-10"><a href="#cb126-10"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb126-11"><a href="#cb126-11"></a>        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>), </span>
<span id="cb126-12"><a href="#cb126-12"></a>        <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>),</span>
<span id="cb126-13"><a href="#cb126-13"></a>        <span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">margin =</span> <span class="kw">margin</span>(<span class="dt">t=</span><span class="dv">0</span>, <span class="dt">r=</span><span class="dv">10</span>,<span class="dt">b=</span><span class="dv">0</span>,<span class="dt">l=</span><span class="dv">0</span>)),</span>
<span id="cb126-14"><a href="#cb126-14"></a>        <span class="dt">panel.border =</span> <span class="kw">element_rect</span>(<span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">fill=</span><span class="ot">NA</span>, <span class="dt">size=</span>.<span class="dv">5</span>)) <span class="op">+</span></span>
<span id="cb126-15"><a href="#cb126-15"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">text =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">12</span>)) <span class="op">+</span></span>
<span id="cb126-16"><a href="#cb126-16"></a><span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;Coral&quot;</span> =<span class="st"> &quot;Coral Cay&quot;</span>,<span class="st">&quot;Roldan&quot;</span> =<span class="st"> &quot;Cayo Roldan&quot;</span>)) <span class="op">+</span></span>
<span id="cb126-17"><a href="#cb126-17"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb126-18"><a href="#cb126-18"></a>                     <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">6</span>))</span>
<span id="cb126-19"><a href="#cb126-19"></a>cp</span></code></pre></div>
<p><img src="field_files/figure-html5/unnamed-chunk-86-1.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body-outset">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="fine-tuning">Fine Tuning</h3>
<h4 id="maximum-quantum-yield-by-site">Maximum Quantum Yield by site</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1"></a>ypp &lt;-<span class="st"> </span>yield_plot <span class="op">+</span></span>
<span id="cb127-2"><a href="#cb127-2"></a><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">color =</span> <span class="st">&quot;black&quot;</span>, <span class="kw">aes</span>(<span class="dt">fill =</span> site), <span class="dt">width =</span> <span class="fl">0.4</span>) <span class="op">+</span></span>
<span id="cb127-3"><a href="#cb127-3"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;skyblue3&quot;</span>, <span class="st">&quot;coral3&quot;</span>), <span class="dt">name =</span> <span class="ot">NULL</span>, </span>
<span id="cb127-4"><a href="#cb127-4"></a>                    <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;Normoxic&quot;</span>, <span class="st">&quot;Hypoxic&quot;</span>)) <span class="op">+</span></span>
<span id="cb127-5"><a href="#cb127-5"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">1</span>, <span class="dt">position =</span> jitter) <span class="op">+</span></span>
<span id="cb127-6"><a href="#cb127-6"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;Site&quot;</span>, <span class="dt">y =</span> (<span class="st">&quot;Maximum Quantum Yield&quot;</span>)) <span class="op">+</span></span>
<span id="cb127-7"><a href="#cb127-7"></a><span class="st">  </span><span class="kw">theme_classic</span>() <span class="op">+</span></span>
<span id="cb127-8"><a href="#cb127-8"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;top&quot;</span>, <span class="dt">legend.direction =</span> <span class="st">&quot;vertical&quot;</span>, <span class="co">#</span></span>
<span id="cb127-9"><a href="#cb127-9"></a>        <span class="dt">legend.title=</span> <span class="kw">element_text</span>(<span class="dt">color =</span> <span class="st">&quot;transparent&quot;</span>),</span>
<span id="cb127-10"><a href="#cb127-10"></a>        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>, <span class="dt">colour=</span><span class="st">&quot;black&quot;</span>), </span>
<span id="cb127-11"><a href="#cb127-11"></a>        <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">8</span>, <span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb127-12"><a href="#cb127-12"></a>        <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb127-13"><a href="#cb127-13"></a>        <span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">9</span>, </span>
<span id="cb127-14"><a href="#cb127-14"></a>                                    <span class="dt">margin =</span> <span class="kw">margin</span>(<span class="dt">t =</span> <span class="dv">0</span>, <span class="dt">r =</span> <span class="dv">10</span>, </span>
<span id="cb127-15"><a href="#cb127-15"></a>                                                    <span class="dt">b =</span> <span class="dv">0</span>, <span class="dt">l =</span> <span class="dv">0</span>)),</span>
<span id="cb127-16"><a href="#cb127-16"></a>        <span class="dt">panel.border =</span> <span class="kw">element_rect</span>(<span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, </span>
<span id="cb127-17"><a href="#cb127-17"></a>                                    <span class="dt">fill =</span> <span class="ot">NA</span>, <span class="dt">size =</span> <span class="fl">0.5</span>)) <span class="op">+</span></span>
<span id="cb127-18"><a href="#cb127-18"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">12</span>)) <span class="op">+</span></span>
<span id="cb127-19"><a href="#cb127-19"></a><span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;Coral&quot;</span> =<span class="st"> &quot;Coral Cay&quot;</span>, <span class="st">&quot;Roldan&quot;</span> =<span class="st"> &quot;Cayo Roldan&quot;</span>)) <span class="op">+</span></span>
<span id="cb127-20"><a href="#cb127-20"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="dt">limits =</span> <span class="kw">c</span>(<span class="fl">0.5</span>,<span class="fl">0.7</span>))</span></code></pre></div>
</div>
<h4 id="zooxanthellae-counts-by-site">Zooxanthellae Counts by site</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1"></a>yzield_zoox &lt;-<span class="st"> </span><span class="kw">ggplot</span>(means.all, <span class="kw">aes</span>(<span class="dt">x =</span> site, <span class="dt">y =</span> zoox_mean))</span>
<span id="cb128-2"><a href="#cb128-2"></a></span>
<span id="cb128-3"><a href="#cb128-3"></a>zpp &lt;-<span class="st"> </span>yield_zoox<span class="op">+</span></span>
<span id="cb128-4"><a href="#cb128-4"></a><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">color=</span><span class="st">&quot;black&quot;</span>,</span>
<span id="cb128-5"><a href="#cb128-5"></a>               <span class="kw">aes</span>(<span class="dt">fill=</span>site),</span>
<span id="cb128-6"><a href="#cb128-6"></a>               <span class="dt">width=</span><span class="fl">0.4</span>) <span class="op">+</span></span>
<span id="cb128-7"><a href="#cb128-7"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;skyblue3&quot;</span>,<span class="st">&quot;coral3&quot;</span>), <span class="dt">name=</span> <span class="ot">NULL</span>, </span>
<span id="cb128-8"><a href="#cb128-8"></a>                    <span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;Normoxic&quot;</span>,<span class="st">&quot;Hypoxic&quot;</span>)) <span class="op">+</span></span>
<span id="cb128-9"><a href="#cb128-9"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="dv">1</span>, <span class="dt">position=</span>jitter) <span class="op">+</span></span>
<span id="cb128-10"><a href="#cb128-10"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;Site&quot;</span>, <span class="dt">y=</span>(<span class="st">&quot;Endosymbiont Density (cells cm^-2)&quot;</span>)) <span class="op">+</span></span>
<span id="cb128-11"><a href="#cb128-11"></a><span class="st">  </span><span class="kw">theme_classic</span>() <span class="op">+</span></span>
<span id="cb128-12"><a href="#cb128-12"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb128-13"><a href="#cb128-13"></a>        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>, <span class="dt">colour=</span><span class="st">&quot;black&quot;</span>), </span>
<span id="cb128-14"><a href="#cb128-14"></a>        <span class="dt">axis.title.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">12</span>),</span>
<span id="cb128-15"><a href="#cb128-15"></a>        <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">8</span>, <span class="dt">colour=</span><span class="st">&quot;black&quot;</span>),</span>
<span id="cb128-16"><a href="#cb128-16"></a>        <span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">9</span>, <span class="dt">margin =</span> <span class="kw">margin</span>(<span class="dt">t=</span><span class="dv">0</span>, <span class="dt">r=</span><span class="dv">10</span>,<span class="dt">b=</span><span class="dv">0</span>,<span class="dt">l=</span><span class="dv">0</span>)),</span>
<span id="cb128-17"><a href="#cb128-17"></a>        <span class="dt">panel.border =</span> <span class="kw">element_rect</span>(<span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">fill=</span><span class="ot">NA</span>, <span class="dt">size=</span>.<span class="dv">5</span>)) <span class="op">+</span></span>
<span id="cb128-18"><a href="#cb128-18"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">text =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">12</span>)) <span class="op">+</span></span>
<span id="cb128-19"><a href="#cb128-19"></a><span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;Coral&quot;</span> =<span class="st"> &quot;Coral Cay&quot;</span>,<span class="st">&quot;Roldan&quot;</span> =<span class="st"> &quot;Cayo Roldan&quot;</span>)) <span class="op">+</span></span>
<span id="cb128-20"><a href="#cb128-20"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">labels =</span> scientific,</span>
<span id="cb128-21"><a href="#cb128-21"></a>                     <span class="dt">breaks=</span><span class="kw">seq</span>(<span class="dv">0</span>,<span class="fl">1.5E6</span>, <span class="fl">4.5E5</span>),</span>
<span id="cb128-22"><a href="#cb128-22"></a>                     <span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb128-23"><a href="#cb128-23"></a>                     <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="fl">1.5E6</span>))</span></code></pre></div>
</div>
<h4 id="chlorophyll-a-chla-1">Chlorophyll a (Chla)</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1"></a>chla_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(means.all, <span class="kw">aes</span>(<span class="dt">x =</span> site, <span class="dt">y =</span> chla_mean))</span>
<span id="cb129-2"><a href="#cb129-2"></a></span>
<span id="cb129-3"><a href="#cb129-3"></a>cpp &lt;-<span class="st"> </span>chla_plot <span class="op">+</span></span>
<span id="cb129-4"><a href="#cb129-4"></a><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">color=</span><span class="st">&quot;black&quot;</span>, <span class="kw">aes</span>(<span class="dt">fill=</span>site), <span class="dt">width=</span><span class="fl">0.4</span>) <span class="op">+</span></span>
<span id="cb129-5"><a href="#cb129-5"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;skyblue3&quot;</span>,<span class="st">&quot;coral3&quot;</span>), </span>
<span id="cb129-6"><a href="#cb129-6"></a>                    <span class="dt">name =</span> <span class="ot">NULL</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;Normoxic&quot;</span>, <span class="st">&quot;Hypoxic&quot;</span>)) <span class="op">+</span></span>
<span id="cb129-7"><a href="#cb129-7"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="dv">1</span>, <span class="dt">position=</span>jitter) <span class="op">+</span><span class="st"> </span></span>
<span id="cb129-8"><a href="#cb129-8"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;Site&quot;</span>, <span class="dt">y =</span> (<span class="st">&quot;Chl a Concentration (ug cm^-2)&quot;</span>)) <span class="op">+</span></span>
<span id="cb129-9"><a href="#cb129-9"></a><span class="st">  </span><span class="kw">theme_classic</span>() <span class="op">+</span></span>
<span id="cb129-10"><a href="#cb129-10"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb129-11"><a href="#cb129-11"></a>        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>, <span class="dt">colour=</span><span class="st">&quot;black&quot;</span>), </span>
<span id="cb129-12"><a href="#cb129-12"></a>        <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">8</span>, <span class="dt">colour=</span><span class="st">&quot;black&quot;</span>),</span>
<span id="cb129-13"><a href="#cb129-13"></a>        <span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">9</span>, </span>
<span id="cb129-14"><a href="#cb129-14"></a>                                    <span class="dt">margin =</span> <span class="kw">margin</span>(<span class="dt">t =</span> <span class="dv">0</span>, <span class="dt">r =</span> <span class="dv">10</span>, </span>
<span id="cb129-15"><a href="#cb129-15"></a>                                                    <span class="dt">b =</span> <span class="dv">0</span>, <span class="dt">l =</span> <span class="dv">0</span>)),</span>
<span id="cb129-16"><a href="#cb129-16"></a>        <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb129-17"><a href="#cb129-17"></a>        <span class="dt">panel.border =</span> <span class="kw">element_rect</span>(<span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">fill=</span><span class="ot">NA</span>, <span class="dt">size=</span>.<span class="dv">5</span>)) <span class="op">+</span></span>
<span id="cb129-18"><a href="#cb129-18"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">text =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">12</span>)) <span class="op">+</span></span>
<span id="cb129-19"><a href="#cb129-19"></a><span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;Coral&quot;</span> =<span class="st"> &quot;Coral Cay&quot;</span>,<span class="st">&quot;Roldan&quot;</span> =<span class="st"> &quot;Cayo Roldan&quot;</span>)) <span class="op">+</span></span>
<span id="cb129-20"><a href="#cb129-20"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb129-21"><a href="#cb129-21"></a>                     <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">6</span>))</span></code></pre></div>
</div>
<h2 id="panel-figure">Panel Figure</h2>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="field_files/figure-html5/unnamed-chunk-92-1.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="source-code" class="appendix">Source Code</h2>
<p>The source code for this page can be accessed on GitHub by <a href="https://github.com/hypocolypse/web/blob/master/field.Rmd">clicking this link</a>.</p>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/hypocolypse/web/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/hypocolypse/web/">https://github.com/hypocolypse/web/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p> Copyright 2021 metacrobe &amp; The HYPOCOLYPSE Project.</p>
<p>Site constructed using <a href="https://rstudio.github.io/distill/">Distill for R Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
