<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

  <!--radix_placeholder_meta_tags-->
  <title>: No 6. MAG02 Phylogenomic Analysis</title>
  
  <meta property="description" itemprop="description" content="In this section, we create a workflow to compare MAG02 with publicly available genomes. MAG02 was only found in the hypoxic sample."/>
  
  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>
  <link rel="icon" type="image/vnd.microsoft.icon" href="assets/favicon.ico"/>
  
  <meta name="article:author" content="Jarrod J Scott"/>
  
  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content=": No 6. MAG02 Phylogenomic Analysis"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="In this section, we create a workflow to compare MAG02 with publicly available genomes. MAG02 was only found in the hypoxic sample."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content=""/>
  
  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content=": No 6. MAG02 Phylogenomic Analysis"/>
  <meta property="twitter:description" content="In this section, we create a workflow to compare MAG02 with publicly available genomes. MAG02 was only found in the hypoxic sample."/>
  
  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Snakemake—a scalable bioinformatics workflow engine;citation_publication_date=2012;citation_volume=28;citation_author=Johannes Köster;citation_author=Sven Rahmann"/>
  <meta name="citation_reference" content="citation_title=MUSCLE: Multiple sequence alignment with high accuracy and high throughput;citation_publication_date=2004;citation_volume=32;citation_author=Robert C Edgar"/>
  <meta name="citation_reference" content="citation_title=High throughput ani analysis of 90K prokaryotic genomes reveals clear species boundaries;citation_publication_date=2018;citation_volume=9;citation_author=Chirag Jain;citation_author=Luis M Rodriguez-R;citation_author=Adam M Phillippy;citation_author=Konstantinos T Konstantinidis;citation_author=Srinivas Aluru"/>
  <meta name="citation_reference" content="citation_title=IQ-tree 2: New models and efficient methods for phylogenetic inference in the genomic era;citation_publication_date=2020;citation_volume=37;citation_author=Bui Quang Minh;citation_author=Heiko A Schmidt;citation_author=Olga Chernomor;citation_author=Dominik Schrempf;citation_author=Michael D Woodhams;citation_author=Arndt Von Haeseler;citation_author=Robert Lanfear"/>
  <meta name="citation_reference" content="citation_title=ModelFinder: Fast model selection for accurate phylogenetic estimates;citation_publication_date=2017;citation_volume=14;citation_author=Subha Kalyaanamoorthy;citation_author=Bui Quang Minh;citation_author=Thomas KF Wong;citation_author=Arndt Haeseler;citation_author=Lars S Jermiin"/>
  <meta name="citation_reference" content="citation_title=GToTree: A user-friendly workflow for phylogenomics;citation_publication_date=2019;citation_volume=35;citation_author=Michael D Lee"/>
  <meta name="citation_reference" content="citation_title=UFBoot2: Improving the ultrafast bootstrap approximation;citation_publication_date=2017;citation_volume=35;citation_author=Diep Thi Hoang;citation_author=Olga Chernomor;citation_author=Arndt Haeseler;citation_author=Bui Quang Minh;citation_author=Le Sy Vinh"/>
  <meta name="citation_reference" content="citation_title=A standardized bacterial taxonomy based on genome phylogeny substantially revises the tree of life;citation_publication_date=2018;citation_volume=36;citation_author=Donovan H Parks;citation_author=Maria Chuvochina;citation_author=David W Waite;citation_author=Christian Rinke;citation_author=Adam Skarshewski;citation_author=Pierre-Alain Chaumeil;citation_author=Philip Hugenholtz"/>
  <!--radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["No 6. MAG02 Phylogenomic Analysis"]},{"type":"character","attributes":{},"value":["In this section, we create a workflow to compare MAG02 with publicly available genomes. MAG02 was only found in the hypoxic sample.\n"]},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name"]}},"value":[{"type":"character","attributes":{},"value":["Jarrod J Scott"]}]}]},{"type":"character","attributes":{},"value":["assets/cite.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  
  <script type="application/javascript">
  
    window.headroom_prevent_pin = false;
  
    window.document.addEventListener("DOMContentLoaded", function (event) {
  
      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');
  
      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });
  
      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });
  
      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>
  
  <style type="text/css">
  
  /* Theme (user-documented overrideables for nav appearance) */
  
  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #455a64;
    font-size: 15px;
    font-weight: 300;
  }
  
  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }
  
  .distill-site-nav a:hover {
    color: white;
  }
  
  @media print {
    .distill-site-nav {
      display: none;
    }
  }
  
  .distill-site-header {
  
  }
  
  .distill-site-footer {
  
  }
  
  
  /* Site Header */
  
  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }
  
  
  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }
  
  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }
  
  .distill-site-header .title {
    font-size: 18px;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }
  
  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }
  
  
  
  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }
  
  
  .distill-site-header .nav-toggle {
    display: none;
  }
  
  .nav-dropdown {
    display: inline-block;
    position: relative;
  }
  
  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }
  
  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }
  
  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .nav-dropdown-active {
    display: block;
  }
  
  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }
  
  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }
  
  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }
  
  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative;}
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }
  
  /* Site Footer */
  
  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }
  
  /* Headroom */
  
  d-title {
    padding-top: 6rem;
  }
  
  @media print {
    d-title {
      padding-top: 4rem;
    }
  }
  
  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }
  
  .headroom--transition {
    transition: all .4s ease-in-out;
  }
  
  .headroom--unpinned {
    top: -100px;
  }
  
  .headroom--pinned {
    top: 0;
  }
  
  </style>
  
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->
  
  <style type="text/css">
  
  body {
    background-color: white;
  }
  
  .pandoc-table {
    width: 100%;
  }
  
  .pandoc-table>caption {
    margin-bottom: 10px;
  }
  
  .pandoc-table th:not([align]) {
    text-align: left;
  }
  
  .pagedtable-footer {
    font-size: 15px;
  }
  
  .html-widget {
    margin-bottom: 2.0em;
  }
  
  .l-screen-inset {
    padding-right: 16px;
  }
  
  .l-screen .caption {
    margin-left: 10px;
  }
  
  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .shaded-content {
    background: white;
  }
  
  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }
  
  .hidden {
    display: none !important;
  }
  
  d-article {
    padding-bottom: 30px;
  }
  
  d-appendix {
    padding-top: 30px;
  }
  
  d-article>p>img {
    width: 100%;
  }
  
  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }
  
  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  /* CSS for table of contents */
  
  .d-toc {
    color: rgba(0,0,0,0.8);
    font-size: 0.8em;
    line-height: 1em;
  }
  
  .d-toc-header {
    font-size: 0.6rem;
    font-weight: 400;
    color: rgba(0, 0, 0, 0.5);
    text-transform: uppercase;
    margin-top: 0;
    margin-bottom: 1.3em;
  }
  
  .d-toc a {
    border-bottom: none;
  }
  
  .d-toc ul {
    padding-left: 0;
  }
  
  .d-toc li>ul {
    padding-top: 0.8em;
    padding-left: 16px;
    margin-bottom: 0.6em;
  }
  
  .d-toc ul,
  .d-toc li {
    list-style-type: none;
  }
  
  .d-toc li {
    margin-bottom: 0.9em;
  }
  
  .d-toc-separator {
    margin-top: 20px;
    margin-bottom: 2em;
  }
  
  .d-article-with-toc {
    border-top: none;
    padding-top: 0;
  }
  
  
  
  /* Tweak code blocks (note that this CSS is repeated above in an injection
     into the d-code shadow dom) */
  
  d-code {
    overflow-x: auto !important;
  }
  
  pre.d-code code.d-code {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }
  
  pre.text-output {
  
    font-size: 12px;
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  @media(min-width: 768px) {
  
  d-code {
    overflow-x: visible !important;
  }
  
  pre.d-code code.d-code  {
      padding-left: 18px;
      font-size: 14px;
  }
  pre.text-output {
    font-size: 14px;
  }
  }
  
  /* Figure */
  
  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }
  
  .figure img {
    width: 100%;
  }
  
  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }
  
  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }
  
  
  
  /* Tweak 1000px media break to show more text */
  
  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }
  
    .grid {
      grid-column-gap: 16px;
    }
  
    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }
  
  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }
  
    .grid {
      grid-column-gap: 32px;
    }
  }
  
  
  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */
  
  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }
  
  
  /* Social footer */
  
  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }
  
  .disqus-comments {
    margin-right: 30px;
  }
  
  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
  
  #disqus_thread {
    margin-top: 30px;
  }
  
  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }
  
  .article-sharing a:hover {
    border-bottom: none;
  }
  
  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .subscribe p {
    margin-bottom: 0.5em;
  }
  
  
  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }
  
  
  /* Improve display for browsers without grid (IE/Edge <= 15) */
  
  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }
  
  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }
  
  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }
  
  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }
  
  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }
  
  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
  
  
  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }
  
  .downlevel .footnotes ol {
    padding-left: 13px;
  }
  
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }
  
  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }
  
  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }
  
  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  </style>
  
  <script type="application/javascript">
  
  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }
  
  // show body when load is complete
  function on_load_complete() {
  
    // set body to visible
    document.body.style.visibility = 'visible';
  
    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }
  
    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }
  
  function init_distill() {
  
    init_common();
  
    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);
  
    // create d-title
    $('.d-title').changeElementType('d-title');
  
    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);
  
    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();
  
    // move posts container into article
    $('.posts-container').appendTo($('d-article'));
  
    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');
  
    // create d-bibliography
    var bibliography = $('<d-bibliography></d-bibliography>');
    $('#distill-bibliography').wrap(bibliography);
  
    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;
  
    // replace citations with <d-cite>
    $('.citation').each(function(i, val) {
      appendix = true;
      var cites = $(this).attr('data-cites').split(" ");
      var dt_cite = $('<d-cite></d-cite>');
      dt_cite.attr('key', cites.join());
      $(this).replaceWith(dt_cite);
    });
    // remove refs
    $('#refs').remove();
  
    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();
  
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-toc a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });
  
    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');
  
    // replace code blocks with d-code
    $('pre>code').each(function(i, val) {
      var code = $(this);
      var pre = code.parent();
      var clz = "";
      var language = pre.attr('class');
      if (language) {
        // map unknown languages to "clike" (without this they just dissapear)
        if ($.inArray(language, ["bash", "clike", "css", "go", "html",
                                 "javascript", "js", "julia", "lua", "markdown",
                                 "markup", "mathml", "python", "svg", "xml"]) == -1)
          language = "clike";
        language = ' language="' + language + '"';
        var dt_code = $('<d-code block' + language + clz + '></d-code>');
        dt_code.text(code.text());
        pre.replaceWith(dt_code);
      } else {
        code.addClass('text-output').unwrap().changeElementType('pre');
      }
    });
  
    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {
  
      // capture layout
      var layout = $(this).attr('data-layout');
  
      // apply layout to markdown level block elements
      var elements = $(this).children().not('d-code, pre.text-output, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });
  
  
      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });
  
    // load distill framework
    load_distill_framework();
  
    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {
  
      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;
  
      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');
  
      // table of contents
      if (have_authors) // adjust border if we are in authors
        $('.d-toc').parent().addClass('d-article-with-toc');
  
      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');
  
      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }
  
      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');
  
      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");
  
      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }
  
       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }
  
      // inject pre code styles (can't do this with a global stylesheet b/c a shadow root is used)
      $('d-code').each(function(i, val) {
        var style = document.createElement('style');
        style.innerHTML = 'pre code { padding-left: 10px; font-size: 12px; border-left: 2px solid rgba(0,0,0,0.1); } ' +
                          '@media(min-width: 768px) { pre code { padding-left: 18px; font-size: 14px; } }';
        if (this.shadowRoot)
          this.shadowRoot.appendChild(style);
      });
  
      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();
  
      // clear polling timer
      clearInterval(tid);
  
      // show body now that everything is ready
      on_load_complete();
    }
  
    var tid = setInterval(distill_post_process, 50);
    distill_post_process();
  
  }
  
  function init_downlevel() {
  
    init_common();
  
     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));
  
    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
  
    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();
  
    // remove toc
    $('.d-toc-header').remove();
    $('.d-toc').remove();
    $('.d-toc-separator').remove();
  
    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });
  
  
    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);
  
    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);
  
    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();
  
    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();
  
    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));
  
    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });
  
    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));
  
    $('body').addClass('downlevel');
  
    on_load_complete();
  }
  
  
  function init_common() {
  
    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};
  
        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });
  
        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);
  
    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});
  
    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });
  
      }
    });
  
    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });
  
    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }
  
    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');
  
    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");
  
    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();
  
    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }
  
  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });
  
  </script>
  
  <!--/radix_placeholder_distill-->
  <script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
  <script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
  <link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
  <script src="site_libs/datatables-binding-0.13/datatables.js"></script>
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
  <script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
  <link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
  <script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
  <link href="site_libs/dt-ext-fixedcolumns-1.10.20/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
  <script src="site_libs/dt-ext-fixedcolumns-1.10.20/js/dataTables.fixedColumns.min.js"></script>
  <link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
  <script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Roboto:300|Nunito+Sans:300|Inconsolata|Source+Sans+Pro');
  /* controls general properties of site*/
  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    background-color: #FAFAFA;
  }
  
  d-title h1{
  font-size: 2.5em;
  font-family: Roboto;
  font-weight: 500;
  }
  
  d-article h1, h2 {
    font-family: Roboto;
  }
  
  d-article h3 {
    margin-top: 1em;
  }
  
  blockquote p {
  	font-size: 1.3em;
  	font-family: 'Lato';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }
  
  d-code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    border-left: 5px solid #006ddb;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
  }
  
  pre.text-output {
    margin-top: 0px;
     color: #333333;
     background-color: #FAFAFA;
     font-size: 0.8em
   }
  
  pre {
    border: 1px solid #ccc;
    padding: 0px 10px 10px 18px;
    margin: 1em 0px;
    border-radius: 5px;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    text-transform: none;
  }
  
  d-article li {
    margin: 0px 0px 4px;
  }
  
  /*************************************************
  *  code for side by side krona plot
  **************************************************/
  
  .column {
    float: left;
    width: 48%;
    padding: 5px;
  }
  
  /* Clear floats after image containers */
  .row::after {
    content: "";
    clear: both;
    display: table;
  }
  
  div.krona {
    width: 100%;
  	height: 90%;
    background-color: white;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    margin-bottom: 25px;
  	margin-top: 5px;
  }
  
  div.container-krona {
    text-align: center;
    padding: 1px 10px;
  	color: #414141;
  	font-size: 1.48em;
  	line-height: 0.8em;
  	font-family: 'Inconsolata';
  }
  
  div.container-krona p {
    margin: 20px;
  }
  
  div.krona img {
    border:1px solid white;
    margin: 25px;
  }
  
  img {
    margin: 25px;
  }
  
  /** LIGHTBOX MARKUP **/
  .thumbnail {
    max-width: 100%;
    position: relative;
  }
  
  .lightbox {
  	/** Default lightbox to hidden */
  	display: none;
  
  	/** Position and style */
  	position: fixed;
  	z-index: 999;
  	width: 100%;
  	height: 100%;
  	text-align: center;
  	top: 0;
  	left: 0;
  	background: rgba(0,0,0,0.8);
  }
  
  .lightbox img {
  	/** Pad the lightbox image */
  	position: fixed;
  	top: 0;
  	left: 0;
  	right: 0;
  	bottom: 0;
  	max-width: 0%;
  	max-height: 0%;
  	margin: auto;
  	box-sizing: border-box;
  }
  
  .lightbox:target {
  	/** Remove default browser outline */
  	outline: none;
  
  	/** Unhide lightbox **/
  	display: block;
  }
  
  .lightbox:target img {
  	max-height: 100%;
  	max-width: 100%;
  }
  
  .distill-site-header .logo img {
    margin-top: 10px;
    max-height: 30px;
    margin-bottom: 10px;
  }
  
  .distill-site-nav {
      color: rgba(255, 255, 255, 0.8);
      background-color: #143D59;
      font-size: 16px;
      font-weight: 300;
    }
  }
  
  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"No 6. MAG02 Phylogenomic Analysis","description":"In this section, we create a workflow to compare MAG02 with publicly available genomes. MAG02 was only found in the hypoxic sample.","authors":[{"author":"Jarrod J Scott","authorURL":"#","affiliation":"&nbsp;","affiliationURL":"#"}]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a class="logo" href="index.html">
<img src="assets/icon.png"/>
</a>
<a href="index.html" class="title"></a>
</div>
<div class="nav-right">
<a href="index.html">
<i class="fa fa-home"></i>
</a>
<a href="intro.html">Introduction</a>
<a href="field.html">A. Field Analyses</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
B. 16S rRNA
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">16S rRNA Processing</span>
<a href="16s-dada2.html">1. DADA2</a>
<a href="16s-data-prep.html">2. Data Prep</a>
<hr/>
<span class="nav-dropdown-header">16S rRNA Analyses</span>
<a href="16s-water.html">3. Diversity</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
C. Metagenomic
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Setup</span>
<a href="mg-setup.html">1. Working Envrionment</a>
<a href="mg-databases.html">2. Annotation Databases</a>
<hr/>
<span class="nav-dropdown-header">Read Processing</span>
<a href="mg-workflow-1.html">3. Assembly &amp; Annotations</a>
<a href="mg-workflow-2.html">4. Assembley &amp; Annotation Summary</a>
<hr/>
<span class="nav-dropdown-header">MAG Analysis</span>
<a href="mg-binning.html">5. Binning</a>
<a href="mag02-phylogenomics.html">6. MAG02 Phylogenomics</a>
</div>
</div>
<a href="data-availability.html">Data</a>
<a href="https://github.com/hypocolypse/web/">
<i class="fa fa-github fa-lg"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>No 6. MAG02 Phylogenomic Analysis</h1>
<p>In this section, we create a workflow to compare MAG02 with publicly available genomes. MAG02 was only found in the hypoxic sample.</p>
</div>

<div class="d-byline">
  Jarrod J Scott  
  

</div>

<div class="d-article">
<h3 class="d-toc-header">Table of Contents</h3>
<nav class="d-toc" id="TOC">
<ul>
<li><a href="#getting-genomes-from-ncbi">Getting Genomes from NCBI</a><ul>
<li><a href="#getting-taxids-to-download">Getting TaxIDs to Download</a></li>
<li><a href="#downloading-genomes-from-ncbi">Downloading Genomes from NCBI</a></li>
<li><a href="#processing-ncbi-genomes">Processing NCBI Genomes</a></li>
</ul></li>
<li><a href="#running-snakemake-contigs-workflow">Running Snakemake Contigs Workflow</a></li>
<li><a href="#curate-genome-collection">Curate Genome Collection</a><ul>
<li><a href="#estimate-taxonomy">Estimate Taxonomy</a></li>
<li><a href="#completeness">Completeness</a></li>
<li><a href="#screen-for-16s-rrna">Screen for 16S rRNA</a></li>
</ul></li>
<li><a href="#dereplicate-genome-collection">Dereplicate Genome Collection</a></li>
<li><a href="#outgroup-workflow">Outgroup Workflow</a></li>
<li><a href="#phylogenomics">Phylogenomics</a><ul>
<li><a href="#pangenome-analysis">Pangenome Analysis</a></li>
</ul></li>
<li><a href="#s-rrna-gene-analysis-from-mag02">16S rRNA gene analysis from MAG02</a></li>
<li><a href="#conclusion">Conclusion</a><ul>
<li><a href="#source-code">Source Code</a></li>
</ul></li>
</ul>
</nav>
<hr class="d-toc-separator"/>
<p>What we want to do next is compare MAG02<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> to publicly available genomes. We know from our binning and binning summary that MAG02 has a lot of hits to <em>Arcobacter</em>, an Epsilonproteobacteria. However, when we run <code>anvi-estimate-scg-taxonomy</code> we get something a little different</p>
<pre class="bash"><code>
anvi-estimate-scg-taxonomy -c 18_NR-MAGs-SPLIT/WATER_MAG_00002/CONTIGS.db</code></pre>
<div class="l-body-outset">
<table>
<colgroup>
<col style="width: 9%" />
<col style="width: 6%" />
<col style="width: 9%" />
<col style="width: 74%" />
</colgroup>
<thead>
<tr class="header">
<th>MAG</th>
<th>total_scgs</th>
<th>supporting_scgs</th>
<th>taxonomy</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>WATER_MAG_00002</td>
<td>20</td>
<td>9</td>
<td>Bacteria / Campylobacterota / Campylobacteria / Campylobacterales / Arcobacteraceae / Poseidonibacter / Poseidonibacter sp002470185</td>
</tr>
</tbody>
</table>
</div>
<p>Hum. <em>Poseidonibacter</em>. A little disagreement between the SCG classification and the short-read classification. Not too big of a deal since <em>Poseidonibacter</em> is in the Arcobacteraceae.</p>
<h1 id="getting-genomes-from-ncbi">Getting Genomes from NCBI</h1>
<p>For this workflow we largely follow the tutorial on <a href="http://merenlab.org/2019/03/14/ncbi-genome-download-magic/">Accessing and including NCBI genomes in ’omics analyses in anvi’o</a> by Alon Shaiber and Meren.</p>
<p>We need genomes to compare to our MAG. As the tutorial explains, we can either use the taxa name itself or the TaxID. But before that we need to get the <a href="https://github.com/kblin/ncbi-genome-download#contributed-scripts-gimme_taxapy"><code>gimme_taxa.py</code></a> <em>helper script</em> by <a href="http://orcid.org/0000-0002-9569-6738">by Joe R. J. Healey</a> that allows us to download genomes from NCBI. Slick.</p>
<pre class="bash"><code>
wget https://raw.githubusercontent.com/kblin/ncbi-genome-download/master/contrib/gimme_taxa.py</code></pre>
<p>We can start with the taxa names themselves.</p>
<h2 id="getting-taxids-to-download">Getting TaxIDs to Download</h2>
<pre class="bash"><code>
python gimme_taxa.py Poseidonibacter \
                     -o Poseidonibacter-for-ngd.txt

python gimme_taxa.py Arcobacter \
                     -o Arcobacter-for-ngd.txt</code></pre>
<p>We only got 1 hit for <a href="files/mag02-phylogenomics/Poseidonibacter-TaxIDs-for-ngd.txt"><em>Poseidonibacter</em></a> but the script returned 48 for <a href="files/mag02-phylogenomics/Arcobacter-TaxIDs-for-ngd.txt"><em>Arcobacter</em></a>.</p>
<p>What about the TaxID?</p>
<p>We need to head to NCBI’s <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi">Taxonomy Browser</a> and get the <em>parent TaxID</em> by searching for the taxon name of interest, in this case <em>Poseidonibacter</em> and <em>Arcobacter</em>.</p>
<p><em>Poseidonibacter</em> is <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Tree&amp;id=2321187">TaxID 2321187</a> and <em>Arcobacter</em> is <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&amp;id=28196">TaxID 28196</a>.</p>
<pre class="bash"><code>
python gimme_taxa.py 2321187 \
                     -o Poseidonibacter-TaxIDs-for-ngd.txt

python gimme_taxa.py 28196 \
                     -o Arcobacter-TaxIDs-for-ngd.txt</code></pre>
<p>In this case the results of using TaxID and taxon name were identical, however that is not always the case, so it is good to check.</p>
<p>On the taxonomy browser, both genera are part of the <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Undef&amp;id=2321108"><em>Arcobacter</em> group</a>. Since there is some ambiguity as to the identification of MAG02, we will instead use the entire group.</p>
<pre class="bash"><code>
python gimme_taxa.py 2321108 \
                     -o Arcobacter_Group-TaxIDs-for-ngd.txt</code></pre>
<p>This time we get <a href="files/mag02-phylogenomics/Arcobacter_Group-TaxIDs-for-ngd.txt">198 hits</a>. Since we only need the TaxIDs, we can rerun the command like so:</p>
<pre class="bash"><code>
python gimme_taxa.py 2321108 \
                     -o Arcobacter_Group-TaxIDs-for-ngd-just-IDs.txt \
                     --just-taxids</code></pre>
<p>As the command implies, we only get a list of TaxIDs. Moving on,</p>
<h2 id="downloading-genomes-from-ncbi">Downloading Genomes from NCBI</h2>
<p>To download genomes using the TaxIDs, we need a collection of scripts called <code>ncbi-genome-download</code>. See the <a href="https://github.com/kblin/ncbi-genome-download#contributed-scripts-gimme_taxapy">GitHub repo</a> for installation instructions. We have a choice of where to look for genomes using the <code>-s</code> flag—either Genbank or Refseq</p>
<pre class="bash"><code>
ncbi-genome-download -t Arcobacter_Group-TaxIDs-for-ngd-just-IDs.txt \
                     bacteria \
                     -o 00_Arcobacter_Genbank \
                     --metadata Arcobacter-NCBI-METADATA.txt \
                     -s genbank</code></pre>
<p>We tested both options and found that the <code>-s genbank</code> returned 121 genomes and <code>-s refseq</code> returned 95. We compared the two lists and found nothing unique in the Refseq database so we will continue with the GenBank set.</p>
<p>After all of this hullabaloo, turns out <em>Poseidonibacter</em> was not part of the download. Curious. Anyway, we ran an additional instance of the command to get these genomes. The command did not work with the TaxID so we had to use the genus name instead. At this point, can you believe anything I say? Well, you shouldn’t.</p>
<pre class="bash"><code>
ncbi-genome-download bacteria  \
                     -g Poseidonibacter \
                     -o 00_Arcobacter_Genbank \
                     --metadata Poseidonibacter-NCBI-METADATA.txt \
                     -s genbank</code></pre>
<h2 id="processing-ncbi-genomes">Processing NCBI Genomes</h2>
<p>A <a href="files/mag02-phylogenomics/Arcobacter-NCBI-METADATA.txt">metadata file</a> is also created by <code>ncbi-genome-download</code>.</p>
<pre class="bash"><code>
column -t Arcobacter-NCBI-METADATA.txt | head -n 4</code></pre>
<pre><code>
assembly_accession  bioproject  biosample   wgs_master  excluded_from_refseq    refseq_category relation_to_type_material   taxid   species_taxid   organism_name   infraspecific_name  isolate version_status  assembly_level  release_type    genome_rep  seq_rel_date    asm_name    submitter   gbrs_paired_asm paired_asm_comp ftp_path    local_filename
GCA_000014025.1 PRJNA16319  SAMN02604276            representative genome   assembly from type material 367737  28197   Arcobacter butzleri RM4018  strain=RM4018       latest  Complete Genome Major   Full    9/27/07 ASM1402v1   USDA - Agricultural Research Service, USA   GCF_000014025.1 identical   ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/014/025/GCA_000014025.1_ASM1402v1    ./Arcobacter_Genbank/genbank/bacteria/GCA_000014025.1/GCA_000014025.1_ASM1402v1_genomic.gbff.gz
GCA_000185325.1 PRJNA53195  SAMN00210774    AEPT00000000.1      na      888827  28197   Arcobacter butzleri JV22    strain=JV22     latest  Scaffold    Major   Full    1/14/11 ASM18532v1  Baylor College of Medicine  GCF_000185325.1 identical   ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/185/325/GCA_000185325.1_ASM18532v1   ./Arcobacter_Genbank/genbank/bacteria/GCA_000185325.1/GCA_000185325.1_ASM18532v1_genomic.gbff.gz
GCA_000215345.3 PRJNA67167  SAMN02604193            na      1036172 28197   Arcobacter butzleri 7h1h    strain=7h1h     latest  Complete Genome Major   Full    7/24/13 ASM21534v3  University of Liverpool GCF_000215345.2 identical   ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/215/345/GCA_000215345.3_ASM21534v3   ./Arcobacter_Genbank/genbank/bacteria/GCA_000215345.3/GCA_000215345.3_ASM21534v3_genomic.gbff.gz</code></pre>
<p>Anvi’o has a script that takes this metadata file and generates a <code>fasta.txt</code> file in the format compatible with anvi’o workflows. We will use the <code>--exclude-gene-calls-from-fasta-txt</code> because we want to do our own gene calling.</p>
<pre class="bash"><code>
anvi-script-process-genbank-metadata -m Arcobacter-NCBI-METADATA.txt \
                                     -o 01_Arcobacter-NCBI-GENOMES \
                                     --output-fasta-txt Arcobacter-fasta.txt
                                     --exclude-gene-calls-from-fasta-txt</code></pre>
<p>This command returns a directory, <code>01_Arcobacter-NCBI-GENOMES</code> full of fasta files and the file <a href="files/mag02-phylogenomics/Arcobacter-fasta.txt"><code>Arcobacter-fasta.txt</code></a>, which contains the name and fasta file path for each genome.</p>
<pre><code>
name    path
Aliiarcobacter_faecis_GCA_000585155_1   /pool/genomics/stri_istmobiome/data/HYPOXIA_DATA/HYPOXIA/19_COMPARATIVE_GENOMICS/01_Arcobacter-NCBI-GENOMES/Aliiarcobacter_faecis_GCA_000585155_1-contigs.fa
Aliiarcobacter_faecis_GCA_013201705_1   /pool/genomics/stri_istmobiome/data/HYPOXIA_DATA/HYPOXIA/19_COMPARATIVE_GENOMICS/01_Arcobacter-NCBI-GENOMES/Aliiarcobacter_faecis_GCA_013201705_1-contigs.fa
Arcobacter_anaerophilus_GCA_000935065_1 /pool/genomics/stri_istmobiome/data/HYPOXIA_DATA/HYPOXIA/19_COMPARATIVE_GENOMICS/01_Arcobacter-NCBI-GENOMES/Arcobacter_anaerophilus_GCA_000935065_1-contigs.fa</code></pre>
<p>We need to add our MAG to the fasta directory and its name to the <code>Arcobacter-fasta.txt</code> file. We will also change the names produced by <code>anvi-script-process-genbank-metadata</code> since they are not very meaningful. We can use the <code>Arcobacter-NCBI-METADATA.txt</code> file, specifically the <code>organism_name</code>, <code>infraspecific_name</code>, and/or <code>isolate</code> columns to cobble together new genome names. I wish I could say I had a clever way to do this task, but really it was just some copying and pasting. Sadly, I am not as cool as Alon and Meren. Anyway, here is the final <a href=""><code>Arcobacter-NCBI-METADATA.txt</code></a> file with our new names added and the <a href="files/mag02-phylogenomics/Arcobacter-fasta-rename.txt"><code>Arcobacter-fasta-rename</code></a> names file with MAG02 added.</p>
<blockquote>
<p>If you change the names they MUST be unique and contain no spaces, periods, dashes, or other weird characters. Letters, numbers, and underscores only.</p>
</blockquote>
<h1 id="running-snakemake-contigs-workflow">Running Snakemake Contigs Workflow</h1>
<p>Now it is time for some comparisons. First we will run the <em>contigs</em> workflow since this will produce annotated contig databases and properly formatted fasta files for all genomes. We will need these for phylogenomics. Ok, renamed file in hand, we use Snakemake workflow<span class="citation" data-cites="koster2012snakemake">(Köster and Rahmann <a href="#ref-koster2012snakemake">2012</a>)</span> to run the built-in anvi’o <a href="http://merenlab.org/2018/07/09/anvio-snakemake-workflows/#contigs-workflow">contigs workflow</a>. If you do not know what workflows are available you can run this command:</p>
<pre class="bash"><code>
anvi-run-workflow --list-workflows</code></pre>
<pre><code>
WARNING
===============================================
If you publish results from this workflow, please do not forget to cite
snakemake (doi:10.1093/bioinformatics/bts480)

Available workflows ..........................: contigs, metagenomics, pangenomics, phylogenomics</code></pre>
<p>Next, we need a config file for the <strong>contigs workflow</strong>. This file tells anvi’o what commands to run and how to run them. We run this command to generate a default config file, which can be modified as you see fit.</p>
<pre class="bash"><code>
anvi-run-workflow -w contigs --get-default-config contigs-default-config.json</code></pre>
<p><a href="files/mag02-phylogenomics/contigs-default-config.json">Click here</a> to see the file or grab it from the folded code below. Some commands we chose not to run but they remain in the workflow for posterity.</p>
<details>
<summary>Show/hide JSON-formatted configuration file.</summary>
<pre><code>
{
    "fasta_txt": "Arcobacter-fasta-rename.txt",
    "anvi_gen_contigs_database": {
        "--project-name": "{group}",
        "--description": "",
        "--skip-gene-calling": "",
        "--external-gene-calls": "",
        "--ignore-internal-stop-codons": "",
        "--skip-mindful-splitting": "",
        "--contigs-fasta": "",
        "--split-length": "",
        "--kmer-size": "",
        "--skip-predict-frame": "",
        "--prodigal-translation-table": "",
        "threads": ""
    },
    "centrifuge": {
        "threads": 2,
        "run": "",
        "db": ""
    },
    "anvi_run_hmms": {
        "run": true,
        "threads": 5,
        "--installed-hmm-profile": "",
        "--hmm-profile-dir": "",
        "--also-scan-trnas": ""
    },
    "anvi_run_ncbi_cogs": {
        "run": true,
        "threads": 5,
        "--cog-data-dir": "/pool/genomics/stri_istmobiome/dbs/cog_db/",
        "--sensitive": "",
        "--temporary-dir-path": "/pool/genomics/stri_istmobiome/dbs/cog_db/tmp/",
        "--search-with": ""
    },
    "anvi_run_scg_taxonomy": {
        "run": true,
        "threads": 6,
        "--scgs-taxonomy-data-dir": "/pool/genomics/stri_istmobiome/dbs/scgs-taxonomy-data/tmp/"
    },
    "anvi_run_trna_scan": {
        "run": true,
        "threads": 6,
        "--trna-cutoff-score": ""
    },
    "anvi_script_reformat_fasta": {
        "run": true,
        "--prefix": "{group}",
        "--simplify-names": true,
        "--keep-ids": "",
        "--exclude-ids": "",
        "--min-len": "",
        "threads": ""
    },
    "emapper": {
        "--database": "bact",
        "--usemem": true,
        "--override": true,
        "path_to_emapper_dir": "",
        "threads": ""
    },
    "anvi_script_run_eggnog_mapper": {
        "--use-version": "0.12.6",
        "run": "",
        "--cog-data-dir": "",
        "--drop-previous-annotations": "",
        "threads": ""
    },
    "gen_external_genome_file": {
        "threads": ""
    },
    "export_gene_calls_for_centrifuge": {
        "threads": ""
    },
    "anvi_import_taxonomy_for_genes": {
        "threads": ""
    },
    "annotate_contigs_database": {
        "threads": ""
    },
    "anvi_get_sequences_for_gene_calls": {
        "threads": ""
    },
    "gunzip_fasta": {
        "threads": ""
    },
    "reformat_external_gene_calls_table": {
        "threads": ""
    },
    "reformat_external_functions": {
        "threads": ""
    },
    "import_external_functions": {
        "threads": ""
    },
    "anvi_run_pfams": {
        "run": true,
        "--pfam-data-dir": "/pool/genomics/stri_istmobiome/dbs/pfam_db/",
        "threads": ""
    },
    "output_dirs": {
        "FASTA_DIR": "02_FASTA",
        "CONTIGS_DIR": "03_CONTIGS",
        "LOGS_DIR": "00_LOGS"
    },
    "max_threads": "",
    "config_version": "1",
    "workflow_name": "contigs"
}
</code></pre>
</details>
<p><br/></p>
<p>We can also run a couple of sanity checks to make sure everything is good before starting the job. How about a workflow graph to visually inspect the steps?</p>
<pre class="bash"><code>
anvi-run-workflow -w contigs -c contigs-default-config.json --save-workflow-graph</code></pre>
<pre><code>
Config Error: Well, the fasta.txt entry name contains characters that anvi&#39;o does not like :/
              Please limit the characters to ASCII letters, digits, and the underscore (&#39;_&#39;)
              character.</code></pre>
<p>Hum? What did I tell you, don’t listen to me. Turns out I had dashes (<code>-</code>) in some genome names in the <code>Arcobacter-fasta-rename.txt</code> file. I took those out and reran the command.</p>
<pre><code>
WARNING
===============================================
If you publish results from this workflow, please do not forget to cite
snakemake (doi:10.1093/bioinformatics/bts480)


WARNING
===============================================
We are initiating parameters for the contigs workflow


WARNING
===============================================
You chose to define --temporary-dir-path for the rule anvi_run_ncbi_cogs in the
config file as /pool/genomics/stri_istmobiome/dbs/cog_db/tmp/. while this is
allowed, know that you are doing so at your own risk. The reason this is risky
is because this rule uses a wildcard/wildcards and hence is probably running
more than once, and this might cause a problem. In case you wanted to know,
these are the wildcards used by this rule: {group}

Workflow DOT file ............................: workflow.dot
Workflow PNG file ............................: workflow.png</code></pre>
<p>Success. Let’s have a look at the workflow. Only four genomes are shown for clarity but each step will be conducted for all genomes.</p>
<blockquote>
<p>Click on the image to zoom in or download a copy.</p>
</blockquote>
<details>
<summary>Show/hide DAG R Code script</summary>
<pre><code>
````
```{r workflow2, eval=FALSE}
dag <- grViz ("
digraph boxes_and_circles {
  graph [layout = dot, align=center]

  node [shape = rectangle, style = 'rounded,filled' fontname=sans, fontsize=12, penwidth=4]
  edge[penwidth=4, color=grey];


0[label = 'contigs_target_rule', color = 'grey'];
1[label = 'annotate_contigs_database', color = '#009E73'];
2[label = 'annotate_contigs_database', color = '#009E73'];
3[label = 'annotate_contigs_database', color = '#009E73'];
4[label = 'annotate_contigs_database', color = '#009E73'];
5[label = 'anvi_gen_contigs_database', color = '#009E73'];
6[label = 'anvi_run_hmms', color = '#CC79A7'];
7[label = 'anvi_run_ncbi_cogs', color = '#E69F00'];
8[label = 'anvi_run_scg_taxonomy', color = '#CC79A7'];
9[label = 'anvi_run_trna_scan', color = '#CC79A7'];
10[label = 'anvi_run_pfams', color = '#E69F00'];
11[label = 'anvi_gen_contigs_database', color = '#009E73'];
12[label = 'anvi_run_hmms', color = '#CC79A7'];
13[label = 'anvi_run_ncbi_cogs', color = '#E69F00'];
14[label = 'anvi_run_scg_taxonomy', color = '#CC79A7'];
15[label = 'anvi_run_trna_scan', color = '#CC79A7'];
16[label = 'anvi_run_pfams', color = '#E69F00'];
17[label = 'anvi_gen_contigs_database', color = '#009E73'];
18[label = 'anvi_run_hmms', color = '#CC79A7'];
19[label = 'anvi_run_ncbi_cogs', color = '#E69F00'];
20[label = 'anvi_run_scg_taxonomy', color = '#CC79A7'];
21[label = 'anvi_run_trna_scan', color = '#CC79A7'];
22[label = 'anvi_run_pfams', color = '#E69F00'];
23[label = 'anvi_gen_contigs_database', color = '#009E73'];
24[label = 'anvi_run_hmms', color = '#CC79A7'];
25[label = 'anvi_run_ncbi_cogs', color = '#E69F00'];
26[label = 'anvi_run_scg_taxonomy', color = '#CC79A7'];
27[label = 'anvi_run_trna_scan', color = '#CC79A7'];
28[label = 'anvi_run_pfams', color = '#E69F00'];
29[label = 'anvi_script_reformat_fasta', color = '#56B4E9'];
30[label = 'anvi_script_reformat_fasta', color = '#56B4E9'];
31[label = 'anvi_script_reformat_fasta', color = '#56B4E9'];
32[label = 'anvi_script_reformat_fasta', color = '#56B4E9'];
33[label = 'anvi_script_reformat_fasta_prefix_only\n Arcobacter_canalis', color = '#56B4E9'];
34[label = 'anvi_script_reformat_fasta_prefix_only\n Arcobacter_marinus', color = '#56B4E9'];
35[label = 'anvi_script_reformat_fasta_prefix_only\n Poseidonibacter', color = '#56B4E9'];
36[label = 'anvi_script_reformat_fasta_prefix_only\n MAG02', color = '#56B4E9'];



1->0; 2->0; 3->0; 4->0; 5->1; 6->1; 7->1;
8->1; 9->1; 10->1; 11->2; 12->2; 13->2; 14->2;
15->2; 16->2; 17->3; 18->3; 19->3; 20->3;
21->3; 22->3; 23->4; 24->4; 25->4; 26->4;
27->4; 28->4; 29->5; 5->6; 5->7; 6->8; 5->8;
5->9; 5->10; 30->11; 11->12; 11->13; 12->14;
11->14; 11->15; 11->16; 31->17; 17->18; 17->19;
18->20; 17->20; 17->21; 17->22; 32->23; 23->24;
23->25; 24->26; 23->26; 23->27; 23->28; 33->29;
34->30; 35->31; 36->32;

    graph [nodesep = 0.1]
{ rank=same; 33, 34, 35, 36 }

}
")

export_svg(dag) %>%
  charToRaw() %>%
  rsvg() %>%
  png::writePNG("figures/mag02-phylogenomics/workflow.png")
```
````
</code></pre>
</details>
<div class="l-body-outset">
<figure>
<a href="figures/mag02-phylogenomics/workflow.png"> <img src="figures/mag02-phylogenomics/workflow.png" class="thumbnail"> </a>
<figcaption>
<strong>DAG (directed acyclic graph) of the contigs workflow.</strong>Colors indicate broad divisions of workflow: sky blue, short-read prep &amp; co-assembly; sky blue, formatting; blueish green, databases; reddish purple, SCG annotation; orange, functional classification.
</figcaption>
</figure>
</div>
<p>Yup. Looks good and makes sense. And a dry run just to be certain</p>
<pre class="bash"><code>
anvi-run-workflow -w contigs -c contigs-default-config.json --dry-run</code></pre>
<p>No complaints. Things are looking good. We are ready to run the workflow.</p>
<p>For all of you HYDRA users, here is the job script to run the contigs workflow.</p>
<details>
<summary>Show/hide HYDRA SNAKEMAKE job script</summary>
<pre><code>
# /bin/sh
# ----------------Parameters---------------------- #
#$ -S /bin/sh
#$ -pe mthread 20
#$ -q sThC.q
#$ -l mres=120G,h_data=6G,h_vmem=6G
#$ -cwd
#$ -j y
#$ -N job_01b_run_contigs_workflow
#$ -o hydra_logs/job_01b_run_contigs_workflow.log
#$ -M scottjj@si.edu
#
# ----------------Modules------------------------- #
module load gcc/4.9.2
#
# ----------------Your Commands------------------- #
#
echo + `date` job $JOB_NAME started in $QUEUE with jobID=$JOB_ID on $HOSTNAME
echo + NSLOTS = $NSLOTS
#
### CALLING ANVIO ###
export PATH=/home/scottjj/miniconda3:$PATH
export PATH=/home/scottjj/miniconda3/bin:$PATH
source activate anvio-master
#
#############################################
### Making sure everything is cool
which python
python --version
source /home/scottjj/virtual-envs/anvio-master/bin/activate
which python
python --version
which anvi-interactive
diamond --version
anvi-self-test -v
#############################################
#
## pfam tmp
rm -r /pool/genomics/stri_istmobiome/dbs/pfam_db/tmp_data/
mkdir -p /pool/genomics/stri_istmobiome/dbs/pfam_db/tmp_data/
TMPDIR="/pool/genomics/stri_istmobiome/dbs/pfam_db/tmp_data/"
#
anvi-run-workflow -w contigs -c contigs-default-config.json --additional-params --jobs 20 --resources nodes=20 --keep-going --rerun-incomplete --unlock
anvi-run-workflow -w contigs -c contigs-default-config.json --additional-params --jobs 20 --resources nodes=20 --keep-going --rerun-incomplete
#
echo = `date` job $JOB_NAME don
</code></pre>
</details>
<p><br/></p>
<p>Finished? Sweet. What we have now are contig databases for 124 <em>Arcobacter</em>-group genomes—databases that contain gene calls, functional annotations, HMM profiles, etc. Remember, our goal is to better understand MAG02 by asking what it is related to and where related genomes are found in nature. Before continuing, we need to ask two questions about our genome collection and then address each of them in turn.</p>
<ol type="1">
<li>How complete are these genomes? We will rely on single copy genes for phylogenomic analysis so there is no reason to keep genomes that are missing too many genes.<br />
</li>
<li>Do we <em>need</em> all of these genomes? It is likely that some of these genomes are highly similar to each other. What we are after is a representative sampling of <em>Arcobacter</em> genomic diversity.</li>
</ol>
<h1 id="curate-genome-collection">Curate Genome Collection</h1>
<p>The commands we ran above retrieved 124 genomes from the <em>Arcobacter</em> group. Here we perform a series of simple steps to eliminate unwanted or unecessary genomes from the collection.</p>
<blockquote>
<p>We start with 125 genomes. This includes MAG02</p>
</blockquote>
<p>We have two goals here. First, we need to get rid of low completion genomes since they will compromise our ability to use single copy core genes for phylogenomics. Second, we want to keep only genomes that have a 16S rRNA sequence. Why? Good question. Because we found an ASV in our 16S rRNA analysis that is 100% identical across it’s length to a 16S rRNA sequence retrieved from MAG02. In addition to the phylogenomic analysis we will also conduct a 16S rRNA analysis of MAG02 with publicly available genomes to see how the topologies compare.</p>
<h2 id="estimate-taxonomy">Estimate Taxonomy</h2>
<p>Microbial taxonomy is a fluid system, meaning that classifications change often. One thing we can do is look at the predicted taxonomy for each genome even though they each came prepackaged with a taxonomic designation. We already know <a href="mg-binning.html#esimate-mag-taxonomy">MAG02’s predicted taxonomy is <em>Poseidonibacter</em></a> but what about the rest</p>
<p>To do this we will again run the command <code>anvi-estimate-scg-taxonomy</code>, which uses 22 ribosomal genes<span class="citation" data-cites="parks2018gtgb">(Parks et al. <a href="#ref-parks2018gtgb">2018</a>)</span> from the <a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database (GTDB)</a> to estimate taxonomy. We added these classifications earlier when we contigs workflow. The workflow ran its own instance of <code>anvi-run-scg-taxonomy</code> for each genome. We can use a nifty <code>for</code> loop and then concatenate the output files and make a handy table. Pretty sure there is a better way :/</p>
<pre class="bash"><code>
for genome in `cat list_genomes.txt`; 
do 
       anvi-estimate-scg-taxonomy \
       -c 03_CONTIGS/$genome-contigs.db \
       -o 09_SCG_TAXONOMY/$genome-SCG.txt; 
done</code></pre>
<p><br/></p>
<div class="layout-chunk" data-layout="l-page">
<div id="rlmoheh55qr4vfp52n7o" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="rlmoheh55qr4vfp52n7o">{"x":{"filter":"none","extensions":["Buttons","FixedColumns"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">Completion estimates of Arcobacter genomes.<\/caption>","autoHideNavigation":true,"data":[["Aliiarcobacter_faecis_AF1078","Aliiarcobacter_faecis_CCUG_66484","Arcobacter_anaerophilus_DSM_24636_A","Arcobacter_anaerophilus_DSM_24636_B","Arcobacter_anaerophilus_IR_1","Arcobacter_aquimarinus_CECT_8442","Arcobacter_aquimarinus_W63","Arcobacter_butzleri_7h1h","Arcobacter_butzleri_ED_1","Arcobacter_butzleri_JV22_A","Arcobacter_butzleri_JV22_B","Arcobacter_butzleri_L348","Arcobacter_butzleri_L349","Arcobacter_butzleri_L350","Arcobacter_butzleri_L351","Arcobacter_butzleri_L352","Arcobacter_butzleri_L353","Arcobacter_butzleri_L354","Arcobacter_butzleri_L355","Arcobacter_butzleri_RM4018","Arcobacter_canalis_F138_33","Arcobacter_canalis_LMG_29148","Arcobacter_cibarius_LMG_21996","Arcobacter_cloacae_CECT_7834","Arcobacter_cloacae_F26","Arcobacter_cloacae_LMG_26153","Arcobacter_cryaerophilus_ATCC_43158_ATCC_43158","Arcobacter_cryaerophilus_ATCC_43158_LMG_9904","Arcobacter_cryaerophilus_D2610","Arcobacter_defluvii_CECT_7697","Arcobacter_defluvii_LMG_25694","Arcobacter_ellisii_CECT_7837","Arcobacter_ellisii_LMG_26155","Arcobacter_halophilus_CCUG_53805","Arcobacter_halophilus_DSM_18005","Arcobacter_halophilus_F166_45","Arcobacter_lanthieri_AF1430","Arcobacter_lanthieri_AF1440","Arcobacter_lanthieri_AF1581","Arcobacter_lanthieri_LMG_28516","Arcobacter_lekithochrous_DSM_100870","Arcobacter_lekithochrous_LFT_1_7","Arcobacter_lekithochrous_LMG_28652","Arcobacter_marinus_CECT_7727","Arcobacter_marinus_F140_37","Arcobacter_marinus_JCM_15502","Arcobacter_marinus_MARC_MIP3H16","Arcobacter_marinus_SH_4D_Col1","Arcobacter_molluscorum_LMG_25693_CECT_7696","Arcobacter_molluscorum_LMG_25693_F98_3","Arcobacter_mytili_LMG_24559_CECT_7386","Arcobacter_mytili_LMG_24559_LMG_24559","Arcobacter_nitrofigilis_DSM_7299_DSM_7299","Arcobacter_pacificus_LMG_26638","Arcobacter_peruensis_PSE_93","_Arcobacter_porcinus_117434","_Arcobacter_porcinus_213","_Arcobacter_porcinus_216","_Arcobacter_porcinus_CCUG_56899","_Arcobacter_porcinus_DU19","Arcobacter_skirrowii_CCUG_10374_CCUG_10374_A","Arcobacter_skirrowii_CCUG_10374_CCUG_10374_B","Arcobacter_skirrowii_CCUG_10374_LMG_6621","Arcobacter_sp_31_11_sub10_T18","Arcobacter_sp_AF1028","Arcobacter_sp_BM102","Arcobacter_sp_BM504","Arcobacter_sp_CECT_8983","Arcobacter_sp_CECT_8985","Arcobacter_sp_CECT_8986","Arcobacter_sp_CECT_8987","Arcobacter_sp_CECT_8989","Arcobacter_sp_CECT_9188","Arcobacter_sp_CECT_9230","Arcobacter_sp_CECT_9299","Arcobacter_sp_CPC16","Arcobacter_sp_CPC309","Arcobacter_sp_CSH_ALOHA_XIX_Bin_5","Arcobacter_sp_DOLZORAL124_29_21","Arcobacter_sp_EP1","Arcobacter_sp_F155","Arcobacter_sp_F2176","Arcobacter_sp_FW59","Arcobacter_sp_HD9_500m_PIT_SAG02","Arcobacter_sp_HD9_500m_PIT_SAG03","Arcobacter_sp_HyVt_597","Arcobacter_sp_LA11","Arcobacter_sp_LMG_29976","Arcobacter_sp_LPB0137","Arcobacter_sp_L","Arcobacter_sp_NORP14","Arcobacter_sp_NORP36","Arcobacter_sp_RW43_9","Arcobacter_sp_SM1702","Arcobacter_sp_SZUA_449","Arcobacter_sp_SZUA_456","Arcobacter_sp_UBA1348","Arcobacter_sp_UBA2431","Arcobacter_sp_UBA2434","Arcobacter_sp_UBA2438","Arcobacter_sp_UBA2443","Arcobacter_sp_UBA4038","Arcobacter_sp_UBA6788","Arcobacter_sp_UBA6789","Arcobacter_sp_UBA7394","Arcobacter_suis_CECT_7833","Arcobacter_thereius_LMG_24486_A","Arcobacter_thereius_LMG_24486_B","Arcobacter_trophiarum_LMG_25534_A","Arcobacter_trophiarum_LMG_25534_B","Arcobacter_venerupis_CECT7836","Arcobacter_venerupis_LMG_26156","Halarcobacter_bivalviorum_CECT_7835","Halarcobacter_bivalviorum_F118_4","Halarcobacter_bivalviorum_LMG_26154","Halarcobacter_ebronensis_CECT_8441","Halarcobacter_ebronensis_CECT_8993","Halarcobacter_ebronensis_LMG_27922","Halarcobacter_mediterraneus_F156_34","MAG02","Poseidonibacter_sp_JOD_M_6","Poseidonibacter_sp_SJOD_M_33","Poseidonibacter_sp_SJOD_M_5","Pseudoarcobacter_acticola_KCTC_52212","Pseudoarcobacter_caeni_RW17_10"],[22,22,21,21,21,19,19,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,19,19,19,22,22,22,22,22,23,22,22,22,22,22,22,22,22,22,23,22,22,22,22,22,22,22,22,16,16,22,15,22,22,22,22,22,22,22,22,22,22,22,21,22,16,20,20,14,17,21,20,21,7,3,10,4,22,17,22,20,0,5,1,21,22,22,22,21,22,22,21,21,0,22,22,22,22,22,16,19,18,22,21,22,22,21,19,21,21,19,15,15,16,17,16,15,20,22,22,22,20,22],[16,16,21,21,21,19,19,22,22,22,22,22,22,22,22,22,22,22,22,22,21,21,15,19,19,19,21,21,20,16,16,11,11,21,21,21,21,21,20,21,13,13,11,21,21,21,21,21,22,22,5,5,20,3,22,13,13,12,14,12,21,21,21,22,16,21,20,16,11,16,4,17,19,15,19,6,2,9,4,22,3,17,15,0,5,1,20,19,22,16,21,22,22,19,21,0,21,21,21,21,21,15,12,11,14,12,14,14,14,14,12,12,4,15,15,7,7,7,3,9,16,16,16,11,22],["Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","None","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","None","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria"],["Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","None","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","None","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota","Campylobacterota"],["Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","None","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","None","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria","Campylobacteria"],["Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","None","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","None","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales","Campylobacterales"],["Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","None","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Sulfurimonadaceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","None","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae","Arcobacteraceae"],["Aliarcobacter","Aliarcobacter","Halarcobacter","Halarcobacter","Halarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Malaciobacter","Malaciobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Malaciobacter","Malaciobacter","Malaciobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Poseidonibacter","Poseidonibacter","Poseidonibacter","Malaciobacter","Malaciobacter","Malaciobacter","Malaciobacter","Malaciobacter","Malaciobacter","Malaciobacter","Malaciobacter","Malaciobacter","Arcobacter","Halarcobacter","Poseidonibacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","NORP36","Aliarcobacter","Halarcobacter","Arcobacter","None","Malaciobacter","Malaciobacter","UBA4036","None","Aliarcobacter","Aliarcobacter","Aliarcobacter","Poseidonibacter","Poseidonibacter","NORP36","None","Arcobacter","Halarcobacter","Arcobacter","Aliarcobacter","None","NORP36","UBA4036","Halarcobacter","Aliarcobacter","Poseidonibacter","Aliarcobacter","GCA-2733885","NORP36","Aliarcobacter","Poseidonibacter","Halarcobacter","None","Malaciobacter","Malaciobacter","Malaciobacter","Malaciobacter","Malaciobacter","Malaciobacter","Aliarcobacter","Aliarcobacter","Poseidonibacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","Aliarcobacter","None","None","Halarcobacter","Halarcobacter","Halarcobacter","Halarcobacter","Poseidonibacter","Poseidonibacter","Poseidonibacter","Poseidonibacter","Aliarcobacter","Aliarcobacter"],["Aliarcobacter_faecis","Aliarcobacter_faecis","Halarcobacter_anaerophilus","Halarcobacter_anaerophilus","Halarcobacter_anaerophilus","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","Malaciobacter_canalis","Malaciobacter_canalis","Aliarcobacter_cibarius","None","None","None","Aliarcobacter_cryaerophilus","Aliarcobacter_cryaerophilus","Aliarcobacter_cryaerophilus","Aliarcobacter_sp000284235","Aliarcobacter_sp000284235","Aliarcobacter_sp002452915","Aliarcobacter_sp002452915","Malaciobacter_halophilus","Malaciobacter_halophilus","Malaciobacter_canalis","Aliarcobacter_lanthieri","Aliarcobacter_lanthieri","Aliarcobacter_lanthieri","Aliarcobacter_lanthieri","Poseidonibacter_lekithochrous","Poseidonibacter_lekithochrous","Poseidonibacter_lekithochrous","Malaciobacter_canalis","Malaciobacter_canalis","Malaciobacter_canalis","Malaciobacter_canalis","Malaciobacter_canalis","Malaciobacter_molluscorum","Malaciobacter_molluscorum","Malaciobacter_halophilus","Malaciobacter_halophilus","Arcobacter_nitrofigilis","Halarcobacter_sp003252105","Poseidonibacter_sp001956695","Aliarcobacter_porcinus","Aliarcobacter_porcinus","Aliarcobacter_porcinus","Aliarcobacter_porcinus","Aliarcobacter_porcinus","Aliarcobacter_skirrowii","Aliarcobacter_skirrowii","Aliarcobacter_skirrowii","NORP36_sp002163065","Aliarcobacter_faecis","Halarcobacter_sp002869565","Arcobacter_sp002869535","None","Malaciobacter_molluscorum","Malaciobacter_molluscorum","UBA4036_sp002382325","None","Aliarcobacter_lanthieri","Aliarcobacter_lanthieri","Aliarcobacter_skirrowii","Poseidonibacter_sp002685075","Poseidonibacter_sp001956695","NORP36_sp002733685","None","Arcobacter_sp001655195","Halarcobacter_sp001895145","Arcobacter_nitrofigilis","Aliarcobacter_lanthieri","None","NORP36_sp002163065","UBA4036_sp002382325","Halarcobacter_sp001895145","Aliarcobacter_cryaerophilus","Poseidonibacter_sp001956695","Aliarcobacter_sp000284235","GCA-2733885_sp002733885","NORP36_sp002733685","None","Poseidonibacter_sp001956695","Halarcobacter_sp003252105","None","Malaciobacter_canalis","Malaciobacter_canalis","Malaciobacter_canalis","Malaciobacter_canalis","Malaciobacter_canalis","Malaciobacter_canalis","Aliarcobacter_sp002452915","Aliarcobacter_sp000284235","Poseidonibacter_sp002470185","Aliarcobacter_sp003063245","Aliarcobacter_thereius","Aliarcobacter_thereius","Aliarcobacter_cryaerophilus","Aliarcobacter_cryaerophilus","Aliarcobacter_sp003063245","Aliarcobacter_sp003063245","Aliarcobacter_skirrowii","None","None","Halarcobacter_anaerophilus","Halarcobacter_anaerophilus","Halarcobacter_anaerophilus","Halarcobacter_anaerophilus","None","Poseidonibacter_sp001956695","Poseidonibacter_sp001956695","Poseidonibacter_sp001956695","Aliarcobacter_sp003063245","Aliarcobacter_sp003063245"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>genome<\/th>\n      <th>total_scgs<\/th>\n      <th>support_scgs<\/th>\n      <th>t_domain<\/th>\n      <th>t_phylum<\/th>\n      <th>t_class<\/th>\n      <th>t_order<\/th>\n      <th>t_family<\/th>\n      <th>t_genus<\/th>\n      <th>t_species<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":0},{"className":"dt-right","targets":[1,2]}],"dom":"Blfrtip","buttons":["csv","copy"],"pageLength":5,"lengthMenu":[5,10,50,125],"scrollX":true,"scrollCollapse":true,"scrollY":true,"paging":true,"fixedColumns":{"leftColumns":1,"rightColumns":0},"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<h2 id="completeness">Completeness</h2>
<p>Let’s start by checking the completeness (and redundancy) of each genome based on the presence of single copy core genes (SCG) using the HMM profiles, which were added to the contig database for each genome during the workflow. We can either run the same command 125 times and concatenate the outputs, or we can create an <em>external-genomes</em> file—a two-column TAB-delimited flat text file that lists anvi’o contigs databases. The first item in the header line reads ‘<em>name</em>’, and the second reads ‘<em>contigs_db_path</em>’. Each line in the file describes a single entry, where the first column is the name of the genome (or MAG), and the second column is the anvi’o contigs database generated for this genome.</p>
<aside>
You can access our original <a href="files/mag02-phylogenomics/external-genomes.txt">external-genomes file here</a>.
</aside>
<pre class="bash"><code>
anvi-estimate-genome-completeness \
                     -e external-genomes.txt 
                     -o completness.txt</code></pre>
<p>Just like that, a table of basic stats for every genome in the collection. First notice the completion estimate of MAG02 is around 93%, which is good, but its total length is among the smallest, meaning we have a pretty incomplete genome. Not so good but it is what it is.</p>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="15f785tpvzzud9tgdwd7" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="15f785tpvzzud9tgdwd7">{"x":{"filter":"none","extensions":["Buttons","FixedColumns"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">Completion estimates of Arcobacter genomes.<\/caption>","autoHideNavigation":true,"data":[["Aliiarcobacter_faecis_AF1078","Aliiarcobacter_faecis_CCUG_66484","Arcobacter_anaerophilus_DSM_24636_A","Arcobacter_anaerophilus_DSM_24636_B","Arcobacter_anaerophilus_IR_1","Arcobacter_aquimarinus_CECT_8442","Arcobacter_aquimarinus_W63","Arcobacter_butzleri_7h1h","Arcobacter_butzleri_ED_1","Arcobacter_butzleri_JV22_A","Arcobacter_butzleri_JV22_B","Arcobacter_butzleri_L348","Arcobacter_butzleri_L349","Arcobacter_butzleri_L350","Arcobacter_butzleri_L351","Arcobacter_butzleri_L352","Arcobacter_butzleri_L353","Arcobacter_butzleri_L354","Arcobacter_butzleri_L355","Arcobacter_butzleri_RM4018","Arcobacter_canalis_F138_33","Arcobacter_canalis_LMG_29148","Arcobacter_cibarius_LMG_21996","Arcobacter_cloacae_CECT_7834","Arcobacter_cloacae_F26","Arcobacter_cloacae_LMG_26153","Arcobacter_cryaerophilus_ATCC_43158_ATCC_43158","Arcobacter_cryaerophilus_ATCC_43158_LMG_9904","Arcobacter_cryaerophilus_D2610","Arcobacter_defluvii_CECT_7697","Arcobacter_defluvii_LMG_25694","Arcobacter_ellisii_CECT_7837","Arcobacter_ellisii_LMG_26155","Arcobacter_halophilus_CCUG_53805","Arcobacter_halophilus_DSM_18005","Arcobacter_halophilus_F166_45","Arcobacter_lanthieri_AF1430","Arcobacter_lanthieri_AF1440","Arcobacter_lanthieri_AF1581","Arcobacter_lanthieri_LMG_28516","Arcobacter_lekithochrous_DSM_100870","Arcobacter_lekithochrous_LFT_1_7","Arcobacter_lekithochrous_LMG_28652","Arcobacter_marinus_CECT_7727","Arcobacter_marinus_F140_37","Arcobacter_marinus_JCM_15502","Arcobacter_marinus_MARC_MIP3H16","Arcobacter_marinus_SH_4D_Col1","Arcobacter_molluscorum_LMG_25693_CECT_7696","Arcobacter_molluscorum_LMG_25693_F98_3","Arcobacter_mytili_LMG_24559_CECT_7386","Arcobacter_mytili_LMG_24559_LMG_24559","Arcobacter_nitrofigilis_DSM_7299_DSM_7299","Arcobacter_pacificus_LMG_26638","Arcobacter_peruensis_PSE_93","_Arcobacter_porcinus_117434","_Arcobacter_porcinus_213","_Arcobacter_porcinus_216","_Arcobacter_porcinus_CCUG_56899","_Arcobacter_porcinus_DU19","Arcobacter_skirrowii_CCUG_10374_CCUG_10374_A","Arcobacter_skirrowii_CCUG_10374_CCUG_10374_B","Arcobacter_skirrowii_CCUG_10374_LMG_6621","Arcobacter_sp_31_11_sub10_T18","Arcobacter_sp_AF1028","Arcobacter_sp_BM102","Arcobacter_sp_BM504","Arcobacter_sp_CECT_8983","Arcobacter_sp_CECT_8985","Arcobacter_sp_CECT_8986","Arcobacter_sp_CECT_8987","Arcobacter_sp_CECT_8989","Arcobacter_sp_CECT_9188","Arcobacter_sp_CECT_9230","Arcobacter_sp_CECT_9299","Arcobacter_sp_CPC16","Arcobacter_sp_CPC309","Arcobacter_sp_CSH_ALOHA_XIX_Bin_5","Arcobacter_sp_DOLZORAL124_29_21","Arcobacter_sp_EP1","Arcobacter_sp_F155","Arcobacter_sp_F2176","Arcobacter_sp_FW59","Arcobacter_sp_HD9_500m_PIT_SAG02","Arcobacter_sp_HD9_500m_PIT_SAG03","Arcobacter_sp_HyVt_597","Arcobacter_sp_LA11","Arcobacter_sp_L","Arcobacter_sp_LMG_29976","Arcobacter_sp_LPB0137","Arcobacter_sp_NORP14","Arcobacter_sp_NORP36","Arcobacter_sp_RW43_9","Arcobacter_sp_SM1702","Arcobacter_sp_SZUA_449","Arcobacter_sp_SZUA_456","Arcobacter_sp_UBA1348","Arcobacter_sp_UBA2431","Arcobacter_sp_UBA2434","Arcobacter_sp_UBA2438","Arcobacter_sp_UBA2443","Arcobacter_sp_UBA4038","Arcobacter_sp_UBA6788","Arcobacter_sp_UBA6789","Arcobacter_sp_UBA7394","Arcobacter_suis_CECT_7833","Arcobacter_thereius_LMG_24486_A","Arcobacter_thereius_LMG_24486_B","Arcobacter_trophiarum_LMG_25534_A","Arcobacter_trophiarum_LMG_25534_B","Arcobacter_venerupis_CECT7836","Arcobacter_venerupis_LMG_26156","Halarcobacter_bivalviorum_CECT_7835","Halarcobacter_bivalviorum_F118_4","Halarcobacter_bivalviorum_LMG_26154","Halarcobacter_ebronensis_CECT_8441","Halarcobacter_ebronensis_CECT_8993","Halarcobacter_ebronensis_LMG_27922","Halarcobacter_mediterraneus_F156_34","MAG02","Poseidonibacter_sp_JOD_M_6","Poseidonibacter_sp_SJOD_M_33","Poseidonibacter_sp_SJOD_M_5","Pseudoarcobacter_acticola_KCTC_52212","Pseudoarcobacter_caeni_RW17_10"],["BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BLANK","BLANK","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BLANK","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA"],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0.4,0.2,0.6,0.6,1,1,1,1,1,0.8,0.4,1,1,1,1,1,1,1,1,1,0.7,1,1,1,1,1,0.7,0.8,0.7,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0.9,1,1,1,1,1],[97.18,97.18,97.18,97.18,97.18,95.77,97.18,97.18,97.18,97.18,97.18,95.77,97.18,97.18,97.18,97.18,95.77,95.77,95.77,97.18,97.18,97.18,97.18,97.18,97.18,97.18,98.59,98.59,98.59,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,97.18,98.59,98.59,98.59,92.96,97.18,98.59,97.18,97.18,98.59,97.18,95.77,97.18,97.18,97.18,98.59,50.7,42.25,61.97,67.61,97.18,97.18,97.18,97.18,0,0,30.99,98.59,97.18,98.59,97.18,97.18,97.18,97.18,97.18,97.18,0,97.18,97.18,97.18,97.18,97.18,74.65,81.69,71.83,94.37,98.59,97.18,97.18,98.59,98.59,97.18,97.18,97.18,97.18,95.77,97.18,97.18,97.18,98.59,92.96,97.18,97.18,97.18,97.18,97.18],[1.41,1.41,0,0,1.41,2.82,2.82,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,2.82,1.41,0,0,0,1.41,1.41,1.41,1.41,1.41,4.23,2.82,0,0,0,1.41,0,0,0,0,0,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,1.41,2.82,1.41,0,1.41,1.41,1.41,1.41,0,0,1.41,1.41,1.41,1.41,1.41,5.63,4.23,5.63,1.41,2.82,1.41,4.23,1.41,0,0,2.82,1.41,1.41,7.04,1.41,1.41,0,1.41,0,2.82,0,0,0,0,0,0,0,1.41,1.41,4.23,1.41,1.41,1.41,4.23,5.63,2.82,1.41,12.68,1.41,1.41,1.41,4.23,0,2.82,8.45,1.41,1.41,1.41,2.82,2.82],[133,122,156,150,161,132,126,112,112,128,114,205,132,145,151,125,117,133,123,117,144,141,110,218,130,140,105,105,102,187,151,239,139,139,166,154,117,117,115,113,177,586,231,215,142,144,142,141,140,203,188,151,159,131,138,95,85,85,100,102,98,111,145,203,126,163,130,144,151,145,128,263,160,141,112,171,185,316,147,350,177,211,241,18,58,582,168,148,405,142,157,157,112,169,145,57,129,129,130,136,128,96,209,128,270,131,95,95,95,323,363,160,295,139,134,222,239,158,157,504,218,213,238,150,140],[2496885,2448998,2975533,3016922,3247305,2461253,2520808,2253233,2256675,2302726,2297844,2471357,2277932,2285472,2290244,2233660,2148038,2221793,2220910,2341251,2782711,2829476,2201349,2775030,2509393,2810923,2108340,2107083,2055914,2938417,3021240,2799964,2799949,2812536,2746292,2965780,2236207,2287768,2260089,2276457,3568672,3607848,3502068,2866788,2779622,2917098,2901897,2818794,2800582,2756560,2973749,3037595,3192235,2657991,2816102,1967723,1781662,1778869,2018219,1883443,1969846,1953172,1973791,1955517,2414790,3490316,2683134,2737685,2788337,2648654,2533736,2927358,2210469,2426792,1843089,2679798,2906968,2450282,1186383,3005869,2798678,3272117,2578341,162008,397249,2123335,3098976,2947662,2191175,2866429,3146531,3322907,2215371,2919928,3023535,386814,2843595,2845524,2838103,2829374,2758171,2070544,2386140,2111771,2539241,2639269,1909306,1915080,1915613,2004889,3277178,3209085,2745053,2707754,2684688,3147783,3227488,3170683,2828874,1180544,2858436,2958551,3206717,3019071,2415164]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>genome<\/th>\n      <th>domain<\/th>\n      <th>confidence<\/th>\n      <th>%_completion<\/th>\n      <th>%_redundancy<\/th>\n      <th>num_contigs<\/th>\n      <th>total_length<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":0},{"className":"dt-right","targets":[2,3,4,5,6]}],"dom":"Blfrtip","buttons":["csv","copy"],"pageLength":5,"lengthMenu":[5,10,50,125],"scrollX":true,"scrollCollapse":true,"scrollY":true,"paging":true,"fixedColumns":{"leftColumns":1,"rightColumns":0},"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Next, notice that 11 genomes have completion estimates lower than MAG02 (92.96%), in fact several have completion estimates of <code>0</code>. These genome will be of no use to us since the phylogenomic analysis is based on SCG. These genomes are the first on our list to go. Another way to look at this is to generate a matrix of HMM hits across all genomes. For that we use the same external-genomes file but we need to also identify the HMM source we want anvi’o to return. Hum, what are the available HMM sources? Run the command with the <code>--list-hmm-sources</code> flag and anvi’o will tell you which HMM sources are common to all contig databases. You also need to give it an output file name or else the command will fail.</p>
<aside>
Genomes with lower completion than MAG02: Arcobacter_sp_CPC16, Arcobacter_sp_CPC309, Arcobacter_sp_CSH_ALOHA_XIX_Bin_5, Arcobacter_sp_DOLZORAL124_29_21, Arcobacter_sp_HD9_500m_PIT_SAG02, Arcobacter_sp_HD9_500m_PIT_SAG03, Arcobacter_sp_HyVt_597, Arcobacter_sp_SZUA_456, Arcobacter_sp_UBA4038, Arcobacter_sp_UBA6788, Arcobacter_sp_UBA6789.
</aside>
<pre class="bash"><code>
anvi-script-gen-hmm-hits-matrix-across-genomes 
                  -e external-genomes.txt \
                  -o dummy_out.txt \
                  --list-hmm-sources</code></pre>
<pre><code>
HMM SOURCES COMMON TO ALL 125 GENOMES
===============================================
* Archaea_76 [type: singlecopy] [num genes: 976]
* Ribosomal_RNAs [type: Ribosomal_RNAs] [num genes: 244]
* Bacteria_71 [type: singlecopy] [num genes: 885]
* Transfer_RNAs [type: Transfer_RNAs] [num genes: 547]
* Protista_83 [type: singlecopy] [num genes: 1741]</code></pre>
<p>The HMM source we are interested in is <code>Bacteria_71</code><span class="citation" data-cites="lee2019gtotree">(Lee <a href="#ref-lee2019gtotree">2019</a>)</span>. Now, rerun the command and we get a table that lists the number of times each of the 71 HMM genes were found in each of the 125 genomes.</p>
<pre class="bash"><code>
anvi-script-gen-hmm-hits-matrix-across-genomes 
                  -e external-genomes.txt \
                  -o Bacteria_71-HMM_HITS.txt \
                  --hmm-source Bacteria_71</code></pre>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="tdy2xm8lh2st5qbsvcr6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="tdy2xm8lh2st5qbsvcr6">{"x":{"filter":"none","extensions":["Buttons","FixedColumns"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">HMM hits for Arcobacter genomes.<\/caption>","autoHideNavigation":true,"data":[["Aliiarcobacter_faecis_AF1078","Aliiarcobacter_faecis_CCUG_66484","Arcobacter_anaerophilus_DSM_24636_A","Arcobacter_anaerophilus_DSM_24636_B","Arcobacter_anaerophilus_IR_1","Arcobacter_aquimarinus_CECT_8442","Arcobacter_aquimarinus_W63","Arcobacter_butzleri_7h1h","Arcobacter_butzleri_ED_1","Arcobacter_butzleri_JV22_A","Arcobacter_butzleri_JV22_B","Arcobacter_butzleri_L348","Arcobacter_butzleri_L349","Arcobacter_butzleri_L350","Arcobacter_butzleri_L351","Arcobacter_butzleri_L352","Arcobacter_butzleri_L353","Arcobacter_butzleri_L354","Arcobacter_butzleri_L355","Arcobacter_butzleri_RM4018","Arcobacter_canalis_F138_33","Arcobacter_canalis_LMG_29148","Arcobacter_cibarius_LMG_21996","Arcobacter_cloacae_CECT_7834","Arcobacter_cloacae_F26","Arcobacter_cloacae_LMG_26153","Arcobacter_cryaerophilus_ATCC_43158_ATCC_43158","Arcobacter_cryaerophilus_ATCC_43158_LMG_9904","Arcobacter_cryaerophilus_D2610","Arcobacter_defluvii_CECT_7697","Arcobacter_defluvii_LMG_25694","Arcobacter_ellisii_CECT_7837","Arcobacter_ellisii_LMG_26155","Arcobacter_halophilus_CCUG_53805","Arcobacter_halophilus_DSM_18005","Arcobacter_halophilus_F166_45","Arcobacter_lanthieri_AF1430","Arcobacter_lanthieri_AF1440","Arcobacter_lanthieri_AF1581","Arcobacter_lanthieri_LMG_28516","Arcobacter_lekithochrous_DSM_100870","Arcobacter_lekithochrous_LFT_1_7","Arcobacter_lekithochrous_LMG_28652","Arcobacter_marinus_CECT_7727","Arcobacter_marinus_F140_37","Arcobacter_marinus_JCM_15502","Arcobacter_marinus_MARC_MIP3H16","Arcobacter_marinus_SH_4D_Col1","Arcobacter_molluscorum_LMG_25693_CECT_7696","Arcobacter_molluscorum_LMG_25693_F98_3","Arcobacter_mytili_LMG_24559_CECT_7386","Arcobacter_mytili_LMG_24559_LMG_24559","Arcobacter_nitrofigilis_DSM_7299_DSM_7299","Arcobacter_pacificus_LMG_26638","Arcobacter_peruensis_PSE_93","Arcobacter_skirrowii_CCUG_10374_CCUG_10374_A","Arcobacter_skirrowii_CCUG_10374_CCUG_10374_B","Arcobacter_skirrowii_CCUG_10374_LMG_6621","Arcobacter_sp_31_11_sub10_T18","Arcobacter_sp_AF1028","Arcobacter_sp_BM102","Arcobacter_sp_BM504","Arcobacter_sp_CECT_8983","Arcobacter_sp_CECT_8985","Arcobacter_sp_CECT_8986","Arcobacter_sp_CECT_8987","Arcobacter_sp_CECT_8989","Arcobacter_sp_CECT_9188","Arcobacter_sp_CECT_9230","Arcobacter_sp_CECT_9299","Arcobacter_sp_CPC16","Arcobacter_sp_CPC309","Arcobacter_sp_CSH_ALOHA_XIX_Bin_5","Arcobacter_sp_DOLZORAL124_29_21","Arcobacter_sp_EP1","Arcobacter_sp_F155","Arcobacter_sp_F2176","Arcobacter_sp_FW59","Arcobacter_sp_HD9_500m_PIT_SAG02","Arcobacter_sp_HD9_500m_PIT_SAG03","Arcobacter_sp_HyVt_597","Arcobacter_sp_L","Arcobacter_sp_LA11","Arcobacter_sp_LMG_29976","Arcobacter_sp_LPB0137","Arcobacter_sp_NORP14","Arcobacter_sp_NORP36","Arcobacter_sp_RW43_9","Arcobacter_sp_SM1702","Arcobacter_sp_SZUA_449","Arcobacter_sp_SZUA_456","Arcobacter_sp_UBA1348","Arcobacter_sp_UBA2431","Arcobacter_sp_UBA2434","Arcobacter_sp_UBA2438","Arcobacter_sp_UBA2443","Arcobacter_sp_UBA4038","Arcobacter_sp_UBA6788","Arcobacter_sp_UBA6789","Arcobacter_sp_UBA7394","Arcobacter_suis_CECT_7833","Arcobacter_thereius_LMG_24486_A","Arcobacter_thereius_LMG_24486_B","Arcobacter_trophiarum_LMG_25534_A","Arcobacter_trophiarum_LMG_25534_B","Arcobacter_venerupis_CECT7836","Arcobacter_venerupis_LMG_26156","Halarcobacter_bivalviorum_CECT_7835","Halarcobacter_bivalviorum_F118_4","Halarcobacter_bivalviorum_LMG_26154","Halarcobacter_ebronensis_CECT_8441","Halarcobacter_ebronensis_CECT_8993","Halarcobacter_ebronensis_LMG_27922","Halarcobacter_mediterraneus_F156_34","MAG02","Poseidonibacter_sp_JOD_M_6","Poseidonibacter_sp_SJOD_M_33","Poseidonibacter_sp_SJOD_M_5","Pseudoarcobacter_acticola_KCTC_52212","Pseudoarcobacter_caeni_RW17_10","_Arcobacter_porcinus_117434","_Arcobacter_porcinus_213","_Arcobacter_porcinus_216","_Arcobacter_porcinus_CCUG_56899","_Arcobacter_porcinus_DU19"],[69,69,69,69,69,68,69,69,69,69,69,68,69,69,69,69,68,68,68,69,69,69,69,69,69,69,70,70,70,69,69,69,69,69,69,69,69,69,69,69,69,69,69,69,69,69,69,69,69,69,69,69,69,69,69,70,70,70,66,69,70,69,69,70,69,68,69,69,69,70,36,30,44,48,69,69,69,69,0,12,22,69,70,70,69,69,69,69,69,69,14,69,69,69,69,69,53,58,51,67,70,69,69,70,70,69,69,69,69,68,69,69,69,70,66,69,69,69,69,69,69,69,69,69,69],[2,2,1,1,1,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,1,1,1,2,2,2,2,2,2,2,1,1,1,2,1,1,1,1,1,2,2,2,2,2,2,1,2,2,2,2,1,1,2,2,2,2,2,2,2,1,1,3,2,2,2,0,0,1,2,2,2,2,2,1,2,1,2,0,1,1,1,1,1,1,2,2,2,2,2,2,2,2,2,2,2,2,2,1,1,1,2,2,2,2,2,2,2,2,2,2,2,2],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,1,0,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,2,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,0,1,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,0,0,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,3,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,0,0,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,0,0,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,0,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,2,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,2,0,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,1,0,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,2,0,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,0,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,1,0,1,1,1,1,0,0,0,1,1,2,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,2,1,1,1,1,0,0,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,0,6,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,0,1,1,1,1,1,1,2,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,0,0,1,0,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,1,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,1,0,0,1,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,0,1,1,1,1,0,1,2,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,1,0,1,1,1,1,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,0,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,2,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,0,0,1,1,1,2,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,0,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,0,1,1,2,1,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,0,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,2,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,2,1,0,0,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,0,1,1,1,1,1,0,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,0,0,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,0,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,2,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,1,1,1,1,0,0,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,1,1,1,1,1,1,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,1,1,1,1,1,1,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,0,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,2,1,1,1,1,1,0,1,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,0,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,0,1,1,1,1,1,1,2,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,1,0,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,0,0,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,0,1,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,0,0,1,0,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,6,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,0,0,0,1,1,2,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,0,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,0,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,0,0,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,0,0,0,1,1,1,1,0,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,0,1,1,1,1,0,0,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,0,1,1,1,1,0,0,0,1,1,2,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>genome_name<\/th>\n      <th>total_hits<\/th>\n      <th>ADK<\/th>\n      <th>AICARFT_IMPCHas<\/th>\n      <th>ATP_synt<\/th>\n      <th>ATP_synt_A<\/th>\n      <th>Adenylsucc_synt<\/th>\n      <th>Chorismate_synt<\/th>\n      <th>EF_TS<\/th>\n      <th>Exonuc_VII_L<\/th>\n      <th>GrpE<\/th>\n      <th>Ham1p_like<\/th>\n      <th>IPPT<\/th>\n      <th>OSCP<\/th>\n      <th>PGK<\/th>\n      <th>Pept_tRNA_hydro<\/th>\n      <th>RBFA<\/th>\n      <th>RNA_pol_L<\/th>\n      <th>RNA_pol_Rpb6<\/th>\n      <th>RRF<\/th>\n      <th>RecO_C<\/th>\n      <th>Ribonuclease_P<\/th>\n      <th>Ribosom_S12_S23<\/th>\n      <th>Ribosomal_L1<\/th>\n      <th>Ribosomal_L13<\/th>\n      <th>Ribosomal_L14<\/th>\n      <th>Ribosomal_L16<\/th>\n      <th>Ribosomal_L17<\/th>\n      <th>Ribosomal_L18p<\/th>\n      <th>Ribosomal_L19<\/th>\n      <th>Ribosomal_L2<\/th>\n      <th>Ribosomal_L20<\/th>\n      <th>Ribosomal_L21p<\/th>\n      <th>Ribosomal_L22<\/th>\n      <th>Ribosomal_L23<\/th>\n      <th>Ribosomal_L27<\/th>\n      <th>Ribosomal_L27A<\/th>\n      <th>Ribosomal_L28<\/th>\n      <th>Ribosomal_L29<\/th>\n      <th>Ribosomal_L3<\/th>\n      <th>Ribosomal_L32p<\/th>\n      <th>Ribosomal_L35p<\/th>\n      <th>Ribosomal_L4<\/th>\n      <th>Ribosomal_L5<\/th>\n      <th>Ribosomal_L6<\/th>\n      <th>Ribosomal_L9_C<\/th>\n      <th>Ribosomal_S10<\/th>\n      <th>Ribosomal_S11<\/th>\n      <th>Ribosomal_S13<\/th>\n      <th>Ribosomal_S15<\/th>\n      <th>Ribosomal_S16<\/th>\n      <th>Ribosomal_S17<\/th>\n      <th>Ribosomal_S19<\/th>\n      <th>Ribosomal_S2<\/th>\n      <th>Ribosomal_S20p<\/th>\n      <th>Ribosomal_S3_C<\/th>\n      <th>Ribosomal_S6<\/th>\n      <th>Ribosomal_S7<\/th>\n      <th>Ribosomal_S8<\/th>\n      <th>Ribosomal_S9<\/th>\n      <th>RsfS<\/th>\n      <th>RuvX<\/th>\n      <th>SecE<\/th>\n      <th>SecG<\/th>\n      <th>SecY<\/th>\n      <th>SmpB<\/th>\n      <th>TsaE<\/th>\n      <th>UPF0054<\/th>\n      <th>YajC<\/th>\n      <th>eIF_1a<\/th>\n      <th>ribosomal_L24<\/th>\n      <th>tRNA_synt_1d<\/th>\n      <th>tRNA_m1G_MT<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":0},{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72]}],"dom":"Blfrtip","buttons":["csv","copy"],"pageLength":5,"lengthMenu":[5,10,50,125],"scrollX":true,"scrollCollapse":true,"scrollY":true,"paging":true,"fixedColumns":{"leftColumns":1,"rightColumns":0},"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><br/> <br/> We added a column (total_hits) that records the total number of unique HMM genes per genome. This time notice that 11 genomes have fewer HMM hits than MAG02 (66). If you look at these two lists of genomes, they should be identical. Seems obvious but it isn’t always. These are two nice ways of looking at genome completeness. Now we can remove these genomes from our data set.</p>
<aside>
Genomes with fewer HMM hits than MAG02: Arcobacter_sp_CPC16, Arcobacter_sp_CPC309, Arcobacter_sp_CSH_ALOHA_XIX_Bin_5, Arcobacter_sp_DOLZORAL124_29_21, Arcobacter_sp_HD9_500m_PIT_SAG02, Arcobacter_sp_HD9_500m_PIT_SAG03, Arcobacter_sp_HyVt_597, Arcobacter_sp_SZUA_456, Arcobacter_sp_UBA4038, Arcobacter_sp_UBA6788, Arcobacter_sp_UBA6789.
</aside>
<blockquote>
<p>Now we have 114 genomes.</p>
</blockquote>
<h2 id="screen-for-16s-rrna">Screen for 16S rRNA</h2>
<p>Remember, we found an ASV in our 16S rRNA analysis that was 100% identical to a 16S rRNA sequence retrieved from MAG02. So we also want to conduct a 16S rRNA analysis of MAG02 against publicly available genomes. It makes no sense (at least to me) to have a MAG02 16S tree that does not contain the same public genomes as the phylogenomic tree. Of the remaining genomes, which ones have a 16S rRNA sequence? We can go back to our old friend <code>anvi-script-gen-hmm-hits-matrix-across-genomes</code> but this time using the setting <code>--hmm-source Ribosomal_RNAs</code>. We also use a new <code>external-genomes</code> file with the 11 low HMM hit genomes removed.</p>
<aside>
The <a href="files/mag02-phylogenomics/external-genomes-rrna-screen.txt">external-genomes file</a> to screen for 16S rRNA.
</aside>
<pre class="bash"><code>
anvi-script-gen-hmm-hits-matrix-across-genomes \
                  -e external-genomes-rrna-screen.txt \
                  -o Ribosomal_RNAs-HMM_HITS.txt \
                  --hmm-source Ribosomal_RNAs</code></pre>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="bgzodib9iiz72hfrwvln" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="bgzodib9iiz72hfrwvln">{"x":{"filter":"none","extensions":["Buttons","FixedColumns"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">rRNA hits for Arcobacter genomes.<\/caption>","autoHideNavigation":true,"data":[["Aliiarcobacter_faecis_AF1078","Aliiarcobacter_faecis_CCUG_66484","Arcobacter_anaerophilus_DSM_24636_A","Arcobacter_anaerophilus_DSM_24636_B","Arcobacter_anaerophilus_IR_1","Arcobacter_aquimarinus_CECT_8442","Arcobacter_aquimarinus_W63","Arcobacter_butzleri_7h1h","Arcobacter_butzleri_ED_1","Arcobacter_butzleri_JV22_A","Arcobacter_butzleri_JV22_B","Arcobacter_butzleri_L348","Arcobacter_butzleri_L349","Arcobacter_butzleri_L350","Arcobacter_butzleri_L351","Arcobacter_butzleri_L352","Arcobacter_butzleri_L353","Arcobacter_butzleri_L354","Arcobacter_butzleri_L355","Arcobacter_butzleri_RM4018","Arcobacter_canalis_F138_33","Arcobacter_canalis_LMG_29148","Arcobacter_cibarius_LMG_21996","Arcobacter_cloacae_CECT_7834","Arcobacter_cloacae_F26","Arcobacter_cloacae_LMG_26153","Arcobacter_cryaerophilus_ATCC_43158_ATCC_43158","Arcobacter_cryaerophilus_ATCC_43158_LMG_9904","Arcobacter_cryaerophilus_D2610","Arcobacter_defluvii_CECT_7697","Arcobacter_defluvii_LMG_25694","Arcobacter_ellisii_CECT_7837","Arcobacter_ellisii_LMG_26155","Arcobacter_halophilus_CCUG_53805","Arcobacter_halophilus_DSM_18005","Arcobacter_halophilus_F166_45","Arcobacter_lanthieri_AF1430","Arcobacter_lanthieri_AF1440","Arcobacter_lanthieri_AF1581","Arcobacter_lanthieri_LMG_28516","Arcobacter_lekithochrous_DSM_100870","Arcobacter_lekithochrous_LFT_1_7","Arcobacter_lekithochrous_LMG_28652","Arcobacter_marinus_CECT_7727","Arcobacter_marinus_F140_37","Arcobacter_marinus_JCM_15502","Arcobacter_marinus_MARC_MIP3H16","Arcobacter_marinus_SH_4D_Col1","Arcobacter_molluscorum_LMG_25693_CECT_7696","Arcobacter_molluscorum_LMG_25693_F98_3","Arcobacter_mytili_LMG_24559_CECT_7386","Arcobacter_mytili_LMG_24559_LMG_24559","Arcobacter_nitrofigilis_DSM_7299_DSM_7299","Arcobacter_pacificus_LMG_26638","Arcobacter_peruensis_PSE_93","Arcobacter_skirrowii_CCUG_10374_CCUG_10374_A","Arcobacter_skirrowii_CCUG_10374_CCUG_10374_B","Arcobacter_skirrowii_CCUG_10374_LMG_6621","Arcobacter_sp_31_11_sub10_T18","Arcobacter_sp_AF1028","Arcobacter_sp_BM102","Arcobacter_sp_BM504","Arcobacter_sp_CECT_8983","Arcobacter_sp_CECT_8985","Arcobacter_sp_CECT_8986","Arcobacter_sp_CECT_8987","Arcobacter_sp_CECT_8989","Arcobacter_sp_CECT_9188","Arcobacter_sp_CECT_9230","Arcobacter_sp_CECT_9299","Arcobacter_sp_EP1","Arcobacter_sp_F155","Arcobacter_sp_F2176","Arcobacter_sp_FW59","Arcobacter_sp_L","Arcobacter_sp_LA11","Arcobacter_sp_LMG_29976","Arcobacter_sp_LPB0137","Arcobacter_sp_NORP14","Arcobacter_sp_NORP36","Arcobacter_sp_RW43_9","Arcobacter_sp_SM1702","Arcobacter_sp_SZUA_449","Arcobacter_sp_UBA1348","Arcobacter_sp_UBA2431","Arcobacter_sp_UBA2434","Arcobacter_sp_UBA2438","Arcobacter_sp_UBA2443","Arcobacter_sp_UBA7394","Arcobacter_suis_CECT_7833","Arcobacter_thereius_LMG_24486_A","Arcobacter_thereius_LMG_24486_B","Arcobacter_trophiarum_LMG_25534_A","Arcobacter_trophiarum_LMG_25534_B","Arcobacter_venerupis_CECT7836","Arcobacter_venerupis_LMG_26156","Halarcobacter_bivalviorum_CECT_7835","Halarcobacter_bivalviorum_F118_4","Halarcobacter_bivalviorum_LMG_26154","Halarcobacter_ebronensis_CECT_8441","Halarcobacter_ebronensis_CECT_8993","Halarcobacter_ebronensis_LMG_27922","Halarcobacter_mediterraneus_F156_34","MAG02","Poseidonibacter_sp_JOD_M_6","Poseidonibacter_sp_SJOD_M_33","Poseidonibacter_sp_SJOD_M_5","Pseudoarcobacter_acticola_KCTC_52212","Pseudoarcobacter_caeni_RW17_10","_Arcobacter_porcinus_117434","_Arcobacter_porcinus_213","_Arcobacter_porcinus_216","_Arcobacter_porcinus_CCUG_56899","_Arcobacter_porcinus_DU19"],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[7,5,1,4,4,1,5,5,5,1,5,1,3,2,3,6,1,1,5,5,1,6,6,1,1,5,5,5,5,1,5,1,5,6,1,1,1,6,9,5,8,1,1,1,1,6,0,1,6,1,0,6,4,5,6,4,1,1,0,6,0,0,1,1,1,1,1,1,1,1,0,1,1,1,5,1,1,6,0,0,1,1,0,0,0,0,0,0,0,5,3,4,5,1,1,6,1,1,4,1,1,3,1,1,1,1,1,6,1,1,1,1,4,1],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[7,5,1,4,4,1,5,5,5,1,5,1,3,2,3,1,1,1,5,5,1,6,3,1,1,5,5,5,5,1,5,1,5,6,1,1,1,6,10,5,8,1,1,1,1,6,1,1,6,1,0,6,4,5,6,4,1,1,0,6,1,1,1,1,1,1,1,1,1,1,0,1,1,1,5,1,1,6,1,0,1,1,0,0,0,0,0,0,0,5,3,4,5,1,1,6,1,1,4,1,1,3,1,1,1,1,1,6,1,1,1,1,4,1],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>genome_name<\/th>\n      <th>Archaeal_16S_rRNA<\/th>\n      <th>Archaeal_23S_rRNA<\/th>\n      <th>Archaeal_5S_rRNA<\/th>\n      <th>Archaeal_5_8S_rRNA<\/th>\n      <th>Bacterial_16S_rRNA<\/th>\n      <th>Bacterial_23S_rRNA<\/th>\n      <th>Bacterial_5S_rRNA<\/th>\n      <th>Eukaryotic_28S_rRNA<\/th>\n      <th>Eukaryotic_5S_rRNA<\/th>\n      <th>Eukaryotic_5_8S_rRNA<\/th>\n      <th>Mitochondrial_12S_rRNA<\/th>\n      <th>Mitochondrial_16S_rRNA<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":0},{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12]}],"dom":"Blfrtip","buttons":["csv","copy"],"pageLength":5,"lengthMenu":[5,10,50,115],"scrollX":true,"scrollCollapse":true,"scrollY":true,"paging":true,"fixedColumns":{"leftColumns":1,"rightColumns":0},"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<p><br/></p>
<p>There were 11 genomes without 16S rRNA genes. We can now remove these from the genome collection and generate a new <code>external-genomes</code> file containing only genomes with 16S rRNA sequences—<code>external-genomes-rrna.txt</code>. I hope you are starting to see that the <code>external-genomes</code> file is an important ally.</p>
<blockquote>
<p>Now we have 103 genomes.</p>
</blockquote>
<p>Using the command <code>anvi-script-gen-hmm-hits-matrix-across-genomes</code> allows us to retrieve the 16S rRNA sequences from each genome. Since several genomes have more than one copy, we use the <code>--return-best-hit</code> flag to return a single copy only. We add the flag <code>--align-with muscle</code> to generate a multiple sequence alignment for our inspection. Incidentally we could have used the flag <code>--min-num-bins-gene-occurs</code> and set it to 1 in this command, which would have removed any genomes without a 16S rRNA sequence. I figured that out after the fact.</p>
<aside>
The <a href="files/mag02-phylogenomics/external-genomes-rrna.txt">external-genomes file</a> to get 16S rRNA genes.
</aside>
<pre class="bash"><code>
anvi-get-sequences-for-hmm-hits 
                     -e external-genomes-rrna.txt \
                     --hmm-sources Ribosomal_RNAs \
                     --gene-names Bacterial_16S_rRNA \
                     --no-wrap \
                     --return-best-hit \
                     --align-with muscle \
                     -o sequences-for-16s-rrna-hits.fa</code></pre>
<aside>
Without 16S rRNA: Arcobacter_mytili_LMG_24559_CECT_7386, Arcobacter_sp_31_11_sub10_T18, Arcobacter_sp_EP1, Arcobacter_sp_NORP36, Arcobacter_sp_SZUA_449, Arcobacter_sp_UBA1348, Arcobacter_sp_UBA2431, Arcobacter_sp_UBA2434, Arcobacter_sp_UBA2438, Arcobacter_sp_UBA2443, Arcobacter_sp_UBA7394, Arcobacter_sp_BM504.
</aside>
<p>The sequence from Arcobacter_sp_BM504 was pretty short. Bye bye Arcobacter_sp_BM504.</p>
<blockquote>
<p>Now we have 101 genomes.</p>
</blockquote>
<p>Up to this point we have a collection consisting of genomes that have high completion <em>and</em> a full-length 16S rRNA sequence. What about redundant genomes? Surely there are some genomes that are highly similar, and we really only need a few representatives of each.</p>
<aside>
Note. Arcobacter_sp_NORP14 failed subsequent the composition Chi-squared test in the IQ-TREE analysis below. This genome was removed.
</aside>
<h1 id="dereplicate-genome-collection">Dereplicate Genome Collection</h1>
<p>Finally, we can dereplicate the genome collection using the command <code>anvi-dereplicate-genomes</code> to assess the number of genome clusters. This command has a lot of options for fine-tuning but since we are interested in a quick and dirty assessment, we’ll keep it simple. We can either give anvi’o an <code>external-genomes</code> file like we just did or a <code>fasta-text-file</code> similar to the one from the <a href="#running-snakemake-contigs-workflow">Snakemake workflow</a> above. Remember, the <code>external-genomes</code> file provides names and paths to <em>contig.dbs</em> and the <code>fasta-text-file</code> provides names and paths to <em>fasta</em> files. Either way, we first need to remove the 11 genomes that we are cutting from the analysis. Let’s stick with the <code>external-genomes</code> file since we will use this downstream anyway.</p>
<p>Since accuracy is not paramount we use <code>--program fastANI</code><span class="citation" data-cites="jain2018high">(Jain et al. <a href="#ref-jain2018high">2018</a>)</span> because, well, it’s fast. We also need to choose a <code>--similarity-threshold</code>. We tested the number of clusters for 0.90, 0.95, 0.97, 0.98, and 0.99. Overkill? Maybe, but this will give us a good sense of replication in our genome collection. Here is the command with 0.97.</p>
<aside>
The <a href="files/mag02-phylogenomics/external-genomes-dereplicate.txt">external-genomes file</a> for dereplication.
</aside>
<pre class="bash"><code>
mkdir 06_DEREPLICATE
anvi-dereplicate-genomes -e external-genomes-rrna.txt 
                         --program fastANI 
                         --similarity-threshold 0.99 
                         -o 06_DEREPLICATE/DR_99 -T $NSLOTS </code></pre>
<p>The command produces numerous output files, but let’s focus on the <code>CLUSTER_REPORT.txt</code> file, which tells us the <strong>a</strong>) number of clusters, <strong>b</strong>) number of genomes in each cluster, <strong>c</strong>) representative genome for the cluster, and <strong>d</strong>) all genomes in the cluster.</p>
<p>How many clusters did we get for each <code>--similarity-threshold</code>?</p>
<ul>
<li>0.900: 38 clusters</li>
<li>0.950: 44 clusters</li>
<li>0.970: 52 clusters</li>
<li>0.980: 60 clusters</li>
<li>0.990: 73 clusters</li>
<li>0.995: 76 clusters</li>
</ul>
<p>We will use the genomes from the <code>--similarity-threshold 0.99</code> to get rid of only genomes that are nearly identical.</p>
<blockquote>
<p>Final collection contains 73 genomes.</p>
</blockquote>
<h1 id="outgroup-workflow">Outgroup Workflow</h1>
<p>We decided to use <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&amp;id=57665"><em>Sulfurospirillum</em></a>, a member of the Campylobacteraceae, as an outgroup for our tree. We basically ran the same pipeline as above, with a few minor changes. The code is included here but the discussion will be kept to a minimum. I am sure by this point you are tired of hearing me drone on and on.</p>
<h4 id="get-taxids">1. Get TaxIDs</h4>
<pre class="bash"><code>
python gimme_taxa.py 57665 -o Sulfurospirillum-TaxIDs-for-ngd.txt

python gimme_taxa.py 57665 -o Sulfurospirillum-TaxIDs-for-ngd-just-IDs.txt \
                           --just-taxids</code></pre>
<h4 id="download-genomes">2. Download genomes</h4>
<pre class="bash"><code>
ncbi-genome-download -t Sulfurospirillum-TaxIDs-for-ngd-just-IDs.txt \
                      bacteria \
                      -o 00_Sulfurospirillum_Genbank \
                      --metadata Sulfurospirillum-NCBI-METADATA.txt \
                      -s genbank \
                      -l complete,chromosome</code></pre>
<p>We received 10 <em>Sulfurospirillum</em> genomes. We used the <code>-l complete,chromosome</code> flag so we wouldn’t need to sift through a bunch of data. We only need a few genomes for the outgroup.</p>
<h4 id="format-genomes">3. Format genomes</h4>
<pre class="bash"><code>
anvi-script-process-genbank-metadata -m Sulfurospirillum-NCBI-METADATA.txt \
                                     -o 01_Sulfurospirillum-NCBI-GENOMES \
                                     --output-fasta-txt Sulfurospirillum-fasta.txt \
                                     --exclude-gene-calls-from-fasta-txt</code></pre>
<h4 id="run-snakemake-workflow">4. Run Snakemake workflow</h4>
<p>You know the routine. Same steps performed for the <em>Arcobacter</em>.</p>
<h4 id="completeness-1">5. Completeness</h4>
<pre class="bash"><code>
anvi-estimate-genome-completeness -e external-genomes.txt \
                                  -o completness-outgroup.txt</code></pre>
<aside>
Nice genomes. The lowest <a href="files/mag02-phylogenomics/outgroup-completness.txt">completion estimate</a> is 95.77%.
</aside>
<h4 id="screen-for-16s-rrna-1">6. Screen for 16S rRNA</h4>
<pre class="bash"><code>
anvi-script-gen-hmm-hits-matrix-across-genomes 
                      -e external-genomes-outgroup.txt \
                      -o Ribosomal_RNAs-HMM_HITS-outgroup.txt \
                      --hmm-source Ribosomal_RNAs</code></pre>
<aside>
And they all have <a href="files/mag02-phylogenomics/Ribosomal_RNAs-HMM_HITS-outgroup.txt">16S rRNA genes</a>.
</aside>
<h4 id="retrieve-16s-rrna-sequences">7. Retrieve 16S rRNA sequences</h4>
<pre class="bash"><code>
anvi-get-sequences-for-hmm-hits -e external-genomes-outgroup.txt \
                                --hmm-sources Ribosomal_RNAs  \
                                --gene-names Bacterial_16S_rRNA \
                                --no-wrap --return-best-hit \
                                --align-with muscle \
                                -o sequences-for-16s-rrna-hits-outgroup.fa</code></pre>
<p>Which are full-length.</p>
<h4 id="dereplicate">8. Dereplicate</h4>
<pre class="bash"><code>
mkdir 06_DEREPLICATE
anvi-dereplicate-genomes -e external-genomes-outgroup.txt 
                         --program fastANI 
                         --similarity-threshold 0.95 
                         -o 06_DEREPLICATE/DR_95 -T $NSLOTS</code></pre>
<p>Good to go. Let’s use <em>Sulfurospirillum halorespirans</em> (DSM 13726), <em>Sulfurospirillum deleyianum</em> (DSM 6946), <em>Sulfurospirillum multivorans</em> (DSM 12446), and <em>Sulfurospirillum barnesii</em> (SES 3) as the outgroup since these were all ‘unique’ at 95% similarity.</p>
<h1 id="phylogenomics">Phylogenomics</h1>
<p>Finally, we are ready for phylogenomic analysis. We first need to make a new <em>external-genomes</em> file that contains the <em>Arcobacter</em> representatives and the <em>Sulfurospirillum</em> outgroup. Then we need to setup a phylogenomic Snakemake workflow. Like before, we start by getting a <em>default config</em> file for the workflow.</p>
<pre class="bash"><code>
anvi-run-workflow -w phylogenomics 
                  --get-default-config phylogenomics-default-config.json</code></pre>
<p>Remember when ran the contigs workflow above? Most of those steps can actually be run in the phylogenomic workflow itself. The thing is that it does not really give us the opportunity to curate our collection <em>before</em> phylogenomic analysis. Because we ran the cotigs workflow first, there are many steps in the phylogenomic workflow that we simply do not need. So, we chop out the chunks of the <code>phylogenomics-default-config.json</code> that are unnecessary.</p>
<p>Seems like a good idea to select SCG that are in all of the genomes. Of the 71 genes in the HMM profile, <a href="files/mag02-phylogenomics/arco-phylo-Bacteria_71-HMM_HITS.txt">63 were present in all 77 genomes</a>. We used these for our analysis though we could easily select a subset like all Ribosomal genes.</p>
<pre class="bash"><code>
anvi-script-gen-hmm-hits-matrix-across-genomes 
                   -e arco-phylo-external-genomes.txt \
                   -o arco-phylo-Bacteria_71-HMM_HITS.txt \
                   --hmm-source Bacteria_71</code></pre>
<aside>
Here is the <a href="files/mag02-phylogenomics/phylogenomics-config.json">modified config</a> file &amp; the <a href="files/mag02-phylogenomics/arco-phylo-external-genomes.txt">final external genomes</a> file.
</aside>
<p>And now here are the Snakemake commands.</p>
<pre class="bash"><code>
anvi-run-workflow -w phylogenomics 
                  -c phylogenomics-config.json 
                  --additional-params 
                  --jobs 5 
                  --resources nodes=5 
                  --keep-going --rerun-incomplete --unlock
anvi-run-workflow -w phylogenomics 
                  -c phylogenomics-config.json 
                  --additional-params 
                  --jobs 5 --resources nodes=5 
                  --keep-going --rerun-incomplete</code></pre>
<blockquote>
<p>Phylogenomic analysis of 73 <em>Arcobacter</em> and 4 <em>Sulfurospirillum</em> genomes.</p>
</blockquote>
<p>And away she goes. A straight-up phylogenomic approach using SCG (<em>picked somewhat at random</em>) as we just did may not be the best approach for scrutinizing closely related genomes. This is because we are not taking into account the degree of <a href="http://merenlab.org/2016/11/08/pangenomics-v2/#concept-of-homogeneity">homogeneity</a> within gene clusters. A better approach is to calculate homogeneity indices and use the results to guide our choice of genes clusters for the phylogenomic analysis. For this, we will use anvi’o pangenomic analysis.</p>
<h2 id="pangenome-analysis">Pangenome Analysis</h2>
<p>For a pangenomic analysis we first need to generate a <a href="http://merenlab.org/2016/11/08/pangenomics-v2/#generating-an-anvio-genomes-storage"><em>genome storage</em></a> database, which is a special anvi’o database. Whatever you choose to name this database, <strong>it must have the extension <code>-GENOMES.db</code></strong>. We generally followed this <a href="http://merenlab.org/2016/11/08/pangenomics-v2/">anvi’o workflow for microbial pangenomics</a> when designing our workflow.</p>
<pre class="bash"><code>
anvi-gen-genomes-storage -e arco-phylo-external-genomes.txt \
                         -o ARCOBACTER-GENOMES.db</code></pre>
<p>And then we can run the pangenome analysis itself. We use the <code>--min-occurrence</code> flag set to <code>3</code> to ignore singleton and doubleton gene clusters.</p>
<pre class="bash"><code>
anvi-pan-genome -g ARCOBACTER-GENOMES.db \
                --min-occurrence 3 \
                --project-name ARCOBACTER -T $NSLOTS</code></pre>
<p>Anvi’o will create gene cluster using the MCL (Markov Cluster) algorithm<span class="citation" data-cites="jain2018high">(Jain et al. <a href="#ref-jain2018high">2018</a>)</span> and calculate homogeneity indices. Then we can visualize the results using the anvi’o interactive interface.</p>
<pre class="bash"><code>
anvi-display-pan -g ARCOBACTER-GENOMES.db -p ARCOBACTER-PAN.db</code></pre>
<p>Once inside the interactive, we performed the following steps:</p>
<ol type="1">
<li>Hit <strong>Draw</strong> and the pangenome will render in the brwoser window.</li>
<li>Go to the <strong>Search</strong> pane and click on the <em>Search gene clusters using filters</em> dropdown.</li>
<li>Set the following parameters:
<ul>
<li>Min number of genomes gene cluster occurs = 77</li>
<li>Min number of genes from each genome = 1</li>
<li>Max number of genes from each genome = 1</li>
<li>Max functional homogeneity index = 0.90</li>
<li>Min geometric homogeneity index = 0.99</li>
</ul></li>
<li>Click <em>Search</em>.</li>
<li>Scroll down and hit <em>Highlight splits on tree</em>. Zoom into that area on the outside of the tree. The red lines are the highlights.</li>
</ol>
<div class="l-body-outset">
<figure>
<a href="figures/mag02-phylogenomics/better_core.png"> <img src="figures/mag02-phylogenomics/better_core.png" class="thumbnail"> </a>
<figcaption>
<strong>Process of using anvi’o interactive interface to delect a better core.</strong> Follow these steps to modify gene collection for phylogenomic inference. Make certain to inspect gene clusters.
</figcaption>
</figure>
</div>
<ol start="6" type="1">
<li>Right click on a gene cluster and hit <em>Inspect gene cluster</em>.</li>
<li>Check out the cluster, inspect others, make sure they look good.</li>
<li>Head back to the button under Highlight and hit <em>Append splits to the selected bin</em>.</li>
<li>Go to the <strong>Bins</strong> tab, rename the bin if you wish, and click <em>Store bin collection</em>.</li>
</ol>
<p>The pangenome database now has a new collection of SCG and a new phylogenomic tree. The new collection has 17 gene clusters and 1309 genes. 1309 divided by 77 (number of genomes) equals 17. Sweet.</p>
<h4 id="optional-on-the-fly-tree-generation.">Optional, on the fly tree generation.</h4>
<p>If you wish, you can use the curated collection of gene clusters to generate a new tree right in the interface.</p>
<ol start="10" type="1">
<li>Head over to the <strong>Layers</strong> tab and click <em>Generate new phylogenetic tree</em>.</li>
<li>In the popup window make sure you like the settings and hit <em>Generate</em>.</li>
</ol>
<p>Rather than using the tree generated <em>on the fly</em> by anvi’o, let’s get the sequences from the curated bin collection and do the analysis ourselves.</p>
<pre class="bash"><code>
anvi-get-sequences-for-gene-clusters -p ARCOBACTER-PAN.db \
                                     -g ARCOBACTER-GENOMES.db \
                                     -C default -b better_core \
                                     --concatenate-gene-clusters \
                                     --align-with muscle \
                                     -o ARCOBACTER-proteins-better-core.fa</code></pre>
<p>Under this scenario, anvi’o retrieves, aligns (in this case with MUSCLE<span class="citation" data-cites="edgar2004muscle">(Edgar <a href="#ref-edgar2004muscle">2004</a>)</span>), and then concatenates the sequences.</p>
<p>It is important to note that everything we just in the interface can instead be done in the command line. If you look at the help menu for <code>anvi-get-sequences-for-gene-clusters</code> you will see that you can apply the filtering steps described above directly in the command line without the pangenomic interactive. Personally, I like to use the interactive because I can <em>visually</em> inspect each gene cluster to make sure nothing fishy is going on. The parameters you choose will depend on your question and the peculiarities of your data set, so play around.</p>
<p>Just for the fun of it, let’s have a look at what this entails without running interactive.</p>
<pre class="bash"><code>
anvi-get-sequences-for-gene-clusters \
                        -p ARCOBACTER-PAN.db \
                        -g ARCOBACTER-GENOMES.db \
                        -o ARCOBACTER-proteins-better-core_no_interactive.fa \
                        --concatenate-gene-clusters \
                        --min-num-genomes-gene-cluster-occurs 77  \
                        --min-num-genes-from-each-genome 1 \
                        --max-num-genes-from-each-genome 1 \
                        --max-functional-homogeneity-index 0.90 \
                        --min-geometric-homogeneity-index 0.99</code></pre>
<p>And the last part of the output looks like this:</p>
<pre><code>
INFO
===============================================
Your filters resulted in 17 gene clusters that contain a total of 1309 genes.
for downstream analyses. Just so you know.

Sequence type ...............................................: Amino acid
Output file for phylogenomics ...............................: ARCOBACTER-proteins-better-core_no_interactive.fa</code></pre>
<p>As expected, 17 gene clusters encompassing a total of 1309 genes. <em>Note, the order of genes in the concatenate file may differ but the content should be the same.</em></p>
<p>Anyway, regardless of the approach you choose, now we can use your curated SCG fasta file to construct a new phylogenomic tree. You can either do this is anvi’o, which will use <a href="http://www.iqtree.org/">IQ-TREE</a><span class="citation" data-cites="minh2020iq">(Minh et al. <a href="#ref-minh2020iq">2020</a>)</span> or you can run a program outside anvi’o</p>
<pre class="bash"><code>
anvi-gen-phylogenomic-tree -f ARCOBACTER-proteins-better-core.fa 
                           -o ARCOBACTER-proteins-better-core.tre</code></pre>
<p>Instead, we will use <a href="http://www.iqtree.org/">IQ-TREE</a><span class="citation" data-cites="minh2020iq">(Minh et al. <a href="#ref-minh2020iq">2020</a>)</span> (v. 2.0.3) natively with <em>ModelFinder Plus</em><span class="citation" data-cites="kalyaanamoorthy2017modelfinder">(Kalyaanamoorthy et al. <a href="#ref-kalyaanamoorthy2017modelfinder">2017</a>)</span> to perform extended model selection followed by tree inference. <em>ModelFinder</em> computes the log-likelihoods of an initial parsimony tree for many different models then computes Akaike information criterion (AIC), corrected Akaike information criterion (AICc), and the Bayesian information criterion (BIC) to choose the substitution model that minimizes the BIC score.</p>
<aside>
It was here that Arcobacter_sp_NORP14 failed the composition Chi-squared test in the IQ-TREE. This genome was removed retroactively.
</aside>
<pre class="bash"><code>
iqtree -s ARCOBACTER-proteins-better-core.fa 
                  --prefix arcobacterMFP 
                  -m MFP</code></pre>
<p><em>ModelFinder</em> tested 49 models (it will test up to 546 protein models) and selected LG+F+R4 as the best-fit model based on BIC. We subsequently ran this model with 5000 ultrafast bootstrap approximations<span class="citation" data-cites="hoang2017ultrafast">(Hoang et al. <a href="#ref-hoang2017ultrafast">2017</a>)</span>.</p>
<pre class="bash"><code>
iqtree -s ARCOBACTER-proteins-better-core.fa 
       --prefix arcobacterLG+F+R4-5000  
       -m LG+F+R4 -bb 5000 -T $NSLOTS</code></pre>
<h1 id="s-rrna-gene-analysis-from-mag02">16S rRNA gene analysis from MAG02</h1>
<p>Next, we performed the same process for 16S rRNA genes from all the genomes in the phylogenomic analysis.</p>
<pre class="bash"><code>
anvi-get-sequences-for-hmm-hits 
                     -e arco-phylo-external-genomes.txt \
                     --hmm-sources Ribosomal_RNAs \
                     --gene-names Bacterial_16S_rRNA \
                     --no-wrap \
                     --return-best-hit \
                     --align-with muscle \
                     -o sequences-for-16s-rrna-hits.fa</code></pre>
<p>The fasta output of this command needs a little TLC before continuing because anvi’o adds a bunch of info to the deflines about the gene. Good info to have but if we use this file in an external program like RAxML, all of this info becomes the name and I am sure you would not like to look at a tree with leaves like this:</p>
<div class="l-body-outset">
<p>|&gt;<em>Bacterial_16S_rRNA_Ribosomal_RNAs_2c53c82ce374e7c1bd22dd51720f5f90b9f43fdfb6b3b3d6cb0623b9 bin_id:Aliiarcobacter_faecis_AF1078|source:Ribosomal_RNAs|e_value:0|</em></p>
</div>
<p>We used <a href="http://www.barebones.com/products/bbedit/index.html">BBEdit</a> and some grep search and replace to extract the sample names.</p>
<p>Next, we again ran <a href="http://www.iqtree.org/">IQ-TREE</a><span class="citation" data-cites="minh2020iq">(Minh et al. <a href="#ref-minh2020iq">2020</a>)</span> (v. 2.0.3) with <em>ModelFinder Plus</em><span class="citation" data-cites="kalyaanamoorthy2017modelfinder">(Kalyaanamoorthy et al. <a href="#ref-kalyaanamoorthy2017modelfinder">2017</a>)</span> to perform extended model selection followed by tree inference.</p>
<pre class="bash"><code>
iqtree -s sequences-for-16s-rrna-hits-rename.fa 
                  --prefix arcobacter-16s-MFP 
                  -m MFP</code></pre>
<p><em>ModelFinder</em> tested 30 models (it will test up to 286 DNA models) and selected TVM+F+I+G4 as the best-fit model based on BIC. We subsequently ran this model with 5000 ultrafast bootstrap approximations<span class="citation" data-cites="hoang2017ultrafast">(Hoang et al. <a href="#ref-hoang2017ultrafast">2017</a>)</span>.</p>
<pre class="bash"><code>
iqtree -s sequences-for-16s-rrna-hits-rename.fa 
                  --prefix arcobacter-16s-TVM+F+I+G4 
                  -m TVM+F+I+G4 -bb 5000</code></pre>
<p>If you are unfamiliar with IQ-TREE, check out the <a href="http://www.iqtree.org/doc/Tutorial">tutorial</a>. It’s a really nice program.</p>
<h1 id="conclusion">Conclusion</h1>
<p>Dandy. Now we have a phylogenomic tree of MAG02 with 72 other <em>Arcobacter</em> genomes and 4 from <em>Sulfurospirillum</em> as the outgroup. We also have a phylogenetic tree of near-full length 16S rRNA genes from the same genomes.</p>
<div class="post-nav">
<div class="post-nav-item">
<div class="meta-nav">
Previous
</div>
<p><a href="mg-binning.html" rel="next">N<sup><u>o</u></sup> 5. Binning MAGs</a></p>
</div>
</div>
<div class="post-nav">
<div class="post-nav-item">
<div class="meta-nav">
Next
</div>
<p><a href="mag02-phylogenomics.html" rel="prev">N<sup><u>o</u></sup> 7. SOMETHING</a></p>
</div>
</div>
<h2 id="source-code" class="appendix">Source Code</h2>
<p>The source code for this page can be accessed on GitHub by <a href="https://github.com/hypocolypse/web/blob/master/mag02-phylogenomics.Rmd">clicking this link</a>.</p>
<div id="refs" class="references">
<div id="ref-edgar2004muscle">
<p>Edgar, Robert C. 2004. “MUSCLE: Multiple Sequence Alignment with High Accuracy and High Throughput.” <em>Nucleic Acids Research</em> 32 (5): 1792–7. <a href="https://doi.org/10.1093/nar/gkh340">https://doi.org/10.1093/nar/gkh340</a>.</p>
</div>
<div id="ref-hoang2017ultrafast">
<p>Hoang, Diep Thi, Olga Chernomor, Arndt von Haeseler, Bui Quang Minh, and Le Sy Vinh. 2017. “UFBoot2: Improving the Ultrafast Bootstrap Approximation.” <em>Molecular Biology and Evolution</em> 35 (2): 518–22. <a href="https://doi.org/10.1093/molbev/msx281">https://doi.org/10.1093/molbev/msx281</a>.</p>
</div>
<div id="ref-jain2018high">
<p>Jain, Chirag, Luis M Rodriguez-R, Adam M Phillippy, Konstantinos T Konstantinidis, and Srinivas Aluru. 2018. “High Throughput Ani Analysis of 90K Prokaryotic Genomes Reveals Clear Species Boundaries.” <em>Nature Communications</em> 9 (1): 1–8. <a href="https://doi.org/10.1038/s41467-018-07641-9">https://doi.org/10.1038/s41467-018-07641-9</a>.</p>
</div>
<div id="ref-kalyaanamoorthy2017modelfinder">
<p>Kalyaanamoorthy, Subha, Bui Quang Minh, Thomas KF Wong, Arndt von Haeseler, and Lars S Jermiin. 2017. “ModelFinder: Fast Model Selection for Accurate Phylogenetic Estimates.” <em>Nature Methods</em> 14 (6): 587. <a href="https://doi.org/10.1038/nmeth.4285">https://doi.org/10.1038/nmeth.4285</a>.</p>
</div>
<div id="ref-koster2012snakemake">
<p>Köster, Johannes, and Sven Rahmann. 2012. “Snakemake—a Scalable Bioinformatics Workflow Engine.” <em>Bioinformatics</em> 28 (19): 2520–2. <a href="https://doi.org/10.1093/bioinformatics/bts480">https://doi.org/10.1093/bioinformatics/bts480</a>.</p>
</div>
<div id="ref-lee2019gtotree">
<p>Lee, Michael D. 2019. “GToTree: A User-Friendly Workflow for Phylogenomics.” <em>Bioinformatics</em> 35 (20): 4162–4. <a href="https://doi.org/10.1093/bioinformatics/btz188">https://doi.org/10.1093/bioinformatics/btz188</a>.</p>
</div>
<div id="ref-minh2020iq">
<p>Minh, Bui Quang, Heiko A Schmidt, Olga Chernomor, Dominik Schrempf, Michael D Woodhams, Arndt Von Haeseler, and Robert Lanfear. 2020. “IQ-Tree 2: New Models and Efficient Methods for Phylogenetic Inference in the Genomic Era.” <em>Molecular Biology and Evolution</em> 37 (5): 1530–4. <a href="https://doi.org/10.1093/molbev/msaa015">https://doi.org/10.1093/molbev/msaa015</a>.</p>
</div>
<div id="ref-parks2018gtgb">
<p>Parks, Donovan H, Maria Chuvochina, David W Waite, Christian Rinke, Adam Skarshewski, Pierre-Alain Chaumeil, and Philip Hugenholtz. 2018. “A Standardized Bacterial Taxonomy Based on Genome Phylogeny Substantially Revises the Tree of Life.” <em>Nature Biotechnology</em> 36 (10): 1546–1696. <a href="https://doi.org/10.1038/nbt.4229">https://doi.org/10.1038/nbt.4229</a>.</p>
</div>
</div>
<section class="footnotes">
<hr />
<ol>
<li id="fn1"><p>In the analysis, this MAG is named WATER_MAG_00002. For brevity we refer to it henceforth as MAG02<a href="#fnref1" class="footnote-back">↩</a></p></li>
</ol>
</section>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/hypocolypse/web/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/hypocolypse/web/">https://github.com/hypocolypse/web/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<script id="distill-bibliography" type="text/bibtex">
@article{callahan2016dada2,
  title={DADA2: high-resolution sample inference from Illumina amplicon data},
  author={Callahan, Benjamin J and McMurdie, Paul J and Rosen, Michael J and Han, Andrew W and Johnson, Amy Jo A and Holmes, Susan P},
  journal={Nature Methods},
  volume={13},
  number={7},
  pages={581},
  year={2016},
  url = {https://doi.org/10.1038/nmeth.3869},
}

@article{martin2011cutadapt,
  title={Cutadapt removes adapter sequences from high-throughput sequencing reads},
  author={Martin, Marcel},
  journal={EMBnet. journal},
  volume={17},
  number={1},
  pages={10--12},
  year={2011},
  url = {https://doi.org/10.14806/ej.17.1.200}
}

@article{quast2012silva,
  title={The SILVA ribosomal RNA gene database project: improved data processing and web-based tools},
  author={Quast, Christian and Pruesse, Elmar and Yilmaz, Pelin and Gerken, Jan and Schweer, Timmy and Yarza, Pablo and Peplies, J{\"o}rg and Gl{\"o}ckner, Frank Oliver},
  journal={Nucleic Acids Research},
  volume={41},
  number={D1},
  pages={D590--D596},
  year={2012},
  url = {https://doi.org/10.1093/nar/gks1219}
}

@article{mcmurdie2013phyloseq,
  title={phyloseq: an R package for reproducible interactive analysis and graphics of microbiome census data},
  author={McMurdie, Paul J and Holmes, Susan},
  journal={PLoS One},
  volume={8},
  number={4},
  pages={e61217},
  year={2013},
  url = {https://doi.org/10.1371/journal.pone.0061217}
}

@misc{wong2011points,
  title={Points of view: Color blindness},
  author={Wong, Bang},
  journal={Nature Methods},
  volume={8},
  number={6},
  pages={441},
  year={2011},
  url = {https://doi.org/10.1038/nmeth.1618}
}

@misc{lahti2017microbiome,
  author = {Lahti, Leo and Shetty Sudarshan and others},
  title = {Tools for microbiome analysis in R. Version 1.9.97},
  year = {2017},
  url = {https://github.com/microbiome/microbiome},
}

@article{eren2015anvi,
  title={Anvi’o: an advanced analysis and visualization platform for ‘omics data},
  author={Eren, A Murat and Esen, {\"O}zcan C and Quince, Christopher and Vineis, Joseph H and Morrison, Hilary G and Sogin, Mitchell L and Delmont, Tom O},
  journal={PeerJ},
  volume={3},
  pages={e1319},
  year={2015},
  url = {https://doi.org/10.7717/peerj.1319}
}

@article{bolger2014trimmomatic,
  title={Trimmomatic: a flexible trimmer for Illumina sequence data},
  author={Bolger, Anthony M and Lohse, Marc and Usadel, Bjoern},
  journal={Bioinformatics},
  volume={30},
  number={15},
  pages={2114--2120},
  year={2014},
  url = {https://doi.org/10.1093/bioinformatics/btu170}
}

@article{koster2012snakemake,
  title={Snakemake—a scalable bioinformatics workflow engine},
  author={K{\"o}ster, Johannes and Rahmann, Sven},
  journal={Bioinformatics},
  volume={28},
  number={19},
  pages={2520--2522},
  year={2012},
  url = {https://doi.org/10.1093/bioinformatics/bts480}
}

@article{menzel2016fast,
  title={Fast and sensitive taxonomic classification for metagenomics with Kaiju},
  author={Menzel, Peter and Ng, Kim Lee and Krogh, Anders},
  journal={Nature Communications},
  volume={7},
  pages={11257},
  year={2016},
  url = {https://doi.org/10.1038/ncomms11257}
}

@article{roux2015virsorter,
  title={VirSorter: mining viral signal from microbial genomic data},
  author={Roux, Simon and Enault, Francois and Hurwitz, Bonnie L and Sullivan, Matthew B},
  journal={PeerJ},
  volume={3},
  pages={e985},
  year={2015},
  url = {https://doi.org/10.1093/bioinformatics/btu170}
}

@article{kim2016centrifuge,
  title={Centrifuge: rapid and sensitive classification of metagenomic sequences},
  author={Kim, Daehwan and Song, Li and Breitwieser, Florian P and Salzberg, Steven L},
  journal={Genome Research},
  volume={26},
  number={12},
  pages={1721--1729},
  year={2016},
  url = {https://doi.org/10.1101/gr.210641.116.}
}

@article{hyatt2010prodigal,
  title={Prodigal: prokaryotic gene recognition and translation initiation site identification},
  author={Hyatt, Doug and Chen, Gwo-Liang and LoCascio, Philip F and Land, Miriam L and Larimer, Frank W and Hauser, Loren J},
  journal={BMC Bioinformatics},
  volume={11},
  number={1},
  pages={119},
  year={2010},
  url = {https://doi.org/10.1186/1471-2105-11-119}
}

@article{edgar2004muscle,
  title={MUSCLE: multiple sequence alignment with high accuracy and high throughput},
  author={Edgar, Robert C},
  journal={Nucleic Acids Research},
  volume={32},
  number={5},
  pages={1792--1797},
  year={2004},
  url = {https://doi.org/10.1093/nar/gkh340}
}

@article{li2015megahit,
  title={MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph},
  author={Li, Dinghua and Liu, Chi-Man and Luo, Ruibang and Sadakane, Kunihiko and Lam, Tak-Wah},
  journal={Bioinformatics},
  volume={31},
  number={10},
  pages={1674--1676},
  year={2015},
  url = {https://doi.org/10.1093/bioinformatics/btv033}
}

@article{langmead2012fast,
  title={Fast gapped-read alignment with Bowtie 2},
  author={Langmead, Ben and Salzberg, Steven L},
  journal={Nature Methods},
  volume={9},
  number={4},
  pages={357},
  year={2012},
  url = {https://doi.org/10.1038/nmeth.1923}
}

@article{li2009fast,
  title={Fast and accurate short read alignment with Burrows--Wheeler transform},
  author={Li, Heng and Durbin, Richard},
  journal={Bioinformatics},
  volume={25},
  number={14},
  pages={1754--1760},
  year={2009},
  url = {https://doi.org/10.1093/bioinformatics/btp324}
}

@article{li2009sequence,
  title={The sequence alignment/map format and SAMtools},
  author={Li, Heng and Handsaker, Bob and Wysoker, Alec and Fennell, Tim and Ruan, Jue and Homer, Nils and Marth, Gabor and Abecasis, Goncalo and Durbin, Richard},
  journal={Bioinformatics},
  volume={25},
  number={16},
  pages={2078--2079},
  year={2009},
  url = {https://doi.org/10.1093/bioinformatics/btp352}
}

@article{breitwieser2018krakenuniq,
  title={KrakenUniq: confident and fast metagenomics classification using unique k-mer counts},
  author={Breitwieser, FP and Baker, DN and Salzberg, Steven L},
  journal={Genome Biology},
  volume={19},
  number={1},
  pages={1--10},
  year={2018},
  url = {https://doi.org/10.5281/zenodo.1412647}
}

@article{eren2013filtering,
  title={A filtering method to generate high quality short reads using Illumina paired-end technology},
  author={Eren, A Murat and Vineis, Joseph H and Morrison, Hilary G and Sogin, Mitchell L},
  journal={PLoS One},
  volume={8},
  number={6},
  year={2013},
  url = {https://doi.org/10.1371/journal.pone.0066643}
}

@article{alneberg2014binning,
  title={Binning metagenomic contigs by coverage and composition},
  author={Alneberg, Johannes and Bjarnason, Brynjar Sm{\'a}ri and De Bruijn, Ino and Schirmer, Melanie and Quick, Joshua and Ijaz, Umer Z and Lahti, Leo and Loman, Nicholas J and Andersson, Anders F and Quince, Christopher},
  journal={Nature Methods},
  volume={11},
  number={11},
  pages={1144--1146},
  year={2014},
  url = {https://doi.org/10.1038/nmeth.3103}
}

@article{sieber2018recovery,
  title={Recovery of genomes from metagenomes via a dereplication, aggregation and scoring strategy},
  author={Sieber, Christian MK and Probst, Alexander J and Sharrar, Allison and Thomas, Brian C and Hess, Matthias and Tringe, Susannah G and Banfield, Jillian F},
  journal={Nature Microbiology},
  volume={3},
  number={7},
  pages={836--843},
  year={2018},
}

@article{kang2019metabat,
  title={MetaBAT 2: an adaptive binning algorithm for robust and efficient genome reconstruction from metagenome assemblies},
  author={Kang, Dongwan D and Li, Feng and Kirton, Edward and Thomas, Ashleigh and Egan, Rob and An, Hong and Wang, Zhong},
  journal={PeerJ},
  volume={7},
  pages={e7359},
  year={2019},
}
@article{wu2016maxbin,
  title={MaxBin 2.0: an automated binning algorithm to recover genomes from multiple metagenomic datasets},
  author={Wu, Yu-Wei and Simmons, Blake A and Singer, Steven W},
  journal={Bioinformatics},
  volume={32},
  number={4},
  pages={605--607},
  year={2016},
}

@article{ondov2011krona,
  title={Interactive metagenomic visualization in a Web browser},
  author={Ondov, Brian D and Bergman, Nicholas H and Phillippy, Adam M},
  journal={BMC Bioinformatics},
  volume={12},
  number={1},
  pages={385},
  year={2011},
  url = {https://doi.org/10.1186/1471-2105-12-385}
}

@article{buchfink2015fast,
  title={Fast and sensitive protein alignment using DIAMOND},
  author={Buchfink, Benjamin and Xie, Chao and Huson, Daniel H},
  journal={Nature Methods},
  volume={12},
  number={1},
  pages={59--60},
  year={2015},
  url = {https://doi.org/10.1038/nmeth.3176}
}

@article{delmont2018nitrogen,
  title={Nitrogen-fixing populations of Planctomycetes and Proteobacteria are abundant in surface ocean metagenomes},
  author={Delmont, Tom O and Quince, Christopher and Shaiber, Alon and Esen, {\"O}zcan C and Lee, Sonny TM and Rapp{\'e}, Michael S and McLellan, Sandra L and L{\"u}cker, Sebastian and Eren, A Murat},
  journal={Nature Microbiology},
  volume={3},
  number={7},
  pages={804--813},
  year={2018},
  url = {https://doi.org/10.1038/s41564-018-0176-9}
}

@article{dongen2000cluster,
  title={A cluster algorithm for graphs},
  author={Dongen, Stijn},
  year={2000},
  publisher={CWI (Centre for Mathematics and Computer Science)},
  url = {https://www.micans.org/mcl/index.html?sec_software}
}

@article{jain2018high,
  title={High throughput ANI analysis of 90K prokaryotic genomes reveals clear species boundaries},
  author={Jain, Chirag and Rodriguez-R, Luis M and Phillippy, Adam M and Konstantinidis, Konstantinos T and Aluru, Srinivas},
  journal={Nature Communications},
  volume={9},
  number={1},
  pages={1--8},
  year={2018},
  url={https://doi.org/10.1038/s41467-018-07641-9}
}

@article{stamatakis2014raxml,
  title={RAxML version 8: a tool for phylogenetic analysis and post-analysis of large phylogenies},
  author={Stamatakis, Alexandros},
  journal={Bioinformatics},
  volume={30},
  number={9},
  pages={1312--1313},
  year={2014},
  url={https://doi.org/10.1093/bioinformatics/btu033}
}

@article{minh2020iq,
  title={IQ-TREE 2: New models and efficient methods for phylogenetic inference in the genomic era},
  author={Minh, Bui Quang and Schmidt, Heiko A and Chernomor, Olga and Schrempf, Dominik and Woodhams, Michael D and Von Haeseler, Arndt and Lanfear, Robert},
  journal={Molecular Biology and Evolution},
  volume={37},
  number={5},
  pages={1530--1534},
  year={2020},
  url={https://doi.org/10.1093/molbev/msaa015}
}

@article{kalyaanamoorthy2017modelfinder,
  title={ModelFinder: fast model selection for accurate phylogenetic estimates},
  author={Kalyaanamoorthy, Subha and Minh, Bui Quang and Wong, Thomas KF and von Haeseler, Arndt and Jermiin, Lars S},
  journal={Nature Methods},
  volume={14},
  number={6},
  pages={587},
  year={2017},
  url={https://doi.org/10.1038/nmeth.4285}
}

@article{quince2011removing,
  title={Removing noise from pyrosequenced amplicons},
  author={Quince, Christopher and Lanzen, Anders and Davenport, Russell J and Turnbaugh, Peter J},
  journal={BMC bioinformatics},
  volume={12},
  number={1},
  pages={38},
  year={2011},
  url={https://doi.org/10.1186/1471-2105-12-38}
}
@article{caporaso2011global,
  title={Global patterns of 16S rRNA diversity at a depth of millions of sequences per sample},
  author={Caporaso, J Gregory and Lauber, Christian L and Walters, William A and Berg-Lyons, Donna and Lozupone, Catherine A and Turnbaugh, Peter J and Fierer, Noah and Knight, Rob},
  journal={Proceedings of the National Academy of Sciences},
  volume={108},
  number={Supplement 1},
  pages={4516--4522},
  year={2011},
  url={https://doi.org/10.1073/pnas.1000080107}
}
@article{kanehisa2000kegg,
  title={KEGG: kyoto encyclopedia of genes and genomes},
  author={Kanehisa, Minoru and Goto, Susumu},
  journal={Nucleic Acids Research},
  volume={28},
  number={1},
  pages={27--30},
  year={2000},
  url={https://doi.org/10.1093/nar/28.1.27}
}

@article{kanehisa2016blastkoala,
  title={BlastKOALA and GhostKOALA: KEGG tools for functional characterization of genome and metagenome sequences},
  author={Kanehisa, Minoru and Sato, Yoko and Morishima, Kanae},
  journal={Journal of Molecular Biology},
  volume={428},
  number={4},
  pages={726--731},
  year={2016},
  url={https://doi.org/10.1016/j.jmb.2015.11.006}
}

@article{minoche2011evaluation,
  title={Evaluation of genomic high-throughput sequencing data generated on Illumina HiSeq and genome analyzer systems},
  author={Minoche, Andr{\'e} E and Dohm, Juliane C and Himmelbauer, Heinz},
  journal={Genome Biology},
  volume={12},
  number={11},
  pages={R112},
  year={2011},
  url={https://doi.org/10.1186/gb-2011-12-11-r112}
}

@article{lee2019gtotree,
  title={GToTree: a user-friendly workflow for phylogenomics},
  author={Lee, Michael D},
  journal={Bioinformatics},
  volume={35},
  number={20},
  pages={4162--4164},
  year={2019},
  url={https://doi.org/10.1093/bioinformatics/btz188}
}

@article{wang2007naive,
  title={Naive Bayesian classifier for rapid assignment of rRNA sequences into the new bacterial taxonomy},
  author={Wang, Qiong and Garrity, George M and Tiedje, James M and Cole, James R},
  journal={Applied & Environmental Microbiology},
  volume={73},
  number={16},
  pages={5261--5267},
  year={2007},
  url={https://doi.org/10.1128/AEM.00062-07}
}

@article{hoang2017ultrafast,
  author = {Hoang, Diep Thi and Chernomor, Olga and von Haeseler, Arndt and Minh, Bui Quang and Vinh, Le Sy},
  title = {UFBoot2: Improving the Ultrafast Bootstrap Approximation},
  journal = {Molecular Biology and Evolution},
  volume = {35},
  number = {2},
  pages = {518-522},
  year = {2017},
  url = {https://doi.org/10.1093/molbev/msx281}
}

@article{parada2016every,
  title={Every base matters: assessing small subunit rRNA primers for marine microbiomes with mock communities, time series and global field samples},
  author={Parada, Alma E and Needham, David M and Fuhrman, Jed A},
  journal={Environmental microbiology},
  volume={18},
  number={5},
  pages={1403--1414},
  year={2016},
  url = {https://doi.org/10.1111/1462-2920.13023}
}

@article{parks2018gtgb,
  title={A standardized bacterial taxonomy based on genome phylogeny substantially revises the tree of life},
  author={Parks, Donovan H and Chuvochina, Maria and Waite, David W and Rinke, Christian and Skarshewski, Adam and Chaumeil, Pierre-Alain and Hugenholtz, Philip},
  journal={Nature Biotechnology},
  volume={36},
  number={10},
  pages={1546-1696},
  year={2018},
  url = {https://doi.org/10.1038/nbt.4229}
}
</script>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p>© Copyright 2020 The HYPOCOLYPSE Project.</p>
<p>Site constructed using <a href="https://rstudio.github.io/distill/">Distill for R Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
