<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

  <!--radix_placeholder_meta_tags-->
  <title>: No 5. Automated Binning</title>
  
  <meta property="description" itemprop="description" content="In this section of the workflow we use four different tools for automated binning of our assembled contigs. Our goal is to rebuild genomes from the metagenomic assembly. These are called metagenome assembled genomes, or MAGs."/>
  
  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>
  <link rel="icon" type="image/vnd.microsoft.icon" href="assets/favicon.ico"/>
  
  <meta name="article:author" content="Jarrod J Scott"/>
  
  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content=": No 5. Automated Binning"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="In this section of the workflow we use four different tools for automated binning of our assembled contigs. Our goal is to rebuild genomes from the metagenomic assembly. These are called metagenome assembled genomes, or MAGs."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content=""/>
  
  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content=": No 5. Automated Binning"/>
  <meta property="twitter:description" content="In this section of the workflow we use four different tools for automated binning of our assembled contigs. Our goal is to rebuild genomes from the metagenomic assembly. These are called metagenome assembled genomes, or MAGs."/>
  
  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Binning metagenomic contigs by coverage and composition;citation_publication_date=2014;citation_volume=11;citation_author=Johannes Alneberg;citation_author=Brynjar Smári Bjarnason;citation_author=Ino De Bruijn;citation_author=Melanie Schirmer;citation_author=Joshua Quick;citation_author=Umer Z Ijaz;citation_author=Leo Lahti;citation_author=Nicholas J Loman;citation_author=Anders F Andersson;citation_author=Christopher Quince"/>
  <meta name="citation_reference" content="citation_title=Recovery of genomes from metagenomes via a dereplication, aggregation and scoring strategy;citation_publication_date=2018;citation_volume=3;citation_author=Christian MK Sieber;citation_author=Alexander J Probst;citation_author=Allison Sharrar;citation_author=Brian C Thomas;citation_author=Matthias Hess;citation_author=Susannah G Tringe;citation_author=Jillian F Banfield"/>
  <meta name="citation_reference" content="citation_title=MetaBAT 2: An adaptive binning algorithm for robust and efficient genome reconstruction from metagenome assemblies;citation_publication_date=2019;citation_volume=7;citation_author=Dongwan D Kang;citation_author=Feng Li;citation_author=Edward Kirton;citation_author=Ashleigh Thomas;citation_author=Rob Egan;citation_author=Hong An;citation_author=Zhong Wang"/>
  <meta name="citation_reference" content="citation_title=MaxBin 2.0: An automated binning algorithm to recover genomes from multiple metagenomic datasets;citation_publication_date=2016;citation_volume=32;citation_author=Yu-Wei Wu;citation_author=Blake A Simmons;citation_author=Steven W Singer"/>
  <!--radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["No 5. Automated Binning"]},{"type":"character","attributes":{},"value":["In this section of the workflow we use four different tools for automated binning of our assembled contigs. Our goal is to rebuild genomes from the metagenomic assembly. These are called metagenome assembled genomes, or MAGs.\n"]},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name"]}},"value":[{"type":"character","attributes":{},"value":["Jarrod J Scott"]}]}]},{"type":"character","attributes":{},"value":["assets/cite.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  
  <script type="application/javascript">
  
    window.headroom_prevent_pin = false;
  
    window.document.addEventListener("DOMContentLoaded", function (event) {
  
      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');
  
      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });
  
      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });
  
      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>
  
  <style type="text/css">
  
  /* Theme (user-documented overrideables for nav appearance) */
  
  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #455a64;
    font-size: 15px;
    font-weight: 300;
  }
  
  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }
  
  .distill-site-nav a:hover {
    color: white;
  }
  
  @media print {
    .distill-site-nav {
      display: none;
    }
  }
  
  .distill-site-header {
  
  }
  
  .distill-site-footer {
  
  }
  
  
  /* Site Header */
  
  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }
  
  
  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }
  
  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }
  
  .distill-site-header .title {
    font-size: 18px;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }
  
  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }
  
  
  
  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }
  
  
  .distill-site-header .nav-toggle {
    display: none;
  }
  
  .nav-dropdown {
    display: inline-block;
    position: relative;
  }
  
  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }
  
  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }
  
  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .nav-dropdown-active {
    display: block;
  }
  
  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }
  
  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }
  
  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }
  
  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative;}
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }
  
  /* Site Footer */
  
  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }
  
  /* Headroom */
  
  d-title {
    padding-top: 6rem;
  }
  
  @media print {
    d-title {
      padding-top: 4rem;
    }
  }
  
  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }
  
  .headroom--transition {
    transition: all .4s ease-in-out;
  }
  
  .headroom--unpinned {
    top: -100px;
  }
  
  .headroom--pinned {
    top: 0;
  }
  
  </style>
  
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->
  
  <style type="text/css">
  
  body {
    background-color: white;
  }
  
  .pandoc-table {
    width: 100%;
  }
  
  .pandoc-table>caption {
    margin-bottom: 10px;
  }
  
  .pandoc-table th:not([align]) {
    text-align: left;
  }
  
  .pagedtable-footer {
    font-size: 15px;
  }
  
  .html-widget {
    margin-bottom: 2.0em;
  }
  
  .l-screen-inset {
    padding-right: 16px;
  }
  
  .l-screen .caption {
    margin-left: 10px;
  }
  
  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .shaded-content {
    background: white;
  }
  
  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }
  
  .hidden {
    display: none !important;
  }
  
  d-article {
    padding-bottom: 30px;
  }
  
  d-appendix {
    padding-top: 30px;
  }
  
  d-article>p>img {
    width: 100%;
  }
  
  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }
  
  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  /* CSS for table of contents */
  
  .d-toc {
    color: rgba(0,0,0,0.8);
    font-size: 0.8em;
    line-height: 1em;
  }
  
  .d-toc-header {
    font-size: 0.6rem;
    font-weight: 400;
    color: rgba(0, 0, 0, 0.5);
    text-transform: uppercase;
    margin-top: 0;
    margin-bottom: 1.3em;
  }
  
  .d-toc a {
    border-bottom: none;
  }
  
  .d-toc ul {
    padding-left: 0;
  }
  
  .d-toc li>ul {
    padding-top: 0.8em;
    padding-left: 16px;
    margin-bottom: 0.6em;
  }
  
  .d-toc ul,
  .d-toc li {
    list-style-type: none;
  }
  
  .d-toc li {
    margin-bottom: 0.9em;
  }
  
  .d-toc-separator {
    margin-top: 20px;
    margin-bottom: 2em;
  }
  
  .d-article-with-toc {
    border-top: none;
    padding-top: 0;
  }
  
  
  
  /* Tweak code blocks (note that this CSS is repeated above in an injection
     into the d-code shadow dom) */
  
  d-code {
    overflow-x: auto !important;
  }
  
  pre.d-code code.d-code {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }
  
  pre.text-output {
  
    font-size: 12px;
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  @media(min-width: 768px) {
  
  d-code {
    overflow-x: visible !important;
  }
  
  pre.d-code code.d-code  {
      padding-left: 18px;
      font-size: 14px;
  }
  pre.text-output {
    font-size: 14px;
  }
  }
  
  /* Figure */
  
  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }
  
  .figure img {
    width: 100%;
  }
  
  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }
  
  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }
  
  
  
  /* Tweak 1000px media break to show more text */
  
  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }
  
    .grid {
      grid-column-gap: 16px;
    }
  
    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }
  
  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }
  
    .grid {
      grid-column-gap: 32px;
    }
  }
  
  
  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */
  
  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }
  
  
  /* Social footer */
  
  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }
  
  .disqus-comments {
    margin-right: 30px;
  }
  
  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
  
  #disqus_thread {
    margin-top: 30px;
  }
  
  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }
  
  .article-sharing a:hover {
    border-bottom: none;
  }
  
  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .subscribe p {
    margin-bottom: 0.5em;
  }
  
  
  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }
  
  
  /* Improve display for browsers without grid (IE/Edge <= 15) */
  
  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }
  
  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }
  
  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }
  
  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }
  
  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }
  
  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
  
  
  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }
  
  .downlevel .footnotes ol {
    padding-left: 13px;
  }
  
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }
  
  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }
  
  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }
  
  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  </style>
  
  <script type="application/javascript">
  
  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }
  
  // show body when load is complete
  function on_load_complete() {
  
    // set body to visible
    document.body.style.visibility = 'visible';
  
    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }
  
    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }
  
  function init_distill() {
  
    init_common();
  
    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);
  
    // create d-title
    $('.d-title').changeElementType('d-title');
  
    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);
  
    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();
  
    // move posts container into article
    $('.posts-container').appendTo($('d-article'));
  
    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');
  
    // create d-bibliography
    var bibliography = $('<d-bibliography></d-bibliography>');
    $('#distill-bibliography').wrap(bibliography);
  
    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;
  
    // replace citations with <d-cite>
    $('.citation').each(function(i, val) {
      appendix = true;
      var cites = $(this).attr('data-cites').split(" ");
      var dt_cite = $('<d-cite></d-cite>');
      dt_cite.attr('key', cites.join());
      $(this).replaceWith(dt_cite);
    });
    // remove refs
    $('#refs').remove();
  
    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();
  
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-toc a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });
  
    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');
  
    // replace code blocks with d-code
    $('pre>code').each(function(i, val) {
      var code = $(this);
      var pre = code.parent();
      var clz = "";
      var language = pre.attr('class');
      if (language) {
        // map unknown languages to "clike" (without this they just dissapear)
        if ($.inArray(language, ["bash", "clike", "css", "go", "html",
                                 "javascript", "js", "julia", "lua", "markdown",
                                 "markup", "mathml", "python", "svg", "xml"]) == -1)
          language = "clike";
        language = ' language="' + language + '"';
        var dt_code = $('<d-code block' + language + clz + '></d-code>');
        dt_code.text(code.text());
        pre.replaceWith(dt_code);
      } else {
        code.addClass('text-output').unwrap().changeElementType('pre');
      }
    });
  
    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {
  
      // capture layout
      var layout = $(this).attr('data-layout');
  
      // apply layout to markdown level block elements
      var elements = $(this).children().not('d-code, pre.text-output, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });
  
  
      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });
  
    // load distill framework
    load_distill_framework();
  
    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {
  
      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;
  
      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');
  
      // table of contents
      if (have_authors) // adjust border if we are in authors
        $('.d-toc').parent().addClass('d-article-with-toc');
  
      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');
  
      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }
  
      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');
  
      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");
  
      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }
  
       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }
  
      // inject pre code styles (can't do this with a global stylesheet b/c a shadow root is used)
      $('d-code').each(function(i, val) {
        var style = document.createElement('style');
        style.innerHTML = 'pre code { padding-left: 10px; font-size: 12px; border-left: 2px solid rgba(0,0,0,0.1); } ' +
                          '@media(min-width: 768px) { pre code { padding-left: 18px; font-size: 14px; } }';
        if (this.shadowRoot)
          this.shadowRoot.appendChild(style);
      });
  
      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();
  
      // clear polling timer
      clearInterval(tid);
  
      // show body now that everything is ready
      on_load_complete();
    }
  
    var tid = setInterval(distill_post_process, 50);
    distill_post_process();
  
  }
  
  function init_downlevel() {
  
    init_common();
  
     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));
  
    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
  
    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();
  
    // remove toc
    $('.d-toc-header').remove();
    $('.d-toc').remove();
    $('.d-toc-separator').remove();
  
    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });
  
  
    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);
  
    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);
  
    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();
  
    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();
  
    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));
  
    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });
  
    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));
  
    $('body').addClass('downlevel');
  
    on_load_complete();
  }
  
  
  function init_common() {
  
    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};
  
        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });
  
        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);
  
    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});
  
    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });
  
      }
    });
  
    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });
  
    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }
  
    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');
  
    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");
  
    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();
  
    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }
  
  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });
  
  </script>
  
  <!--/radix_placeholder_distill-->
  <script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
  <script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
  <link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
  <script src="site_libs/datatables-binding-0.13/datatables.js"></script>
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
  <script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
  <link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
  <script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
  <link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
  <script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Roboto:300|Nunito+Sans:300|Inconsolata|Source+Sans+Pro');
  /* controls general properties of site*/
  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    background-color: #FAFAFA;
  }
  
  d-title h1{
  font-size: 2.5em;
  font-family: Roboto;
  font-weight: 500;
  }
  
  d-article h1, h2 {
    font-family: Roboto;
  }
  
  d-article h3 {
    margin-top: 1em;
  }
  
  blockquote p {
  	font-size: 1.3em;
  	font-family: 'Lato';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }
  
  d-code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    border-left: 5px solid #FC6464;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
  }
  
  pre.text-output {
    margin-top: 0px;
     color: #333333;
     background-color: #FAFAFA;
     font-size: 0.8em
   }
  
  pre {
    border: 1px solid #ccc;
    padding: 0px 10px 10px 18px;
    margin: 1em 0px;
    border-radius: 5px;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    text-transform: none;
  }
  
  d-article li {
    margin: 0px 0px 4px;
  }
  
  /*************************************************
  *  code for side by side krona plot
  **************************************************/
  
  .column {
    float: left;
    width: 48%;
    padding: 5px;
  }
  
  /* Clear floats after image containers */
  .row::after {
    content: "";
    clear: both;
    display: table;
  }
  
  div.krona {
    width: 100%;
  	height: 90%;
    background-color: white;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    margin-bottom: 25px;
  	margin-top: 5px;
  }
  
  div.container-krona {
    text-align: center;
    padding: 1px 10px;
  	color: #414141;
  	font-size: 1.48em;
  	line-height: 0.8em;
  	font-family: 'Inconsolata';
  }
  
  div.container-krona p {
    margin: 20px;
  }
  
  div.krona img {
    border:1px solid white;
    margin: 25px;
  }
  
  img {
    margin: 25px;
  }
  
  blockquote p {
  	font-size: 1.3em;
  	font-family: 'Lato';
  }
  
  /** LIGHTBOX MARKUP **/
  .thumbnail {
    max-width: 100%;
    position: relative;
  }
  
  .lightbox {
  	/** Default lightbox to hidden */
  	display: none;
  
  	/** Position and style */
  	position: fixed;
  	z-index: 999;
  	width: 100%;
  	height: 100%;
  	text-align: center;
  	top: 0;
  	left: 0;
  	background: rgba(0,0,0,0.8);
  }
  
  .lightbox img {
  	/** Pad the lightbox image */
  	position: fixed;
  	top: 0;
  	left: 0;
  	right: 0;
  	bottom: 0;
  	max-width: 0%;
  	max-height: 0%;
  	margin: auto;
  	box-sizing: border-box;
  }
  
  .lightbox:target {
  	/** Remove default browser outline */
  	outline: none;
  
  	/** Unhide lightbox **/
  	display: block;
  }
  
  .lightbox:target img {
  	max-height: 100%;
  	max-width: 100%;
  }
  
  .distill-site-header .logo img {
    margin-top: 10px;
    max-height: 30px;
    margin-bottom: 10px;
  }
  
  .distill-site-nav {
      color: rgba(255, 255, 255, 0.8);
      background-color: #143D59;
      font-size: 16px;
      font-weight: 300;
    }
  }
  
  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"No 5. Automated Binning","description":"In this section of the workflow we use four different tools for automated binning of our assembled contigs. Our goal is to rebuild genomes from the metagenomic assembly. These are called metagenome assembled genomes, or MAGs.","authors":[{"author":"Jarrod J Scott","authorURL":"#","affiliation":"&nbsp;","affiliationURL":"#"}]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a class="logo" href="index.html">
<img src="assets/icon.png"/>
</a>
<a href="index.html" class="title"></a>
</div>
<div class="nav-right">
<a href="index.html">
<i class="fa fa-home"></i>
</a>
<a href="intro.html">Introduction</a>
<a href="field.html">A. Field Analyses</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
B. 16S rRNA
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">16S rRNA Processing</span>
<a href="16s-dada2.html">1. DADA2</a>
<a href="16s-data-prep.html">2. Data Prep</a>
<hr/>
<span class="nav-dropdown-header">16S rRNA Analyses</span>
<a href="16s-water.html">3. Water Diversity</a>
<hr/>
<a href="session-info.html">Session Info</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
C. Metagenomic
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Setup</span>
<a href="mg-setup.html">1. Working Envrionment</a>
<a href="mg-databases.html">2. Annotation Databases</a>
<hr/>
<span class="nav-dropdown-header">Read Processing</span>
<a href="mg-workflow-1.html">3. Assembly &amp; Annotations</a>
<a href="mg-workflow-2.html">4. Assembley &amp; Annotation Summary</a>
<hr/>
<span class="nav-dropdown-header">Binning</span>
<a href="mg-auto-binning.html">5. Automated Binning</a>
</div>
</div>
<a href="synthesis.html">D. Synthesis</a>
<a href="data-availability.html">Data</a>
<a href="https://github.com/hypocolypse/web/">
<i class="fa fa-github fa-lg"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>No 5. Automated Binning</h1>
<p>In this section of the workflow we use four different tools for automated binning of our assembled contigs. Our goal is to rebuild genomes from the metagenomic assembly. These are called metagenome assembled genomes, or MAGs.</p>
</div>

<div class="d-byline">
  Jarrod J Scott  
  

</div>

<div class="d-article">
<h3 class="d-toc-header">Table of Contents</h3>
<nav class="d-toc" id="TOC">
<ul>
<li><a href="#automated-binning-tools">Automated Binning Tools</a></li>
<li><a href="#run-tools">Run Tools</a></li>
<li><a href="#inspect-results">Inspect Results</a><ul>
<li><a href="#concoct">CONCOCT</a></li>
<li><a href="#concoct-5">CONCOCT 5</a></li>
<li><a href="#metabat2">MetaBAT2</a></li>
<li><a href="#maxbin2">MaxBIN2</a></li>
<li><a href="#dastool">DASTOOL</a></li>
</ul></li>
<li><a href="#source-code">Source Code</a></li>
</ul>
</nav>
<hr class="d-toc-separator"/>
<h2 id="automated-binning-tools">Automated Binning Tools</h2>
<ol type="1">
<li><a href="https://github.com/BinPro/CONCOCT">CONCOCT</a> (Clustering cONtigs with COverage and ComposiTion)<span class="citation" data-cites="alneberg2014binning">(Alneberg et al. <a href="#ref-alneberg2014binning">2014</a>)</span> is a program for unsupervised binning of metagenomic contigs by using nucleotide composition, coverage data in multiple samples and linkage data from paired end reads.</li>
<li><a href="https://bitbucket.org/berkeleylab/metabat">MetaBAT2</a><span class="citation" data-cites="kang2019metabat">(Kang et al. <a href="#ref-kang2019metabat">2019</a>)</span> is a statistical framework for reconstructing genomes from metagenomic data.</li>
<li><a href="https://sourceforge.net/projects/maxbin2/">MaxBIN2</a><span class="citation" data-cites="wu2016maxbin">(Wu, Simmons, and Singer <a href="#ref-wu2016maxbin">2016</a>)</span> is an automatic tool for binning metagenomics sequences.</li>
<li><a href="https://github.com/cmks/DAS_Tool">DASTOOL</a><span class="citation" data-cites="sieber2018recovery">(Sieber et al. <a href="#ref-sieber2018recovery">2018</a>)</span> is an automated method that integrates the results of a number of binning algorithms to calculate an optimized, non-redundant set of bins from a single assembly.</li>
</ol>
<h2 id="run-tools">Run Tools</h2>
<p>The latest iteration of anvi’o (v6 as of this writing) ports several popular automated binning algorithms into its ecosystem. We will bin the water meatgenomic assembly using CONCOCT, MetaBAT2, and MaxBIN2. Then we use DASTOOL to integrate the results of the other binning tools. The anvio commands below will a) bin the assembly using the specified tool and b) add the results to a collection in the <code>PROFILE.db</code>. For these steps we use the command <code>anvi-cluster-contigs</code>.</p>
<p>Here we run CONCOCT in two different ways. The first is using the defaults and the second is where we speciffy the number of bins. To my knowledge CONCOCT is the only one of these tools that allows you to specify the number of clusters.</p>
<pre class="bash"><code>
anvi-cluster-contigs -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -T $NSLOTS -C  CONCOCT --driver concoct --just-do-it
anvi-cluster-contigs -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -T $NSLOTS -C  CONCOCT_5 --driver concoct --just-do-it --clusters 5</code></pre>
<p>Next the command for MetaBAT2.</p>
<pre class="bash"><code>
anvi-cluster-contigs -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -T $NSLOTS -C  MetaBAT2 --driver metabat2 --just-do-it --minContig 1500</code></pre>
<p>And MaxBIN2</p>
<pre class="bash"><code>
anvi-cluster-contigs -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -T $NSLOTS -C  MaxBIN2 --driver maxbin2 --just-do-it --min-contig-length 1000</code></pre>
<p>Now we can run DASTOOL to aggregate the results. Since we ran two instances of CONCOCT we need to run both in DASTOOL against the results of MaxBIN2 and MetaBAT2.</p>
<pre class="bash"><code>
anvi-cluster-contigs -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -T $NSLOTS -C  DASTOOL --driver dastool --just-do-it --search-engine diamond -S CONCOCT,MaxBIN2,MetaBAT2 --debug
anvi-cluster-contigs -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -T $NSLOTS -C  DASTOOL_5 --driver dastool --just-do-it --search-engine diamond -S CONCOCT_5,MaxBIN2,MetaBAT2 --debug</code></pre>
<details>
<summary>Show/hide HYDRA CONTIG AUTO BINNING job script</summary>
<pre><code>
# /bin/sh
# ----------------Parameters---------------------- #
#$ -S /bin/sh
#$ -pe mthread 5
#$ -q sThC.q
#$ -l mres=25G,h_data=5G,h_vmem=5G
#$ -cwd
#$ -j y
#$ -N water_job_11_cluster_contigs
#$ -o hydra_logs/job_11_cluster_contigs_water.log
#$ -M scottjj@si.edu
#
# ----------------Modules------------------------- #
module load gcc/4.9.2
#
# ----------------Your Commands------------------- #
#
echo + `date` job $JOB_NAME started in $QUEUE with jobID=$JOB_ID on $HOSTNAME
echo + NSLOTS = $NSLOTS
#
# ----------------CALLING ANVIO------------------- #
#
export PATH=/home/scottjj/miniconda3:$PATH
export PATH=/home/scottjj/miniconda3/bin:$PATH
export PATH=/home/scottjj/miniconda3/envs:$PATH
source activate anvio-master
#
# ----------------CHECKING EVERYTHING------------------- #
#
which python
python --version
source /home/scottjj/virtual-envs/anvio-master/bin/activate
which python
python --version
which anvi-interactive
diamond --version
anvi-self-test -v
#
# ----------------BINNING CONDA------------------- #
#
source activate binning
conda activate binning
which run_MaxBin.pl
which concoct
#
# ----------------SETUP TEMP DIRECTORIES------------------- #
#
rm -r /pool/genomics/stri_istmobiome/dbs/tmp_data_WATER/
mkdir -p /pool/genomics/stri_istmobiome/dbs/tmp_data_WATER/
TMPDIR="/pool/genomics/stri_istmobiome/dbs/tmp_data_WATER/"
#
# ----------------CONCOCT------------------- #
#
anvi-cluster-contigs -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -T $NSLOTS -C  CONCOCT --driver concoct --just-do-it --debug
anvi-cluster-contigs -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -T $NSLOTS -C  CONCOCT_5 --driver concoct --just-do-it --clusters 5 --debug
#
# ----------------MetaBAT2------------------- #
#
anvi-cluster-contigs -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -T $NSLOTS -C  MetaBAT2 --driver metabat2 --just-do-it --debug --minContig 1500
#
# ----------------MaxBIN2------------------- #
#
anvi-cluster-contigs -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -T $NSLOTS -C  MaxBIN2 --driver maxbin2 --just-do-it --min-contig-length 1000 --debug
#
# ----------------DASTOOL------------------- #
#
anvi-cluster-contigs -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -T $NSLOTS -C  DASTOOL --driver dastool --just-do-it --search-engine diamond -S CONCOCT,MaxBIN2,MetaBAT2 --debug
anvi-cluster-contigs -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -T $NSLOTS -C  DASTOOL_5 --driver dastool --just-do-it --search-engine diamond -S CONCOCT_5,MaxBIN2,MetaBAT2 --debug
#
echo = `date` job $JOB_NAME done
</code></pre>
</details>
<h2 id="inspect-results">Inspect Results</h2>
<p>Let’s take a look at the results of the different binning strategies using the command <code>anvi-export-collection</code> with the <code>--list-collections</code> flag. Anvio won’t do anything here except show us what collections are in the <code>PROFILE.db</code>.</p>
<pre class="bash"><code>
anvi-export-collection -p 06_MERGED/WATER/PROFILE.db --list-collections</code></pre>
<p>And here is the output of that command. The VIRSORTER collection was imported during the annotation phase and can be ignored for now.</p>
<pre class="bash"><code>
COLLECTIONS FOUND
===============================================
* VIRSORTER (2840 bins, representing 2857 items).
* CONCOCT (71 bins, representing 23716 items).
* CONCOCT_5 (5 bins, representing 23716 items).
* METABAT2 (5 bins, representing 1442 items).
* MAXBIN2 (16 bins, representing 22489 items).
* DASTOOL (7 bins, representing 4775 items).
* DASTOOL_5 (2 bins, representing 956 items).</code></pre>
<p>Here we can see the number of bins and the total number of contigs included in each collection. Remember the original assembly had 23,758 contigs (at a minimum length of 1000bp). We can take a quick look at the estimates of genome completion and redundancy using domain-specific single-copy core genes for each collection using <code>anvi-estimate-genome-completeness</code>.</p>
<p>One thing to remember is how each algorithm names bins, or else this can get confusing. CONCOCT adds the prefix <code>Bin_</code>, MaxBin adds the prefix <code>MAXBIN_</code>, and MetaBAT adds the prefix <code>METABAT__</code> (with two underscores). DAS Tool then adds its own <code>Bin_</code> prefix to the parent name. For example DAS Tool <code>Bin_Bin_11</code> is CONCOCT bin <code>Bin_11</code> while DASTOOL <code>Bin_METABAT__2</code> is MetaBAT2 bin <code>METABAT__2</code>.</p>
<h3 id="concoct">CONCOCT</h3>
<p>CONCOCT bins with default parameters.</p>
<pre class="bash"><code>
anvi-estimate-genome-completeness -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -C CONCOCT</code></pre>
<p>Here are the results for the 71 CONCOCT bins.</p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="htmlwidget-943a55846508a9f81bdc" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-943a55846508a9f81bdc">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">Table 9.1: CONCOCT binning results<\/caption>","autoHideNavigation":true,"data":[["Bin_18","Bin_39","Bin_23","Bin_66","Bin_65","Bin_7","Bin_31","Bin_8","Bin_5","Bin_55","Bin_68","Bin_52","Bin_63","Bin_14","Bin_17","Bin_32","Bin_15","Bin_22","Bin_4","Bin_28","Bin_69","Bin_41","Bin_19","Bin_60","Bin_47","Bin_42","Bin_11","Bin_33","Bin_58","Bin_35","Bin_20","Bin_0","Bin_43","Bin_67","Bin_53","Bin_13","Bin_9","Bin_54","Bin_1","Bin_64","Bin_59","Bin_30","Bin_51","Bin_38","Bin_56","Bin_25","Bin_48","Bin_37","Bin_61","Bin_29","Bin_36","Bin_3","Bin_6","Bin_62","Bin_70","Bin_2","Bin_34","Bin_46","Bin_12","Bin_44","Bin_57","Bin_27","Bin_26","Bin_49","Bin_40","Bin_50","Bin_21","Bin_16","Bin_10","Bin_24","Bin_45"],["BLANK","BACTERIA","BLANK","BACTERIA","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BACTERIA","BLANK","BLANK","BACTERIA","BLANK","BLANK","BLANK","BACTERIA","ARCHAEA","BLANK","BLANK","BACTERIA","BLANK","ARCHAEA","BACTERIA","BACTERIA","BLANK","BACTERIA","BLANK","BLANK","BLANK","BLANK","BACTERIA","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK","BLANK"],[1,0.9,1,0.4,0.8,1,1,1,1,1,1,0.7,1,1,1,1,0.6,1,1,0.3,0.2,1,1,0.4,1,0.4,0.7,0.9,1,0.9,1,0.7,1,1,0.5,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[0,98.59,0,42.25,0,0,0,0,0,0,0,0,97.18,0,0,91.55,0,0,0,15.49,42.11,0,0,42.25,0,51.32,83.1,94.37,0,90.14,0,0,0,0,49.3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,363.38,0,0,0,0,0,0,0,0,0,0,105.63,0,0,12.68,0,0,0,2.82,22.37,0,0,1.41,0,6.58,9.86,9.86,0,8.45,0,0,0,0,1.41,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[1909,2873,1020,428,161,286,347,3130,44,84,234,1072,1357,458,97,875,319,320,794,943,1064,919,388,745,139,508,1144,673,92,495,61,196,108,123,183,17,1,9,10,2,13,7,2,2,1,3,3,4,9,2,8,3,3,1,1,1,5,1,1,1,1,3,2,2,3,1,1,1,1,1,1],[2878395,4230640,1641401,593993,195308,647813,567474,4784769,88132,145516,540286,1403974,2085721,963542,581617,1683488,418832,646136,4422671,1523468,1418866,1787227,566964,1021857,219759,809567,1876732,2091590,249714,1077361,101352,363481,354152,152664,263671,27463,1100,20178,13209,2172,40513,8773,5618,2865,1203,3422,4454,4307,10731,3419,13061,4152,3865,1225,1283,1196,7297,1131,1396,1288,1062,4625,2500,2269,3137,1010,1135,1256,1847,2240,1290]],"container":"<table class=\"FixedColumns\">\n  <thead>\n    <tr>\n      <th>bin name<\/th>\n      <th>domain<\/th>\n      <th>confidence<\/th>\n      <th>% completion<\/th>\n      <th>% redundancy<\/th>\n      <th>num_splits<\/th>\n      <th>total length<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":0},{"className":"dt-right","targets":[2,3,4,5,6]}],"dom":"Brltip","buttons":["csv","copy","colvis"],"pageLength":5,"FixedColumns":{"leftColumns":2,"rightColumns":1},"scrollY":false,"scroller":true,"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<h3 id="concoct-5">CONCOCT 5</h3>
<p>CONCOCT bins with the <code>--clusters</code> flag equal to 5.</p>
<pre class="bash"><code>
anvi-estimate-genome-completeness -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -C CONCOCT_5</code></pre>
<p>Here are the results for the 5 CONCOCT bins with the <code>--clusters</code> flag equal to 5.</p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="htmlwidget-28f5cb61d4bf48e78591" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-28f5cb61d4bf48e78591">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">Table 9.2: CONCOCT 5 fored cluster binning results<\/caption>","autoHideNavigation":true,"data":[["Bin_0","Bin_2","Bin_4","Bin_1","Bin_3"],["BACTERIA","BLANK","BACTERIA","ARCHAEA","BACTERIA"],[0.9,1,0.8,0.2,1],[100,0,98.59,82.89,98.59],[467.61,0,263.38,110.53,160.56],[6461,6525,3996,4691,2043],[10394293,14323882,7392044,7140550,3355056]],"container":"<table class=\"FixedColumns\">\n  <thead>\n    <tr>\n      <th>bin name<\/th>\n      <th>domain<\/th>\n      <th>confidence<\/th>\n      <th>% completion<\/th>\n      <th>% redundancy<\/th>\n      <th>num_splits<\/th>\n      <th>total length<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":0},{"className":"dt-right","targets":[2,3,4,5,6]}],"dom":"Brti","buttons":["csv","copy","colvis"],"pageLength":5,"FixedColumns":{"leftColumns":2,"rightColumns":1},"scrollY":false,"scroller":true,"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<h3 id="metabat2">MetaBAT2</h3>
<pre class="bash"><code>
anvi-estimate-genome-completeness -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -C METABAT2</code></pre>
<p>Here are the results for the 5 MetaBAT2 bins.</p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="htmlwidget-a3760ded907df2533572" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a3760ded907df2533572">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">Table 9.3: MetaBAT2 binning results<\/caption>","autoHideNavigation":true,"data":[["METABAT__4","METABAT__5","METABAT__2","METABAT__1","METABAT__3"],["BACTERIA","BACTERIA","BACTERIA","BACTERIA","BLANK"],[0.5,0.4,0.9,0.4,0.9],[46.48,57.75,85.92,45.07,0],[4.23,4.23,2.82,2.82,0],[164,357,295,284,342],[367788,967826,1560594,777991,770834]],"container":"<table class=\"FixedColumns\">\n  <thead>\n    <tr>\n      <th>bin name<\/th>\n      <th>domain<\/th>\n      <th>confidence<\/th>\n      <th>% completion<\/th>\n      <th>% redundancy<\/th>\n      <th>num_splits<\/th>\n      <th>total length<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":0},{"className":"dt-right","targets":[2,3,4,5,6]}],"dom":"Brti","buttons":["csv","copy","colvis"],"pageLength":5,"FixedColumns":{"leftColumns":2,"rightColumns":1},"scrollY":false,"scroller":true,"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<h3 id="maxbin2">MaxBIN2</h3>
<pre class="bash"><code>
anvi-estimate-genome-completeness -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -C MAXBIN2</code></pre>
<p>Here are the results for the 16 MaxBIN2 bins.</p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="htmlwidget-f353840f458973aaf47d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f353840f458973aaf47d">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">Table 9.4: MaxBIN2 binning results<\/caption>","autoHideNavigation":true,"data":[["MAXBIN__004","MAXBIN__001","MAXBIN__002","MAXBIN__013","MAXBIN__009","MAXBIN__011","MAXBIN__014","MAXBIN__007","MAXBIN__003","MAXBIN__006","MAXBIN__010","MAXBIN__005","MAXBIN__012","MAXBIN__008","MAXBIN__015","MAXBIN__016"],["BACTERIA","BLANK","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BLANK","BLANK","BACTERIA","BACTERIA","EUKARYA","BLANK","BACTERIA","BACTERIA"],[0.6,1,1,0.7,0.9,0.5,0.5,0.6,0.6,0.6,0.7,0.7,0.3,1,0.6,0.8],[56.34,0,95.77,97.18,87.32,53.52,52.11,71.83,0,0,73.24,71.83,10.84,0,67.61,83.1],[46.48,0,29.58,66.2,71.83,9.86,5.63,32.39,0,0,98.59,11.27,2.41,0,36.62,21.13],[614,496,721,2258,955,921,1059,734,3033,1308,4056,854,1885,2504,430,661],[951606,2522035,2200677,3407557,1537630,1691060,1620255,1089442,5865919,2231954,6904709,1640820,2815608,4780066,593428,1154935]],"container":"<table class=\"FixedColumns\">\n  <thead>\n    <tr>\n      <th>bin name<\/th>\n      <th>domain<\/th>\n      <th>confidence<\/th>\n      <th>% completion<\/th>\n      <th>% redundancy<\/th>\n      <th>num_splits<\/th>\n      <th>total length<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":0},{"className":"dt-right","targets":[2,3,4,5,6]}],"dom":"Brltip","buttons":["csv","copy","colvis"],"pageLength":5,"lengthMenu":[5,20],"FixedColumns":{"leftColumns":2,"rightColumns":1},"scrollY":false,"scroller":true,"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<h3 id="dastool">DASTOOL</h3>
<p>And finnally we can see what happens when we let DASTOOL choose bins from the combination of the other three tools.</p>
<p>First the results using the default CONCOCT settings.</p>
<pre class="bash"><code>
anvi-estimate-genome-completeness -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -C DASTOOL</code></pre>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="htmlwidget-8f24fd2299602be33767" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-8f24fd2299602be33767">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">Table 9.5: DASTOOL binning results<\/caption>","autoHideNavigation":true,"data":[["Bin_METABAT__2","Bin_Bin_32_sub","Bin_Bin_11","Bin_Bin_35","Bin_MAXBIN__016","Bin_Bin_53_sub","Bin_MAXBIN__013_sub"],["BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA"],[0.9,0.9,0.7,0.9,0.8,0.4,0.5],[85.92,90.14,83.1,90.14,83.1,42.25,59.15],[2.82,12.68,9.86,8.45,21.13,1.41,11.27],[295,874,1144,495,661,163,1143],[1560594,1680114,1876732,1077361,1154935,228083,1585419]],"container":"<table class=\"FixedColumns\">\n  <thead>\n    <tr>\n      <th>bin name<\/th>\n      <th>domain<\/th>\n      <th>confidence<\/th>\n      <th>% completion<\/th>\n      <th>% redundancy<\/th>\n      <th>num_splits<\/th>\n      <th>total length<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":0},{"className":"dt-right","targets":[2,3,4,5,6]}],"dom":"Brti","buttons":["csv","copy","colvis"],"pageLength":7,"FixedColumns":{"leftColumns":2,"rightColumns":1},"scrollY":false,"scroller":true,"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[7,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<p>Here we see that DASTOOL used at least 1 bin from each of the three methods.</p>
<p>And then the results using the CONCOCT <code>--clusters</code> flag equal to 5.</p>
<pre class="bash"><code>
anvi-estimate-genome-completeness -c 03_CONTIGS/WATER-contigs.db -p 06_MERGED/WATER/PROFILE.db -C DASTOOL_5</code></pre>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="htmlwidget-5c2ac1b4d9b27d6287f3" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5c2ac1b4d9b27d6287f3">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">Table 9.6: DASTOOL binning results for CONCOCT forced cluster 5<\/caption>","autoHideNavigation":true,"data":[["Bin_METABAT__2","Bin_MAXBIN__016"],["BACTERIA","BACTERIA"],[0.9,0.8],[85.92,83.1],[2.82,21.13],[295,661],[1560594,1154935]],"container":"<table class=\"FixedColumns\">\n  <thead>\n    <tr>\n      <th>bin name<\/th>\n      <th>domain<\/th>\n      <th>confidence<\/th>\n      <th>% completion<\/th>\n      <th>% redundancy<\/th>\n      <th>num_splits<\/th>\n      <th>total length<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":0},{"className":"dt-right","targets":[2,3,4,5,6]}],"dom":"Brti","buttons":["csv","copy","colvis"],"pageLength":7,"FixedColumns":{"leftColumns":2,"rightColumns":1},"scrollY":false,"scroller":true,"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[7,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<p>However, here DASTOOL did not choose any of the CONCOCT bins.</p>
<div class="post-nav">
<div class="post-nav-item">
<div class="meta-nav">
Previous
</div>
<p><a href="mg-workflow-2.html" rel="next">N<sup><u>o</u></sup> 4. Assembley &amp; Annotation Summary</a></p>
</div>
</div>
<div class="post-nav">
<div class="post-nav-item">
<div class="meta-nav">
Next
</div>
<p><a href="mg-workflow-2.html" rel="prev">N<sup><u>o</u></sup> 6. SOMETHING</a></p>
</div>
</div>
<h2 id="source-code" class="appendix">Source Code</h2>
<p>The source code for this page can be accessed on GitHub by <a href="https://github.com/hypocolypse/web/blob/master/mg-auto-binning.Rmd">clicking this link</a>.</p>
<div id="refs" class="references">
<div id="ref-alneberg2014binning">
<p>Alneberg, Johannes, Brynjar Smári Bjarnason, Ino De Bruijn, Melanie Schirmer, Joshua Quick, Umer Z Ijaz, Leo Lahti, Nicholas J Loman, Anders F Andersson, and Christopher Quince. 2014. “Binning Metagenomic Contigs by Coverage and Composition.” <em>Nature Methods</em> 11 (11): 1144–6. <a href="https://doi.org/10.1038/nmeth.3103">https://doi.org/10.1038/nmeth.3103</a>.</p>
</div>
<div id="ref-kang2019metabat">
<p>Kang, Dongwan D, Feng Li, Edward Kirton, Ashleigh Thomas, Rob Egan, Hong An, and Zhong Wang. 2019. “MetaBAT 2: An Adaptive Binning Algorithm for Robust and Efficient Genome Reconstruction from Metagenome Assemblies.” <em>PeerJ</em> 7: e7359.</p>
</div>
<div id="ref-sieber2018recovery">
<p>Sieber, Christian MK, Alexander J Probst, Allison Sharrar, Brian C Thomas, Matthias Hess, Susannah G Tringe, and Jillian F Banfield. 2018. “Recovery of Genomes from Metagenomes via a Dereplication, Aggregation and Scoring Strategy.” <em>Nature Microbiology</em> 3 (7): 836–43.</p>
</div>
<div id="ref-wu2016maxbin">
<p>Wu, Yu-Wei, Blake A Simmons, and Steven W Singer. 2016. “MaxBin 2.0: An Automated Binning Algorithm to Recover Genomes from Multiple Metagenomic Datasets.” <em>Bioinformatics</em> 32 (4): 605–7.</p>
</div>
</div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/hypocolypse/web/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/hypocolypse/web/">https://github.com/hypocolypse/web/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<script id="distill-bibliography" type="text/bibtex">
@article{callahan2016dada2,
  title={DADA2: high-resolution sample inference from Illumina amplicon data},
  author={Callahan, Benjamin J and McMurdie, Paul J and Rosen, Michael J and Han, Andrew W and Johnson, Amy Jo A and Holmes, Susan P},
  journal={Nature Methods},
  volume={13},
  number={7},
  pages={581},
  year={2016},
  url = {https://doi.org/10.1038/nmeth.3869},
}

@article{martin2011cutadapt,
  title={Cutadapt removes adapter sequences from high-throughput sequencing reads},
  author={Martin, Marcel},
  journal={EMBnet. journal},
  volume={17},
  number={1},
  pages={10--12},
  year={2011},
  url = {https://doi.org/10.14806/ej.17.1.200}
}

@article{quast2012silva,
  title={The SILVA ribosomal RNA gene database project: improved data processing and web-based tools},
  author={Quast, Christian and Pruesse, Elmar and Yilmaz, Pelin and Gerken, Jan and Schweer, Timmy and Yarza, Pablo and Peplies, J{\"o}rg and Gl{\"o}ckner, Frank Oliver},
  journal={Nucleic Acids Research},
  volume={41},
  number={D1},
  pages={D590--D596},
  year={2012},
  url = {https://doi.org/10.1093/nar/gks1219}
}

@article{mcmurdie2013phyloseq,
  title={phyloseq: an R package for reproducible interactive analysis and graphics of microbiome census data},
  author={McMurdie, Paul J and Holmes, Susan},
  journal={PLoS One},
  volume={8},
  number={4},
  pages={e61217},
  year={2013},
  url = {https://doi.org/10.1371/journal.pone.0061217}
}

@misc{wong2011points,
  title={Points of view: Color blindness},
  author={Wong, Bang},
  journal={Nature Methods},
  volume={8},
  number={6},
  pages={441},
  year={2011},
  url = {https://doi.org/10.1038/nmeth.1618}
}

@misc{lahti2017microbiome,
  author = {Lahti, Leo and Shetty Sudarshan and others},
  title = {Tools for microbiome analysis in R. Version 1.9.97},
  year = {2017},
  url = {https://github.com/microbiome/microbiome},
}

@article{eren2015anvi,
  title={Anvi’o: an advanced analysis and visualization platform for ‘omics data},
  author={Eren, A Murat and Esen, {\"O}zcan C and Quince, Christopher and Vineis, Joseph H and Morrison, Hilary G and Sogin, Mitchell L and Delmont, Tom O},
  journal={PeerJ},
  volume={3},
  pages={e1319},
  year={2015},
  url = {https://doi.org/10.7717/peerj.1319}
}
@article{bolger2014trimmomatic,
  title={Trimmomatic: a flexible trimmer for Illumina sequence data},
  author={Bolger, Anthony M and Lohse, Marc and Usadel, Bjoern},
  journal={Bioinformatics},
  volume={30},
  number={15},
  pages={2114--2120},
  year={2014},
  url = {https://doi.org/10.1093/bioinformatics/btu170}
}
@article{koster2012snakemake,
  title={Snakemake—a scalable bioinformatics workflow engine},
  author={K{\"o}ster, Johannes and Rahmann, Sven},
  journal={Bioinformatics},
  volume={28},
  number={19},
  pages={2520--2522},
  year={2012},
  url = {https://doi.org/10.1093/bioinformatics/bts480}
}
@article{menzel2016fast,
  title={Fast and sensitive taxonomic classification for metagenomics with Kaiju},
  author={Menzel, Peter and Ng, Kim Lee and Krogh, Anders},
  journal={Nature Communications},
  volume={7},
  pages={11257},
  year={2016},
  url = {https://doi.org/10.1038/ncomms11257}
}
@article{roux2015virsorter,
  title={VirSorter: mining viral signal from microbial genomic data},
  author={Roux, Simon and Enault, Francois and Hurwitz, Bonnie L and Sullivan, Matthew B},
  journal={PeerJ},
  volume={3},
  pages={e985},
  year={2015},
  url = {https://doi.org/10.1093/bioinformatics/btu170}
}
@article{kim2016centrifuge,
  title={Centrifuge: rapid and sensitive classification of metagenomic sequences},
  author={Kim, Daehwan and Song, Li and Breitwieser, Florian P and Salzberg, Steven L},
  journal={Genome Research},
  volume={26},
  number={12},
  pages={1721--1729},
  year={2016},
  url = {https://doi.org/10.1101/gr.210641.116.}
}
@article{hyatt2010prodigal,
  title={Prodigal: prokaryotic gene recognition and translation initiation site identification},
  author={Hyatt, Doug and Chen, Gwo-Liang and LoCascio, Philip F and Land, Miriam L and Larimer, Frank W and Hauser, Loren J},
  journal={BMC Bioinformatics},
  volume={11},
  number={1},
  pages={119},
  year={2010},
  url = {https://doi.org/10.1186/1471-2105-11-119}
}
@article{edgar2004muscle,
  title={MUSCLE: multiple sequence alignment with high accuracy and high throughput},
  author={Edgar, Robert C},
  journal={Nucleic Acids Research},
  volume={32},
  number={5},
  pages={1792--1797},
  year={2004},
  url = {https://doi.org/10.1093/nar/gkh340}
}
@article{li2015megahit,
  title={MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph},
  author={Li, Dinghua and Liu, Chi-Man and Luo, Ruibang and Sadakane, Kunihiko and Lam, Tak-Wah},
  journal={Bioinformatics},
  volume={31},
  number={10},
  pages={1674--1676},
  year={2015},
  url = {https://doi.org/10.1093/bioinformatics/btv033}
}
@article{langmead2012fast,
  title={Fast gapped-read alignment with Bowtie 2},
  author={Langmead, Ben and Salzberg, Steven L},
  journal={Nature Methods},
  volume={9},
  number={4},
  pages={357},
  year={2012},
  url = {https://doi.org/10.1038/nmeth.1923}
}
@article{li2009fast,
  title={Fast and accurate short read alignment with Burrows--Wheeler transform},
  author={Li, Heng and Durbin, Richard},
  journal={Bioinformatics},
  volume={25},
  number={14},
  pages={1754--1760},
  year={2009},
  url = {https://doi.org/10.1093/bioinformatics/btp324}
}
@article{li2009sequence,
  title={The sequence alignment/map format and SAMtools},
  author={Li, Heng and Handsaker, Bob and Wysoker, Alec and Fennell, Tim and Ruan, Jue and Homer, Nils and Marth, Gabor and Abecasis, Goncalo and Durbin, Richard},
  journal={Bioinformatics},
  volume={25},
  number={16},
  pages={2078--2079},
  year={2009},
  url = {https://doi.org/10.1093/bioinformatics/btp352}
}

@article{breitwieser2018krakenuniq,
  title={KrakenUniq: confident and fast metagenomics classification using unique k-mer counts},
  author={Breitwieser, FP and Baker, DN and Salzberg, Steven L},
  journal={Genome Biology},
  volume={19},
  number={1},
  pages={1--10},
  year={2018},
  url = {https://doi.org/10.5281/zenodo.1412647}
}

@article{eren2013filtering,
  title={A filtering method to generate high quality short reads using Illumina paired-end technology},
  author={Eren, A Murat and Vineis, Joseph H and Morrison, Hilary G and Sogin, Mitchell L},
  journal={PLoS One},
  volume={8},
  number={6},
  year={2013},
  url = {https://doi.org/10.1371/journal.pone.0066643}
}

@article{alneberg2014binning,
  title={Binning metagenomic contigs by coverage and composition},
  author={Alneberg, Johannes and Bjarnason, Brynjar Sm{\'a}ri and De Bruijn, Ino and Schirmer, Melanie and Quick, Joshua and Ijaz, Umer Z and Lahti, Leo and Loman, Nicholas J and Andersson, Anders F and Quince, Christopher},
  journal={Nature Methods},
  volume={11},
  number={11},
  pages={1144--1146},
  year={2014},
  url = {https://doi.org/10.1038/nmeth.3103}
}
@article{sieber2018recovery,
  title={Recovery of genomes from metagenomes via a dereplication, aggregation and scoring strategy},
  author={Sieber, Christian MK and Probst, Alexander J and Sharrar, Allison and Thomas, Brian C and Hess, Matthias and Tringe, Susannah G and Banfield, Jillian F},
  journal={Nature Microbiology},
  volume={3},
  number={7},
  pages={836--843},
  year={2018},
}

@article{kang2019metabat,
  title={MetaBAT 2: an adaptive binning algorithm for robust and efficient genome reconstruction from metagenome assemblies},
  author={Kang, Dongwan D and Li, Feng and Kirton, Edward and Thomas, Ashleigh and Egan, Rob and An, Hong and Wang, Zhong},
  journal={PeerJ},
  volume={7},
  pages={e7359},
  year={2019},
}
@article{wu2016maxbin,
  title={MaxBin 2.0: an automated binning algorithm to recover genomes from multiple metagenomic datasets},
  author={Wu, Yu-Wei and Simmons, Blake A and Singer, Steven W},
  journal={Bioinformatics},
  volume={32},
  number={4},
  pages={605--607},
  year={2016},
}

@article{ondov2011krona,
  title={Interactive metagenomic visualization in a Web browser},
  author={Ondov, Brian D and Bergman, Nicholas H and Phillippy, Adam M},
  journal={BMC Bioinformatics},
  volume={12},
  number={1},
  pages={385},
  year={2011},
  url = {https://doi.org/10.1186/1471-2105-12-385}
}

@article{buchfink2015fast,
  title={Fast and sensitive protein alignment using DIAMOND},
  author={Buchfink, Benjamin and Xie, Chao and Huson, Daniel H},
  journal={Nature Methods},
  volume={12},
  number={1},
  pages={59--60},
  year={2015},
  url = {https://doi.org/10.1038/nmeth.3176}
}
@article{delmont2018nitrogen,
  title={Nitrogen-fixing populations of Planctomycetes and Proteobacteria are abundant in surface ocean metagenomes},
  author={Delmont, Tom O and Quince, Christopher and Shaiber, Alon and Esen, {\"O}zcan C and Lee, Sonny TM and Rapp{\'e}, Michael S and McLellan, Sandra L and L{\"u}cker, Sebastian and Eren, A Murat},
  journal={Nature Microbiology},
  volume={3},
  number={7},
  pages={804--813},
  year={2018},
  url = {https://doi.org/10.1038/s41564-018-0176-9}
}
</script>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p>© Copyright 2020 The HYPOCOLYPSE Project.</p>
<p>Site constructed using <a href="https://rstudio.github.io/distill/">Distill for R Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
