<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

  <!--radix_placeholder_meta_tags-->
  <title>: No 7. KEGG/KOFam Analysis</title>
  
  <meta property="description" itemprop="description" content="Diving deeper into funtional annotation. This is a work in progress and is not complete."/>
  
  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>
  <link rel="icon" type="image/vnd.microsoft.icon" href="assets/favicon.ico"/>
  
  
  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content=": No 7. KEGG/KOFam Analysis"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Diving deeper into funtional annotation. This is a work in progress and is not complete."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content=""/>
  
  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content=": No 7. KEGG/KOFam Analysis"/>
  <meta property="twitter:description" content="Diving deeper into funtional annotation. This is a work in progress and is not complete."/>
  
  <!--/radix_placeholder_meta_tags-->
  
  <!--radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["No 7. KEGG/KOFam Analysis"]},{"type":"character","attributes":{},"value":["Diving deeper into funtional annotation. This is a work in progress and is not complete.\n"]},{"type":"character","attributes":{},"value":["assets/cite.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>
  
  <script type="application/javascript">
  
    window.headroom_prevent_pin = false;
  
    window.document.addEventListener("DOMContentLoaded", function (event) {
  
      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');
  
      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });
  
      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });
  
      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>
  
  <style type="text/css">
  
  /* Theme (user-documented overrideables for nav appearance) */
  
  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #455a64;
    font-size: 15px;
    font-weight: 300;
  }
  
  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }
  
  .distill-site-nav a:hover {
    color: white;
  }
  
  @media print {
    .distill-site-nav {
      display: none;
    }
  }
  
  .distill-site-header {
  
  }
  
  .distill-site-footer {
  
  }
  
  
  /* Site Header */
  
  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }
  
  
  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }
  
  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }
  
  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }
  
  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }
  
  
  
  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }
  
  
  .distill-site-header .nav-toggle {
    display: none;
  }
  
  .nav-dropdown {
    display: inline-block;
    position: relative;
  }
  
  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }
  
  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }
  
  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .nav-dropdown-active {
    display: block;
  }
  
  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }
  
  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }
  
  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }
  
  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }
  
  /* Site Footer */
  
  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }
  
  /* Headroom */
  
  d-title {
    padding-top: 6rem;
  }
  
  @media print {
    d-title {
      padding-top: 4rem;
    }
  }
  
  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }
  
  .headroom--transition {
    transition: all .4s ease-in-out;
  }
  
  .headroom--unpinned {
    top: -100px;
  }
  
  .headroom--pinned {
    top: 0;
  }
  
  </style>
  
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>
  
  <script type="application/javascript">
  
  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }
  
  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }
  
  function createIndex() {
    var options = {
      keys: [
        "title",
        "categories",
        "description",
        "contents"
      ]
    };
    return new window.Fuse([],options);
  }
  
  function createFuseIndex() {
  
    // create fuse index
    var options = { keys: ["title", "description", "contents"] };
    var fuse = new window.Fuse([], options);
  
    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });
  
        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }
  
  window.document.addEventListener("DOMContentLoaded", function (event) {
  
    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;
  
    createFuseIndex()
      .then(function(fuse) {
  
        // make search box visible
        searchEl.classList.remove('hidden');
  
        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
              keys: [
                { name: 'title', weight: 20 },
                { name: 'categories', weight: 15 },
                { name: 'description', weight: 10 },
                { name: 'contents', weight: 5 },
              ],
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
              .filter(function(item) { return !!item.description; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description}
                  </div>
                  <div class="search-item-preview">
                    <img src="${suggestion.preview ? offsetURL(suggestion.preview) : ''}"</img>
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
      })
      .catch(function(error) {
        console.log(error);
      });
  });
  
  </script>
  
  <style type="text/css">
  
  /* Algolioa Autocomplete */
  
  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #455a64 solid;
    width: 180px;
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      visibility: hidden
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }
  
  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }
  
  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    margin-block-start: 0;
    margin-block-end: 5px;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }
  
  </style>
  
  
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->
  
  <style type="text/css">
  
  body {
    background-color: white;
  }
  
  .pandoc-table {
    width: 100%;
  }
  
  .pandoc-table>caption {
    margin-bottom: 10px;
  }
  
  .pandoc-table th:not([align]) {
    text-align: left;
  }
  
  .pagedtable-footer {
    font-size: 15px;
  }
  
  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }
  
  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }
  
  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }
  
  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }
  
  .html-widget {
    margin-bottom: 2.0em;
  }
  
  .l-screen-inset {
    padding-right: 16px;
  }
  
  .l-screen .caption {
    margin-left: 10px;
  }
  
  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .shaded-content {
    background: white;
  }
  
  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }
  
  .hidden {
    display: none !important;
  }
  
  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }
  
  d-appendix {
    padding-top: 30px;
  }
  
  d-article>p>img {
    width: 100%;
  }
  
  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }
  
  d-article h3 {
    margin-top: 1.5rem;
  }
  
  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }
  
  /* Tweak code blocks */
  
  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }
  
  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: hidden;
  }
  
  d-article div.sourceCode {
    background-color: white;
  }
  
  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }
  
  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  d-article pre a {
    border-bottom: none;
  }
  
  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }
  
  @media(min-width: 768px) {
  
  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }
  
  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }
  
  d-article pre {
    font-size: 14px;
  }
  
  }
  
  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  /* CSS for d-contents */
  
  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }
  
  @media(min-width: 1000px) {
    .d-contents {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }
  
  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }
  
  .d-contents li {
    list-style-type: none
  }
  
  .d-contents nav > ul {
    padding-left: 0;
  }
  
  .d-contents ul {
    padding-left: 1em
  }
  
  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }
  
  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }
  
  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }
  
  .d-contents nav > ul > li > a {
    font-weight: 600;
  }
  
  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }
  
  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }
  
  
  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }
  
  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }
  
  
  /* Figure */
  
  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }
  
  .figure img {
    width: 100%;
  }
  
  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }
  
  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }
  
  /* Citations */
  
  d-article .citation {
    color: inherit;
    cursor: inherit;
  }
  
  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }
  
  
  /* Tweak 1000px media break to show more text */
  
  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }
  
    .grid {
      grid-column-gap: 16px;
    }
  
    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }
  
  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }
  
    .grid {
      grid-column-gap: 32px;
    }
  }
  
  
  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */
  
  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }
  
  
  /* Anchor.js */
  
  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }
  
  /* Social footer */
  
  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }
  
  .disqus-comments {
    margin-right: 30px;
  }
  
  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
  
  #disqus_thread {
    margin-top: 30px;
  }
  
  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }
  
  .article-sharing a:hover {
    border-bottom: none;
  }
  
  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .subscribe p {
    margin-bottom: 0.5em;
  }
  
  
  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }
  
  
  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .custom p {
    margin-bottom: 0.5em;
  }
  
  
  /* Improve display for browsers without grid (IE/Edge <= 15) */
  
  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }
  
  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }
  
  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }
  
  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }
  
  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }
  
  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
  
  
  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }
  
  .downlevel .footnotes ol {
    padding-left: 13px;
  }
  
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }
  
  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }
  
  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }
  
  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  </style>
  
  <script type="application/javascript">
  
  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }
  
  // show body when load is complete
  function on_load_complete() {
  
    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }
  
  
    // set body to visible
    document.body.style.visibility = 'visible';
  
    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }
  
    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }
  
  function init_distill() {
  
    init_common();
  
    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);
  
    // create d-title
    $('.d-title').changeElementType('d-title');
  
    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);
  
    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();
  
    // move posts container into article
    $('.posts-container').appendTo($('d-article'));
  
    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');
  
    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;
  
    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();
  
    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));
  
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });
  
    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');
  
    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {
  
      // capture layout
      var layout = $(this).attr('data-layout');
  
      // apply layout to markdown level block elements
      var elements = $(this).children().not('div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });
  
  
      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });
  
    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();
  
    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a[aria-hidden="true"]').remove();
  
    // load distill framework
    load_distill_framework();
  
    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {
  
      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;
  
      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');
  
      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');
  
      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');
  
      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });
  
      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }
  
      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');
  
      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");
  
      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }
  
       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }
  
      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();
  
      // clear polling timer
      clearInterval(tid);
  
      // show body now that everything is ready
      on_load_complete();
    }
  
    var tid = setInterval(distill_post_process, 50);
    distill_post_process();
  
  }
  
  function init_downlevel() {
  
    init_common();
  
     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));
  
    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
  
    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();
  
    // remove toc
    $('.d-contents').remove();
  
    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });
  
  
    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);
  
    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);
  
    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();
  
    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();
  
    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));
  
    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });
  
    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));
  
    $('body').addClass('downlevel');
  
    on_load_complete();
  }
  
  
  function init_common() {
  
    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};
  
        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });
  
        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);
  
    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});
  
    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });
  
      }
    });
  
    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });
  
    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }
  
    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');
  
    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");
  
    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();
  
    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }
  
  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });
  
  </script>
  
  <!--/radix_placeholder_distill-->
  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Roboto:300|Nunito+Sans:300|Inconsolata|Source+Sans+Pro|Work+Sans:200,300,400,500');
  /********************************/
  /* controls general properties of site*/
  /********************************/
  
  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    background-color: #FEFEFE;
  }
  
  d-title h1{
  font-size: 2.1em;
  font-family: 'Work Sans';
  font-weight: 300;
  }
  
  d-article h1 {
    font-size: 1.8em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 300;
  }
  
  d-article h2 {
    font-size: 1.65em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 500;
  }
  
  d-article h3 {
    margin-top: 1em;
  }
  d-byline .authors-affiliations {
    grid-column-end: span 3;
    grid-column-gap: 8px;
  }
  
  a.name:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }
  
  a.affiliation:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }
  
  blockquote p {
  	font-size: 1.3em;
  	font-family: 'Lato';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }
  
  d-article li {
    margin: 0px 0px 4px;
  }
  
  /********************************/
  /* controls CODE*/
  /********************************/
  
  /** pre.sourceCode controls code block without lang **/
  
  /** Controls r code  **/
  
  pre.sourceCode.r {
    margin-top: 0px;
     color: #333333;
     background-color: #E8E8E8;
     border-radius: 3px;
     font-size: 0.80em;
     font-family: 'Monaco', 'PT Mono', monospace;
     overflow-x: scroll !important;
     border-left: 6px solid #006ddb;
     border-top: 1px solid #B7B7B7;
     border-right: 1px solid #B7B7B7;
     border-bottom: 1px solid #B7B7B7;
     padding: 18px 0px 18px 20px;
   }
  
   pre.sourceCode.bash {
     margin-top: 0px;
      color: #333333;
      background-color: #E8E8E8;
      border-radius: 3px;
      font-size: 0.80em;
      font-family: 'Monaco', 'PT Mono', monospace;
      overflow-x: scroll !important;
      border-left: 6px solid #D55E00;
      border-top: 1px solid #B7B7B7;
      border-right: 1px solid #B7B7B7;
      border-bottom: 1px solid #B7B7B7;
      padding: 18px 0px 18px 20px;
    }
  /** Controls code output **/
  
  d-article pre {
     margin-top: 0px;
      color: #333333;
      background-color: #FAFAFA;
      font-size: 0.8em;
      border: 1px solid #ccc;
      overflow-x: scroll !important;
      padding:  10px 10px 10px 18px;
      border-radius: 5px;
  
    }
  
  /** Controls inline code  **/
  
  p code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    margin-bottom: 20px;
    color: #BF2727;
  }
  
  d-article aside code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-family: 'Monaco', 'PT Mono', monospace;
    color: #BF2727;
  }
  
  /** OLD code b4 distill update
  d-code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    border-left: 5px solid #006ddb;
    border-top: 1px solid #B7B7B7;
    border-right: 1px solid #B7B7B7;
    border-bottom: 1px solid #B7B7B7;
  }
  
  code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.9em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }
  
  pre.text-output {
    margin-top: 0px;
     color: #333333;
     background-color: #FAFAFA;
     font-size: 0.8em
   }
  
  pre {
    border: 1px solid #ccc;
    padding: 0px 10px 10px 18px;
    margin: 1em 0px;
    border-radius: 5px;
  }
  
  **/
  
  /********************************/
  /* controls NAV BAR and HEADERS*/
  /********************************/
  
  .nav-dropdown-content .nav-dropdown-header {
    text-transform: none;
  }
  
  .distill-site-header .logo img {
    margin-top: 10px;
    max-height: 30px;
    margin-bottom: 10px;
  }
  
  .distill-site-nav {
      color: rgba(255, 255, 255, 0.8);
      background-color: #143D59;
      font-size: 16px;
      font-weight: 300;
    }
  }
  
  .article-banner {
    width: auto;
    height: auto;
    border: 0;
  }
  
  img[src="assets/banner.png"] {
    border: 0;
    padding: 0;
    width: 100%;
    margin: 100px 10px 0px 25px;
  }
  
  /*************************************************
  *  code for side by side krona plot
  **************************************************/
  
  .column {
    float: left;
    width: 48%;
    padding: 5px;
  }
  
  /* Clear floats after image containers */
  .row::after {
    content: "";
    clear: both;
    display: table;
  }
  
  div.krona {
    width: 100%;
  	height: 90%;
    background-color: white;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    margin-bottom: 25px;
  	margin-top: 5px;
  }
  
  div.container-krona {
    text-align: center;
    padding: 1px 10px;
  	color: #414141;
  	font-size: 1.48em;
  	line-height: 0.8em;
  	font-family: 'Inconsolata';
  }
  
  div.container-krona p {
    margin: 20px;
  }
  
  div.krona img {
    border:1px solid white;
    margin: 25px;
  }
  
  img {
    margin: 25px;
  }
  
  /*************************************************
  *  code for LIGHTBOX MARKUP
  **************************************************/
  
  .thumbnail {
    max-width: 100%;
    position: relative;
  }
  
  .lightbox {
  	/** Default lightbox to hidden */
  	display: none;
  
  	/** Position and style */
  	position: fixed;
  	z-index: 999;
  	width: 100%;
  	height: 100%;
  	text-align: center;
  	top: 0;
  	left: 0;
  	background: rgba(0,0,0,0.8);
  }
  
  .lightbox img {
  	/** Pad the lightbox image */
  	position: fixed;
  	top: 0;
  	left: 0;
  	right: 0;
  	bottom: 0;
  	max-width: 0%;
  	max-height: 0%;
  	margin: auto;
  	box-sizing: border-box;
  }
  
  .lightbox:target {
  	/** Remove default browser outline */
  	outline: none;
  
  	/** Unhide lightbox **/
  	display: block;
  }
  
  .lightbox:target img {
  	max-height: 100%;
  	max-width: 100%;
  }
  
  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"No 7. KEGG/KOFam Analysis","description":"Diving deeper into funtional annotation. This is a work in progress and is not complete.","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a class="logo" href="index.html">
<img src="assets/icon.png" alt="Logo"/>
</a>
<a href="index.html" class="title"></a>
</div>
<div class="nav-right">
<a href="index.html">
<i class="fa fa-home" aria-hidden="true"></i>
</a>
<a href="intro.html">Site Overview</a>
<a href="field.html">A. Field Analyses</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
B. 16S rRNA
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">16S rRNA Processing</span>
<a href="16s-dada2.html">1. DADA2</a>
<a href="16s-data-prep.html">2. Data Prep</a>
<hr/>
<span class="nav-dropdown-header">16S rRNA Analyses</span>
<a href="16s-water.html">3. Diversity</a>
<a href="16s-da-asvs.html">4. Differentially Abundant ASVs</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
C. Metagenomic
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Setup</span>
<a href="mg-setup.html">1. Working Envrionment</a>
<a href="mg-databases.html">2. Annotation Databases</a>
<hr/>
<span class="nav-dropdown-header">Read Processing</span>
<a href="mg-workflow-1.html">3. Assembly &amp; Annotations</a>
<a href="mg-workflow-2.html">4. Assembley &amp; Annotation Summary</a>
<hr/>
<span class="nav-dropdown-header">MAG Analysis</span>
<a href="mg-binning.html">5. Binning</a>
<a href="mag02-phylogenomics.html">6. MAG02 Phylogenomics</a>
<a href="mg-function.html">7. KEGG/KOFam Analysis</a>
</div>
</div>
<a href="data-availability.html">Data</a>
<a href="https://github.com/hypocolypse/web/">
<i class="fa fa-github fa-lg" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>No 7. KEGG/KOFam Analysis</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p><p>Diving deeper into funtional annotation. This is a work in progress and is not complete.</p></p>
</div>


<div class="d-article">
<div class="d-contents">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#step-1.-create-contigs-database">Step 1. Create contigs database</a></li>
<li><a href="#add-annotations-to-new-database">Add annotations to new database</a></li>
<li><a href="#profile-metagenomes">Profile metagenomes</a></li>
<li><a href="#adding-additional-items-info">Adding additional items info</a></li>
<li><a href="#genes-in-modules-only">Genes in modules only</a></li>
<li><a href="#source-code">Source Code</a></li>
</ul>
</nav>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The objective of this section is to visualize all of the <em>genes</em> that have KEGG-KOfam annotations and assess their distribution across metagenomes. Our goal is to create a contig.db that contains only genes annotated by KEGG-KOfams and a corresponding profile.db that contains coverage and detection of genes across each metagenome. We also will generate files that have other important information about each gene such as taxonomy, KOfam functions(s), and KEGG modules (if any).</p>
<h2 id="step-1.-create-contigs-database">Step 1. Create contigs database</h2>
<p>In order to create a basic contigs.db of genes, we need a) a <strong>fasta file</strong> and b) an <a href="https://github.com/merenlab/anvio/blob/master/anvio/tests/sandbox/example_external_gene_calls.txt"><strong>external gene calls file</strong></a>, both of which <em>only</em> contain KOfam annotated genes. The final command will look something like this:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1"></a><span class="ex">anvi-gen-contigs-database</span> --contigs-fasta kofam_genes_rn.fa \</span>
<span id="cb1-2"><a href="#cb1-2"></a>                          --output-db-path KOFAMS-contigs.db  \</span>
<span id="cb1-3"><a href="#cb1-3"></a>                          --external-gene-calls kofam_external_gene_calls.txt \</span></code></pre></div>
<h3 id="kofam-gene-annotation">KOfam gene annotation</h3>
<p>First we run the annotation on the full contig.db</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1"></a><span class="ex">anvi-run-kegg-kofams</span> -c 03_CONTIGS/WATER-contigs.db \</span>
<span id="cb2-2"><a href="#cb2-2"></a>                     --kegg-data-dir /PATH/to/KEGG/db \</span>
<span id="cb2-3"><a href="#cb2-3"></a>                     -T <span class="va">$NSLOTS</span></span></code></pre></div>
<p>And here is the final portion of the output from the command.</p>
<pre><code>Number of raw hits ...........................: 631,174
Number of weak hits removed ..................: 619,356
Number of hits in annotation dict  ...........: 11,818
Gene functions ...............................: 11818 function calls from 1 sources for 11588 unique gene calls has been added to the contigs database.
Gene functions ...............................: 3240 function calls from 1 sources for 3198 unique gene calls has been added to the contigs database.
Gene functions ...............................: 3240 function calls from 1 sources for 3198 unique gene calls has been added to the contigs database.</code></pre>
<p>As you can see, 11,818 KOfam function calls were added from 11,588 unique gene calls, meaning a small percentage of genes had multiple KOfam hits. This is unavoidable and we will need to deal with that at some point, but not now. If you run something like <code>anvi-export-gene-calls -c contigs.db</code>, we find out the assembly has 65,102 gene calls, which tells us that ~18% of the genes have KOfam annotations. What the output also tells us is that 3240 KEGG Module calls were also added to the data base.</p>
<p>Next, we export all the KOfam function calls. To keep things organized, we can keep this entire analysis in a separate directory (<code>METABOLISM</code>) and store the files we need to build and annotate in the subdirectory <code>00_HELPER_FILES</code>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1"></a><span class="fu">mkdir</span> -p METABOLISM/00_HELPER_FILES/</span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="ex">anvi-export-functions</span> -c 03_CONTIGS/WATER-contigs.db \</span>
<span id="cb4-3"><a href="#cb4-3"></a>                      --annotation-sources KOfam \</span>
<span id="cb4-4"><a href="#cb4-4"></a>                      -o METABOLISM/00_HELPER_FILES/KOfam.txt</span></code></pre></div>
<p>Again, this file has 11,818 hits from 11,588 <em>unique</em> gene calls. The file is tab-delimited and contains the KOfam <code>accession</code> number, <code>function</code>, and <code>e_value</code> for each <code>gene_callers_id</code>. We need a list of <em>unique</em> gene IDs to build a contig database so for now we select the hit with the lowest <code>e_value</code>. It really doesn’t matter at this point but just in case we decide latter to only present a single function call per gene, this approach seems to make the most sense.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>kofam &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/00_HELPER_FILES/KOfam.txt&quot;</span>, </span>
<span id="cb5-2"><a href="#cb5-2"></a>                    <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>,  <span class="dt">quote =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb5-3"><a href="#cb5-3"></a>kofam &lt;-<span class="st"> </span>kofam <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">ko_function =</span> function.)</span>
<span id="cb5-4"><a href="#cb5-4"></a>kofam<span class="op">$</span>ko_function &lt;-<span class="st"> </span>stringr<span class="op">::</span><span class="kw">str_replace</span>(kofam<span class="op">$</span>ko_function, <span class="st">&quot;&#39;&quot;</span>, <span class="st">&quot;&quot;</span>)</span>
<span id="cb5-5"><a href="#cb5-5"></a>kofam<span class="op">$</span>gene_callers_id &lt;-<span class="st"> </span><span class="kw">as.factor</span>(kofam<span class="op">$</span>gene_callers_id)</span>
<span id="cb5-6"><a href="#cb5-6"></a>kofam &lt;-<span class="st"> </span>kofam[<span class="kw">order</span>(kofam<span class="op">$</span>gene_callers_id, kofam<span class="op">$</span>e_value), ]</span>
<span id="cb5-7"><a href="#cb5-7"></a>kofam_unique &lt;-<span class="st"> </span>kofam[<span class="op">!</span><span class="kw">duplicated</span>(kofam<span class="op">$</span>gene_callers_id), ]</span>
<span id="cb5-8"><a href="#cb5-8"></a><span class="kw">write.table</span>(kofam_unique, <span class="st">&quot;metabolism/00_HELPER_FILES/KOfam_unique.txt&quot;</span>, </span>
<span id="cb5-9"><a href="#cb5-9"></a>            <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</span>
<span id="cb5-10"><a href="#cb5-10"></a><span class="kw">head</span>(kofam_unique)</span></code></pre></div>
</div>
<p>Same file but with duplicate gene IDs removed.</p>
<h3 id="get-gene-sequences-for-kofam-hits">Get gene sequences for KOfam hits</h3>
<p>Next, we create a file containing a list of the just the unique <code>gene_callers_id</code>s so that we can parse out the KOfam annotated gene sequence.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>kofam_genes &lt;-<span class="st"> </span>kofam_unique</span>
<span id="cb6-2"><a href="#cb6-2"></a>kofam_genes[, <span class="kw">c</span>(<span class="st">&#39;accession&#39;</span>, <span class="st">&#39;ko_function&#39;</span>, <span class="st">&#39;e_value&#39;</span>, <span class="st">&#39;source&#39;</span>)] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="ot">NULL</span>)</span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="kw">write.table</span>(kofam_genes, <span class="st">&quot;metabolism/00_HELPER_FILES/kofam_genes_list.txt&quot;</span>, </span>
<span id="cb6-4"><a href="#cb6-4"></a>            <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>, <span class="dt">col.names =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
</div>
<p>We can now use this file to get the sequences for each gene caller ID that has a KOfam hit from the original contigs.db.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1"></a><span class="ex">anvi-get-sequences-for-gene-calls</span> \</span>
<span id="cb7-2"><a href="#cb7-2"></a>                        -c 03_CONTIGS/WATER-contigs.db \</span>
<span id="cb7-3"><a href="#cb7-3"></a>                        --gene-caller-ids kofam_genes_list.txt \</span>
<span id="cb7-4"><a href="#cb7-4"></a>                        --wrap 0 \</span>
<span id="cb7-5"><a href="#cb7-5"></a>                        -o METABOLISM/00_HELPER_FILES/kofam_genes.fa</span></code></pre></div>
<h3 id="generate-external-gene-calls-file">Generate external gene calls file</h3>
<p>We need to tweak the deflines of the fasta file before building the new database but we will do that in a minute. For now assume the fasta file is fine as is. First, we create the <code>--external-gene-calls</code> file. This file allows us to skip many of the steps anvi’o normally performs when building a database, like contig splitting and gene calling. Remember, we already did this with the assembly, which is where these genes came from in the first place. Our goal is to create a database of gene calls containing information as close to the original database as possible so that the two databases are comparable.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1"></a><span class="ex">anvi-export-gene-calls</span> -c 03_CONTIGS/WATER-contigs.db \</span>
<span id="cb8-2"><a href="#cb8-2"></a>                       --gene-caller prodigal \</span>
<span id="cb8-3"><a href="#cb8-3"></a>                       -o METABOLISM/00_HELPER_FILES/kofam_gene_calls.txt</span></code></pre></div>
<p>This command gives us a file containing all 65,102 gene calls. We can use the list of unique gene calls created above (<code>kofam_genes_list.txt</code>) to pull out the relevant IDs.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>all_genes &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/00_HELPER_FILES/kofam_gene_calls.txt&quot;</span>, </span>
<span id="cb9-2"><a href="#cb9-2"></a>                        <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb9-3"><a href="#cb9-3"></a>all_genes &lt;-<span class="st"> </span>all_genes <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">gene_callers_id =</span> <span class="kw">as.character</span>(gene_callers_id))</span>
<span id="cb9-4"><a href="#cb9-4"></a>kofam_genes_tab &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(kofam_genes, all_genes, </span>
<span id="cb9-5"><a href="#cb9-5"></a>                                     <span class="dt">by =</span> <span class="st">&quot;gene_callers_id&quot;</span>)</span>
<span id="cb9-6"><a href="#cb9-6"></a><span class="kw">head</span>(kofam_genes_tab)</span></code></pre></div>
</div>
<p>We need to modify this file a bit to make it compatible with what anvi’o needs to build the database. You see, anvi’o is expecting <em>contigs</em> not genes. When we give anvi’o an external gene calls file for <code>anvi-gen-contigs-database</code> we are basically saying that we performed our own gene calling on the contigs in the fasta file. Anvi’o will look at the <code>contig</code> ID in the external gene calls file and compare those IDs to the fasta deflines. Problem is that the deflines are named by genes, not contigs. We could just make up new names but then we lose the original contig information and gene IDs for each call. So instead, we add the original gene caller id to the original contig name. This not only preserves the contig information but also avoids any duplication of contig IDs, since many genes are found on the same contig.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>kofam_genes_tab &lt;-<span class="st"> </span>kofam_genes_tab <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">unite</span>(<span class="st">&quot;contig&quot;</span>, contig, gene_callers_id, </span>
<span id="cb10-2"><a href="#cb10-2"></a>                                              <span class="dt">remove =</span> <span class="ot">FALSE</span>, <span class="dt">sep =</span> <span class="st">&quot;_gene_&quot;</span>) </span>
<span id="cb10-3"><a href="#cb10-3"></a>kofam_genes_tab &lt;-<span class="st"> </span>kofam_genes_tab[, <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">7</span>,<span class="dv">8</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">3</span>,<span class="dv">6</span>,<span class="dv">9</span>,<span class="dv">10</span>)]</span>
<span id="cb10-4"><a href="#cb10-4"></a><span class="kw">head</span>(kofam_genes_tab)</span></code></pre></div>
</div>
<p>Better. We still have the original contig IDs, now appended with the gene ID. If we just give anvi’o this file, it will look at gene caller id <code>9</code>, and see that this gene came from position 443–896 on contig <code>WATER_000000000004_gene_9</code>. The problem is that when anvi’o looks in the fasta file at the corresponding sequence, it will not find a sequence greater than or equal to 896 bps. Instead, it will find a sequence that is 453 bps long, since this is the length of the <em>gene</em> and not the <em>contig</em>. Make sense? So we need to trick anvi’o a bit by making it think each “contig” just happens to be the exact length of a given gene. To do this, we need to adjust the start and stop values. If you don’t do this you will get an error saying something like <code>IndexError: list assignment index out of range</code>. This is because of the length discrepency just described. To circumvent this problem, we can set all start values to <code>0</code> and adjust the stop values to the full gene length.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>kofam_genes_tab &lt;-<span class="st"> </span>kofam_genes_tab <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">stop =</span> (stop <span class="op">-</span><span class="st"> </span>start), <span class="dt">.after =</span> <span class="dv">4</span>) </span>
<span id="cb11-2"><a href="#cb11-2"></a>kofam_genes_tab<span class="op">$</span>start &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb11-3"><a href="#cb11-3"></a><span class="kw">head</span>(kofam_genes_tab)</span>
<span id="cb11-4"><a href="#cb11-4"></a><span class="kw">write.table</span>(kofam_genes_tab, <span class="st">&quot;metabolism/00_HELPER_FILES/kofam_external_gene_calls.txt&quot;</span>, </span>
<span id="cb11-5"><a href="#cb11-5"></a>            <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
</div>
<p>Back to gene <code>9</code>, which is still 453 bp long, but now starts at the beginning of the sequence.</p>
<blockquote>
<p>This is the new <code>external-gene-calls</code> file.</p>
</blockquote>
<h3 id="fix-fasta-deflines">Fix fasta deflines</h3>
<p>If you take a peak at the <code>kofam_genes.fa</code> gene fasta file we generated above, you will see the deflines are named after the genes, not contigs. Again, avni’o is expecting contig IDs. Time to play another trick. We will take the new contig IDs, appended with the gene IDs, and use those in our fasta file instead of the gene IDs. Things get a bit weird here since we are trying to substitute the new <em>contig</em> name for the original gene caller id in the fasta file.</p>
<p>We start by converting the fasta file to a data frame and then ordering the data frame by the sequence name so they are in the same order as the contig names in the <code>external-gene-calls file</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>kofam_fna2 &lt;-<span class="st"> </span><span class="kw">readDNAStringSet</span>(<span class="st">&quot;metabolism/00_HELPER_FILES/kofam_genes.fa&quot;</span>)</span>
<span id="cb12-2"><a href="#cb12-2"></a>kofam_names &lt;-<span class="st">  </span><span class="kw">names</span>(kofam_fna2)</span>
<span id="cb12-3"><a href="#cb12-3"></a>kofam_seqs &lt;-<span class="st">  </span><span class="kw">paste</span>(kofam_fna2)</span>
<span id="cb12-4"><a href="#cb12-4"></a>fasta_df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(kofam_names, kofam_seqs)</span>
<span id="cb12-5"><a href="#cb12-5"></a>fasta_df<span class="op">$</span>kofam_names &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">as.character</span>(fasta_df<span class="op">$</span>kofam_names))</span>
<span id="cb12-6"><a href="#cb12-6"></a>fasta_df &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(fasta_df, kofam_names)</span></code></pre></div>
</div>
<p>Next, add the contigs column from the <code>external-gene-calls file</code> to the data frame and make sure the names match.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>fasta_df &lt;-<span class="st"> </span>fasta_df <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">new_id =</span> kofam_genes_tab<span class="op">$</span>contig)</span>
<span id="cb13-2"><a href="#cb13-2"></a>fasta_df &lt;-<span class="st"> </span>fasta_df[, <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>)]</span>
<span id="cb13-3"><a href="#cb13-3"></a>fasta_df<span class="op">$</span>kofam_names &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb13-4"><a href="#cb13-4"></a>fasta_df<span class="op">$</span>new_id &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;&gt;&quot;</span>, fasta_df<span class="op">$</span>new_id)</span>
<span id="cb13-5"><a href="#cb13-5"></a><span class="kw">write.table</span>(fasta_df, <span class="st">&quot;metabolism/00_HELPER_FILES/kofam_genes_rn.fa&quot;</span>,</span>
<span id="cb13-6"><a href="#cb13-6"></a>            <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\r</span><span class="st">&quot;</span>, <span class="dt">col.names =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>,</span>
<span id="cb13-7"><a href="#cb13-7"></a>            <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">fileEncoding =</span> <span class="st">&quot;UTF-8&quot;</span>)</span>
<span id="cb13-8"><a href="#cb13-8"></a><span class="kw">head</span>(fasta_df)</span></code></pre></div>
</div>
<blockquote>
<p>This is the new <code>contigs-fasta</code> file.</p>
</blockquote>
<p>And that’s it. It is <strong>a really good idea to double check the two files</strong> to make certain everything looks OK.</p>
<h3 id="make-the-contig.db-of-kofam-genes">Make the contig.db of KOfam genes</h3>
<p>And finally we can create the database. We want anvi’o to do as little as possible so we use some additional flags to prevent any unnecessary processing.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1"></a><span class="ex">anvi-gen-contigs-database</span> --contigs-fasta 00_HELPER_FILES/kofam_genes_rn.fa \</span>
<span id="cb14-2"><a href="#cb14-2"></a>                          --project-name KOFAMS \</span>
<span id="cb14-3"><a href="#cb14-3"></a>                          --output-db-path 01_CONTIGS/KOFAMS-contigs.db  \</span>
<span id="cb14-4"><a href="#cb14-4"></a>                          --external-gene-calls 00_HELPER_FILES/kofam_external_gene_calls.txt \</span>
<span id="cb14-5"><a href="#cb14-5"></a>                          --ignore-internal-stop-codons \</span>
<span id="cb14-6"><a href="#cb14-6"></a>                          --skip-predict-frame</span></code></pre></div>
<p>And here is the final output of the command…</p>
<pre><code>Input FASTA file .............................: kofam_genes_rn.fa
Name .........................................: KOFAMS
Description ..................................: No description is given
External gene calls ..........................: 11588 gene calls recovered and will be processed.

WARNING
===============================================
Anvi&#39;o found amino acid sequences in your external gene calls file that match to
11588 of 11588 gene in it and will use these amino acid sequences for
everything.


EXTERNAL GENE CALLS PARSER REPORT
===============================================
Num gene calls in file .......................: 11,588
Non-coding gene calls ........................: 0
Partial gene calls ...........................: 5,748
Num amino acid sequences provided ............: 11,588
  - For complete gene calls ..................: 5,840
  - For partial gene calls ...................: 5,748
Frames predicted .............................: 0
  - For complete gene calls ..................: 0
  - For partial gene calls ...................: 0
Gene calls marked as NONCODING ...............: 0
  - For complete gene calls ..................: 0
  - For partial gene calls ...................: 0
Gene calls with internal stops ...............: 0
  - For complete gene calls ..................: 0
  - For partial gene calls ...................: 0


CONTIGS DB CREATE REPORT
===============================================
Split Length .................................: 20,000
K-mer size ...................................: 4
Skip gene calling? ...........................: False
External gene calls provided? ................: True
External gene calls file have AA sequences? ..: True
Proper frames will be predicted? .............: False
Ignoring internal stop codons? ...............: True
Splitting pays attention to gene calls? ......: True
Contigs with at least one gene call ..........: 11588 of 11588 (100.0%)
Contigs database .............................: A new database, KOFAMS-contigs.db, has been created.
Number of contigs ............................: 11,588
Number of splits .............................: 11,588
Total number of nucleotides ..................: 9,249,465
Gene calling step skipped ....................: False
Splits broke genes (non-mindful mode) ........: False
Desired split length (what the user wanted) ..: 20,000
Average split length (what anvi&#39;o gave back) .: (Anvi&#39;o did not create any splits)</code></pre>
<h2 id="add-annotations-to-new-database">Add annotations to new database</h2>
<p>We could rerun the KOfam annotation, but since we already did that to get these genes in the first place, there really is no need. If you do not know what annotations are in your original contigs database, run the command like this.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1"></a><span class="ex">anvi-export-functions</span> -c 03_CONTIGS/WATER-contigs.db \</span>
<span id="cb16-2"><a href="#cb16-2"></a>                     --list-annotation-sources</span></code></pre></div>
<pre><code>FUNCTIONAL ANNOTATION SOURCES FOUND
===============================================
* COG_CATEGORY
* COG_FUNCTION
* Pfam
* KOfam
* Transfer_RNAs
* KEGG_Module</code></pre>
<p>OK, so we have <code>KOfam</code> and <code>KEGG_Module</code> annotations.</p>
<p>Remember, to get annotations you need to run this:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1"></a><span class="ex">anvi-export-functions</span> --contigs-db 03_CONTIGS/WATER-contigs.db \</span>
<span id="cb18-2"><a href="#cb18-2"></a>                      --annotation-sources KOfam \</span>
<span id="cb18-3"><a href="#cb18-3"></a>                      --output-file METABOLISM/00_HELPER_FILES/KOfam.txt</span></code></pre></div>
<p>And then import the annotation into the new database.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1"></a><span class="ex">anvi-import-functions</span> --contigs-db 01_CONTIGS/KOFAMS-contigs.db \</span>
<span id="cb19-2"><a href="#cb19-2"></a>                      --input-files 00_HELPER_FILES/KOfam.txt</span></code></pre></div>
<pre><code>Gene functions ...............................: 11818 function calls from 1 sources for 11588 unique gene calls has been added to the contigs database.</code></pre>
<p>We can also import the <code>KEGG_Module</code> annotation. First, export the information from the original contigs.db.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1"></a><span class="ex">anvi-export-functions</span> --contigs-db 03_CONTIGS/WATER-contigs.db \</span>
<span id="cb21-2"><a href="#cb21-2"></a>                      --annotation-sources KEGG_Module \</span>
<span id="cb21-3"><a href="#cb21-3"></a>                      -output-file METABOLISM/00_HELPER_FILES/KEGG_Module.txt</span></code></pre></div>
<p>There are two things we need to change in the <code>KEGG_Module.txt</code> file before importing the functions into the new contigs.db. The first is related to anvi’o itself, specifically the output file from the command above contains the value of <code>NONE</code> for every entry in the <code>e_value</code> column. When we try to import the function calls into our new database, anvi’o will throw an error because it needs an actual e value. We can just set the value to <code>0</code> to make anvi’o happy. The other problem is that when we read the file into R, the column called <code>function</code> is changed to <code>function.</code> with a period since <code>function</code> has a special meaning in R. So we need to change it back to the original column name. In order to import gene functions, <a href="http://merenlab.org/2016/06/18/importing-functions/#simple-matrix">anvi’o <em>needs</em> very specific column headers</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a>kegg_mod &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/00_HELPER_FILES/KEGG_Module.txt&quot;</span>, </span>
<span id="cb22-2"><a href="#cb22-2"></a>                       <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb22-3"><a href="#cb22-3"></a>kegg_mod<span class="op">$</span>e_value &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb22-4"><a href="#cb22-4"></a><span class="kw">colnames</span>(kegg_mod) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;gene_callers_id&quot;</span>, <span class="st">&quot;source&quot;</span>, </span>
<span id="cb22-5"><a href="#cb22-5"></a>                        <span class="st">&quot;accession&quot;</span>, <span class="st">&quot;function&quot;</span>, <span class="st">&quot;e_value&quot;</span>)</span>
<span id="cb22-6"><a href="#cb22-6"></a><span class="kw">write.table</span>(kegg_mod, <span class="st">&quot;metabolism/00_HELPER_FILES/KEGG_Module_mod.txt&quot;</span>, </span>
<span id="cb22-7"><a href="#cb22-7"></a>            <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
</div>
<p>And now we import the modified annotation file. To run this you need to be in the <code>METABOLISM</code> directory.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1"></a><span class="ex">anvi-import-functions</span> --contigs-db 01_CONTIGS/KOFAMS-contigs.db \</span>
<span id="cb23-2"><a href="#cb23-2"></a>                      --input-files 00_HELPER_FILES/KEGG_Module_mod.txt</span></code></pre></div>
<pre><code>Gene functions ...............................: 3240 function calls from 1 sources for 3198 unique gene calls has been added to the contigs database.</code></pre>
<h2 id="profile-metagenomes">Profile metagenomes</h2>
<p>Next, we profile each metagenome against the new gene database. Profiling quantifies things like coverage, detection, as well as single-nucleotide, single-codon, and single-amino acid for each metagenome. The default input for profiling is a BAM file, so let’s map short reads to the database. variants, etc.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="#cb25-1"></a><span class="fu">mkdir</span> 02_MAPPING_TO_KOFAMS</span>
<span id="cb25-2"><a href="#cb25-2"></a><span class="ex">bowtie2-build</span> kofam_genes_rn.fa 02_MAPPING_TO_KOFAMS/KOFAMS</span>
<span id="cb25-3"><a href="#cb25-3"></a></span>
<span id="cb25-4"><a href="#cb25-4"></a><span class="kw">for</span> <span class="ex">sample</span> in <span class="kw">`</span><span class="fu">cat</span> samples_WATER.txt<span class="kw">`</span></span>
<span id="cb25-5"><a href="#cb25-5"></a><span class="kw">do</span></span>
<span id="cb25-6"><a href="#cb25-6"></a>    <span class="ex">bowtie2</span>  -x 02_MAPPING_TO_KOFAMS/KOFAMS \</span>
<span id="cb25-7"><a href="#cb25-7"></a>             -1 00_QC/<span class="va">$sample</span>-QUALITY_PASSED_R1.fastq.gz \</span>
<span id="cb25-8"><a href="#cb25-8"></a>             -2 00_QC/<span class="va">$sample</span>-QUALITY_PASSED_R2.fastq.gz \</span>
<span id="cb25-9"><a href="#cb25-9"></a>             -S 02_MAPPING_TO_KOFAMS/<span class="va">$sample</span>-in-KOFAMS.sam</span>
<span id="cb25-10"><a href="#cb25-10"></a>             <span class="ex">--no-unal</span> --threads <span class="va">$NSLOTS</span> </span>
<span id="cb25-11"><a href="#cb25-11"></a>             </span>
<span id="cb25-12"><a href="#cb25-12"></a>    <span class="ex">samtools</span> view -F 4 -bS 02_MAPPING_TO_KOFAMS/<span class="va">$sample</span>-in-KOFAMS.sam <span class="op">&gt;</span> \</span>
<span id="cb25-13"><a href="#cb25-13"></a>                           02_MAPPING_TO_KOFAMS/<span class="va">$sample</span>-in-KOFAMS-RAW.bam</span>
<span id="cb25-14"><a href="#cb25-14"></a></span>
<span id="cb25-15"><a href="#cb25-15"></a>    <span class="co"># sort and index the BAM file:</span></span>
<span id="cb25-16"><a href="#cb25-16"></a>    <span class="ex">samtools</span> sort 02_MAPPING_TO_KOFAMS/<span class="va">$sample</span>-in-KOFAMS-RAW.bam \</span>
<span id="cb25-17"><a href="#cb25-17"></a>               -o 02_MAPPING_TO_KOFAMS/<span class="va">$sample</span>-in-KOFAMS.bam</span>
<span id="cb25-18"><a href="#cb25-18"></a>    <span class="ex">samtools</span> index 02_MAPPING_TO_KOFAMS/<span class="va">$sample</span>-in-KOFAMS.bam</span>
<span id="cb25-19"><a href="#cb25-19"></a></span>
<span id="cb25-20"><a href="#cb25-20"></a>    <span class="co"># remove temporary files:</span></span>
<span id="cb25-21"><a href="#cb25-21"></a>    <span class="fu">rm</span> 02_MAPPING_TO_KOFAMS/<span class="va">$sample</span>-in-KOFAMS.sam \</span>
<span id="cb25-22"><a href="#cb25-22"></a>       02_MAPPING_TO_KOFAMS/<span class="va">$sample</span>-in-KOFAMS-RAW.bam</span>
<span id="cb25-23"><a href="#cb25-23"></a><span class="kw">done</span></span></code></pre></div>
<p>Then we profile each metagenome.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb26-1"><a href="#cb26-1"></a><span class="fu">mkdir</span> 03_KOFAMS_PROFILES</span>
<span id="cb26-2"><a href="#cb26-2"></a><span class="kw">for</span> <span class="ex">sample</span> in <span class="kw">`</span><span class="fu">cat</span> samples_WATER.txt<span class="kw">`</span></span>
<span id="cb26-3"><a href="#cb26-3"></a><span class="kw">do</span></span>
<span id="cb26-4"><a href="#cb26-4"></a>    <span class="ex">anvi-profile</span> --contigs-db KOFAMS-contigs.db \</span>
<span id="cb26-5"><a href="#cb26-5"></a>                 --input-file 02_MAPPING_TO_KOFAMS/<span class="va">$sample</span>-in-KOFAMS.bam \</span>
<span id="cb26-6"><a href="#cb26-6"></a>                 --profile-SCVs \</span>
<span id="cb26-7"><a href="#cb26-7"></a>                 --num-threads 1 \</span>
<span id="cb26-8"><a href="#cb26-8"></a>                 --min-contig-length 10 \</span>
<span id="cb26-9"><a href="#cb26-9"></a>                 --output-dir 03_KOFAMS_PROFILES/<span class="va">$sample</span>-in-KOFAMS</span>
<span id="cb26-10"><a href="#cb26-10"></a><span class="kw">done</span></span></code></pre></div>
<p>And finally merge all of the profiles so that we can make comparisons across metagenomes.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="#cb27-1"></a><span class="ex">anvi-merge</span> 03_KOFAMS_PROFILES/*-in-KOFAMS/PROFILE.db \</span>
<span id="cb27-2"><a href="#cb27-2"></a>           --contigs-db KOFAMS-contigs.db \</span>
<span id="cb27-3"><a href="#cb27-3"></a>           --output-dir 04_KOFAM_PROFILES-MERGED/ </span></code></pre></div>
<p>All set. We have a contigs.db of all KOfam annotated genes and a profile.db. We could just run a visualiztion on the data we have so far, but what we want to do instead in to create some additional files to enhance the visualiztion.</p>
<h2 id="adding-additional-items-info">Adding additional items info</h2>
<p>Most of the time you use the interactive interface you are looking at two types of data frames—<strong>items</strong>, which in this case are the genes and <strong>layers</strong>, which in this case are the samples. <a href="http://merenlab.org/2017/12/11/additional-data-tables/">Click here for more details</a>. We can add basically any type of data we wish for either of these data frames. We will do this piecemeal and most of what we add is already somewhere in our various databases, just not yet in a format that is accessible by the interactive.</p>
<p>So let’s start by generating additional info for the <em>items</em>, that is the genes.</p>
<h3 id="adding-gene-classifications">Adding gene classifications</h3>
<p>To add taxonomic info for each gene call we need to two tables that can be generated from the original contigs database.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="#cb28-1"></a><span class="ex">anvi-export-table</span> 03_CONTIGS/WATER-contigs.db \</span>
<span id="cb28-2"><a href="#cb28-2"></a>                  --table genes_taxonomy \</span>
<span id="cb28-3"><a href="#cb28-3"></a>                  --output-file METABOLISM/00_HELPER_FILES/genes_taxonomy.txt</span>
<span id="cb28-4"><a href="#cb28-4"></a><span class="ex">anvi-export-table</span> 03_CONTIGS/WATER-contigs.db \</span>
<span id="cb28-5"><a href="#cb28-5"></a>                  --table taxon_names \</span>
<span id="cb28-6"><a href="#cb28-6"></a>                  --output-file METABOLISM/00_HELPER_FILES/taxon_names.txt</span></code></pre></div>
<p>The first file (<code>genes_taxonomy.txt</code>) provides a taxon ids for all genes in the database. Again, we could rerun the classification on just the KOfam annotated genes but why bother. The second file (<code>taxon_names.txt</code>) is a lookup table for each taxon id. What we need to do is join these two table so that each gene id is coupled with a taxon id and a lineage. Then merge these data with of list of mock contig names in the in the KOfam database.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a>tax_names &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/00_HELPER_FILES/taxon_names.txt&quot;</span>, </span>
<span id="cb29-2"><a href="#cb29-2"></a>                        <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">quote =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb29-3"><a href="#cb29-3"></a>gene_tax &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/00_HELPER_FILES/genes_taxonomy.txt&quot;</span>, </span>
<span id="cb29-4"><a href="#cb29-4"></a>                       <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">quote =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb29-5"><a href="#cb29-5"></a>gene_tax_names &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(gene_tax, tax_names, <span class="dt">by =</span> <span class="st">&quot;taxon_id&quot;</span>)</span>
<span id="cb29-6"><a href="#cb29-6"></a>kofam_genes_n &lt;-<span class="st"> </span>kofam_genes</span>
<span id="cb29-7"><a href="#cb29-7"></a>kofam_genes_n<span class="op">$</span>gene_callers_id &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">as.character</span>(kofam_genes_n<span class="op">$</span>gene_callers_id))</span>
<span id="cb29-8"><a href="#cb29-8"></a>kofam_with_tax &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(kofam_genes_n, gene_tax_names, </span>
<span id="cb29-9"><a href="#cb29-9"></a>                                   <span class="dt">by =</span> <span class="st">&quot;gene_callers_id&quot;</span>)</span>
<span id="cb29-10"><a href="#cb29-10"></a><span class="kw">head</span>(kofam_with_tax)</span>
<span id="cb29-11"><a href="#cb29-11"></a><span class="kw">write.table</span>(kofam_with_tax, <span class="st">&quot;metabolism/00_HELPER_FILES/kofam_with_tax.txt&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, </span>
<span id="cb29-12"><a href="#cb29-12"></a>            <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
</div>
<p>As you can see, we now have each gene associated with a taxonmic lineage (when one was identified). We could remove genes with <code>NA</code> from the taxonomy file but anvi’o does not seem to have a problem with it so let’s leave it be. Now we can import the taxonomy files into the new contigs database.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb30-1"><a href="#cb30-1"></a><span class="ex">anvi-import-taxonomy-for-genes</span> --contigs-db 01_CONTIGS/KOFAMS-contigs.db \</span>
<span id="cb30-2"><a href="#cb30-2"></a>                               --input-files 00_HELPER_FILES/kofam_with_tax.txt \</span>
<span id="cb30-3"><a href="#cb30-3"></a>                               --parser default_matrix</span></code></pre></div>
<p>Here is the output.</p>
<pre><code>Total num hits found .........................: 11,588
Num gene caller ids in the db ................: 11,588
Num gene caller ids in the incoming data .....: 11,588
Taxon names table ............................: Updated with 1467 unique taxon names
Genes taxonomy table .........................: Taxonomy stored for 11588 gene calls
Splits taxonomy ..............................: Input data from &quot;default_matrix&quot; annotated 11588 of 11588 splits (100.0%) with taxonomy.</code></pre>
<h3 id="adding-functional-annotations">Adding functional annotations</h3>
<p>Even though we have this information, adding to the interactive is tricky because some genes have multiple annotations. Our main purpose for the visualiztion is exploratory analysis so we do not need to capture everything, but this is still a concern, especially when we are trying to generate the files we need. To narrow down our efforts, let’s focus on retrieving three pieces of information.</p>
<ol type="1">
<li><p>Which KOfam hits are part of complete Pathway Modules?</p></li>
<li><p>What module(s) is/are they a part of?</p></li>
<li><p>What hits are found in which MAGs?</p></li>
<li><p>Get module information for each gene call.</p></li>
</ol>
<p>We use the command called <code>anvi-estimate-metabolism</code> with the <code>--kegg-output-modes modules</code> flag. This will generate a file (<code>all_kofam_hits.txt</code>) that has the gene caller ID, the KO accession number, an e-value, and what module(s) the gene belongs to.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb32-1"><a href="#cb32-1"></a><span class="ex">anvi-estimate-metabolism</span> --contigs-db 03_CONTIGS/WATER-contigs.db \</span>
<span id="cb32-2"><a href="#cb32-2"></a>                         --kegg-data-dir ~/Desktop/kegg-kofams \</span>
<span id="cb32-3"><a href="#cb32-3"></a>                         --metagenome-mode \</span>
<span id="cb32-4"><a href="#cb32-4"></a>                         --output-file-prefix METABOLISM/00_HELPER_FILES/all  \</span>
<span id="cb32-5"><a href="#cb32-5"></a>                         --kegg-output-modes kofam_hits                         </span></code></pre></div>
<ol start="2" type="1">
<li>Select only unique hits.</li>
</ol>
<p>As before, in cases of multiple KO hits for a single gene call, we select the hit with the lowest e-value. We could do what we did above or we can use the table with unique gene hits that is stored in the variable <code>kofam_unique</code>. What we do here is merge the two tables by gene caller id <em>and</em> KO accession number. This way we get only the top hits from the <code>all_modules.txt</code> file.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a>genes_all_ko &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/00_HELPER_FILES/all_kofam_hits.txt&quot;</span>, </span>
<span id="cb33-2"><a href="#cb33-2"></a>                        <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">quote =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb33-3"><a href="#cb33-3"></a>genes_all_ko[, <span class="kw">c</span>(<span class="st">&#39;unique_id&#39;</span>, <span class="st">&#39;metagenome_name&#39;</span>, </span>
<span id="cb33-4"><a href="#cb33-4"></a>            <span class="st">&#39;contig&#39;</span>, <span class="st">&#39;source&#39;</span>)] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="ot">NULL</span>)</span>
<span id="cb33-5"><a href="#cb33-5"></a>genes_all_ko &lt;-<span class="st"> </span>genes_all_ko[, <span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">1</span>, <span class="dv">3</span>)]</span>
<span id="cb33-6"><a href="#cb33-6"></a>genes_all_ko_alt &lt;-<span class="st"> </span>genes_all_ko</span>
<span id="cb33-7"><a href="#cb33-7"></a>kofam_unique1 &lt;-<span class="st"> </span>kofam_unique</span>
<span id="cb33-8"><a href="#cb33-8"></a>kofam_unique1[, <span class="kw">c</span>(<span class="st">&#39;source&#39;</span>, <span class="st">&#39;ko_function&#39;</span>, <span class="st">&#39;e_value&#39;</span> )] &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb33-9"><a href="#cb33-9"></a>kofam_unique1<span class="op">$</span>gene_callers_id &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">as.character</span>(kofam_unique1<span class="op">$</span>gene_callers_id))</span>
<span id="cb33-10"><a href="#cb33-10"></a>genes_all_ko &lt;-<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(kofam_unique1, genes_all_ko, </span>
<span id="cb33-11"><a href="#cb33-11"></a>                                  <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;gene_callers_id&quot;</span> =<span class="st"> &quot;gene_caller_id&quot;</span> , </span>
<span id="cb33-12"><a href="#cb33-12"></a>                                         <span class="st">&quot;accession&quot;</span> =<span class="st"> &quot;ko&quot;</span>))</span>
<span id="cb33-13"><a href="#cb33-13"></a></span>
<span id="cb33-14"><a href="#cb33-14"></a><span class="co">### ALT TO USE ALL GENE CALLS </span><span class="al">###</span></span>
<span id="cb33-15"><a href="#cb33-15"></a>kofam1 &lt;-<span class="st"> </span>kofam</span>
<span id="cb33-16"><a href="#cb33-16"></a>kofam1[, <span class="kw">c</span>(<span class="st">&#39;source&#39;</span>, <span class="st">&#39;ko_function&#39;</span>, <span class="st">&#39;e_value&#39;</span> )] &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb33-17"><a href="#cb33-17"></a>kofam1<span class="op">$</span>gene_callers_id &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">as.character</span>(kofam1<span class="op">$</span>gene_callers_id))</span>
<span id="cb33-18"><a href="#cb33-18"></a>genes_all_ko_alt &lt;-<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(kofam1, genes_all_ko_alt, </span>
<span id="cb33-19"><a href="#cb33-19"></a>                                  <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;gene_callers_id&quot;</span> =<span class="st"> &quot;gene_caller_id&quot;</span> , </span>
<span id="cb33-20"><a href="#cb33-20"></a>                                         <span class="st">&quot;accession&quot;</span> =<span class="st"> &quot;ko&quot;</span>))</span></code></pre></div>
</div>
<ol start="3" type="1">
<li>Remove genes with no module hit.</li>
</ol>
<p>OK. now we have a data frame that contains gene IDs, KO hits, and module IDs (if any). Some KOs are found in multiple modules while some are not found in any modules. We will deal with the multiple module cases in a minute. For now, let’s remove any hits <em>not</em> found in a module. Anvi’o doesn’t need information for all genes and removing genes with no hits will make parsing the data a tiny bit easier.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a><span class="co">#genes_ko_only &lt;- genes_all_ko[!(genes_all_ko$modules_with_ko==&quot;None&quot;),]</span></span>
<span id="cb34-2"><a href="#cb34-2"></a><span class="co">#genes_ko_only &lt;- genes_ko_only %&gt;% dplyr::rename(kegg_module = modules_with_ko)</span></span>
<span id="cb34-3"><a href="#cb34-3"></a><span class="co">#genes_ko_only_long &lt;- genes_ko_only %&gt;% tidyr::separate_rows(kegg_module)</span></span>
<span id="cb34-4"><a href="#cb34-4"></a></span>
<span id="cb34-5"><a href="#cb34-5"></a><span class="co">### ALT TO USE ALL GENE CALLS </span><span class="al">###</span></span>
<span id="cb34-6"><a href="#cb34-6"></a>genes_ko_only_alt &lt;-<span class="st"> </span>genes_all_ko_alt[<span class="op">!</span>(genes_all_ko_alt<span class="op">$</span>modules_with_ko<span class="op">==</span><span class="st">&quot;None&quot;</span>),]</span>
<span id="cb34-7"><a href="#cb34-7"></a>genes_ko_only_alt &lt;-<span class="st"> </span>genes_ko_only_alt <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">kegg_module =</span> modules_with_ko)</span>
<span id="cb34-8"><a href="#cb34-8"></a>genes_ko_only_long_alt &lt;-<span class="st"> </span>genes_ko_only_alt <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">separate_rows</span>(kegg_module)</span></code></pre></div>
</div>
<p>And we get a list of 3,210 genes that have module affiliations. All we know right now is the module numbers, not the names.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb35-1"><a href="#cb35-1"></a><span class="ex">anvi-estimate-metabolism</span> --contigs-db 03_CONTIGS/WATER-contigs.db </span>
<span id="cb35-2"><a href="#cb35-2"></a>                         <span class="ex">--kegg-data-dir</span> ~/Desktop/kegg-kofams/ </span>
<span id="cb35-3"><a href="#cb35-3"></a>                         <span class="ex">--metagenome-mode</span> </span>
<span id="cb35-4"><a href="#cb35-4"></a>                         <span class="ex">--output-file-prefix</span> METABOLISM/00_HELPER_FILES/all  </span>
<span id="cb35-5"><a href="#cb35-5"></a>                         <span class="ex">--kegg-output-modes</span> modules</span></code></pre></div>
<ol start="4" type="1">
<li>Get module information</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a>all_mod &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/00_HELPER_FILES/all_modules.txt&quot;</span>, </span>
<span id="cb36-2"><a href="#cb36-2"></a>                        <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">quote =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb36-3"><a href="#cb36-3"></a>all_mod[, <span class="kw">c</span>(<span class="st">&#39;module_definition&#39;</span>, <span class="st">&#39;metagenome_name&#39;</span>, </span>
<span id="cb36-4"><a href="#cb36-4"></a>            <span class="st">&#39;module_class&#39;</span>, <span class="st">&#39;module_is_complete&#39;</span>, </span>
<span id="cb36-5"><a href="#cb36-5"></a>            <span class="st">&#39;module_completeness&#39;</span>, <span class="st">&#39;gene_caller_ids_in_module&#39;</span>, </span>
<span id="cb36-6"><a href="#cb36-6"></a>            <span class="st">&#39;kofam_hits_in_module&#39;</span>)] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="ot">NULL</span>)</span>
<span id="cb36-7"><a href="#cb36-7"></a>all_mod &lt;-<span class="st"> </span>all_mod[, <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">3</span>)]</span>
<span id="cb36-8"><a href="#cb36-8"></a>mod_tab &lt;-<span class="st"> </span>all_mod[,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)]</span>
<span id="cb36-9"><a href="#cb36-9"></a>cat_tab &lt;-<span class="st"> </span>all_mod[,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)]</span>
<span id="cb36-10"><a href="#cb36-10"></a>sub_tab &lt;-<span class="st"> </span>all_mod[,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">4</span>)]</span>
<span id="cb36-11"><a href="#cb36-11"></a>name_tab &lt;-<span class="st">  </span>all_mod[,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">5</span>)]</span></code></pre></div>
</div>
<ol start="5" type="1">
<li>Combine genes &amp; module data</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a>genes_mod &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(genes_ko_only_long_alt, mod_tab, </span>
<span id="cb37-2"><a href="#cb37-2"></a>                              <span class="dt">by =</span> <span class="st">&quot;kegg_module&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb37-3"><a href="#cb37-3"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">drop_na</span>(<span class="st">&quot;unique_id&quot;</span>)</span>
<span id="cb37-4"><a href="#cb37-4"></a>genes_mod &lt;-<span class="st"> </span>genes_mod <span class="op">%&gt;%</span><span class="st"> </span>tibble<span class="op">::</span><span class="kw">rowid_to_column</span>(<span class="st">&quot;unique_row_id&quot;</span>)</span>
<span id="cb37-5"><a href="#cb37-5"></a>genes_mod_ko &lt;-<span class="st"> </span>genes_mod</span>
<span id="cb37-6"><a href="#cb37-6"></a>genes_mod &lt;-<span class="st"> </span>genes_mod[, <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">4</span>)]</span>
<span id="cb37-7"><a href="#cb37-7"></a>genes_mod &lt;-<span class="st"> </span>genes_mod <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate_if</span>(is.numeric, </span>
<span id="cb37-8"><a href="#cb37-8"></a>                                            as.character, </span>
<span id="cb37-9"><a href="#cb37-9"></a>                                            is.factor, </span>
<span id="cb37-10"><a href="#cb37-10"></a>                                            as.character) <span class="op">%&gt;%</span></span>
<span id="cb37-11"><a href="#cb37-11"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;path&quot;</span>, unique_row_id<span class="op">:</span>kegg_module, </span>
<span id="cb37-12"><a href="#cb37-12"></a>               <span class="dt">remove =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span>  <span class="st">&quot;XXX&quot;</span>) <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb37-13"><a href="#cb37-13"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="kw">c</span>(<span class="st">&quot;path&quot;</span>), <span class="dt">values_from =</span> <span class="kw">c</span>(<span class="st">&quot;path&quot;</span>)) <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb37-14"><a href="#cb37-14"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;kegg_module&quot;</span>, <span class="op">-</span><span class="st">&quot;gene_callers_id&quot;</span>, </span>
<span id="cb37-15"><a href="#cb37-15"></a>               <span class="dt">na.rm =</span> <span class="ot">TRUE</span>, <span class="dt">remove =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;!!!&quot;</span>)</span>
<span id="cb37-16"><a href="#cb37-16"></a>n_reps &lt;-<span class="st"> </span><span class="kw">max</span>(<span class="kw">str_count</span>(genes_mod<span class="op">$</span>kegg_module, <span class="st">&quot;!!!&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="dv">2</span></span>
<span id="cb37-17"><a href="#cb37-17"></a></span>
<span id="cb37-18"><a href="#cb37-18"></a>genes_mod &lt;-<span class="st"> </span>genes_mod <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">separate</span>(<span class="st">&#39;kegg_module&#39;</span>, </span>
<span id="cb37-19"><a href="#cb37-19"></a>                                    <span class="kw">paste</span>(<span class="st">&quot;module&quot;</span>, <span class="dv">1</span><span class="op">:</span>n_reps, <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>), </span>
<span id="cb37-20"><a href="#cb37-20"></a>                                    <span class="dt">sep =</span> <span class="st">&quot;!!!&quot;</span>, <span class="dt">extra =</span> <span class="st">&quot;drop&quot;</span>)</span>
<span id="cb37-21"><a href="#cb37-21"></a></span>
<span id="cb37-22"><a href="#cb37-22"></a><span class="co">#######################################</span></span>
<span id="cb37-23"><a href="#cb37-23"></a>genes_cat &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(genes_ko_only_long_alt, cat_tab, </span>
<span id="cb37-24"><a href="#cb37-24"></a>                              <span class="dt">by =</span> <span class="st">&quot;kegg_module&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb37-25"><a href="#cb37-25"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">drop_na</span>(<span class="st">&quot;unique_id&quot;</span>)</span>
<span id="cb37-26"><a href="#cb37-26"></a>genes_cat[, <span class="kw">c</span>(<span class="st">&quot;accession&quot;</span>, <span class="st">&quot;kegg_module&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="ot">NULL</span>)</span>
<span id="cb37-27"><a href="#cb37-27"></a>genes_cat &lt;-<span class="st"> </span>genes_cat <span class="op">%&gt;%</span><span class="st"> </span>tibble<span class="op">::</span><span class="kw">rowid_to_column</span>(<span class="st">&quot;unique_row_id&quot;</span>)</span>
<span id="cb37-28"><a href="#cb37-28"></a>genes_cat &lt;-<span class="st"> </span>genes_cat[, <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">4</span>)]</span>
<span id="cb37-29"><a href="#cb37-29"></a></span>
<span id="cb37-30"><a href="#cb37-30"></a>genes_cat &lt;-<span class="st"> </span>genes_cat <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate_if</span>(is.numeric, </span>
<span id="cb37-31"><a href="#cb37-31"></a>                                            as.character, </span>
<span id="cb37-32"><a href="#cb37-32"></a>                                            is.factor, </span>
<span id="cb37-33"><a href="#cb37-33"></a>                                            as.character) <span class="op">%&gt;%</span></span>
<span id="cb37-34"><a href="#cb37-34"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;path&quot;</span>, unique_row_id<span class="op">:</span>module_category, </span>
<span id="cb37-35"><a href="#cb37-35"></a>               <span class="dt">remove =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span>  <span class="st">&quot;XXX&quot;</span>) <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb37-36"><a href="#cb37-36"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="kw">c</span>(<span class="st">&quot;path&quot;</span>), <span class="dt">values_from =</span> <span class="kw">c</span>(<span class="st">&quot;path&quot;</span>)) <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb37-37"><a href="#cb37-37"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;module_category&quot;</span>, <span class="op">-</span><span class="st">&quot;gene_callers_id&quot;</span>, </span>
<span id="cb37-38"><a href="#cb37-38"></a>               <span class="dt">na.rm =</span> <span class="ot">TRUE</span>, <span class="dt">remove =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;!!!&quot;</span>)</span>
<span id="cb37-39"><a href="#cb37-39"></a>n_reps &lt;-<span class="st"> </span><span class="kw">max</span>(<span class="kw">str_count</span>(genes_cat<span class="op">$</span>module_category, <span class="st">&quot;!!!&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="dv">2</span></span>
<span id="cb37-40"><a href="#cb37-40"></a></span>
<span id="cb37-41"><a href="#cb37-41"></a>genes_cat &lt;-<span class="st"> </span>genes_cat <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">separate</span>(<span class="st">&#39;module_category&#39;</span>, </span>
<span id="cb37-42"><a href="#cb37-42"></a>                                    <span class="kw">paste</span>(<span class="st">&quot;category&quot;</span>, <span class="dv">1</span><span class="op">:</span>n_reps, <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>), </span>
<span id="cb37-43"><a href="#cb37-43"></a>                                    <span class="dt">sep =</span> <span class="st">&quot;!!!&quot;</span>, <span class="dt">extra =</span> <span class="st">&quot;drop&quot;</span>)</span>
<span id="cb37-44"><a href="#cb37-44"></a><span class="co">#######################################</span></span>
<span id="cb37-45"><a href="#cb37-45"></a>genes_sub &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(genes_ko_only_long_alt, sub_tab, </span>
<span id="cb37-46"><a href="#cb37-46"></a>                              <span class="dt">by =</span> <span class="st">&quot;kegg_module&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb37-47"><a href="#cb37-47"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">drop_na</span>(<span class="st">&quot;unique_id&quot;</span>)</span>
<span id="cb37-48"><a href="#cb37-48"></a>genes_sub[, <span class="kw">c</span>(<span class="st">&quot;accession&quot;</span>, <span class="st">&quot;kegg_module&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="ot">NULL</span>)</span>
<span id="cb37-49"><a href="#cb37-49"></a>genes_sub &lt;-<span class="st"> </span>genes_sub <span class="op">%&gt;%</span><span class="st"> </span>tibble<span class="op">::</span><span class="kw">rowid_to_column</span>(<span class="st">&quot;unique_row_id&quot;</span>)</span>
<span id="cb37-50"><a href="#cb37-50"></a>genes_sub &lt;-<span class="st"> </span>genes_sub[, <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">4</span>)]</span>
<span id="cb37-51"><a href="#cb37-51"></a></span>
<span id="cb37-52"><a href="#cb37-52"></a>genes_sub &lt;-<span class="st"> </span>genes_sub <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate_if</span>(is.numeric, </span>
<span id="cb37-53"><a href="#cb37-53"></a>                                            as.character, </span>
<span id="cb37-54"><a href="#cb37-54"></a>                                            is.factor, </span>
<span id="cb37-55"><a href="#cb37-55"></a>                                            as.character) <span class="op">%&gt;%</span></span>
<span id="cb37-56"><a href="#cb37-56"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;path&quot;</span>, unique_row_id<span class="op">:</span>module_subcategory, </span>
<span id="cb37-57"><a href="#cb37-57"></a>               <span class="dt">remove =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span>  <span class="st">&quot;XXX&quot;</span>) <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb37-58"><a href="#cb37-58"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="kw">c</span>(<span class="st">&quot;path&quot;</span>), <span class="dt">values_from =</span> <span class="kw">c</span>(<span class="st">&quot;path&quot;</span>)) <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb37-59"><a href="#cb37-59"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;module_subcategory&quot;</span>, <span class="op">-</span><span class="st">&quot;gene_callers_id&quot;</span>, </span>
<span id="cb37-60"><a href="#cb37-60"></a>               <span class="dt">na.rm =</span> <span class="ot">TRUE</span>, <span class="dt">remove =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;!!!&quot;</span>)</span>
<span id="cb37-61"><a href="#cb37-61"></a>n_reps &lt;-<span class="st"> </span><span class="kw">max</span>(<span class="kw">str_count</span>(genes_sub<span class="op">$</span>module_subcategory, <span class="st">&quot;!!!&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="dv">2</span></span>
<span id="cb37-62"><a href="#cb37-62"></a></span>
<span id="cb37-63"><a href="#cb37-63"></a>genes_sub &lt;-<span class="st"> </span>genes_sub <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">separate</span>(<span class="st">&#39;module_subcategory&#39;</span>, </span>
<span id="cb37-64"><a href="#cb37-64"></a>                                    <span class="kw">paste</span>(<span class="st">&quot;subcategory&quot;</span>, <span class="dv">1</span><span class="op">:</span>n_reps, <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>), </span>
<span id="cb37-65"><a href="#cb37-65"></a>                                    <span class="dt">sep =</span> <span class="st">&quot;!!!&quot;</span>, <span class="dt">extra =</span> <span class="st">&quot;drop&quot;</span>)</span>
<span id="cb37-66"><a href="#cb37-66"></a></span>
<span id="cb37-67"><a href="#cb37-67"></a><span class="co">#######################################</span></span>
<span id="cb37-68"><a href="#cb37-68"></a>genes_name &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(genes_ko_only_long_alt, name_tab, </span>
<span id="cb37-69"><a href="#cb37-69"></a>                              <span class="dt">by =</span> <span class="st">&quot;kegg_module&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb37-70"><a href="#cb37-70"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">drop_na</span>(<span class="st">&quot;unique_id&quot;</span>)</span>
<span id="cb37-71"><a href="#cb37-71"></a>genes_name[, <span class="kw">c</span>(<span class="st">&quot;accession&quot;</span>, <span class="st">&quot;kegg_module&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="ot">NULL</span>)</span>
<span id="cb37-72"><a href="#cb37-72"></a>genes_name &lt;-<span class="st"> </span>genes_name <span class="op">%&gt;%</span><span class="st"> </span>tibble<span class="op">::</span><span class="kw">rowid_to_column</span>(<span class="st">&quot;unique_row_id&quot;</span>)</span>
<span id="cb37-73"><a href="#cb37-73"></a>genes_name &lt;-<span class="st"> </span>genes_name[, <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">4</span>)]</span>
<span id="cb37-74"><a href="#cb37-74"></a>genes_name &lt;-<span class="st"> </span>genes_name <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate_if</span>(is.numeric, </span>
<span id="cb37-75"><a href="#cb37-75"></a>                                            as.character, </span>
<span id="cb37-76"><a href="#cb37-76"></a>                                            is.factor, </span>
<span id="cb37-77"><a href="#cb37-77"></a>                                            as.character) <span class="op">%&gt;%</span></span>
<span id="cb37-78"><a href="#cb37-78"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;path&quot;</span>, unique_row_id<span class="op">:</span>module_name, </span>
<span id="cb37-79"><a href="#cb37-79"></a>               <span class="dt">remove =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span>  <span class="st">&quot;XXX&quot;</span>) <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb37-80"><a href="#cb37-80"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="kw">c</span>(<span class="st">&quot;path&quot;</span>), <span class="dt">values_from =</span> <span class="kw">c</span>(<span class="st">&quot;path&quot;</span>)) <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb37-81"><a href="#cb37-81"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;module_name&quot;</span>, <span class="op">-</span><span class="st">&quot;gene_callers_id&quot;</span>, </span>
<span id="cb37-82"><a href="#cb37-82"></a>               <span class="dt">na.rm =</span> <span class="ot">TRUE</span>, <span class="dt">remove =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;!!!&quot;</span>)</span>
<span id="cb37-83"><a href="#cb37-83"></a>n_reps &lt;-<span class="st"> </span><span class="kw">max</span>(<span class="kw">str_count</span>(genes_name<span class="op">$</span>module_name, <span class="st">&quot;!!!&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="dv">2</span></span>
<span id="cb37-84"><a href="#cb37-84"></a></span>
<span id="cb37-85"><a href="#cb37-85"></a>genes_name &lt;-<span class="st"> </span>genes_name <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">separate</span>(<span class="st">&#39;module_name&#39;</span>, </span>
<span id="cb37-86"><a href="#cb37-86"></a>                                    <span class="kw">paste</span>(<span class="st">&quot;name&quot;</span>, <span class="dv">1</span><span class="op">:</span>n_reps, <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>), </span>
<span id="cb37-87"><a href="#cb37-87"></a>                                    <span class="dt">sep =</span> <span class="st">&quot;!!!&quot;</span>, <span class="dt">extra =</span> <span class="st">&quot;drop&quot;</span>)</span>
<span id="cb37-88"><a href="#cb37-88"></a><span class="co">#######################################</span></span>
<span id="cb37-89"><a href="#cb37-89"></a>genes_ko &lt;-<span class="st"> </span>genes_mod_ko</span>
<span id="cb37-90"><a href="#cb37-90"></a>genes_ko &lt;-<span class="st"> </span>genes_ko[, <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">3</span>)]</span>
<span id="cb37-91"><a href="#cb37-91"></a>genes_ko &lt;-<span class="st"> </span>genes_ko <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate_if</span>(is.numeric, </span>
<span id="cb37-92"><a href="#cb37-92"></a>                                            as.character, </span>
<span id="cb37-93"><a href="#cb37-93"></a>                                            is.factor, </span>
<span id="cb37-94"><a href="#cb37-94"></a>                                            as.character) <span class="op">%&gt;%</span></span>
<span id="cb37-95"><a href="#cb37-95"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;path&quot;</span>, unique_row_id<span class="op">:</span>accession, </span>
<span id="cb37-96"><a href="#cb37-96"></a>               <span class="dt">remove =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span>  <span class="st">&quot;XXX&quot;</span>) <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb37-97"><a href="#cb37-97"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="kw">c</span>(<span class="st">&quot;path&quot;</span>), <span class="dt">values_from =</span> <span class="kw">c</span>(<span class="st">&quot;path&quot;</span>)) <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb37-98"><a href="#cb37-98"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;ko&quot;</span>, <span class="op">-</span><span class="st">&quot;gene_callers_id&quot;</span>, </span>
<span id="cb37-99"><a href="#cb37-99"></a>               <span class="dt">na.rm =</span> <span class="ot">TRUE</span>, <span class="dt">remove =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;!!!&quot;</span>)</span>
<span id="cb37-100"><a href="#cb37-100"></a>n_reps &lt;-<span class="st"> </span><span class="kw">max</span>(<span class="kw">str_count</span>(genes_ko<span class="op">$</span>ko, <span class="st">&quot;!!!&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="dv">2</span></span>
<span id="cb37-101"><a href="#cb37-101"></a></span>
<span id="cb37-102"><a href="#cb37-102"></a>genes_ko &lt;-<span class="st"> </span>genes_ko <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">separate</span>(<span class="st">&#39;ko&#39;</span>, </span>
<span id="cb37-103"><a href="#cb37-103"></a>                                    <span class="kw">paste</span>(<span class="st">&quot;module&quot;</span>, <span class="dv">1</span><span class="op">:</span>n_reps, <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>), </span>
<span id="cb37-104"><a href="#cb37-104"></a>                                    <span class="dt">sep =</span> <span class="st">&quot;!!!&quot;</span>, <span class="dt">extra =</span> <span class="st">&quot;drop&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1"></a>genes_mod[, <span class="dv">2</span><span class="op">:</span>n_reps] &lt;-<span class="st"> </span><span class="kw">apply</span>(genes_mod[, <span class="dv">2</span><span class="op">:</span>n_reps], <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="kw">gsub</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">d+XXX&quot;</span>, <span class="st">&quot;&quot;</span>, x))</span>
<span id="cb38-2"><a href="#cb38-2"></a>genes_mod &lt;-<span class="st"> </span>genes_mod <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;kegg_module&quot;</span>, <span class="op">-</span><span class="st">&quot;gene_callers_id&quot;</span>, </span>
<span id="cb38-3"><a href="#cb38-3"></a>               <span class="dt">na.rm =</span> <span class="ot">TRUE</span>, <span class="dt">remove =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;!!!&quot;</span>)</span>
<span id="cb38-4"><a href="#cb38-4"></a></span>
<span id="cb38-5"><a href="#cb38-5"></a>genes_cat[, <span class="dv">2</span><span class="op">:</span>n_reps] &lt;-<span class="st"> </span><span class="kw">apply</span>(genes_cat[, <span class="dv">2</span><span class="op">:</span>n_reps], <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="kw">gsub</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">d+XXX&quot;</span>, <span class="st">&quot;&quot;</span>, x))</span>
<span id="cb38-6"><a href="#cb38-6"></a>genes_cat &lt;-<span class="st"> </span>genes_cat <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;module_category&quot;</span>, <span class="op">-</span><span class="st">&quot;gene_callers_id&quot;</span>, </span>
<span id="cb38-7"><a href="#cb38-7"></a>               <span class="dt">na.rm =</span> <span class="ot">TRUE</span>, <span class="dt">remove =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;!!!&quot;</span>)</span>
<span id="cb38-8"><a href="#cb38-8"></a></span>
<span id="cb38-9"><a href="#cb38-9"></a>genes_sub[, <span class="dv">2</span><span class="op">:</span>n_reps] &lt;-<span class="st"> </span><span class="kw">apply</span>(genes_sub[, <span class="dv">2</span><span class="op">:</span>n_reps], <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="kw">gsub</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">d+XXX&quot;</span>, <span class="st">&quot;&quot;</span>, x))</span>
<span id="cb38-10"><a href="#cb38-10"></a>genes_sub &lt;-<span class="st"> </span>genes_sub <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;module_subcategory&quot;</span>, <span class="op">-</span><span class="st">&quot;gene_callers_id&quot;</span>, </span>
<span id="cb38-11"><a href="#cb38-11"></a>               <span class="dt">na.rm =</span> <span class="ot">TRUE</span>, <span class="dt">remove =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;!!!&quot;</span>)</span>
<span id="cb38-12"><a href="#cb38-12"></a></span>
<span id="cb38-13"><a href="#cb38-13"></a>genes_name[, <span class="dv">2</span><span class="op">:</span>n_reps] &lt;-<span class="st"> </span><span class="kw">apply</span>(genes_name[, <span class="dv">2</span><span class="op">:</span>n_reps], <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="kw">gsub</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">d+XXX&quot;</span>, <span class="st">&quot;&quot;</span>, x))</span>
<span id="cb38-14"><a href="#cb38-14"></a>genes_name &lt;-<span class="st"> </span>genes_name <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;module_name&quot;</span>, <span class="op">-</span><span class="st">&quot;gene_callers_id&quot;</span>, </span>
<span id="cb38-15"><a href="#cb38-15"></a>               <span class="dt">na.rm =</span> <span class="ot">TRUE</span>, <span class="dt">remove =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;!!!&quot;</span>)</span>
<span id="cb38-16"><a href="#cb38-16"></a></span>
<span id="cb38-17"><a href="#cb38-17"></a>genes_ko[, <span class="dv">2</span><span class="op">:</span>n_reps] &lt;-<span class="st"> </span><span class="kw">apply</span>(genes_ko[, <span class="dv">2</span><span class="op">:</span>n_reps], <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="kw">gsub</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">d+XXX&quot;</span>, <span class="st">&quot;&quot;</span>, x))</span>
<span id="cb38-18"><a href="#cb38-18"></a>genes_ko &lt;-<span class="st"> </span>genes_ko <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;ko&quot;</span>, <span class="op">-</span><span class="st">&quot;gene_callers_id&quot;</span>, </span>
<span id="cb38-19"><a href="#cb38-19"></a>               <span class="dt">na.rm =</span> <span class="ot">TRUE</span>, <span class="dt">remove =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;!!!&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1"></a>genes_modules &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(genes_ko, genes_mod, <span class="dt">by =</span> <span class="st">&quot;gene_callers_id&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb39-2"><a href="#cb39-2"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(., genes_cat, <span class="dt">by =</span> <span class="st">&quot;gene_callers_id&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb39-3"><a href="#cb39-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(., genes_sub, <span class="dt">by =</span> <span class="st">&quot;gene_callers_id&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb39-4"><a href="#cb39-4"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(., genes_name, <span class="dt">by =</span> <span class="st">&quot;gene_callers_id&quot;</span>) </span>
<span id="cb39-5"><a href="#cb39-5"></a><span class="kw">write.table</span>(genes_modules, <span class="st">&quot;metabolism/00_HELPER_FILES/genes_modules.txt&quot;</span>, </span>
<span id="cb39-6"><a href="#cb39-6"></a>            <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
</div>
<p>Now we have a table of 2842 unique genes and their KEGG module annotations. The last thing we need to do is change the gene caller IDs to “contig IDs” to match the name of the slits in the contig.db. Way back uo the road we created a variable called <code>kofam_genes_tab</code>, a table derived from a gene calls file where we added the gene call onto the end of the contig name. We need to do the same with the new KEGG modules annotation file so we can add this as an additional layer to the visualization. <em>And</em>, we also need to tack on a split name like we did for the fasta file.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1"></a>genes_modules1 &lt;-<span class="st"> </span>genes_modules</span>
<span id="cb40-2"><a href="#cb40-2"></a>kofam_genes_tab1 &lt;-<span class="st"> </span>kofam_genes_tab</span>
<span id="cb40-3"><a href="#cb40-3"></a>add_view_part_<span class="dv">1</span> &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">right_join</span>(kofam_genes_tab1, genes_modules1, <span class="dt">by =</span> <span class="st">&quot;gene_callers_id&quot;</span>)</span>
<span id="cb40-4"><a href="#cb40-4"></a>add_view_part_<span class="dv">1</span>[, <span class="dv">3</span><span class="op">:</span><span class="dv">10</span>] &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb40-5"><a href="#cb40-5"></a>add_view_part_<span class="dv">1</span>[, <span class="dv">1</span>] &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb40-6"><a href="#cb40-6"></a>add_view_part_<span class="dv">1</span> &lt;-<span class="st"> </span>add_view_part_<span class="dv">1</span> <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">split_id =</span> <span class="st">&quot;split_00001&quot;</span>, <span class="dt">.after =</span> <span class="dv">1</span>) <span class="op">%&gt;%</span></span>
<span id="cb40-7"><a href="#cb40-7"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(items, contig, split_id, <span class="dt">remove =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb40-8"><a href="#cb40-8"></a><span class="kw">write.table</span>(add_view_part_<span class="dv">1</span>, <span class="st">&quot;metabolism/05_ANNOTATION_FILES/additional-items-kegg-functions.txt&quot;</span>, </span>
<span id="cb40-9"><a href="#cb40-9"></a>            <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
</div>
<p>Now we can run the following command to import the annotation data into the profile database.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb41-1"><a href="#cb41-1"></a><span class="ex">anvi-import-misc-data</span> 05_ANNOTATION_FILES/additional-items-kegg-functions.txt \</span>
<span id="cb41-2"><a href="#cb41-2"></a>                      --pan-or-profile-db 04_KOFAM_PROFILES-MERGED/PROFILE.db \</span>
<span id="cb41-3"><a href="#cb41-3"></a>                      --target-data-table items</span></code></pre></div>
<ol start="6" type="1">
<li>KO genes from MAGs</li>
</ol>
<div class="sourceCode" id="cb42"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb42-1"><a href="#cb42-1"></a><span class="kw">for</span> <span class="ex">mag</span> in <span class="kw">`</span><span class="fu">cat</span> mags.list<span class="kw">`</span> </span>
<span id="cb42-2"><a href="#cb42-2"></a><span class="kw">do</span> <span class="ex">anvi-estimate-metabolism</span> --contigs-db 03_CONTIGS/WATER-contigs.db \</span>
<span id="cb42-3"><a href="#cb42-3"></a>                            --kegg-data-dir ~/PATH/to/kegg-kofams/ \</span>
<span id="cb42-4"><a href="#cb42-4"></a>                            --profile-db 06_MERGED/WATER/PROFILE.db \</span>
<span id="cb42-5"><a href="#cb42-5"></a>                            --collection-name MICROBIAL_FINAL \</span>
<span id="cb42-6"><a href="#cb42-6"></a>                            --bin-id <span class="va">$mag</span> -O <span class="va">$mag</span>  \</span>
<span id="cb42-7"><a href="#cb42-7"></a>                            --kegg-output-modes kofam_hits</span>
<span id="cb42-8"><a href="#cb42-8"></a><span class="kw">done</span></span>
<span id="cb42-9"><a href="#cb42-9"></a></span>
<span id="cb42-10"><a href="#cb42-10"></a><span class="kw">for</span> <span class="ex">mag</span> in <span class="kw">`</span><span class="fu">cat</span> mags.list<span class="kw">`</span></span>
<span id="cb42-11"><a href="#cb42-11"></a><span class="kw">do</span> <span class="ex">anvi-estimate-metabolism</span> --contigs-db 03_CONTIGS/WATER-contigs.db \</span>
<span id="cb42-12"><a href="#cb42-12"></a>                            --kegg-data-dir ~/PATH/to/kegg-kofams/ \</span>
<span id="cb42-13"><a href="#cb42-13"></a>                            --profile-db 06_MERGED/WATER/PROFILE.db \</span>
<span id="cb42-14"><a href="#cb42-14"></a>                            --collection-name MICROBIAL_FINAL \</span>
<span id="cb42-15"><a href="#cb42-15"></a>                            --bin-id <span class="va">$mag</span> --output-file-prefix <span class="va">$mag</span>  </span>
<span id="cb42-16"><a href="#cb42-16"></a>                            <span class="ex">--kegg-output-modes</span> modules</span>
<span id="cb42-17"><a href="#cb42-17"></a><span class="kw">done</span></span></code></pre></div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1"></a><span class="co">###################### mag_01 ############################</span></span>
<span id="cb43-2"><a href="#cb43-2"></a>mag_<span class="dv">01</span>&lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/06_MAG_KOFAMS/WATER_MAG_00001_kofam_hits.txt&quot;</span>, </span>
<span id="cb43-3"><a href="#cb43-3"></a>                    <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb43-4"><a href="#cb43-4"></a>mag_<span class="dv">01</span>[, <span class="kw">c</span>(<span class="st">&quot;unique_id&quot;</span>, <span class="st">&quot;ko&quot;</span>, <span class="st">&quot;modules_with_ko&quot;</span>, <span class="st">&quot;bin_name&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="ot">NULL</span>)</span>
<span id="cb43-5"><a href="#cb43-5"></a>mag_<span class="dv">01</span><span class="op">$</span>mag_id &lt;-<span class="st"> &quot;MAG01&quot;</span> </span>
<span id="cb43-6"><a href="#cb43-6"></a>mag_<span class="dv">01</span> &lt;-<span class="st"> </span>mag_<span class="dv">01</span> <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(gene_caller_id)</span>
<span id="cb43-7"><a href="#cb43-7"></a>mag_<span class="dv">01</span> &lt;-<span class="st"> </span>mag_<span class="dv">01</span>[<span class="op">!</span><span class="kw">duplicated</span>(mag_<span class="dv">01</span><span class="op">$</span>gene_caller_id), ]</span>
<span id="cb43-8"><a href="#cb43-8"></a>mag_<span class="dv">01</span> &lt;-<span class="st"> </span>mag_<span class="dv">01</span>[, <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">3</span>)]</span>
<span id="cb43-9"><a href="#cb43-9"></a>mag_<span class="dv">01</span> &lt;-<span class="st"> </span>mag_<span class="dv">01</span> <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">unite</span>(int_name, <span class="st">&quot;contig&quot;</span>, <span class="st">&quot;gene_caller_id&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;_gene_&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb43-10"><a href="#cb43-10"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">split_id =</span> <span class="st">&quot;split_00001&quot;</span>, <span class="dt">.after =</span> <span class="dv">2</span>) <span class="op">%&gt;%</span></span>
<span id="cb43-11"><a href="#cb43-11"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(items, <span class="st">&quot;int_name&quot;</span>, <span class="st">&quot;split_id&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb43-12"><a href="#cb43-12"></a><span class="co">###################### mag_02 ############################</span></span>
<span id="cb43-13"><a href="#cb43-13"></a>mag_<span class="dv">02</span>&lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/06_MAG_KOFAMS/WATER_MAG_00002_kofam_hits.txt&quot;</span>, </span>
<span id="cb43-14"><a href="#cb43-14"></a>                    <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb43-15"><a href="#cb43-15"></a>mag_<span class="dv">02</span>[, <span class="kw">c</span>(<span class="st">&quot;unique_id&quot;</span>, <span class="st">&quot;ko&quot;</span>, <span class="st">&quot;modules_with_ko&quot;</span>, <span class="st">&quot;bin_name&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="ot">NULL</span>)</span>
<span id="cb43-16"><a href="#cb43-16"></a>mag_<span class="dv">02</span><span class="op">$</span>mag_id &lt;-<span class="st"> &quot;MAG02&quot;</span> </span>
<span id="cb43-17"><a href="#cb43-17"></a>mag_<span class="dv">02</span> &lt;-<span class="st"> </span>mag_<span class="dv">02</span> <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(gene_caller_id)</span>
<span id="cb43-18"><a href="#cb43-18"></a>mag_<span class="dv">02</span> &lt;-<span class="st"> </span>mag_<span class="dv">02</span>[<span class="op">!</span><span class="kw">duplicated</span>(mag_<span class="dv">02</span><span class="op">$</span>gene_caller_id), ]</span>
<span id="cb43-19"><a href="#cb43-19"></a>mag_<span class="dv">02</span> &lt;-<span class="st"> </span>mag_<span class="dv">02</span>[, <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">3</span>)]</span>
<span id="cb43-20"><a href="#cb43-20"></a>mag_<span class="dv">02</span> &lt;-<span class="st"> </span>mag_<span class="dv">02</span> <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">unite</span>(int_name, <span class="st">&quot;contig&quot;</span>, <span class="st">&quot;gene_caller_id&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;_gene_&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb43-21"><a href="#cb43-21"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">split_id =</span> <span class="st">&quot;split_00001&quot;</span>, <span class="dt">.after =</span> <span class="dv">1</span>) <span class="op">%&gt;%</span></span>
<span id="cb43-22"><a href="#cb43-22"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(items, <span class="st">&quot;int_name&quot;</span>, <span class="st">&quot;split_id&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb43-23"><a href="#cb43-23"></a><span class="co">###################### mag_03 ############################</span></span>
<span id="cb43-24"><a href="#cb43-24"></a>mag_<span class="dv">03</span>&lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/06_MAG_KOFAMS/WATER_MAG_00003_kofam_hits.txt&quot;</span>, </span>
<span id="cb43-25"><a href="#cb43-25"></a>                    <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb43-26"><a href="#cb43-26"></a>mag_<span class="dv">03</span>[, <span class="kw">c</span>(<span class="st">&quot;unique_id&quot;</span>, <span class="st">&quot;ko&quot;</span>, <span class="st">&quot;modules_with_ko&quot;</span>, <span class="st">&quot;bin_name&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="ot">NULL</span>)</span>
<span id="cb43-27"><a href="#cb43-27"></a>mag_<span class="dv">03</span><span class="op">$</span>mag_id &lt;-<span class="st"> &quot;MAG03&quot;</span> </span>
<span id="cb43-28"><a href="#cb43-28"></a>mag_<span class="dv">03</span> &lt;-<span class="st"> </span>mag_<span class="dv">03</span> <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(gene_caller_id)</span>
<span id="cb43-29"><a href="#cb43-29"></a>mag_<span class="dv">03</span> &lt;-<span class="st"> </span>mag_<span class="dv">03</span>[<span class="op">!</span><span class="kw">duplicated</span>(mag_<span class="dv">03</span><span class="op">$</span>gene_caller_id), ]</span>
<span id="cb43-30"><a href="#cb43-30"></a>mag_<span class="dv">03</span> &lt;-<span class="st"> </span>mag_<span class="dv">03</span>[, <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">3</span>)]</span>
<span id="cb43-31"><a href="#cb43-31"></a>mag_<span class="dv">03</span> &lt;-<span class="st"> </span>mag_<span class="dv">03</span> <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">unite</span>(int_name, <span class="st">&quot;contig&quot;</span>, <span class="st">&quot;gene_caller_id&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;_gene_&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb43-32"><a href="#cb43-32"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">split_id =</span> <span class="st">&quot;split_00001&quot;</span>, <span class="dt">.after =</span> <span class="dv">1</span>) <span class="op">%&gt;%</span></span>
<span id="cb43-33"><a href="#cb43-33"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(items, <span class="st">&quot;int_name&quot;</span>, <span class="st">&quot;split_id&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb43-34"><a href="#cb43-34"></a><span class="co">###################### mag_04 ############################</span></span>
<span id="cb43-35"><a href="#cb43-35"></a>mag_<span class="dv">04</span>&lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/06_MAG_KOFAMS/WATER_MAG_00004_kofam_hits.txt&quot;</span>, </span>
<span id="cb43-36"><a href="#cb43-36"></a>                    <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb43-37"><a href="#cb43-37"></a>mag_<span class="dv">04</span>[, <span class="kw">c</span>(<span class="st">&quot;unique_id&quot;</span>, <span class="st">&quot;ko&quot;</span>, <span class="st">&quot;modules_with_ko&quot;</span>, <span class="st">&quot;bin_name&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="ot">NULL</span>)</span>
<span id="cb43-38"><a href="#cb43-38"></a>mag_<span class="dv">04</span><span class="op">$</span>mag_id &lt;-<span class="st"> &quot;MAG04&quot;</span> </span>
<span id="cb43-39"><a href="#cb43-39"></a>mag_<span class="dv">04</span> &lt;-<span class="st"> </span>mag_<span class="dv">04</span> <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(gene_caller_id)</span>
<span id="cb43-40"><a href="#cb43-40"></a>mag_<span class="dv">04</span> &lt;-<span class="st"> </span>mag_<span class="dv">04</span>[<span class="op">!</span><span class="kw">duplicated</span>(mag_<span class="dv">04</span><span class="op">$</span>gene_caller_id), ]</span>
<span id="cb43-41"><a href="#cb43-41"></a>mag_<span class="dv">04</span> &lt;-<span class="st"> </span>mag_<span class="dv">04</span>[, <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">3</span>)]</span>
<span id="cb43-42"><a href="#cb43-42"></a>mag_<span class="dv">04</span> &lt;-<span class="st"> </span>mag_<span class="dv">04</span> <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">unite</span>(int_name, <span class="st">&quot;contig&quot;</span>, <span class="st">&quot;gene_caller_id&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;_gene_&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb43-43"><a href="#cb43-43"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">split_id =</span> <span class="st">&quot;split_00001&quot;</span>, <span class="dt">.after =</span> <span class="dv">1</span>) <span class="op">%&gt;%</span></span>
<span id="cb43-44"><a href="#cb43-44"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(items, <span class="st">&quot;int_name&quot;</span>, <span class="st">&quot;split_id&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb43-45"><a href="#cb43-45"></a><span class="co">###################### mag_05 ############################</span></span>
<span id="cb43-46"><a href="#cb43-46"></a>mag_<span class="dv">05</span>&lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/06_MAG_KOFAMS/WATER_MAG_00005_kofam_hits.txt&quot;</span>, </span>
<span id="cb43-47"><a href="#cb43-47"></a>                    <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb43-48"><a href="#cb43-48"></a>mag_<span class="dv">05</span>[, <span class="kw">c</span>(<span class="st">&quot;unique_id&quot;</span>, <span class="st">&quot;ko&quot;</span>, <span class="st">&quot;modules_with_ko&quot;</span>, <span class="st">&quot;bin_name&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="ot">NULL</span>)</span>
<span id="cb43-49"><a href="#cb43-49"></a>mag_<span class="dv">05</span><span class="op">$</span>mag_id &lt;-<span class="st"> &quot;MAG05&quot;</span> </span>
<span id="cb43-50"><a href="#cb43-50"></a>mag_<span class="dv">05</span> &lt;-<span class="st"> </span>mag_<span class="dv">05</span> <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(gene_caller_id)</span>
<span id="cb43-51"><a href="#cb43-51"></a>mag_<span class="dv">05</span> &lt;-<span class="st"> </span>mag_<span class="dv">05</span>[<span class="op">!</span><span class="kw">duplicated</span>(mag_<span class="dv">05</span><span class="op">$</span>gene_caller_id), ]</span>
<span id="cb43-52"><a href="#cb43-52"></a>mag_<span class="dv">05</span> &lt;-<span class="st"> </span>mag_<span class="dv">05</span>[, <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">3</span>)]</span>
<span id="cb43-53"><a href="#cb43-53"></a>mag_<span class="dv">05</span> &lt;-<span class="st"> </span>mag_<span class="dv">05</span> <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">unite</span>(int_name, <span class="st">&quot;contig&quot;</span>, <span class="st">&quot;gene_caller_id&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;_gene_&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb43-54"><a href="#cb43-54"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">split_id =</span> <span class="st">&quot;split_00001&quot;</span>, <span class="dt">.after =</span> <span class="dv">1</span>) <span class="op">%&gt;%</span></span>
<span id="cb43-55"><a href="#cb43-55"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(items, <span class="st">&quot;int_name&quot;</span>, <span class="st">&quot;split_id&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb43-56"><a href="#cb43-56"></a><span class="co">###################### bin_13 ############################</span></span>
<span id="cb43-57"><a href="#cb43-57"></a>bin_<span class="dv">13</span>&lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/06_MAG_KOFAMS/WATER_Bin_00013_kofam_hits.txt&quot;</span>, </span>
<span id="cb43-58"><a href="#cb43-58"></a>                    <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb43-59"><a href="#cb43-59"></a>bin_<span class="dv">13</span>[, <span class="kw">c</span>(<span class="st">&quot;unique_id&quot;</span>, <span class="st">&quot;ko&quot;</span>, <span class="st">&quot;modules_with_ko&quot;</span>, <span class="st">&quot;bin_name&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="ot">NULL</span>)</span>
<span id="cb43-60"><a href="#cb43-60"></a>bin_<span class="dv">13</span><span class="op">$</span>mag_id &lt;-<span class="st"> &quot;BIN13&quot;</span> </span>
<span id="cb43-61"><a href="#cb43-61"></a>bin_<span class="dv">13</span> &lt;-<span class="st"> </span>bin_<span class="dv">13</span> <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(gene_caller_id)</span>
<span id="cb43-62"><a href="#cb43-62"></a>bin_<span class="dv">13</span> &lt;-<span class="st"> </span>bin_<span class="dv">13</span>[<span class="op">!</span><span class="kw">duplicated</span>(bin_<span class="dv">13</span><span class="op">$</span>gene_caller_id), ]</span>
<span id="cb43-63"><a href="#cb43-63"></a>bin_<span class="dv">13</span> &lt;-<span class="st"> </span>bin_<span class="dv">13</span>[, <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">3</span>)]</span>
<span id="cb43-64"><a href="#cb43-64"></a>bin_<span class="dv">13</span> &lt;-<span class="st"> </span>bin_<span class="dv">13</span> <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">unite</span>(int_name, <span class="st">&quot;contig&quot;</span>, <span class="st">&quot;gene_caller_id&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;_gene_&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb43-65"><a href="#cb43-65"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">split_id =</span> <span class="st">&quot;split_00001&quot;</span>, <span class="dt">.after =</span> <span class="dv">1</span>) <span class="op">%&gt;%</span></span>
<span id="cb43-66"><a href="#cb43-66"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(items, <span class="st">&quot;int_name&quot;</span>, <span class="st">&quot;split_id&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1"></a>add_mag &lt;-<span class="st"> </span><span class="kw">rbind</span>(mag_<span class="dv">01</span>, mag_<span class="dv">02</span>, mag_<span class="dv">03</span>, mag_<span class="dv">04</span>, mag_<span class="dv">05</span>, bin_<span class="dv">13</span>)</span>
<span id="cb44-2"><a href="#cb44-2"></a><span class="kw">write.table</span>(add_mag, <span class="st">&quot;metabolism/05_ANNOTATION_FILES/additional-items-kos-in-mags.txt&quot;</span>, </span>
<span id="cb44-3"><a href="#cb44-3"></a>            <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
</div>
<div class="sourceCode" id="cb45"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb45-1"><a href="#cb45-1"></a><span class="ex">anvi-import-misc-data</span> 05_ANNOTATION_FILES/additional-items-kos-in-mags.txt \</span>
<span id="cb45-2"><a href="#cb45-2"></a>                      --pan-or-profile-db 04_KOFAM_PROFILES-MERGED/PROFILE.db \</span>
<span id="cb45-3"><a href="#cb45-3"></a>                      --target-data-table items</span></code></pre></div>
<h2 id="genes-in-modules-only">Genes in modules only</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1"></a>kofam_fna_mod &lt;-<span class="st"> </span><span class="kw">readDNAStringSet</span>(<span class="st">&quot;metabolism/00_HELPER_FILES/kofam_genes.fa&quot;</span>)</span>
<span id="cb46-2"><a href="#cb46-2"></a>kofam_names_mod &lt;-<span class="st">  </span><span class="kw">names</span>(kofam_fna_mod)</span>
<span id="cb46-3"><a href="#cb46-3"></a>kofam_seqs_mod &lt;-<span class="st">  </span><span class="kw">paste</span>(kofam_fna_mod)</span>
<span id="cb46-4"><a href="#cb46-4"></a>fasta_df_mod &lt;-<span class="st"> </span><span class="kw">data.frame</span>(kofam_names_mod, kofam_seqs_mod)</span>
<span id="cb46-5"><a href="#cb46-5"></a>fasta_df_mod<span class="op">$</span>kofam_names_mod &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">as.character</span>(fasta_df_mod<span class="op">$</span>kofam_names_mod))</span>
<span id="cb46-6"><a href="#cb46-6"></a>fasta_df_mod &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(fasta_df_mod, kofam_names_mod)</span>
<span id="cb46-7"><a href="#cb46-7"></a></span>
<span id="cb46-8"><a href="#cb46-8"></a>genes_modules_list &lt;-<span class="st"> </span>genes_modules1</span>
<span id="cb46-9"><a href="#cb46-9"></a>fasta_df_mod &lt;-<span class="st"> </span>fasta_df_mod <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate_if</span>(is.numeric, </span>
<span id="cb46-10"><a href="#cb46-10"></a>                                            as.character, </span>
<span id="cb46-11"><a href="#cb46-11"></a>                                            is.factor, </span>
<span id="cb46-12"><a href="#cb46-12"></a>                                            as.character) </span>
<span id="cb46-13"><a href="#cb46-13"></a>genes_modules_list[, <span class="kw">c</span>(<span class="st">&quot;ko&quot;</span>, <span class="st">&quot;kegg_module&quot;</span>, <span class="st">&quot;module_category&quot;</span>, </span>
<span id="cb46-14"><a href="#cb46-14"></a>                       <span class="st">&quot;module_subcategory&quot;</span>, <span class="st">&quot;module_name&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="ot">NULL</span>)</span>
<span id="cb46-15"><a href="#cb46-15"></a>fasta_df_mod1 &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(genes_modules_list, fasta_df_mod, </span>
<span id="cb46-16"><a href="#cb46-16"></a>                                                     <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;gene_callers_id&quot;</span> =<span class="st"> &quot;kofam_names_mod&quot;</span>))</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1"></a>fasta_df_mod1 &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(fasta_df_mod1, kofam_genes_tab1, </span>
<span id="cb47-2"><a href="#cb47-2"></a>                                                     <span class="dt">by =</span> <span class="st">&quot;gene_callers_id&quot;</span>)</span>
<span id="cb47-3"><a href="#cb47-3"></a>fasta_df_mod1 &lt;-<span class="st"> </span>fasta_df_mod1[, <span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">2</span>)]</span>
<span id="cb47-4"><a href="#cb47-4"></a>fasta_df_mod1_rn &lt;-<span class="st"> </span>fasta_df_mod1</span>
<span id="cb47-5"><a href="#cb47-5"></a>fasta_df_mod1_rn<span class="op">$</span>contig &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;&gt;&quot;</span>, fasta_df_mod1_rn<span class="op">$</span>contig)</span>
<span id="cb47-6"><a href="#cb47-6"></a><span class="kw">write.table</span>(fasta_df_mod1_rn, <span class="st">&quot;metabolism/00_HELPER_FILES_MOD_ONLY/kofam_genes_rn.fa&quot;</span>,</span>
<span id="cb47-7"><a href="#cb47-7"></a>            <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\r</span><span class="st">&quot;</span>, <span class="dt">col.names =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>,</span>
<span id="cb47-8"><a href="#cb47-8"></a>            <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">fileEncoding =</span> <span class="st">&quot;UTF-8&quot;</span>)</span>
<span id="cb47-9"><a href="#cb47-9"></a><span class="kw">head</span>(fasta_df_mod1_rn)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1"></a>mod_only_list &lt;-<span class="st"> </span>fasta_df_mod1</span>
<span id="cb48-2"><a href="#cb48-2"></a>mod_only_list &lt;-<span class="st"> </span>mod_only_list <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">split_id =</span> <span class="st">&quot;split_00001&quot;</span>, <span class="dt">.after =</span> <span class="dv">1</span>) <span class="op">%&gt;%</span></span>
<span id="cb48-3"><a href="#cb48-3"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unite</span>(items, contig, split_id, <span class="dt">remove =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb48-4"><a href="#cb48-4"></a></span>
<span id="cb48-5"><a href="#cb48-5"></a>mod_only_list<span class="op">$</span>kofam_seqs_mod &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb48-6"><a href="#cb48-6"></a>mod_only_mag &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(mod_only_list, add_mag,  <span class="dt">by =</span> <span class="st">&quot;items&quot;</span>)</span>
<span id="cb48-7"><a href="#cb48-7"></a></span>
<span id="cb48-8"><a href="#cb48-8"></a>gene_mod_only_list &lt;-<span class="st"> </span>mod_only_list <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb48-9"><a href="#cb48-9"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">separate</span>(<span class="st">&quot;items&quot;</span>, </span>
<span id="cb48-10"><a href="#cb48-10"></a>                  <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&quot;V1&quot;</span>, <span class="st">&quot;V2&quot;</span>, <span class="st">&quot;V3&quot;</span>, <span class="st">&quot;gene_callers_id&quot;</span>, <span class="st">&quot;V4&quot;</span>, <span class="st">&quot;V5&quot;</span>),  </span>
<span id="cb48-11"><a href="#cb48-11"></a>                  <span class="dt">sep =</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb48-12"><a href="#cb48-12"></a>gene_mod_only_list[, <span class="kw">c</span>(<span class="st">&quot;V1&quot;</span>, <span class="st">&quot;V2&quot;</span>, <span class="st">&quot;V3&quot;</span>, <span class="st">&quot;V4&quot;</span>, <span class="st">&quot;V5&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="ot">NULL</span>)</span>
<span id="cb48-13"><a href="#cb48-13"></a></span>
<span id="cb48-14"><a href="#cb48-14"></a>gene_mod_only_list<span class="op">$</span>gene_callers_id &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">as.character</span>(gene_mod_only_list<span class="op">$</span>gene_callers_id))</span>
<span id="cb48-15"><a href="#cb48-15"></a>mods_tax &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(gene_mod_only_list, </span>
<span id="cb48-16"><a href="#cb48-16"></a>                                      kofam_with_tax,  </span>
<span id="cb48-17"><a href="#cb48-17"></a>                                      <span class="dt">by =</span> <span class="st">&quot;gene_callers_id&quot;</span>)</span>
<span id="cb48-18"><a href="#cb48-18"></a>kofam_genes_tab1<span class="op">$</span>gene_callers_id &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">as.character</span>(kofam_genes_tab<span class="op">$</span>gene_callers_id))</span>
<span id="cb48-19"><a href="#cb48-19"></a>mod_external_gene &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(gene_mod_only_list, kofam_genes_tab1,  <span class="dt">by =</span> <span class="st">&quot;gene_callers_id&quot;</span>)</span>
<span id="cb48-20"><a href="#cb48-20"></a></span>
<span id="cb48-21"><a href="#cb48-21"></a><span class="kw">write.table</span>(mod_external_gene, <span class="st">&quot;metabolism/00_HELPER_FILES_MOD_ONLY/mod_external_gene.txt&quot;</span>, </span>
<span id="cb48-22"><a href="#cb48-22"></a>            <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</span>
<span id="cb48-23"><a href="#cb48-23"></a><span class="kw">write.table</span>(mod_only_mag, <span class="st">&quot;metabolism/00_HELPER_FILES_MOD_ONLY/mag-mods.txt&quot;</span>, </span>
<span id="cb48-24"><a href="#cb48-24"></a>            <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</span>
<span id="cb48-25"><a href="#cb48-25"></a><span class="kw">write.table</span>(mods_tax, <span class="st">&quot;metabolism/00_HELPER_FILES_MOD_ONLY/mods_tax.txt&quot;</span>, </span>
<span id="cb48-26"><a href="#cb48-26"></a>            <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</span>
<span id="cb48-27"><a href="#cb48-27"></a><span class="kw">write.table</span>(add_view_part_<span class="dv">1</span>, <span class="st">&quot;metabolism/00_HELPER_FILES_MOD_ONLY/mods.txt&quot;</span>, </span>
<span id="cb48-28"><a href="#cb48-28"></a>            <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
</div>
<div class="sourceCode" id="cb49"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb49-1"><a href="#cb49-1"></a><span class="ex">anvi-gen-contigs-database</span> --contigs-fasta 00_HELPER_FILES_MOD_ONLY/kofam_genes_rn.fa --project-name MODS --output-db-path 00_HELPER_FILES_MOD_ONLY/KOFAMS_MODS-contigs.db  --external-gene-calls 00_HELPER_FILES_MOD_ONLY/mod_external_gene.txt --ignore-internal-stop-codons --skip-predict-frame</span>
<span id="cb49-2"><a href="#cb49-2"></a></span>
<span id="cb49-3"><a href="#cb49-3"></a><span class="ex">anvi-import-taxonomy-for-genes</span> --contigs-db 00_HELPER_FILES_MOD_ONLY/KOFAMS_MODS-contigs.db --input-files 00_HELPER_FILES_MOD_ONLY/mods_tax.txt --parser default_matrix</span></code></pre></div>
<p>anvi-import-taxonomy-for-genes –contigs-db KOFAMS_MODS-contigs.db –input-files mods_tax.txt –parser default_matrix</p>
<p>anvi-import-misc-data mods.txt –pan-or-profile-db 03_KOFAM_PROFILES-MERGED/PROFILE.db –target-data-table items anvi-import-misc-data mag-mods.txt –pan-or-profile-db 03_KOFAM_PROFILES-MERGED/PROFILE.db –target-data-table items</p>
<p>distinct(genes_cat, gene_callers_id, .keep_all = T)</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1"></a>mag01_mod &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/06_MAG_KOFAMS/WATER_MAG_00001_modules.txt&quot;</span>, </span>
<span id="cb50-2"><a href="#cb50-2"></a>                    <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb50-3"><a href="#cb50-3"></a>mag01_mod<span class="op">$</span>unique_id &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb50-4"><a href="#cb50-4"></a>mag01_mod<span class="op">$</span>bin_name &lt;-<span class="st"> &quot;MAG01&quot;</span></span>
<span id="cb50-5"><a href="#cb50-5"></a></span>
<span id="cb50-6"><a href="#cb50-6"></a>mag02_mod &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/06_MAG_KOFAMS/WATER_MAG_00002_modules.txt&quot;</span>, </span>
<span id="cb50-7"><a href="#cb50-7"></a>                    <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb50-8"><a href="#cb50-8"></a>mag02_mod<span class="op">$</span>unique_id &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb50-9"><a href="#cb50-9"></a>mag02_mod<span class="op">$</span>bin_name &lt;-<span class="st"> &quot;MAG02&quot;</span></span>
<span id="cb50-10"><a href="#cb50-10"></a></span>
<span id="cb50-11"><a href="#cb50-11"></a>mag03_mod &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/06_MAG_KOFAMS/WATER_MAG_00003_modules.txt&quot;</span>, </span>
<span id="cb50-12"><a href="#cb50-12"></a>                    <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb50-13"><a href="#cb50-13"></a>mag03_mod<span class="op">$</span>unique_id &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb50-14"><a href="#cb50-14"></a>mag03_mod<span class="op">$</span>bin_name &lt;-<span class="st"> &quot;MAG03&quot;</span></span>
<span id="cb50-15"><a href="#cb50-15"></a></span>
<span id="cb50-16"><a href="#cb50-16"></a>mag04_mod &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/06_MAG_KOFAMS/WATER_MAG_00004_modules.txt&quot;</span>, </span>
<span id="cb50-17"><a href="#cb50-17"></a>                    <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb50-18"><a href="#cb50-18"></a>mag04_mod<span class="op">$</span>unique_id &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb50-19"><a href="#cb50-19"></a>mag04_mod<span class="op">$</span>bin_name &lt;-<span class="st"> &quot;MAG04&quot;</span></span>
<span id="cb50-20"><a href="#cb50-20"></a></span>
<span id="cb50-21"><a href="#cb50-21"></a>mag05_mod &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/06_MAG_KOFAMS/WATER_MAG_00005_modules.txt&quot;</span>, </span>
<span id="cb50-22"><a href="#cb50-22"></a>                    <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb50-23"><a href="#cb50-23"></a>mag05_mod<span class="op">$</span>unique_id &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb50-24"><a href="#cb50-24"></a>mag05_mod<span class="op">$</span>bin_name &lt;-<span class="st"> &quot;MAG05&quot;</span></span>
<span id="cb50-25"><a href="#cb50-25"></a></span>
<span id="cb50-26"><a href="#cb50-26"></a>bin13_mod &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/06_MAG_KOFAMS/WATER_Bin_00013_modules.txt&quot;</span>, </span>
<span id="cb50-27"><a href="#cb50-27"></a>                    <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb50-28"><a href="#cb50-28"></a>bin13_mod<span class="op">$</span>unique_id &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb50-29"><a href="#cb50-29"></a>bin13_mod<span class="op">$</span>bin_name &lt;-<span class="st"> &quot;Bin13&quot;</span></span>
<span id="cb50-30"><a href="#cb50-30"></a></span>
<span id="cb50-31"><a href="#cb50-31"></a>all_mods &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/00_HELPER_FILES/all_modules.txt&quot;</span>, </span>
<span id="cb50-32"><a href="#cb50-32"></a>                        <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">quote =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb50-33"><a href="#cb50-33"></a>all_mods &lt;-<span class="st"> </span>all_mods <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">bin_name =</span> metagenome_name)</span>
<span id="cb50-34"><a href="#cb50-34"></a>all_mods<span class="op">$</span>unique_id &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb50-35"><a href="#cb50-35"></a>all_mods<span class="op">$</span>bin_name &lt;-<span class="st"> &quot;ALL&quot;</span></span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1"></a>modules &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(all_mods, mag01_mod, </span>
<span id="cb51-2"><a href="#cb51-2"></a>                            <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;kegg_module&quot;</span>, <span class="st">&quot;module_name&quot;</span>, </span>
<span id="cb51-3"><a href="#cb51-3"></a>                                   <span class="st">&quot;module_class&quot;</span>, <span class="st">&quot;module_category&quot;</span>, </span>
<span id="cb51-4"><a href="#cb51-4"></a>                                   <span class="st">&quot;module_subcategory&quot;</span>), </span>
<span id="cb51-5"><a href="#cb51-5"></a>                            <span class="dt">suffix =</span> <span class="kw">c</span>(<span class="st">&quot;_all&quot;</span>, <span class="st">&quot;_MAG_01&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb51-6"><a href="#cb51-6"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(., mag02_mod, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;kegg_module&quot;</span>, <span class="st">&quot;module_name&quot;</span>, </span>
<span id="cb51-7"><a href="#cb51-7"></a>                                   <span class="st">&quot;module_class&quot;</span>, <span class="st">&quot;module_category&quot;</span>, </span>
<span id="cb51-8"><a href="#cb51-8"></a>                                   <span class="st">&quot;module_subcategory&quot;</span>), <span class="dt">suffix =</span> <span class="kw">c</span>(<span class="st">&quot;XXXX&quot;</span>, <span class="st">&quot;_MAG_02&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb51-9"><a href="#cb51-9"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(., mag03_mod, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;kegg_module&quot;</span>, <span class="st">&quot;module_name&quot;</span>, </span>
<span id="cb51-10"><a href="#cb51-10"></a>                                   <span class="st">&quot;module_class&quot;</span>, <span class="st">&quot;module_category&quot;</span>, </span>
<span id="cb51-11"><a href="#cb51-11"></a>                                   <span class="st">&quot;module_subcategory&quot;</span>), <span class="dt">suffix =</span> <span class="kw">c</span>(<span class="st">&quot;_MAG_02&quot;</span>, <span class="st">&quot;_MAG_03&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb51-12"><a href="#cb51-12"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(., mag04_mod, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;kegg_module&quot;</span>, <span class="st">&quot;module_name&quot;</span>, </span>
<span id="cb51-13"><a href="#cb51-13"></a>                                   <span class="st">&quot;module_class&quot;</span>, <span class="st">&quot;module_category&quot;</span>, </span>
<span id="cb51-14"><a href="#cb51-14"></a>                                   <span class="st">&quot;module_subcategory&quot;</span>), <span class="dt">suffix =</span> <span class="kw">c</span>(<span class="st">&quot;_MAG_03&quot;</span>, <span class="st">&quot;_MAG_04&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb51-15"><a href="#cb51-15"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(., mag05_mod, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;kegg_module&quot;</span>, <span class="st">&quot;module_name&quot;</span>, </span>
<span id="cb51-16"><a href="#cb51-16"></a>                                   <span class="st">&quot;module_class&quot;</span>, <span class="st">&quot;module_category&quot;</span>, </span>
<span id="cb51-17"><a href="#cb51-17"></a>                                   <span class="st">&quot;module_subcategory&quot;</span>), <span class="dt">suffix =</span> <span class="kw">c</span>(<span class="st">&quot;_MAG_04&quot;</span>, <span class="st">&quot;_MAG_05&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb51-18"><a href="#cb51-18"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(., bin13_mod, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;kegg_module&quot;</span>, <span class="st">&quot;module_name&quot;</span>, </span>
<span id="cb51-19"><a href="#cb51-19"></a>                                   <span class="st">&quot;module_class&quot;</span>, <span class="st">&quot;module_category&quot;</span>, </span>
<span id="cb51-20"><a href="#cb51-20"></a>                                   <span class="st">&quot;module_subcategory&quot;</span>), <span class="dt">suffix =</span> <span class="kw">c</span>(<span class="st">&quot;_MAG_05&quot;</span>, <span class="st">&quot;_Bin_13&quot;</span>))</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1"></a>modules &lt;-<span class="st"> </span><span class="kw">rbind</span>(all_mods, mag01_mod, mag02_mod, mag03_mod, mag04_mod, mag05_mod, bin13_mod)</span>
<span id="cb52-2"><a href="#cb52-2"></a>modules[, <span class="kw">c</span>(<span class="st">&quot;module_definition&quot;</span>, <span class="st">&quot;module_class&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="ot">NULL</span>)</span>
<span id="cb52-3"><a href="#cb52-3"></a><span class="kw">colnames</span>(modules)</span>
<span id="cb52-4"><a href="#cb52-4"></a>modules_w &lt;-<span class="st"> </span>modules <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">pivot_wider</span>(</span>
<span id="cb52-5"><a href="#cb52-5"></a>                                          <span class="dt">names_from =</span> <span class="kw">c</span>(<span class="st">&quot;bin_name&quot;</span>), </span>
<span id="cb52-6"><a href="#cb52-6"></a>                                          <span class="dt">values_from =</span> <span class="kw">c</span>(<span class="st">&quot;module_completeness&quot;</span>, </span>
<span id="cb52-7"><a href="#cb52-7"></a>                                                          <span class="st">&quot;module_is_complete&quot;</span>,</span>
<span id="cb52-8"><a href="#cb52-8"></a>                                                          <span class="st">&quot;kofam_hits_in_module&quot;</span>, </span>
<span id="cb52-9"><a href="#cb52-9"></a>                                                          <span class="st">&quot;gene_caller_ids_in_module&quot;</span>),</span>
<span id="cb52-10"><a href="#cb52-10"></a>                                          <span class="dt">names_sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb52-11"><a href="#cb52-11"></a></span>
<span id="cb52-12"><a href="#cb52-12"></a>items_add_data &lt;-<span class="st"> </span>modules_w</span>
<span id="cb52-13"><a href="#cb52-13"></a>items_add_data[, <span class="kw">c</span>(<span class="dv">5</span><span class="op">:</span><span class="kw">max</span>(<span class="kw">ncol</span>(items_add_data)))] &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb52-14"><a href="#cb52-14"></a>items_add_data &lt;-<span class="st"> </span>items_add_data <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="st">&quot;items&quot;</span> =<span class="st"> &quot;kegg_module&quot;</span>)</span>
<span id="cb52-15"><a href="#cb52-15"></a>items_add_data &lt;-<span class="st"> </span>items_add_data[<span class="kw">order</span>(items_add_data<span class="op">$</span>module_category, items_add_data<span class="op">$</span>module_subcategory), ]</span>
<span id="cb52-16"><a href="#cb52-16"></a><span class="kw">write.table</span>(items_add_data, <span class="st">&quot;metabolism/00_HELPER_FILES/module_item_add_data.txt&quot;</span>, </span>
<span id="cb52-17"><a href="#cb52-17"></a>            <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</span>
<span id="cb52-18"><a href="#cb52-18"></a></span>
<span id="cb52-19"><a href="#cb52-19"></a>data_tab &lt;-<span class="st"> </span>modules_w</span>
<span id="cb52-20"><a href="#cb52-20"></a>data_tab &lt;-<span class="st"> </span>data_tab[<span class="kw">order</span>(data_tab<span class="op">$</span>module_category, data_tab<span class="op">$</span>module_subcategory), ]</span>
<span id="cb52-21"><a href="#cb52-21"></a>data_tab[, <span class="kw">c</span>(<span class="dv">2</span><span class="op">:</span><span class="dv">4</span>)] &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb52-22"><a href="#cb52-22"></a>data_tab[, <span class="kw">c</span>(<span class="dv">9</span><span class="op">:</span><span class="kw">max</span>(<span class="kw">ncol</span>(data_tab)))] &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb52-23"><a href="#cb52-23"></a>data_tab &lt;-<span class="st"> </span>data_tab <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">replace</span>(<span class="kw">is.na</span>(.), <span class="dv">0</span>)</span>
<span id="cb52-24"><a href="#cb52-24"></a></span>
<span id="cb52-25"><a href="#cb52-25"></a><span class="kw">names</span>(data_tab) &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="dt">x =</span> <span class="kw">names</span>(data_tab), <span class="dt">pattern =</span> <span class="st">&quot;module_completeness_&quot;</span>, <span class="dt">replacement =</span> <span class="st">&quot;&quot;</span>)  </span>
<span id="cb52-26"><a href="#cb52-26"></a><span class="kw">write.table</span>(data_tab, <span class="st">&quot;metabolism/00_HELPER_FILES/module_data.txt&quot;</span>, </span>
<span id="cb52-27"><a href="#cb52-27"></a>            <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</span>
<span id="cb52-28"><a href="#cb52-28"></a></span>
<span id="cb52-29"><a href="#cb52-29"></a><span class="kw">write.table</span>(data_tab<span class="op">$</span>kegg_module, <span class="st">&quot;metabolism/00_HELPER_FILES/module_order.txt&quot;</span>, </span>
<span id="cb52-30"><a href="#cb52-30"></a>            <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>, <span class="dt">col.names =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
</div>
<p>[1] “unique_row_id” “bin_name” “kegg_module” “module_name”<br />
[5] “module_class” “module_category” “module_subcategory” “module_definition”<br />
[9] “module_completeness” “module_is_complete” “kofam_hits_in_module” “gene_caller_ids_in_module”</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1"></a>coverage &lt;-<span class="st">  </span><span class="kw">read.table</span>(<span class="st">&quot;metabolism/00_HELPER_FILES/WATER-GENE-COVERAGES.txt&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb53-2"><a href="#cb53-2"></a>coverage<span class="op">$</span>key &lt;-<span class="st"> </span><span class="kw">as.factor</span>(coverage<span class="op">$</span>key)</span>
<span id="cb53-3"><a href="#cb53-3"></a>kofam_unique_coveage &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(kofam_unique, coverage, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;gene_callers_id&quot;</span> =<span class="st"> &quot;key&quot;</span>))</span>
<span id="cb53-4"><a href="#cb53-4"></a>kofam_unique_coveage[, <span class="kw">c</span>(<span class="st">&quot;gene_callers_id&quot;</span>, <span class="st">&quot;source&quot;</span>, </span>
<span id="cb53-5"><a href="#cb53-5"></a>                         <span class="st">&quot;ko_function&quot;</span>, <span class="st">&quot;e_value&quot;</span>, </span>
<span id="cb53-6"><a href="#cb53-6"></a>                         <span class="st">&quot;WCCR_1913&quot;</span>, <span class="st">&quot;WCCR_1916&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="ot">NULL</span>)</span>
<span id="cb53-7"><a href="#cb53-7"></a>kofam_unique_coveage1 &lt;-<span class="st"> </span>kofam_unique_coveage[<span class="kw">which</span>(<span class="kw">rowSums</span>(kofam_unique_coveage[,<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>]) <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>),]</span>
<span id="cb53-8"><a href="#cb53-8"></a><span class="kw">write.table</span>(kofam_unique_coveage1, <span class="st">&quot;metabolism/00_HELPER_FILES/kofam_unique_coveage1.txt&quot;</span>, </span>
<span id="cb53-9"><a href="#cb53-9"></a>            <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</span>
<span id="cb53-10"><a href="#cb53-10"></a>kofam_unique_coveage1_u &lt;-<span class="st"> </span>kofam_unique_coveage1[<span class="op">!</span><span class="kw">duplicated</span>(kofam_unique_coveage1<span class="op">$</span>accession),]</span>
<span id="cb53-11"><a href="#cb53-11"></a>kofam_unique_coveage1_u &lt;-<span class="st"> </span>kofam_unique_coveage1_u <span class="op">%&gt;%</span><span class="st"> </span>tibble<span class="op">::</span><span class="kw">remove_rownames</span>() <span class="op">%&gt;%</span></span>
<span id="cb53-12"><a href="#cb53-12"></a><span class="st">  </span>tibble<span class="op">::</span><span class="kw">column_to_rownames</span>(<span class="st">&quot;accession&quot;</span>)</span>
<span id="cb53-13"><a href="#cb53-13"></a></span>
<span id="cb53-14"><a href="#cb53-14"></a>kofam_unique_coveage1_u &lt;-<span class="st"> </span><span class="kw">data.matrix</span>(kofam_unique_coveage1_u[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>], <span class="dt">rownames.force =</span> <span class="ot">NULL</span>)</span>
<span id="cb53-15"><a href="#cb53-15"></a>i &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb53-16"><a href="#cb53-16"></a>pv.out &lt;-<span class="st"> </span><span class="kw">pathview</span>(<span class="dt">gene.data =</span> kofam_unique_coveage1_u[, <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>], </span>
<span id="cb53-17"><a href="#cb53-17"></a>                   <span class="dt">pathway.id =</span> <span class="st">&quot;00920&quot;</span>, </span>
<span id="cb53-18"><a href="#cb53-18"></a>                   <span class="dt">species =</span> <span class="st">&quot;ko&quot;</span>, <span class="dt">out.suffix =</span> <span class="st">&quot;all&quot;</span>, </span>
<span id="cb53-19"><a href="#cb53-19"></a>                   <span class="dt">kegg.native =</span> <span class="ot">TRUE</span>)</span>
<span id="cb53-20"><a href="#cb53-20"></a><span class="kw">str</span>(pv.out)</span>
<span id="cb53-21"><a href="#cb53-21"></a><span class="kw">head</span>(pv.out<span class="op">$</span>plot.data.gene)</span>
<span id="cb53-22"><a href="#cb53-22"></a><span class="co">#result PNG file in current directory</span></span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1"></a>mag01 &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&#39;metabolism/06_MAG_KOFAMS/16s-asv_data-2019.txt&#39;</span>, <span class="dt">header =</span> T)</span>
<span id="cb54-2"><a href="#cb54-2"></a>all_<span class="dv">2019</span> &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(rrna, amf, its)</span>
<span id="cb54-3"><a href="#cb54-3"></a></span>
<span id="cb54-4"><a href="#cb54-4"></a>mag02 &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;mag02.txt&quot;</span>, <span class="dt">header =</span> <span class="ot">FALSE</span>)</span>
<span id="cb54-5"><a href="#cb54-5"></a>mag02 &lt;-<span class="st"> </span><span class="kw">c</span>(mag02<span class="op">$</span>V1)</span>
<span id="cb54-6"><a href="#cb54-6"></a><span class="co">#KEGG view: gene data only</span></span>
<span id="cb54-7"><a href="#cb54-7"></a></span>
<span id="cb54-8"><a href="#cb54-8"></a></span>
<span id="cb54-9"><a href="#cb54-9"></a><span class="co">#load data</span></span>
<span id="cb54-10"><a href="#cb54-10"></a><span class="kw">data</span>(gse16873.d)</span>
<span id="cb54-11"><a href="#cb54-11"></a><span class="kw">data</span>(demo.paths)</span>
<span id="cb54-12"><a href="#cb54-12"></a></span>
<span id="cb54-13"><a href="#cb54-13"></a>test &lt;-<span class="st"> </span><span class="kw">data.frame</span>(gse16873.d)</span>
<span id="cb54-14"><a href="#cb54-14"></a>test &lt;-<span class="st"> </span>test <span class="op">%&gt;%</span><span class="st"> </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;id&quot;</span>)</span>
<span id="cb54-15"><a href="#cb54-15"></a></span>
<span id="cb54-16"><a href="#cb54-16"></a>test[<span class="kw">duplicated</span>(test<span class="op">$</span>id),]</span>
<span id="cb54-17"><a href="#cb54-17"></a><span class="kw">dim</span>(test[<span class="kw">duplicated</span>(test<span class="op">$</span>id),])[<span class="dv">1</span>]</span>
<span id="cb54-18"><a href="#cb54-18"></a></span>
<span id="cb54-19"><a href="#cb54-19"></a>test<span class="op">$</span>id</span>
<span id="cb54-20"><a href="#cb54-20"></a></span>
<span id="cb54-21"><a href="#cb54-21"></a><span class="co">#KEGG view: gene data only</span></span>
<span id="cb54-22"><a href="#cb54-22"></a>i &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb54-23"><a href="#cb54-23"></a>pv.out &lt;-<span class="st"> </span><span class="kw">pathview</span>(<span class="dt">gene.data =</span> gse16873.d[, <span class="dv">1</span>], </span>
<span id="cb54-24"><a href="#cb54-24"></a>                   <span class="dt">pathway.id =</span> demo.paths<span class="op">$</span>sel.paths[i], </span>
<span id="cb54-25"><a href="#cb54-25"></a>                   <span class="dt">species =</span> <span class="st">&quot;hsa&quot;</span>, <span class="dt">out.suffix =</span> <span class="st">&quot;gse16873&quot;</span>, </span>
<span id="cb54-26"><a href="#cb54-26"></a>                   <span class="dt">kegg.native =</span> <span class="ot">TRUE</span>)</span>
<span id="cb54-27"><a href="#cb54-27"></a><span class="kw">str</span>(pv.out)</span>
<span id="cb54-28"><a href="#cb54-28"></a><span class="kw">head</span>(pv.out<span class="op">$</span>plot.data.gene)</span>
<span id="cb54-29"><a href="#cb54-29"></a><span class="co">#result PNG file in current directory</span></span>
<span id="cb54-30"><a href="#cb54-30"></a></span>
<span id="cb54-31"><a href="#cb54-31"></a><span class="co">#Graphviz view: gene data only</span></span>
<span id="cb54-32"><a href="#cb54-32"></a>pv.out &lt;-<span class="st"> </span><span class="kw">pathview</span>(<span class="dt">gene.data =</span> gse16873.d[, <span class="dv">1</span>], </span>
<span id="cb54-33"><a href="#cb54-33"></a>                   <span class="dt">pathway.id =</span> demo.paths<span class="op">$</span>sel.paths[i], </span>
<span id="cb54-34"><a href="#cb54-34"></a>                   <span class="dt">species =</span> <span class="st">&quot;hsa&quot;</span>, <span class="dt">out.suffix =</span> <span class="st">&quot;gse16873&quot;</span>, </span>
<span id="cb54-35"><a href="#cb54-35"></a>                   <span class="dt">kegg.native =</span> <span class="ot">FALSE</span>, <span class="dt">sign.pos =</span> demo.paths<span class="op">$</span>spos[i])</span>
<span id="cb54-36"><a href="#cb54-36"></a><span class="co">#result PDF file in current directory</span></span>
<span id="cb54-37"><a href="#cb54-37"></a></span>
<span id="cb54-38"><a href="#cb54-38"></a><span class="co">#KEGG view: both gene and compound data</span></span>
<span id="cb54-39"><a href="#cb54-39"></a>sim.cpd.data=<span class="kw">sim.mol.data</span>(<span class="dt">mol.type=</span><span class="st">&quot;cpd&quot;</span>, <span class="dt">nmol=</span><span class="dv">3000</span>)</span>
<span id="cb54-40"><a href="#cb54-40"></a>i &lt;-<span class="st"> </span><span class="dv">3</span></span>
<span id="cb54-41"><a href="#cb54-41"></a><span class="kw">print</span>(demo.paths<span class="op">$</span>sel.paths[i])</span>
<span id="cb54-42"><a href="#cb54-42"></a>pv.out &lt;-<span class="st"> </span><span class="kw">pathview</span>(<span class="dt">gene.data =</span> gse16873.d[, <span class="dv">1</span>], <span class="dt">cpd.data =</span> sim.cpd.data, </span>
<span id="cb54-43"><a href="#cb54-43"></a>                   <span class="dt">pathway.id =</span> demo.paths<span class="op">$</span>sel.paths[i], </span>
<span id="cb54-44"><a href="#cb54-44"></a>                   <span class="dt">species =</span> <span class="st">&quot;hsa&quot;</span>, <span class="dt">out.suffix =</span> <span class="st">&quot;gse16873.cpd&quot;</span>, </span>
<span id="cb54-45"><a href="#cb54-45"></a>                   <span class="dt">keys.align =</span> <span class="st">&quot;y&quot;</span>, <span class="dt">kegg.native =</span> <span class="ot">TRUE</span>, </span>
<span id="cb54-46"><a href="#cb54-46"></a>                   <span class="dt">key.pos =</span> demo.paths<span class="op">$</span>kpos1[i])</span>
<span id="cb54-47"><a href="#cb54-47"></a><span class="kw">str</span>(pv.out)</span>
<span id="cb54-48"><a href="#cb54-48"></a><span class="kw">head</span>(pv.out<span class="op">$</span>plot.data.cpd)</span>
<span id="cb54-49"><a href="#cb54-49"></a></span>
<span id="cb54-50"><a href="#cb54-50"></a><span class="co">#multiple states in one graph</span></span>
<span id="cb54-51"><a href="#cb54-51"></a><span class="kw">set.seed</span>(<span class="dv">10</span>)</span>
<span id="cb54-52"><a href="#cb54-52"></a>sim.cpd.data2 =<span class="st"> </span><span class="kw">matrix</span>(<span class="kw">sample</span>(sim.cpd.data, <span class="dv">18000</span>, <span class="dt">replace =</span> <span class="ot">TRUE</span>), <span class="dt">ncol =</span> <span class="dv">6</span>)</span>
<span id="cb54-53"><a href="#cb54-53"></a>pv.out &lt;-<span class="st"> </span><span class="kw">pathview</span>(<span class="dt">gene.data =</span> gse16873.d[, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>], </span>
<span id="cb54-54"><a href="#cb54-54"></a>                   <span class="dt">cpd.data =</span> sim.cpd.data2[, <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>], </span>
<span id="cb54-55"><a href="#cb54-55"></a>                   <span class="dt">pathway.id =</span> demo.paths<span class="op">$</span>sel.paths[i], </span>
<span id="cb54-56"><a href="#cb54-56"></a>                   <span class="dt">species =</span> <span class="st">&quot;hsa&quot;</span>, <span class="dt">out.suffix =</span> <span class="st">&quot;gse16873.cpd.3-2s&quot;</span>, </span>
<span id="cb54-57"><a href="#cb54-57"></a>                   <span class="dt">keys.align =</span> <span class="st">&quot;y&quot;</span>, <span class="dt">kegg.native =</span> <span class="ot">TRUE</span>, </span>
<span id="cb54-58"><a href="#cb54-58"></a>                   <span class="dt">match.data =</span> <span class="ot">FALSE</span>, <span class="dt">multi.state =</span> <span class="ot">TRUE</span>, <span class="dt">same.layer =</span> <span class="ot">TRUE</span>)</span>
<span id="cb54-59"><a href="#cb54-59"></a><span class="kw">str</span>(pv.out)</span>
<span id="cb54-60"><a href="#cb54-60"></a><span class="kw">head</span>(pv.out<span class="op">$</span>plot.data.cpd)</span>
<span id="cb54-61"><a href="#cb54-61"></a></span>
<span id="cb54-62"><a href="#cb54-62"></a><span class="co">#result PNG file in current directory</span></span>
<span id="cb54-63"><a href="#cb54-63"></a></span>
<span id="cb54-64"><a href="#cb54-64"></a><span class="co">##more examples of pathview usages are shown in the vignette.</span></span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1"></a>arco &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&#39;arco-phylo-kegg-modules_modules.txt&#39;</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, </span>
<span id="cb55-2"><a href="#cb55-2"></a>                    <span class="dt">header =</span> <span class="ot">TRUE</span>,  <span class="dt">quote =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb55-3"><a href="#cb55-3"></a>arco_t &lt;-<span class="st"> </span>arco</span>
<span id="cb55-4"><a href="#cb55-4"></a>arco_t[, <span class="kw">c</span>(<span class="st">&quot;unique_id&quot;</span>, <span class="st">&quot;db_name&quot;</span>, <span class="st">&quot;module_name&quot;</span>, <span class="st">&quot;module_class&quot;</span>, <span class="st">&quot;module_category&quot;</span>, <span class="st">&quot;module_subcategory&quot;</span>, <span class="st">&quot;module_definition&quot;</span>, <span class="st">&quot;kofam_hits_in_module&quot;</span>, <span class="st">&quot;gene_caller_ids_in_module&quot;</span>, <span class="st">&quot;module_completeness&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="ot">NULL</span>)</span>
<span id="cb55-5"><a href="#cb55-5"></a></span>
<span id="cb55-6"><a href="#cb55-6"></a>arco_t<span class="op">$</span>module_is_complete &lt;-<span class="st"> </span><span class="kw">str_replace</span>(arco_t<span class="op">$</span>module_is_complete, <span class="st">&quot;True&quot;</span>, <span class="st">&quot;1&quot;</span>)</span>
<span id="cb55-7"><a href="#cb55-7"></a>arco_t<span class="op">$</span>module_is_complete &lt;-<span class="st"> </span><span class="kw">str_replace</span>(arco_t<span class="op">$</span>module_is_complete, <span class="st">&quot;False&quot;</span>, <span class="st">&quot;0&quot;</span>)</span>
<span id="cb55-8"><a href="#cb55-8"></a>arco_t<span class="op">$</span>module_is_complete &lt;-<span class="st"> </span><span class="kw">as.integer</span>(arco_t<span class="op">$</span>module_is_complete)</span>
<span id="cb55-9"><a href="#cb55-9"></a>arco_t_w &lt;-<span class="st"> </span>arco_t <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="kw">c</span>(<span class="st">&quot;kegg_module&quot;</span>), <span class="dt">values_from =</span> <span class="kw">c</span>(<span class="st">&quot;module_is_complete&quot;</span>))</span>
<span id="cb55-10"><a href="#cb55-10"></a>arco_t_w &lt;-<span class="st"> </span>arco_t_w[, <span class="kw">colSums</span>(arco_t_w <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>]</span>
<span id="cb55-11"><a href="#cb55-11"></a>arco_t_w &lt;-<span class="st"> </span><span class="kw">column_to_rownames</span>(arco_t_w, <span class="dt">var =</span> <span class="st">&quot;genome_name&quot;</span>)</span>
<span id="cb55-12"><a href="#cb55-12"></a>arco_t_w &lt;-<span class="st"> </span>arco_t_w[, <span class="kw">colSums</span>(arco_t_w <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) <span class="op">&lt;=</span><span class="st"> </span><span class="dv">70</span>]</span>
<span id="cb55-13"><a href="#cb55-13"></a>arco_t_w &lt;-<span class="st"> </span><span class="kw">rownames_to_column</span>(arco_t_w, <span class="dt">var =</span> <span class="st">&quot;genome_name&quot;</span>)</span>
<span id="cb55-14"><a href="#cb55-14"></a>g_order &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&#39;genome_order.txt&#39;</span>, </span>
<span id="cb55-15"><a href="#cb55-15"></a>                    <span class="dt">header =</span> <span class="ot">TRUE</span>,  <span class="dt">quote =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb55-16"><a href="#cb55-16"></a></span>
<span id="cb55-17"><a href="#cb55-17"></a>arco_t_w_order &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(g_order, arco_t_w, </span>
<span id="cb55-18"><a href="#cb55-18"></a>                                   <span class="dt">by =</span> <span class="st">&quot;genome_name&quot;</span>)</span>
<span id="cb55-19"><a href="#cb55-19"></a>arco_t_l &lt;-<span class="st"> </span>arco_t_w_order <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="dt">cols =</span> <span class="dv">2</span><span class="op">:</span><span class="dv">34</span>, <span class="dt">names_to =</span> <span class="st">&quot;kegg_module&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;module_is_complete&quot;</span>)</span>
<span id="cb55-20"><a href="#cb55-20"></a></span>
<span id="cb55-21"><a href="#cb55-21"></a>arco_t_w_order_no_row_id &lt;-<span class="st"> </span>arco_t_w_order <span class="op">%&gt;%</span><span class="st"> </span>tibble<span class="op">::</span><span class="kw">column_to_rownames</span>(<span class="st">&quot;genome_name&quot;</span>)</span>
<span id="cb55-22"><a href="#cb55-22"></a></span>
<span id="cb55-23"><a href="#cb55-23"></a></span>
<span id="cb55-24"><a href="#cb55-24"></a><span class="kw">heatmaply</span>(arco_t_w_order_no_row_id, </span>
<span id="cb55-25"><a href="#cb55-25"></a>          <span class="dt">Rowv =</span> <span class="ot">NULL</span>, <span class="dt">Colv =</span> <span class="ot">NULL</span>, </span>
<span id="cb55-26"><a href="#cb55-26"></a>          <span class="dt">k_row =</span> <span class="dv">1</span>, <span class="dt">k_col =</span> <span class="dv">1</span>, </span>
<span id="cb55-27"><a href="#cb55-27"></a>          <span class="dt">file =</span> <span class="st">&quot;heatmaply_plot.html&quot;</span>, </span>
<span id="cb55-28"><a href="#cb55-28"></a>          <span class="dt">colors =</span> <span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;red&quot;</span>), </span>
<span id="cb55-29"><a href="#cb55-29"></a>          <span class="dt">hide_colorbar =</span> <span class="ot">TRUE</span>, </span>
<span id="cb55-30"><a href="#cb55-30"></a>          <span class="dt">fontsize_row =</span> <span class="dv">6</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1"></a>coverage &lt;-<span class="st">  </span><span class="kw">read.table</span>(<span class="st">&quot;WATER-GENE-COVERAGES.txt&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb56-2"><a href="#cb56-2"></a>kofam &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;00_HELPER_FILES/KOfam.txt&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, </span>
<span id="cb56-3"><a href="#cb56-3"></a>                    <span class="dt">header =</span> <span class="ot">TRUE</span>,  <span class="dt">quote =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb56-4"><a href="#cb56-4"></a>kofam_cov &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(kofam, coverage, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;gene_callers_id&quot;</span> =<span class="st"> &quot;key&quot;</span>))</span></code></pre></div>
</div>
<div class="sourceCode" id="cb57"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb57-1"><a href="#cb57-1"></a><span class="ex">anvi-search-functions</span> --contigs-db 03_CONTIGS/WATER-contigs.db  --full-report ko_search.txt --annotation-sources KOfam --search-terms K17222,K17223,K17227,K17226,K17224,K17225,K22622,K00170,K00169,K00171,K00172,K00174,K00175,K00240,K01676,K00244,K00246,K00245,K01007,K01682,K00407,K00406,K00404,K00405</span></code></pre></div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1"></a>kofam_search &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;00_HELPER_FILES/ko_search.txt&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, </span>
<span id="cb58-2"><a href="#cb58-2"></a>                    <span class="dt">header =</span> <span class="ot">TRUE</span>,  <span class="dt">quote =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb58-3"><a href="#cb58-3"></a>kofam_seach_cov &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(kofam_search, kofam_cov, <span class="dt">by =</span> <span class="st">&quot;gene_callers_id&quot;</span>)</span>
<span id="cb58-4"><a href="#cb58-4"></a><span class="kw">write.table</span>(kofam_seach_cov, <span class="st">&quot;kofam_seach_cov.txt&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
</div>
<div class="post-nav">
<div class="post-nav-item">
<div class="meta-nav">
Previous
</div>
<p><a href="mag02-phylogenomics.html" rel="next">N<sup><u>o</u></sup> 6. MAG02 Phylogenomic Analysis</a></p>
</div>
</div>
<h2 id="source-code" class="appendix">Source Code</h2>
<p>The source code for this page can be accessed on GitHub by <a href="https://github.com/hypocolypse/web/blob/master/mg-function.Rmd">clicking this link</a>.</p>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/hypocolypse/web/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/hypocolypse/web/">https://github.com/hypocolypse/web/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p>© Copyright 2020 The HYPOCOLYPSE Project.</p>
<p>Site constructed using <a href="https://rstudio.github.io/distill/">Distill for R Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
